{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "   # RNAseq Coho Differential Expression\n",
    "   \n",
    "   This notebook is the first in a series that outlines the steps taken to analyse an RNAseq data set from raw reads to differential expression analysis and Gene Ontology Enrichment analysis. This contains the steps taken to trim-reads and run quality control on the raw dataset. \n",
    "   \n",
    "   The raw data files are too large to post on Github, but can be provided (aspanjer@uw.edu)\n",
    "   \n",
    "   Software used in this analysis:\n",
    "   [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)\n",
    "   [MultiQC](http://multiqc.info)\n",
    "   [Trim galore](http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/)\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "A quick look at the raw data files that are used for the analysis:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/andrewspanjer/RNAseq_Coho\n"
     ]
    }
   ],
   "source": [
    "cd /Users/andrewspanjer/RNAseq_Coho/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A.1.fastq.gz  F.1.fastq.gz  K.1.fastq.gz  P.1.fastq.gz  T.2.fastq.gz\r\n",
      "A.2.fastq.gz  F.2.fastq.gz  K.2.fastq.gz  P.2.fastq.gz  U.1.fastq.gz\r\n",
      "B.1.fastq.gz  G.1.fastq.gz  L.1.fastq.gz  Q.1.fastq.gz  U.2.fastq.gz\r\n",
      "B.2.fastq.gz  G.2.fastq.gz  L.2.fastq.gz  Q.2.fastq.gz  V.1.fastq.gz\r\n",
      "C.1.fastq.gz  H.1.fastq.gz  M.1.fastq.gz  R.1.fastq.gz  V.2.fastq.gz\r\n",
      "C.2.fastq.gz  H.2.fastq.gz  M.2.fastq.gz  R.2.fastq.gz  W.1.fastq.gz\r\n",
      "D.1.fastq.gz  I.1.fastq.gz  N.1.fastq.gz  README.md     W.2.fastq.gz\r\n",
      "D.2.fastq.gz  I.2.fastq.gz  N.2.fastq.gz  S.1.fastq.gz  X.1.fastq.gz\r\n",
      "E.1.fastq.gz  J.1.fastq.gz  O.1.fastq.gz  S.2.fastq.gz  X.2.fastq.gz\r\n",
      "E.2.fastq.gz  J.2.fastq.gz  O.2.fastq.gz  T.1.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Unzip first fastq file to look at the first 10 lines to see how the raw sequencng data looks. Rezip the file to save space... "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "head: A.1.fastq: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!gunzip A.1.fastq.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For a complete description of the data files see the README.md in /data. This file is one of 48 sequencing files representing PE reads of RNA from Juvenile Coho fish livers. Sequencing was performed by the UW Genomic Core facility."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@HWKWKBGXX:1:11101:10000:19471/1\r\n",
      "TTTTTTTGTACAAAAGCTGAGTAAGTTGGCCTGCAAAGCCCTAATTAAATGCCCCCACAACCAATAGAGAAGCAACAATTACAATTGTCAACTCTTTTTGCAGGCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACGAATTCGTATA\r\n",
      "+\r\n",
      "AAAAAEEAEEEEAEEEEA<EEEEE6E//6E/AA<EEEEEE/EEE/EEAEEE<EA/EE/E666<EAAA6<EEEEEEEAEE<E/EEA/A/<EE/EE<EE/<//E//AE/<EAEEEEEEE/6/A/<E/E<E/AA</AAE<A//EEE<A/AE##\r\n",
      "@HWKWKBGXX:1:11101:10002:10211/1\r\n",
      "GAAGTGTTCTAGGGGTACCGCTTCAACAACGATGGGTATGAAGCTATGGTTAGCCCCGCAGATTTCAGAACATTGTCCGTAGAATACTCCAGGTCGAGAGATCGGAAGAGCACACGTCTGAACTCCAGTCACGAATTCGTATCTCGTATG\r\n",
      "+\r\n",
      "6A/AA6/EEEEEAEEEEAAEEEEEEAE/EEAEEEEAEE//E//AEEEEA/EEE6/EEEEEEA//E//</<EEAE/AA6EEEA<AEEEE/6E<6AE<EEEE6/AAEAEAEEAEEE<A/EAA<E/EE/66EEA66AAE/</AE<<66<A###\r\n",
      "@HWKWKBGXX:1:11101:10005:13424/1\r\n",
      "CCACCGCCGGCTTGTACCTGTTGAGGTAGTCAGCCACGCCACATATCGTCGGGCAGTACTTGCCAAATCCATCCCTTGGTGTGCAGTCCTCAGAATAGTCTCCCCTCGTTTGTGCAGAGGAGAGGGAGAACAGCAAAAGAACGCCTGCGG\r\n"
     ]
    }
   ],
   "source": [
    "!head A.1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The data files are structured as follows:\n",
    "\n",
    "line 1:  @ 'instrument':'run number':'flowcell ID':'lane':'tile':'x-pos':'y-pos' 'read':'is filtered':'control number':'index sequence'\n",
    "\n",
    "Line 2: This is the actual sequence\n",
    "\n",
    "Line 3: a line break signaled by '+'\n",
    "\n",
    "Line 4: These are quality codes for each base in the sequence. A guide for these codes can be found here: http://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/QualityScoreEncoding_swBS.htm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!gzip A.1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, I want to look at the quality of my raw data reads..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of A.1.fastq.gz\n",
      "Approx 5% complete for A.1.fastq.gz\n",
      "Approx 10% complete for A.1.fastq.gz\n",
      "Approx 15% complete for A.1.fastq.gz\n",
      "Approx 20% complete for A.1.fastq.gz\n",
      "Approx 25% complete for A.1.fastq.gz\n",
      "Approx 30% complete for A.1.fastq.gz\n",
      "Approx 35% complete for A.1.fastq.gz\n",
      "Approx 40% complete for A.1.fastq.gz\n",
      "Approx 45% complete for A.1.fastq.gz\n",
      "Approx 50% complete for A.1.fastq.gz\n",
      "Approx 55% complete for A.1.fastq.gz\n",
      "Approx 60% complete for A.1.fastq.gz\n",
      "Approx 65% complete for A.1.fastq.gz\n",
      "Approx 70% complete for A.1.fastq.gz\n",
      "Approx 75% complete for A.1.fastq.gz\n",
      "Approx 80% complete for A.1.fastq.gz\n",
      "Approx 85% complete for A.1.fastq.gz\n",
      "Approx 90% complete for A.1.fastq.gz\n",
      "Approx 95% complete for A.1.fastq.gz\n",
      "Analysis complete for A.1.fastq.gz\n",
      "Started analysis of A.2.fastq.gz\n",
      "Approx 5% complete for A.2.fastq.gz\n",
      "Approx 10% complete for A.2.fastq.gz\n",
      "Approx 15% complete for A.2.fastq.gz\n",
      "Approx 20% complete for A.2.fastq.gz\n",
      "Approx 25% complete for A.2.fastq.gz\n",
      "Approx 30% complete for A.2.fastq.gz\n",
      "Approx 35% complete for A.2.fastq.gz\n",
      "Approx 40% complete for A.2.fastq.gz\n",
      "Approx 45% complete for A.2.fastq.gz\n",
      "Approx 50% complete for A.2.fastq.gz\n",
      "Approx 55% complete for A.2.fastq.gz\n",
      "Approx 60% complete for A.2.fastq.gz\n",
      "Approx 65% complete for A.2.fastq.gz\n",
      "Approx 70% complete for A.2.fastq.gz\n",
      "Approx 75% complete for A.2.fastq.gz\n",
      "Approx 80% complete for A.2.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "! scripts/FastQC/fastqc data/*.fastq.gz -o analyses/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A summary of these fastqc reports can be generated for all files using... "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v0.8\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching './analyses/fastQC_before/'\n",
      "[INFO   ]          fastqc : Found 48 reports\n",
      "[INFO   ]         multiqc : Report      : analyses/multiqc_before.html\n",
      "[INFO   ]         multiqc : Data        : analyses/multiqc_before_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc ./analyses/fastQC_before/ -o ./analyses/ -n multiqc_before"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, based on the fastqc results, I want to run a triming program to remove low quality reads, the output of this program is quite large so I put the results on an external harddrive...\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/A.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t273916\tAGATCGGAAGAGC\t1000000\t27.39\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 273916). Second best hit was Nextera (count: 2)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/A.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/A.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/A.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to A.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/A.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/A.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 636.67 s (45 us/read; 1.34 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,188,117\n",
      "Reads with adapters:                 8,628,250 (60.8%)\n",
      "Reads written (passing filters):    14,188,117 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,128,217,550 bp\n",
      "Quality-trimmed:             253,477,958 bp (11.9%)\n",
      "Total written (filtered):  1,692,805,170 bp (79.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 8628250 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 22.0%\n",
      "  C: 31.8%\n",
      "  G: 26.7%\n",
      "  T: 17.6%\n",
      "  none/other: 1.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2666963\t3547029.2\t0\t2666963\n",
      "2\t605006\t886757.3\t0\t605006\n",
      "3\t311682\t221689.3\t0\t311682\n",
      "4\t134798\t55422.3\t0\t134798\n",
      "5\t72606\t13855.6\t0\t72606\n",
      "6\t50292\t3463.9\t0\t50292\n",
      "7\t48711\t866.0\t0\t48711\n",
      "8\t64828\t216.5\t0\t64828\n",
      "9\t99184\t54.1\t0\t99023 161\n",
      "10\t95939\t13.5\t1\t84307 11632\n",
      "11\t106247\t3.4\t1\t84026 22221\n",
      "12\t85523\t0.8\t1\t70206 15317\n",
      "13\t81158\t0.2\t1\t70892 10266\n",
      "14\t78835\t0.2\t1\t63989 14846\n",
      "15\t81403\t0.2\t1\t70671 10732\n",
      "16\t89525\t0.2\t1\t66914 22611\n",
      "17\t81524\t0.2\t1\t68177 13347\n",
      "18\t60023\t0.2\t1\t52571 7452\n",
      "19\t75697\t0.2\t1\t65817 9880\n",
      "20\t59296\t0.2\t1\t50994 8302\n",
      "21\t79066\t0.2\t1\t68810 10256\n",
      "22\t79735\t0.2\t1\t70068 9667\n",
      "23\t80663\t0.2\t1\t69340 11323\n",
      "24\t101759\t0.2\t1\t83477 18282\n",
      "25\t84223\t0.2\t1\t71089 13134\n",
      "26\t80033\t0.2\t1\t66910 13123\n",
      "27\t82922\t0.2\t1\t71978 10944\n",
      "28\t65102\t0.2\t1\t57638 7464\n",
      "29\t96468\t0.2\t1\t81894 14574\n",
      "30\t73598\t0.2\t1\t64183 9415\n",
      "31\t89529\t0.2\t1\t77979 11550\n",
      "32\t76518\t0.2\t1\t67626 8892\n",
      "33\t84584\t0.2\t1\t72100 12484\n",
      "34\t80748\t0.2\t1\t70831 9917\n",
      "35\t72407\t0.2\t1\t62513 9894\n",
      "36\t84060\t0.2\t1\t71988 12072\n",
      "37\t106370\t0.2\t1\t89740 16630\n",
      "38\t97434\t0.2\t1\t82435 14999\n",
      "39\t122763\t0.2\t1\t104111 18652\n",
      "40\t123222\t0.2\t1\t106443 16779\n",
      "41\t149220\t0.2\t1\t131133 18087\n",
      "42\t173180\t0.2\t1\t152240 20940\n",
      "43\t179578\t0.2\t1\t152779 26799\n",
      "44\t113773\t0.2\t1\t98374 15399\n",
      "45\t41599\t0.2\t1\t36175 5424\n",
      "46\t39153\t0.2\t1\t33409 5744\n",
      "47\t38359\t0.2\t1\t33533 4826\n",
      "48\t47016\t0.2\t1\t42363 4653\n",
      "49\t61509\t0.2\t1\t55386 6123\n",
      "50\t52497\t0.2\t1\t45243 7254\n",
      "51\t77420\t0.2\t1\t68109 9311\n",
      "52\t61777\t0.2\t1\t55387 6390\n",
      "53\t47195\t0.2\t1\t42136 5059\n",
      "54\t48056\t0.2\t1\t43137 4919\n",
      "55\t55479\t0.2\t1\t50264 5215\n",
      "56\t41616\t0.2\t1\t37414 4202\n",
      "57\t25225\t0.2\t1\t22504 2721\n",
      "58\t41401\t0.2\t1\t37368 4033\n",
      "59\t54783\t0.2\t1\t49349 5434\n",
      "60\t23684\t0.2\t1\t21125 2559\n",
      "61\t32635\t0.2\t1\t29275 3360\n",
      "62\t34911\t0.2\t1\t31570 3341\n",
      "63\t18399\t0.2\t1\t16432 1967\n",
      "64\t27163\t0.2\t1\t24464 2699\n",
      "65\t32516\t0.2\t1\t29294 3222\n",
      "66\t20593\t0.2\t1\t18499 2094\n",
      "67\t22579\t0.2\t1\t20212 2367\n",
      "68\t23651\t0.2\t1\t21151 2500\n",
      "69\t22885\t0.2\t1\t20149 2736\n",
      "70\t23225\t0.2\t1\t20322 2903\n",
      "71\t27347\t0.2\t1\t23499 3848\n",
      "72\t35890\t0.2\t1\t30498 5392\n",
      "73\t38523\t0.2\t1\t32407 6116\n",
      "74\t41479\t0.2\t1\t33590 7889\n",
      "75\t33751\t0.2\t1\t27278 6473\n",
      "76\t10168\t0.2\t1\t7983 2185\n",
      "77\t1210\t0.2\t1\t925 285\n",
      "78\t1457\t0.2\t1\t1240 217\n",
      "79\t2636\t0.2\t1\t2340 296\n",
      "80\t3640\t0.2\t1\t3241 399\n",
      "81\t4351\t0.2\t1\t3876 475\n",
      "82\t4527\t0.2\t1\t4046 481\n",
      "83\t4630\t0.2\t1\t4174 456\n",
      "84\t4540\t0.2\t1\t4058 482\n",
      "85\t4221\t0.2\t1\t3829 392\n",
      "86\t4120\t0.2\t1\t3710 410\n",
      "87\t3824\t0.2\t1\t3430 394\n",
      "88\t3650\t0.2\t1\t3259 391\n",
      "89\t3383\t0.2\t1\t3033 350\n",
      "90\t2999\t0.2\t1\t2686 313\n",
      "91\t2670\t0.2\t1\t2412 258\n",
      "92\t2432\t0.2\t1\t2168 264\n",
      "93\t2001\t0.2\t1\t1798 203\n",
      "94\t1775\t0.2\t1\t1590 185\n",
      "95\t1672\t0.2\t1\t1507 165\n",
      "96\t1560\t0.2\t1\t1393 167\n",
      "97\t1538\t0.2\t1\t1376 162\n",
      "98\t1451\t0.2\t1\t1310 141\n",
      "99\t1272\t0.2\t1\t1139 133\n",
      "100\t1196\t0.2\t1\t1071 125\n",
      "101\t1048\t0.2\t1\t941 107\n",
      "102\t861\t0.2\t1\t784 77\n",
      "103\t688\t0.2\t1\t625 63\n",
      "104\t642\t0.2\t1\t570 72\n",
      "105\t593\t0.2\t1\t523 70\n",
      "106\t482\t0.2\t1\t428 54\n",
      "107\t525\t0.2\t1\t448 77\n",
      "108\t436\t0.2\t1\t380 56\n",
      "109\t461\t0.2\t1\t412 49\n",
      "110\t402\t0.2\t1\t356 46\n",
      "111\t411\t0.2\t1\t364 47\n",
      "112\t326\t0.2\t1\t294 32\n",
      "113\t319\t0.2\t1\t279 40\n",
      "114\t271\t0.2\t1\t242 29\n",
      "115\t286\t0.2\t1\t250 36\n",
      "116\t296\t0.2\t1\t254 42\n",
      "117\t292\t0.2\t1\t264 28\n",
      "118\t246\t0.2\t1\t209 37\n",
      "119\t272\t0.2\t1\t248 24\n",
      "120\t235\t0.2\t1\t203 32\n",
      "121\t211\t0.2\t1\t182 29\n",
      "122\t202\t0.2\t1\t181 21\n",
      "123\t229\t0.2\t1\t197 32\n",
      "124\t188\t0.2\t1\t170 18\n",
      "125\t191\t0.2\t1\t169 22\n",
      "126\t207\t0.2\t1\t179 28\n",
      "127\t139\t0.2\t1\t114 25\n",
      "128\t112\t0.2\t1\t93 19\n",
      "129\t84\t0.2\t1\t64 20\n",
      "130\t68\t0.2\t1\t56 12\n",
      "131\t90\t0.2\t1\t70 20\n",
      "132\t58\t0.2\t1\t41 17\n",
      "133\t96\t0.2\t1\t86 10\n",
      "134\t105\t0.2\t1\t93 12\n",
      "135\t89\t0.2\t1\t82 7\n",
      "136\t43\t0.2\t1\t38 5\n",
      "137\t44\t0.2\t1\t36 8\n",
      "138\t72\t0.2\t1\t69 3\n",
      "139\t58\t0.2\t1\t50 8\n",
      "140\t99\t0.2\t1\t91 8\n",
      "141\t78\t0.2\t1\t68 10\n",
      "142\t67\t0.2\t1\t54 13\n",
      "143\t57\t0.2\t1\t52 5\n",
      "144\t67\t0.2\t1\t63 4\n",
      "145\t251\t0.2\t1\t243 8\n",
      "146\t188\t0.2\t1\t179 9\n",
      "147\t54\t0.2\t1\t50 4\n",
      "148\t282\t0.2\t1\t206 76\n",
      "149\t48\t0.2\t1\t37 11\n",
      "150\t20247\t0.2\t1\t68 20179\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/A.1.fastq.gz\n",
      "=============================================\n",
      "14188117 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/A.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/A.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to A.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/A.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/A.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 670.96 s (47 us/read; 1.27 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,188,117\n",
      "Reads with adapters:                 7,543,028 (53.2%)\n",
      "Reads written (passing filters):    14,188,117 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,128,217,550 bp\n",
      "Quality-trimmed:             630,200,205 bp (29.6%)\n",
      "Total written (filtered):  1,369,806,672 bp (64.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 7543028 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.4%\n",
      "  C: 31.3%\n",
      "  G: 27.2%\n",
      "  T: 13.7%\n",
      "  none/other: 0.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2967731\t3547029.2\t0\t2967731\n",
      "2\t560929\t886757.3\t0\t560929\n",
      "3\t302409\t221689.3\t0\t302409\n",
      "4\t112285\t55422.3\t0\t112285\n",
      "5\t86881\t13855.6\t0\t86881\n",
      "6\t41311\t3463.9\t0\t41311\n",
      "7\t40726\t866.0\t0\t40726\n",
      "8\t54764\t216.5\t0\t54764\n",
      "9\t77704\t54.1\t0\t77484 220\n",
      "10\t79336\t13.5\t1\t69326 10010\n",
      "11\t68221\t3.4\t1\t51710 16511\n",
      "12\t58245\t0.8\t1\t46957 11288\n",
      "13\t65601\t0.2\t1\t54075 11526\n",
      "14\t56142\t0.2\t1\t48609 7533\n",
      "15\t52642\t0.2\t1\t44516 8126\n",
      "16\t73882\t0.2\t1\t60556 13326\n",
      "17\t74111\t0.2\t1\t63614 10497\n",
      "18\t79164\t0.2\t1\t68157 11007\n",
      "19\t65866\t0.2\t1\t55639 10227\n",
      "20\t72102\t0.2\t1\t63853 8249\n",
      "21\t43786\t0.2\t1\t35462 8324\n",
      "22\t39125\t0.2\t1\t35522 3603\n",
      "23\t44613\t0.2\t1\t39966 4647\n",
      "24\t68735\t0.2\t1\t61362 7373\n",
      "25\t50571\t0.2\t1\t44482 6089\n",
      "26\t69475\t0.2\t1\t59810 9665\n",
      "27\t110827\t0.2\t1\t93618 17209\n",
      "28\t54591\t0.2\t1\t48584 6007\n",
      "29\t101840\t0.2\t1\t79810 22030\n",
      "30\t61493\t0.2\t1\t54186 7307\n",
      "31\t77264\t0.2\t1\t67487 9777\n",
      "32\t57474\t0.2\t1\t50929 6545\n",
      "33\t74178\t0.2\t1\t65473 8705\n",
      "34\t78371\t0.2\t1\t67802 10569\n",
      "35\t57806\t0.2\t1\t51765 6041\n",
      "36\t72945\t0.2\t1\t65158 7787\n",
      "37\t57573\t0.2\t1\t51042 6531\n",
      "38\t56618\t0.2\t1\t50207 6411\n",
      "39\t60104\t0.2\t1\t53330 6774\n",
      "40\t61318\t0.2\t1\t54277 7041\n",
      "41\t55858\t0.2\t1\t49998 5860\n",
      "42\t53793\t0.2\t1\t49771 4022\n",
      "43\t55792\t0.2\t1\t51282 4510\n",
      "44\t63877\t0.2\t1\t59646 4231\n",
      "45\t64926\t0.2\t1\t59324 5602\n",
      "46\t55983\t0.2\t1\t50242 5741\n",
      "47\t50453\t0.2\t1\t46897 3556\n",
      "48\t51413\t0.2\t1\t45004 6409\n",
      "49\t49939\t0.2\t1\t44514 5425\n",
      "50\t33642\t0.2\t1\t30553 3089\n",
      "51\t37681\t0.2\t1\t33835 3846\n",
      "52\t30389\t0.2\t1\t27966 2423\n",
      "53\t35016\t0.2\t1\t32937 2079\n",
      "54\t40578\t0.2\t1\t38106 2472\n",
      "55\t33748\t0.2\t1\t31452 2296\n",
      "56\t37577\t0.2\t1\t35428 2149\n",
      "57\t48084\t0.2\t1\t45295 2789\n",
      "58\t34919\t0.2\t1\t32206 2713\n",
      "59\t26133\t0.2\t1\t24515 1618\n",
      "60\t30568\t0.2\t1\t28680 1888\n",
      "61\t22545\t0.2\t1\t20889 1656\n",
      "62\t24586\t0.2\t1\t22706 1880\n",
      "63\t29573\t0.2\t1\t27674 1899\n",
      "64\t21457\t0.2\t1\t19973 1484\n",
      "65\t18073\t0.2\t1\t16631 1442\n",
      "66\t18598\t0.2\t1\t17128 1470\n",
      "67\t13848\t0.2\t1\t12969 879\n",
      "68\t18157\t0.2\t1\t16403 1754\n",
      "69\t20442\t0.2\t1\t19250 1192\n",
      "70\t21143\t0.2\t1\t19539 1604\n",
      "71\t10186\t0.2\t1\t9484 702\n",
      "72\t10410\t0.2\t1\t9603 807\n",
      "73\t11507\t0.2\t1\t10562 945\n",
      "74\t12561\t0.2\t1\t11597 964\n",
      "75\t14657\t0.2\t1\t13525 1132\n",
      "76\t19556\t0.2\t1\t18230 1326\n",
      "77\t24352\t0.2\t1\t22580 1772\n",
      "78\t20448\t0.2\t1\t18225 2223\n",
      "79\t10121\t0.2\t1\t8597 1524\n",
      "80\t4175\t0.2\t1\t3326 849\n",
      "81\t747\t0.2\t1\t582 165\n",
      "82\t462\t0.2\t1\t416 46\n",
      "83\t697\t0.2\t1\t654 43\n",
      "84\t1130\t0.2\t1\t1069 61\n",
      "85\t1411\t0.2\t1\t1333 78\n",
      "86\t1666\t0.2\t1\t1595 71\n",
      "87\t1603\t0.2\t1\t1522 81\n",
      "88\t1311\t0.2\t1\t1235 76\n",
      "89\t1013\t0.2\t1\t955 58\n",
      "90\t839\t0.2\t1\t786 53\n",
      "91\t729\t0.2\t1\t689 40\n",
      "92\t635\t0.2\t1\t594 41\n",
      "93\t526\t0.2\t1\t485 41\n",
      "94\t434\t0.2\t1\t404 30\n",
      "95\t362\t0.2\t1\t332 30\n",
      "96\t341\t0.2\t1\t318 23\n",
      "97\t332\t0.2\t1\t308 24\n",
      "98\t322\t0.2\t1\t294 28\n",
      "99\t248\t0.2\t1\t231 17\n",
      "100\t242\t0.2\t1\t224 18\n",
      "101\t213\t0.2\t1\t191 22\n",
      "102\t172\t0.2\t1\t158 14\n",
      "103\t131\t0.2\t1\t113 18\n",
      "104\t125\t0.2\t1\t111 14\n",
      "105\t108\t0.2\t1\t99 9\n",
      "106\t97\t0.2\t1\t85 12\n",
      "107\t95\t0.2\t1\t88 7\n",
      "108\t105\t0.2\t1\t91 14\n",
      "109\t82\t0.2\t1\t72 10\n",
      "110\t81\t0.2\t1\t72 9\n",
      "111\t87\t0.2\t1\t79 8\n",
      "112\t68\t0.2\t1\t61 7\n",
      "113\t72\t0.2\t1\t64 8\n",
      "114\t57\t0.2\t1\t46 11\n",
      "115\t74\t0.2\t1\t65 9\n",
      "116\t45\t0.2\t1\t39 6\n",
      "117\t61\t0.2\t1\t49 12\n",
      "118\t43\t0.2\t1\t35 8\n",
      "119\t60\t0.2\t1\t52 8\n",
      "120\t38\t0.2\t1\t30 8\n",
      "121\t42\t0.2\t1\t33 9\n",
      "122\t42\t0.2\t1\t35 7\n",
      "123\t53\t0.2\t1\t43 10\n",
      "124\t37\t0.2\t1\t28 9\n",
      "125\t44\t0.2\t1\t34 10\n",
      "126\t37\t0.2\t1\t30 7\n",
      "127\t37\t0.2\t1\t29 8\n",
      "128\t24\t0.2\t1\t18 6\n",
      "129\t12\t0.2\t1\t10 2\n",
      "130\t16\t0.2\t1\t10 6\n",
      "131\t14\t0.2\t1\t12 2\n",
      "132\t9\t0.2\t1\t8 1\n",
      "133\t11\t0.2\t1\t8 3\n",
      "134\t20\t0.2\t1\t17 3\n",
      "135\t14\t0.2\t1\t7 7\n",
      "136\t11\t0.2\t1\t5 6\n",
      "137\t10\t0.2\t1\t9 1\n",
      "138\t20\t0.2\t1\t18 2\n",
      "139\t8\t0.2\t1\t7 1\n",
      "140\t12\t0.2\t1\t11 1\n",
      "141\t16\t0.2\t1\t11 5\n",
      "142\t12\t0.2\t1\t8 4\n",
      "143\t6\t0.2\t1\t5 1\n",
      "144\t9\t0.2\t1\t6 3\n",
      "145\t29\t0.2\t1\t28 1\n",
      "146\t36\t0.2\t1\t32 4\n",
      "147\t10\t0.2\t1\t7 3\n",
      "148\t31\t0.2\t1\t22 9\n",
      "149\t8\t0.2\t1\t7 1\n",
      "150\t11\t0.2\t1\t5 6\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/A.2.fastq.gz\n",
      "=============================================\n",
      "14188117 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files A.1_trimmed.fq.gz and A.2_trimmed.fq.gz\n",
      "file_1: A.1_trimmed.fq.gz, file_2: A.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: A.1_trimmed.fq.gz and A.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to A.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to A.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to A.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to A.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 14188117\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 2134426 (15.04%)\n",
      "\n",
      "Deleting both intermediate output files A.1_trimmed.fq.gz and A.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/A.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/A.1.fastq.gz ./data_raw/A.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/B.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t368340\tAGATCGGAAGAGC\t1000000\t36.83\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 368340). Second best hit was Nextera (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/B.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/B.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/B.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to B.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/B.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/B.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 896.59 s (55 us/read; 1.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,254,289\n",
      "Reads with adapters:                10,486,233 (64.5%)\n",
      "Reads written (passing filters):    16,254,289 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,438,143,350 bp\n",
      "Quality-trimmed:              74,710,334 bp (3.1%)\n",
      "Total written (filtered):  2,073,925,013 bp (85.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10486233 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.9%\n",
      "  C: 35.4%\n",
      "  G: 27.9%\n",
      "  T: 18.0%\n",
      "  none/other: 1.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1978565\t4063572.2\t0\t1978565\n",
      "2\t653063\t1015893.1\t0\t653063\n",
      "3\t267703\t253973.3\t0\t267703\n",
      "4\t138365\t63493.3\t0\t138365\n",
      "5\t97760\t15873.3\t0\t97760\n",
      "6\t82532\t3968.3\t0\t82532\n",
      "7\t81449\t992.1\t0\t81449\n",
      "8\t99119\t248.0\t0\t99119\n",
      "9\t128210\t62.0\t0\t128009 201\n",
      "10\t145240\t15.5\t1\t132035 13205\n",
      "11\t99492\t3.9\t1\t88900 10592\n",
      "12\t102884\t1.0\t1\t92384 10500\n",
      "13\t128511\t0.2\t1\t116300 12211\n",
      "14\t92023\t0.2\t1\t82507 9516\n",
      "15\t132575\t0.2\t1\t120288 12287\n",
      "16\t67024\t0.2\t1\t59386 7638\n",
      "17\t109993\t0.2\t1\t99427 10566\n",
      "18\t98917\t0.2\t1\t90082 8835\n",
      "19\t116960\t0.2\t1\t106554 10406\n",
      "20\t91860\t0.2\t1\t83450 8410\n",
      "21\t122549\t0.2\t1\t111181 11368\n",
      "22\t128616\t0.2\t1\t116419 12197\n",
      "23\t121476\t0.2\t1\t110238 11238\n",
      "24\t119226\t0.2\t1\t107324 11902\n",
      "25\t117644\t0.2\t1\t106988 10656\n",
      "26\t95636\t0.2\t1\t86519 9117\n",
      "27\t124068\t0.2\t1\t112463 11605\n",
      "28\t114532\t0.2\t1\t104300 10232\n",
      "29\t122589\t0.2\t1\t111179 11410\n",
      "30\t110206\t0.2\t1\t100148 10058\n",
      "31\t136133\t0.2\t1\t123373 12760\n",
      "32\t139608\t0.2\t1\t126791 12817\n",
      "33\t107416\t0.2\t1\t96731 10685\n",
      "34\t148463\t0.2\t1\t135380 13083\n",
      "35\t82025\t0.2\t1\t74646 7379\n",
      "36\t123245\t0.2\t1\t111685 11560\n",
      "37\t136106\t0.2\t1\t122314 13792\n",
      "38\t135123\t0.2\t1\t121705 13418\n",
      "39\t179837\t0.2\t1\t162002 17835\n",
      "40\t130655\t0.2\t1\t118040 12615\n",
      "41\t182366\t0.2\t1\t164673 17693\n",
      "42\t332861\t0.2\t1\t301949 30912\n",
      "43\t329979\t0.2\t1\t298674 31305\n",
      "44\t236578\t0.2\t1\t213978 22600\n",
      "45\t116703\t0.2\t1\t106548 10155\n",
      "46\t56607\t0.2\t1\t51269 5338\n",
      "47\t47175\t0.2\t1\t41445 5730\n",
      "48\t74799\t0.2\t1\t68155 6644\n",
      "49\t129608\t0.2\t1\t118781 10827\n",
      "50\t32997\t0.2\t1\t29137 3860\n",
      "51\t141479\t0.2\t1\t128587 12892\n",
      "52\t106109\t0.2\t1\t96943 9166\n",
      "53\t62219\t0.2\t1\t56443 5776\n",
      "54\t72579\t0.2\t1\t66107 6472\n",
      "55\t121892\t0.2\t1\t111304 10588\n",
      "56\t76202\t0.2\t1\t69559 6643\n",
      "57\t18284\t0.2\t1\t16542 1742\n",
      "58\t57833\t0.2\t1\t52716 5117\n",
      "59\t122281\t0.2\t1\t111698 10583\n",
      "60\t18446\t0.2\t1\t16623 1823\n",
      "61\t51289\t0.2\t1\t46897 4392\n",
      "62\t73851\t0.2\t1\t67365 6486\n",
      "63\t15926\t0.2\t1\t14492 1434\n",
      "64\t41196\t0.2\t1\t37443 3753\n",
      "65\t66247\t0.2\t1\t60354 5893\n",
      "66\t26486\t0.2\t1\t23976 2510\n",
      "67\t31828\t0.2\t1\t29025 2803\n",
      "68\t37500\t0.2\t1\t34174 3326\n",
      "69\t34537\t0.2\t1\t31305 3232\n",
      "70\t33862\t0.2\t1\t30643 3219\n",
      "71\t35547\t0.2\t1\t30902 4645\n",
      "72\t49389\t0.2\t1\t40726 8663\n",
      "73\t63810\t0.2\t1\t51448 12362\n",
      "74\t77993\t0.2\t1\t58719 19274\n",
      "75\t54915\t0.2\t1\t46864 8051\n",
      "76\t4791\t0.2\t1\t4160 631\n",
      "77\t250\t0.2\t1\t212 38\n",
      "78\t1491\t0.2\t1\t1351 140\n",
      "79\t4436\t0.2\t1\t4028 408\n",
      "80\t6952\t0.2\t1\t6320 632\n",
      "81\t8389\t0.2\t1\t7591 798\n",
      "82\t8972\t0.2\t1\t8147 825\n",
      "83\t8954\t0.2\t1\t8124 830\n",
      "84\t8937\t0.2\t1\t8077 860\n",
      "85\t8678\t0.2\t1\t7936 742\n",
      "86\t8268\t0.2\t1\t7525 743\n",
      "87\t7741\t0.2\t1\t7046 695\n",
      "88\t7248\t0.2\t1\t6578 670\n",
      "89\t6629\t0.2\t1\t6006 623\n",
      "90\t5923\t0.2\t1\t5392 531\n",
      "91\t5324\t0.2\t1\t4818 506\n",
      "92\t4779\t0.2\t1\t4299 480\n",
      "93\t4013\t0.2\t1\t3595 418\n",
      "94\t3521\t0.2\t1\t3198 323\n",
      "95\t3301\t0.2\t1\t2968 333\n",
      "96\t3118\t0.2\t1\t2794 324\n",
      "97\t2881\t0.2\t1\t2598 283\n",
      "98\t2860\t0.2\t1\t2607 253\n",
      "99\t2533\t0.2\t1\t2313 220\n",
      "100\t2241\t0.2\t1\t2028 213\n",
      "101\t1838\t0.2\t1\t1676 162\n",
      "102\t1685\t0.2\t1\t1507 178\n",
      "103\t1295\t0.2\t1\t1175 120\n",
      "104\t1167\t0.2\t1\t1041 126\n",
      "105\t984\t0.2\t1\t875 109\n",
      "106\t854\t0.2\t1\t760 94\n",
      "107\t803\t0.2\t1\t699 104\n",
      "108\t755\t0.2\t1\t648 107\n",
      "109\t739\t0.2\t1\t675 64\n",
      "110\t570\t0.2\t1\t505 65\n",
      "111\t491\t0.2\t1\t446 45\n",
      "112\t485\t0.2\t1\t427 58\n",
      "113\t379\t0.2\t1\t322 57\n",
      "114\t317\t0.2\t1\t258 59\n",
      "115\t287\t0.2\t1\t247 40\n",
      "116\t257\t0.2\t1\t212 45\n",
      "117\t259\t0.2\t1\t218 41\n",
      "118\t197\t0.2\t1\t165 32\n",
      "119\t196\t0.2\t1\t152 44\n",
      "120\t153\t0.2\t1\t119 34\n",
      "121\t183\t0.2\t1\t140 43\n",
      "122\t141\t0.2\t1\t106 35\n",
      "123\t122\t0.2\t1\t90 32\n",
      "124\t122\t0.2\t1\t89 33\n",
      "125\t134\t0.2\t1\t103 31\n",
      "126\t109\t0.2\t1\t74 35\n",
      "127\t94\t0.2\t1\t74 20\n",
      "128\t89\t0.2\t1\t59 30\n",
      "129\t79\t0.2\t1\t55 24\n",
      "130\t41\t0.2\t1\t26 15\n",
      "131\t58\t0.2\t1\t29 29\n",
      "132\t42\t0.2\t1\t31 11\n",
      "133\t43\t0.2\t1\t34 9\n",
      "134\t47\t0.2\t1\t34 13\n",
      "135\t50\t0.2\t1\t45 5\n",
      "136\t27\t0.2\t1\t20 7\n",
      "137\t35\t0.2\t1\t25 10\n",
      "138\t38\t0.2\t1\t30 8\n",
      "139\t27\t0.2\t1\t22 5\n",
      "140\t51\t0.2\t1\t44 7\n",
      "141\t56\t0.2\t1\t48 8\n",
      "142\t31\t0.2\t1\t20 11\n",
      "143\t37\t0.2\t1\t29 8\n",
      "144\t15\t0.2\t1\t12 3\n",
      "145\t76\t0.2\t1\t71 5\n",
      "146\t88\t0.2\t1\t82 6\n",
      "147\t69\t0.2\t1\t67 2\n",
      "148\t117\t0.2\t1\t110 7\n",
      "149\t40\t0.2\t1\t23 17\n",
      "150\t99460\t0.2\t1\t36 99424\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/B.1.fastq.gz\n",
      "=============================================\n",
      "16254289 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/B.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/B.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to B.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/B.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/B.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 790.68 s (49 us/read; 1.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,254,289\n",
      "Reads with adapters:                 9,811,233 (60.4%)\n",
      "Reads written (passing filters):    16,254,289 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,438,143,350 bp\n",
      "Quality-trimmed:             271,457,843 bp (11.1%)\n",
      "Total written (filtered):  1,920,410,966 bp (78.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9811233 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.6%\n",
      "  C: 37.0%\n",
      "  G: 33.0%\n",
      "  T: 13.1%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2125248\t4063572.2\t0\t2125248\n",
      "2\t585165\t1015893.1\t0\t585165\n",
      "3\t296179\t253973.3\t0\t296179\n",
      "4\t124209\t63493.3\t0\t124209\n",
      "5\t128464\t15873.3\t0\t128464\n",
      "6\t74744\t3968.3\t0\t74744\n",
      "7\t77165\t992.1\t0\t77165\n",
      "8\t101458\t248.0\t0\t101458\n",
      "9\t135993\t62.0\t0\t135779 214\n",
      "10\t132267\t15.5\t1\t125203 7064\n",
      "11\t80517\t3.9\t1\t72355 8162\n",
      "12\t73620\t1.0\t1\t67302 6318\n",
      "13\t103138\t0.2\t1\t96406 6732\n",
      "14\t97735\t0.2\t1\t91812 5923\n",
      "15\t86634\t0.2\t1\t81813 4821\n",
      "16\t114599\t0.2\t1\t106686 7913\n",
      "17\t124639\t0.2\t1\t117205 7434\n",
      "18\t143430\t0.2\t1\t135578 7852\n",
      "19\t112006\t0.2\t1\t105497 6509\n",
      "20\t143033\t0.2\t1\t136015 7018\n",
      "21\t65309\t0.2\t1\t60467 4842\n",
      "22\t73831\t0.2\t1\t70666 3165\n",
      "23\t86484\t0.2\t1\t82459 4025\n",
      "24\t119964\t0.2\t1\t114072 5892\n",
      "25\t104586\t0.2\t1\t99251 5335\n",
      "26\t120649\t0.2\t1\t113871 6778\n",
      "27\t185501\t0.2\t1\t174541 10960\n",
      "28\t79516\t0.2\t1\t75466 4050\n",
      "29\t146716\t0.2\t1\t135302 11414\n",
      "30\t106613\t0.2\t1\t101619 4994\n",
      "31\t148063\t0.2\t1\t141241 6822\n",
      "32\t73100\t0.2\t1\t69129 3971\n",
      "33\t166765\t0.2\t1\t159847 6918\n",
      "34\t84398\t0.2\t1\t78762 5636\n",
      "35\t83672\t0.2\t1\t80370 3302\n",
      "36\t163941\t0.2\t1\t158300 5641\n",
      "37\t100508\t0.2\t1\t94290 6218\n",
      "38\t158796\t0.2\t1\t152662 6134\n",
      "39\t88271\t0.2\t1\t82052 6219\n",
      "40\t73907\t0.2\t1\t70958 2949\n",
      "41\t122354\t0.2\t1\t117730 4624\n",
      "42\t141137\t0.2\t1\t136302 4835\n",
      "43\t121707\t0.2\t1\t117219 4488\n",
      "44\t86562\t0.2\t1\t82874 3688\n",
      "45\t183794\t0.2\t1\t177424 6370\n",
      "46\t97898\t0.2\t1\t93101 4797\n",
      "47\t126860\t0.2\t1\t122966 3894\n",
      "48\t70972\t0.2\t1\t66317 4655\n",
      "49\t140834\t0.2\t1\t135662 5172\n",
      "50\t41339\t0.2\t1\t39126 2213\n",
      "51\t96659\t0.2\t1\t92977 3682\n",
      "52\t44814\t0.2\t1\t42615 2199\n",
      "53\t76073\t0.2\t1\t73689 2384\n",
      "54\t108226\t0.2\t1\t105342 2884\n",
      "55\t71444\t0.2\t1\t68828 2616\n",
      "56\t73813\t0.2\t1\t71436 2377\n",
      "57\t120158\t0.2\t1\t116995 3163\n",
      "58\t91990\t0.2\t1\t88821 3169\n",
      "59\t38135\t0.2\t1\t36433 1702\n",
      "60\t89406\t0.2\t1\t87328 2078\n",
      "61\t40830\t0.2\t1\t39156 1674\n",
      "62\t44866\t0.2\t1\t42919 1947\n",
      "63\t75234\t0.2\t1\t73243 1991\n",
      "64\t57633\t0.2\t1\t55651 1982\n",
      "65\t25184\t0.2\t1\t23856 1328\n",
      "66\t56523\t0.2\t1\t54802 1721\n",
      "67\t13559\t0.2\t1\t12959 600\n",
      "68\t39463\t0.2\t1\t37801 1662\n",
      "69\t35631\t0.2\t1\t34396 1235\n",
      "70\t70467\t0.2\t1\t68491 1976\n",
      "71\t12163\t0.2\t1\t11625 538\n",
      "72\t22078\t0.2\t1\t21267 811\n",
      "73\t24281\t0.2\t1\t23435 846\n",
      "74\t25168\t0.2\t1\t24321 847\n",
      "75\t27531\t0.2\t1\t26613 918\n",
      "76\t36836\t0.2\t1\t35781 1055\n",
      "77\t62301\t0.2\t1\t60529 1772\n",
      "78\t50449\t0.2\t1\t48331 2118\n",
      "79\t14903\t0.2\t1\t13736 1167\n",
      "80\t3104\t0.2\t1\t2685 419\n",
      "81\t328\t0.2\t1\t286 42\n",
      "82\t427\t0.2\t1\t408 19\n",
      "83\t954\t0.2\t1\t923 31\n",
      "84\t1921\t0.2\t1\t1866 55\n",
      "85\t3032\t0.2\t1\t2940 92\n",
      "86\t4013\t0.2\t1\t3906 107\n",
      "87\t4193\t0.2\t1\t4075 118\n",
      "88\t3591\t0.2\t1\t3491 100\n",
      "89\t2758\t0.2\t1\t2687 71\n",
      "90\t2335\t0.2\t1\t2261 74\n",
      "91\t2008\t0.2\t1\t1948 60\n",
      "92\t1772\t0.2\t1\t1723 49\n",
      "93\t1376\t0.2\t1\t1332 44\n",
      "94\t1143\t0.2\t1\t1120 23\n",
      "95\t1033\t0.2\t1\t991 42\n",
      "96\t952\t0.2\t1\t912 40\n",
      "97\t820\t0.2\t1\t783 37\n",
      "98\t760\t0.2\t1\t735 25\n",
      "99\t659\t0.2\t1\t630 29\n",
      "100\t595\t0.2\t1\t570 25\n",
      "101\t374\t0.2\t1\t361 13\n",
      "102\t346\t0.2\t1\t331 15\n",
      "103\t273\t0.2\t1\t258 15\n",
      "104\t252\t0.2\t1\t236 16\n",
      "105\t232\t0.2\t1\t218 14\n",
      "106\t174\t0.2\t1\t157 17\n",
      "107\t168\t0.2\t1\t161 7\n",
      "108\t131\t0.2\t1\t116 15\n",
      "109\t135\t0.2\t1\t125 10\n",
      "110\t115\t0.2\t1\t102 13\n",
      "111\t88\t0.2\t1\t82 6\n",
      "112\t106\t0.2\t1\t90 16\n",
      "113\t86\t0.2\t1\t74 12\n",
      "114\t66\t0.2\t1\t57 9\n",
      "115\t66\t0.2\t1\t55 11\n",
      "116\t55\t0.2\t1\t38 17\n",
      "117\t60\t0.2\t1\t48 12\n",
      "118\t50\t0.2\t1\t41 9\n",
      "119\t48\t0.2\t1\t34 14\n",
      "120\t56\t0.2\t1\t37 19\n",
      "121\t60\t0.2\t1\t39 21\n",
      "122\t45\t0.2\t1\t36 9\n",
      "123\t31\t0.2\t1\t23 8\n",
      "124\t37\t0.2\t1\t25 12\n",
      "125\t46\t0.2\t1\t26 20\n",
      "126\t40\t0.2\t1\t28 12\n",
      "127\t31\t0.2\t1\t24 7\n",
      "128\t18\t0.2\t1\t12 6\n",
      "129\t17\t0.2\t1\t8 9\n",
      "130\t14\t0.2\t1\t9 5\n",
      "131\t14\t0.2\t1\t7 7\n",
      "132\t4\t0.2\t1\t2 2\n",
      "133\t9\t0.2\t1\t5 4\n",
      "134\t12\t0.2\t1\t9 3\n",
      "135\t10\t0.2\t1\t7 3\n",
      "136\t2\t0.2\t1\t2\n",
      "137\t9\t0.2\t1\t7 2\n",
      "138\t3\t0.2\t1\t2 1\n",
      "139\t6\t0.2\t1\t4 2\n",
      "140\t7\t0.2\t1\t5 2\n",
      "141\t8\t0.2\t1\t7 1\n",
      "142\t3\t0.2\t1\t0 3\n",
      "143\t2\t0.2\t1\t2\n",
      "144\t2\t0.2\t1\t1 1\n",
      "145\t7\t0.2\t1\t5 2\n",
      "146\t9\t0.2\t1\t9\n",
      "147\t6\t0.2\t1\t3 3\n",
      "148\t10\t0.2\t1\t9 1\n",
      "149\t1\t0.2\t1\t1\n",
      "150\t7\t0.2\t1\t5 2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/B.2.fastq.gz\n",
      "=============================================\n",
      "16254289 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files B.1_trimmed.fq.gz and B.2_trimmed.fq.gz\n",
      "file_1: B.1_trimmed.fq.gz, file_2: B.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: B.1_trimmed.fq.gz and B.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to B.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to B.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to B.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to B.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16254289\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1260020 (7.75%)\n",
      "\n",
      "Deleting both intermediate output files B.1_trimmed.fq.gz and B.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/B.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/B.1.fastq.gz ./data_raw/B.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/C.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t409340\tAGATCGGAAGAGC\t1000000\t40.93\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 409340). Second best hit was Nextera (count: 1)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/C.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/C.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/C.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to C.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/C.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/C.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1054.56 s (64 us/read; 0.93 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,358,360\n",
      "Reads with adapters:                10,914,449 (66.7%)\n",
      "Reads written (passing filters):    16,358,360 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,453,754,000 bp\n",
      "Quality-trimmed:              79,606,079 bp (3.2%)\n",
      "Total written (filtered):  2,038,290,431 bp (83.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10914449 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.4%\n",
      "  C: 33.1%\n",
      "  G: 31.3%\n",
      "  T: 17.5%\n",
      "  none/other: 1.8%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1882169\t4089590.0\t0\t1882169\n",
      "2\t618657\t1022397.5\t0\t618657\n",
      "3\t254382\t255599.4\t0\t254382\n",
      "4\t129924\t63899.8\t0\t129924\n",
      "5\t91715\t15975.0\t0\t91715\n",
      "6\t77283\t3993.7\t0\t77283\n",
      "7\t77285\t998.4\t0\t77285\n",
      "8\t93062\t249.6\t0\t93062\n",
      "9\t120536\t62.4\t0\t120348 188\n",
      "10\t133017\t15.6\t1\t122892 10125\n",
      "11\t91940\t3.9\t1\t83320 8620\n",
      "12\t95569\t1.0\t1\t87092 8477\n",
      "13\t119571\t0.2\t1\t110216 9355\n",
      "14\t85299\t0.2\t1\t77592 7707\n",
      "15\t122968\t0.2\t1\t113600 9368\n",
      "16\t63586\t0.2\t1\t57352 6234\n",
      "17\t102868\t0.2\t1\t94366 8502\n",
      "18\t92985\t0.2\t1\t86114 6871\n",
      "19\t109625\t0.2\t1\t101428 8197\n",
      "20\t84072\t0.2\t1\t77544 6528\n",
      "21\t113664\t0.2\t1\t104980 8684\n",
      "22\t118845\t0.2\t1\t109349 9496\n",
      "23\t113995\t0.2\t1\t105028 8967\n",
      "24\t111795\t0.2\t1\t102148 9647\n",
      "25\t110206\t0.2\t1\t101647 8559\n",
      "26\t90789\t0.2\t1\t83476 7313\n",
      "27\t117926\t0.2\t1\t108764 9162\n",
      "28\t108135\t0.2\t1\t100213 7922\n",
      "29\t117068\t0.2\t1\t108105 8963\n",
      "30\t104500\t0.2\t1\t96475 8025\n",
      "31\t130356\t0.2\t1\t120261 10095\n",
      "32\t133647\t0.2\t1\t123563 10084\n",
      "33\t102735\t0.2\t1\t94134 8601\n",
      "34\t142944\t0.2\t1\t132627 10317\n",
      "35\t80615\t0.2\t1\t74559 6056\n",
      "36\t120994\t0.2\t1\t111538 9456\n",
      "37\t136272\t0.2\t1\t124444 11828\n",
      "38\t135868\t0.2\t1\t124580 11288\n",
      "39\t183560\t0.2\t1\t167986 15574\n",
      "40\t134835\t0.2\t1\t123914 10921\n",
      "41\t194547\t0.2\t1\t178787 15760\n",
      "42\t360335\t0.2\t1\t333247 27088\n",
      "43\t381264\t0.2\t1\t351944 29320\n",
      "44\t264392\t0.2\t1\t244153 20239\n",
      "45\t137396\t0.2\t1\t127727 9669\n",
      "46\t61182\t0.2\t1\t56430 4752\n",
      "47\t50634\t0.2\t1\t45333 5301\n",
      "48\t81547\t0.2\t1\t75623 5924\n",
      "49\t142288\t0.2\t1\t132686 9602\n",
      "50\t38069\t0.2\t1\t34139 3930\n",
      "51\t160453\t0.2\t1\t148874 11579\n",
      "52\t120156\t0.2\t1\t111626 8530\n",
      "53\t70270\t0.2\t1\t65058 5212\n",
      "54\t84532\t0.2\t1\t78253 6279\n",
      "55\t145745\t0.2\t1\t135632 10113\n",
      "56\t92590\t0.2\t1\t86349 6241\n",
      "57\t22571\t0.2\t1\t20761 1810\n",
      "58\t71816\t0.2\t1\t66807 5009\n",
      "59\t156895\t0.2\t1\t146560 10335\n",
      "60\t24213\t0.2\t1\t22311 1902\n",
      "61\t68064\t0.2\t1\t63514 4550\n",
      "62\t97873\t0.2\t1\t91136 6737\n",
      "63\t20814\t0.2\t1\t19273 1541\n",
      "64\t55648\t0.2\t1\t51653 3995\n",
      "65\t92250\t0.2\t1\t86079 6171\n",
      "66\t37666\t0.2\t1\t34935 2731\n",
      "67\t45758\t0.2\t1\t42627 3131\n",
      "68\t56122\t0.2\t1\t52314 3808\n",
      "69\t52574\t0.2\t1\t48732 3842\n",
      "70\t53046\t0.2\t1\t49131 3915\n",
      "71\t56687\t0.2\t1\t51343 5344\n",
      "72\t78549\t0.2\t1\t68329 10220\n",
      "73\t102616\t0.2\t1\t88565 14051\n",
      "74\t123519\t0.2\t1\t101384 22135\n",
      "75\t91167\t0.2\t1\t81353 9814\n",
      "76\t7826\t0.2\t1\t7012 814\n",
      "77\t364\t0.2\t1\t312 52\n",
      "78\t2403\t0.2\t1\t2220 183\n",
      "79\t7821\t0.2\t1\t7259 562\n",
      "80\t12712\t0.2\t1\t11802 910\n",
      "81\t15096\t0.2\t1\t14027 1069\n",
      "82\t16343\t0.2\t1\t15225 1118\n",
      "83\t16801\t0.2\t1\t15599 1202\n",
      "84\t16379\t0.2\t1\t15198 1181\n",
      "85\t15633\t0.2\t1\t14529 1104\n",
      "86\t15339\t0.2\t1\t14261 1078\n",
      "87\t14032\t0.2\t1\t12993 1039\n",
      "88\t12997\t0.2\t1\t12118 879\n",
      "89\t12001\t0.2\t1\t11185 816\n",
      "90\t10813\t0.2\t1\t10105 708\n",
      "91\t9797\t0.2\t1\t9110 687\n",
      "92\t8533\t0.2\t1\t7972 561\n",
      "93\t7345\t0.2\t1\t6807 538\n",
      "94\t6459\t0.2\t1\t5987 472\n",
      "95\t5889\t0.2\t1\t5435 454\n",
      "96\t5521\t0.2\t1\t5117 404\n",
      "97\t5261\t0.2\t1\t4897 364\n",
      "98\t4767\t0.2\t1\t4436 331\n",
      "99\t4236\t0.2\t1\t3925 311\n",
      "100\t3650\t0.2\t1\t3385 265\n",
      "101\t3198\t0.2\t1\t2980 218\n",
      "102\t2689\t0.2\t1\t2504 185\n",
      "103\t2308\t0.2\t1\t2140 168\n",
      "104\t1850\t0.2\t1\t1717 133\n",
      "105\t1549\t0.2\t1\t1425 124\n",
      "106\t1393\t0.2\t1\t1271 122\n",
      "107\t1288\t0.2\t1\t1168 120\n",
      "108\t1186\t0.2\t1\t1080 106\n",
      "109\t1056\t0.2\t1\t983 73\n",
      "110\t915\t0.2\t1\t850 65\n",
      "111\t775\t0.2\t1\t713 62\n",
      "112\t552\t0.2\t1\t502 50\n",
      "113\t575\t0.2\t1\t515 60\n",
      "114\t363\t0.2\t1\t332 31\n",
      "115\t338\t0.2\t1\t307 31\n",
      "116\t266\t0.2\t1\t229 37\n",
      "117\t255\t0.2\t1\t227 28\n",
      "118\t203\t0.2\t1\t179 24\n",
      "119\t202\t0.2\t1\t162 40\n",
      "120\t173\t0.2\t1\t148 25\n",
      "121\t173\t0.2\t1\t138 35\n",
      "122\t107\t0.2\t1\t81 26\n",
      "123\t116\t0.2\t1\t85 31\n",
      "124\t95\t0.2\t1\t73 22\n",
      "125\t108\t0.2\t1\t85 23\n",
      "126\t107\t0.2\t1\t78 29\n",
      "127\t96\t0.2\t1\t68 28\n",
      "128\t60\t0.2\t1\t45 15\n",
      "129\t71\t0.2\t1\t52 19\n",
      "130\t49\t0.2\t1\t33 16\n",
      "131\t71\t0.2\t1\t50 21\n",
      "132\t36\t0.2\t1\t32 4\n",
      "133\t62\t0.2\t1\t50 12\n",
      "134\t54\t0.2\t1\t49 5\n",
      "135\t52\t0.2\t1\t45 7\n",
      "136\t30\t0.2\t1\t22 8\n",
      "137\t41\t0.2\t1\t35 6\n",
      "138\t34\t0.2\t1\t31 3\n",
      "139\t22\t0.2\t1\t19 3\n",
      "140\t30\t0.2\t1\t24 6\n",
      "141\t69\t0.2\t1\t65 4\n",
      "142\t33\t0.2\t1\t14 19\n",
      "143\t38\t0.2\t1\t35 3\n",
      "144\t24\t0.2\t1\t22 2\n",
      "145\t100\t0.2\t1\t94 6\n",
      "146\t121\t0.2\t1\t118 3\n",
      "147\t73\t0.2\t1\t68 5\n",
      "148\t220\t0.2\t1\t218 2\n",
      "149\t47\t0.2\t1\t37 10\n",
      "150\t104648\t0.2\t1\t31 104617\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/C.1.fastq.gz\n",
      "=============================================\n",
      "16358360 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/C.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/C.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to C.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/C.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/C.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 743.46 s (45 us/read; 1.32 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,358,360\n",
      "Reads with adapters:                10,349,246 (63.3%)\n",
      "Reads written (passing filters):    16,358,360 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,453,754,000 bp\n",
      "Quality-trimmed:             247,914,555 bp (10.1%)\n",
      "Total written (filtered):  1,914,394,565 bp (78.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10349246 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.2%\n",
      "  C: 33.9%\n",
      "  G: 37.1%\n",
      "  T: 12.6%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2053898\t4089590.0\t0\t2053898\n",
      "2\t555799\t1022397.5\t0\t555799\n",
      "3\t291723\t255599.4\t0\t291723\n",
      "4\t120342\t63899.8\t0\t120342\n",
      "5\t122318\t15975.0\t0\t122318\n",
      "6\t70782\t3993.7\t0\t70782\n",
      "7\t73319\t998.4\t0\t73319\n",
      "8\t94631\t249.6\t0\t94631\n",
      "9\t128197\t62.4\t0\t128004 193\n",
      "10\t122065\t15.6\t1\t115850 6215\n",
      "11\t78022\t3.9\t1\t70115 7907\n",
      "12\t69366\t1.0\t1\t63325 6041\n",
      "13\t98497\t0.2\t1\t92209 6288\n",
      "14\t91455\t0.2\t1\t86245 5210\n",
      "15\t82251\t0.2\t1\t77780 4471\n",
      "16\t110735\t0.2\t1\t103258 7477\n",
      "17\t118978\t0.2\t1\t111989 6989\n",
      "18\t138512\t0.2\t1\t131366 7146\n",
      "19\t104683\t0.2\t1\t98686 5997\n",
      "20\t137126\t0.2\t1\t130994 6132\n",
      "21\t63956\t0.2\t1\t59206 4750\n",
      "22\t69026\t0.2\t1\t66276 2750\n",
      "23\t82250\t0.2\t1\t78750 3500\n",
      "24\t113979\t0.2\t1\t108587 5392\n",
      "25\t101565\t0.2\t1\t96665 4900\n",
      "26\t117634\t0.2\t1\t111163 6471\n",
      "27\t188832\t0.2\t1\t177834 10998\n",
      "28\t77842\t0.2\t1\t73945 3897\n",
      "29\t151579\t0.2\t1\t139698 11881\n",
      "30\t101941\t0.2\t1\t97188 4753\n",
      "31\t150601\t0.2\t1\t144146 6455\n",
      "32\t68731\t0.2\t1\t64858 3873\n",
      "33\t160324\t0.2\t1\t154216 6108\n",
      "34\t123214\t0.2\t1\t114403 8811\n",
      "35\t121642\t0.2\t1\t117012 4630\n",
      "36\t279871\t0.2\t1\t270922 8949\n",
      "37\t109383\t0.2\t1\t104262 5121\n",
      "38\t123919\t0.2\t1\t119357 4562\n",
      "39\t80453\t0.2\t1\t75360 5093\n",
      "40\t153656\t0.2\t1\t148604 5052\n",
      "41\t82690\t0.2\t1\t77727 4963\n",
      "42\t50545\t0.2\t1\t48357 2188\n",
      "43\t139136\t0.2\t1\t135022 4114\n",
      "44\t60423\t0.2\t1\t58022 2401\n",
      "45\t199029\t0.2\t1\t193568 5461\n",
      "46\t116089\t0.2\t1\t111183 4906\n",
      "47\t105155\t0.2\t1\t102262 2893\n",
      "48\t91988\t0.2\t1\t87047 4941\n",
      "49\t161778\t0.2\t1\t156720 5058\n",
      "50\t44025\t0.2\t1\t41870 2155\n",
      "51\t115006\t0.2\t1\t111388 3618\n",
      "52\t45787\t0.2\t1\t43860 1927\n",
      "53\t77609\t0.2\t1\t75642 1967\n",
      "54\t119593\t0.2\t1\t116992 2601\n",
      "55\t90262\t0.2\t1\t87602 2660\n",
      "56\t82836\t0.2\t1\t80570 2266\n",
      "57\t141344\t0.2\t1\t138262 3082\n",
      "58\t126011\t0.2\t1\t122566 3445\n",
      "59\t45785\t0.2\t1\t44178 1607\n",
      "60\t113181\t0.2\t1\t110959 2222\n",
      "61\t53860\t0.2\t1\t52091 1769\n",
      "62\t69124\t0.2\t1\t66825 2299\n",
      "63\t103455\t0.2\t1\t101219 2236\n",
      "64\t77132\t0.2\t1\t75014 2118\n",
      "65\t35264\t0.2\t1\t33713 1551\n",
      "66\t83780\t0.2\t1\t81714 2066\n",
      "67\t17535\t0.2\t1\t16749 786\n",
      "68\t64964\t0.2\t1\t62748 2216\n",
      "69\t53264\t0.2\t1\t51695 1569\n",
      "70\t121489\t0.2\t1\t118764 2725\n",
      "71\t18551\t0.2\t1\t17893 658\n",
      "72\t34498\t0.2\t1\t33430 1068\n",
      "73\t39629\t0.2\t1\t38440 1189\n",
      "74\t41318\t0.2\t1\t40134 1184\n",
      "75\t44960\t0.2\t1\t43645 1315\n",
      "76\t60023\t0.2\t1\t58503 1520\n",
      "77\t108763\t0.2\t1\t106364 2399\n",
      "78\t101312\t0.2\t1\t98058 3254\n",
      "79\t33072\t0.2\t1\t31158 1914\n",
      "80\t7317\t0.2\t1\t6526 791\n",
      "81\t641\t0.2\t1\t553 88\n",
      "82\t790\t0.2\t1\t762 28\n",
      "83\t1835\t0.2\t1\t1778 57\n",
      "84\t3435\t0.2\t1\t3375 60\n",
      "85\t5421\t0.2\t1\t5325 96\n",
      "86\t7231\t0.2\t1\t7098 133\n",
      "87\t7788\t0.2\t1\t7652 136\n",
      "88\t6909\t0.2\t1\t6736 173\n",
      "89\t5114\t0.2\t1\t4982 132\n",
      "90\t4329\t0.2\t1\t4256 73\n",
      "91\t3797\t0.2\t1\t3704 93\n",
      "92\t3171\t0.2\t1\t3097 74\n",
      "93\t2545\t0.2\t1\t2468 77\n",
      "94\t2168\t0.2\t1\t2122 46\n",
      "95\t1863\t0.2\t1\t1800 63\n",
      "96\t1638\t0.2\t1\t1584 54\n",
      "97\t1522\t0.2\t1\t1483 39\n",
      "98\t1321\t0.2\t1\t1285 36\n",
      "99\t1202\t0.2\t1\t1169 33\n",
      "100\t923\t0.2\t1\t891 32\n",
      "101\t780\t0.2\t1\t758 22\n",
      "102\t582\t0.2\t1\t561 21\n",
      "103\t513\t0.2\t1\t498 15\n",
      "104\t382\t0.2\t1\t370 12\n",
      "105\t304\t0.2\t1\t293 11\n",
      "106\t263\t0.2\t1\t253 10\n",
      "107\t247\t0.2\t1\t233 14\n",
      "108\t229\t0.2\t1\t218 11\n",
      "109\t195\t0.2\t1\t185 10\n",
      "110\t173\t0.2\t1\t164 9\n",
      "111\t142\t0.2\t1\t130 12\n",
      "112\t125\t0.2\t1\t115 10\n",
      "113\t103\t0.2\t1\t97 6\n",
      "114\t80\t0.2\t1\t69 11\n",
      "115\t76\t0.2\t1\t61 15\n",
      "116\t60\t0.2\t1\t47 13\n",
      "117\t62\t0.2\t1\t51 11\n",
      "118\t50\t0.2\t1\t39 11\n",
      "119\t49\t0.2\t1\t39 10\n",
      "120\t48\t0.2\t1\t44 4\n",
      "121\t44\t0.2\t1\t34 10\n",
      "122\t38\t0.2\t1\t29 9\n",
      "123\t45\t0.2\t1\t26 19\n",
      "124\t35\t0.2\t1\t25 10\n",
      "125\t42\t0.2\t1\t31 11\n",
      "126\t27\t0.2\t1\t16 11\n",
      "127\t16\t0.2\t1\t11 5\n",
      "128\t18\t0.2\t1\t13 5\n",
      "129\t8\t0.2\t1\t4 4\n",
      "130\t12\t0.2\t1\t6 6\n",
      "131\t20\t0.2\t1\t9 11\n",
      "132\t10\t0.2\t1\t7 3\n",
      "133\t14\t0.2\t1\t10 4\n",
      "134\t10\t0.2\t1\t7 3\n",
      "135\t14\t0.2\t1\t12 2\n",
      "136\t6\t0.2\t1\t2 4\n",
      "137\t4\t0.2\t1\t4\n",
      "138\t1\t0.2\t1\t0 1\n",
      "139\t9\t0.2\t1\t5 4\n",
      "140\t7\t0.2\t1\t6 1\n",
      "141\t8\t0.2\t1\t7 1\n",
      "142\t3\t0.2\t1\t0 3\n",
      "143\t2\t0.2\t1\t2\n",
      "144\t3\t0.2\t1\t3\n",
      "145\t7\t0.2\t1\t5 2\n",
      "146\t16\t0.2\t1\t16\n",
      "147\t12\t0.2\t1\t10 2\n",
      "148\t19\t0.2\t1\t19\n",
      "149\t8\t0.2\t1\t7 1\n",
      "150\t3\t0.2\t1\t3\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/C.2.fastq.gz\n",
      "=============================================\n",
      "16358360 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files C.1_trimmed.fq.gz and C.2_trimmed.fq.gz\n",
      "file_1: C.1_trimmed.fq.gz, file_2: C.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: C.1_trimmed.fq.gz and C.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to C.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to C.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to C.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to C.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16358360\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1071921 (6.55%)\n",
      "\n",
      "Deleting both intermediate output files C.1_trimmed.fq.gz and C.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/C.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/C.1.fastq.gz ./data_raw/C.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/D.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t372267\tAGATCGGAAGAGC\t1000000\t37.23\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 372267). Second best hit was Nextera (count: 1)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/D.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/D.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/D.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to D.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/D.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/D.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 696.60 s (45 us/read; 1.35 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,648,595\n",
      "Reads with adapters:                 9,992,904 (63.9%)\n",
      "Reads written (passing filters):    15,648,595 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,347,289,250 bp\n",
      "Quality-trimmed:              72,274,093 bp (3.1%)\n",
      "Total written (filtered):  2,004,795,928 bp (85.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9992904 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.4%\n",
      "  C: 33.6%\n",
      "  G: 30.5%\n",
      "  T: 17.1%\n",
      "  none/other: 1.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1917808\t3912148.8\t0\t1917808\n",
      "2\t654052\t978037.2\t0\t654052\n",
      "3\t269711\t244509.3\t0\t269711\n",
      "4\t133288\t61127.3\t0\t133288\n",
      "5\t94313\t15281.8\t0\t94313\n",
      "6\t79097\t3820.5\t0\t79097\n",
      "7\t79689\t955.1\t0\t79689\n",
      "8\t95827\t238.8\t0\t95827\n",
      "9\t126212\t59.7\t0\t126028 184\n",
      "10\t135593\t14.9\t1\t126856 8737\n",
      "11\t94961\t3.7\t1\t87140 7821\n",
      "12\t97128\t0.9\t1\t89412 7716\n",
      "13\t121672\t0.2\t1\t113405 8267\n",
      "14\t87618\t0.2\t1\t80887 6731\n",
      "15\t124232\t0.2\t1\t116223 8009\n",
      "16\t63101\t0.2\t1\t57396 5705\n",
      "17\t104914\t0.2\t1\t97670 7244\n",
      "18\t93021\t0.2\t1\t87155 5866\n",
      "19\t110161\t0.2\t1\t103188 6973\n",
      "20\t86621\t0.2\t1\t80776 5845\n",
      "21\t115613\t0.2\t1\t107988 7625\n",
      "22\t119995\t0.2\t1\t111888 8107\n",
      "23\t114083\t0.2\t1\t106606 7477\n",
      "24\t112129\t0.2\t1\t103836 8293\n",
      "25\t110972\t0.2\t1\t103748 7224\n",
      "26\t90937\t0.2\t1\t84645 6292\n",
      "27\t117853\t0.2\t1\t110336 7517\n",
      "28\t107524\t0.2\t1\t100843 6681\n",
      "29\t116948\t0.2\t1\t109085 7863\n",
      "30\t104389\t0.2\t1\t97752 6637\n",
      "31\t129284\t0.2\t1\t120683 8601\n",
      "32\t132274\t0.2\t1\t123842 8432\n",
      "33\t101043\t0.2\t1\t93860 7183\n",
      "34\t141629\t0.2\t1\t133033 8596\n",
      "35\t77511\t0.2\t1\t72561 4950\n",
      "36\t116163\t0.2\t1\t108640 7523\n",
      "37\t129018\t0.2\t1\t119626 9392\n",
      "38\t127685\t0.2\t1\t118595 9090\n",
      "39\t169673\t0.2\t1\t157559 12114\n",
      "40\t124356\t0.2\t1\t115765 8591\n",
      "41\t172583\t0.2\t1\t160860 11723\n",
      "42\t312920\t0.2\t1\t293110 19810\n",
      "43\t315272\t0.2\t1\t294615 20657\n",
      "44\t227713\t0.2\t1\t212996 14717\n",
      "45\t110887\t0.2\t1\t104338 6549\n",
      "46\t53724\t0.2\t1\t50241 3483\n",
      "47\t44916\t0.2\t1\t40974 3942\n",
      "48\t69769\t0.2\t1\t65578 4191\n",
      "49\t121594\t0.2\t1\t114761 6833\n",
      "50\t31592\t0.2\t1\t28774 2818\n",
      "51\t131836\t0.2\t1\t123855 7981\n",
      "52\t97976\t0.2\t1\t92039 5937\n",
      "53\t58568\t0.2\t1\t54873 3695\n",
      "54\t68197\t0.2\t1\t63945 4252\n",
      "55\t111773\t0.2\t1\t105210 6563\n",
      "56\t70556\t0.2\t1\t66559 3997\n",
      "57\t17179\t0.2\t1\t15988 1191\n",
      "58\t54213\t0.2\t1\t51009 3204\n",
      "59\t114202\t0.2\t1\t107949 6253\n",
      "60\t17600\t0.2\t1\t16406 1194\n",
      "61\t48332\t0.2\t1\t45581 2751\n",
      "62\t67407\t0.2\t1\t63522 3885\n",
      "63\t14715\t0.2\t1\t13749 966\n",
      "64\t38350\t0.2\t1\t36059 2291\n",
      "65\t61671\t0.2\t1\t58017 3654\n",
      "66\t24256\t0.2\t1\t22736 1520\n",
      "67\t29347\t0.2\t1\t27645 1702\n",
      "68\t35545\t0.2\t1\t33519 2026\n",
      "69\t31974\t0.2\t1\t29930 2044\n",
      "70\t31601\t0.2\t1\t29565 2036\n",
      "71\t33406\t0.2\t1\t30305 3101\n",
      "72\t45005\t0.2\t1\t38891 6114\n",
      "73\t58183\t0.2\t1\t49561 8622\n",
      "74\t72105\t0.2\t1\t57935 14170\n",
      "75\t51744\t0.2\t1\t46181 5563\n",
      "76\t4519\t0.2\t1\t4087 432\n",
      "77\t240\t0.2\t1\t213 27\n",
      "78\t1301\t0.2\t1\t1207 94\n",
      "79\t4069\t0.2\t1\t3809 260\n",
      "80\t6570\t0.2\t1\t6194 376\n",
      "81\t7801\t0.2\t1\t7322 479\n",
      "82\t8413\t0.2\t1\t7910 503\n",
      "83\t8755\t0.2\t1\t8260 495\n",
      "84\t8403\t0.2\t1\t7878 525\n",
      "85\t8088\t0.2\t1\t7620 468\n",
      "86\t7828\t0.2\t1\t7351 477\n",
      "87\t7491\t0.2\t1\t7024 467\n",
      "88\t6937\t0.2\t1\t6524 413\n",
      "89\t6419\t0.2\t1\t6060 359\n",
      "90\t5776\t0.2\t1\t5404 372\n",
      "91\t5253\t0.2\t1\t4959 294\n",
      "92\t4644\t0.2\t1\t4359 285\n",
      "93\t3905\t0.2\t1\t3619 286\n",
      "94\t3502\t0.2\t1\t3315 187\n",
      "95\t3311\t0.2\t1\t3100 211\n",
      "96\t3107\t0.2\t1\t2913 194\n",
      "97\t2849\t0.2\t1\t2639 210\n",
      "98\t2705\t0.2\t1\t2535 170\n",
      "99\t2452\t0.2\t1\t2282 170\n",
      "100\t2221\t0.2\t1\t2078 143\n",
      "101\t1902\t0.2\t1\t1781 121\n",
      "102\t1519\t0.2\t1\t1414 105\n",
      "103\t1396\t0.2\t1\t1317 79\n",
      "104\t1132\t0.2\t1\t1050 82\n",
      "105\t948\t0.2\t1\t863 85\n",
      "106\t920\t0.2\t1\t846 74\n",
      "107\t815\t0.2\t1\t747 68\n",
      "108\t747\t0.2\t1\t685 62\n",
      "109\t697\t0.2\t1\t619 78\n",
      "110\t623\t0.2\t1\t566 57\n",
      "111\t488\t0.2\t1\t432 56\n",
      "112\t423\t0.2\t1\t388 35\n",
      "113\t364\t0.2\t1\t316 48\n",
      "114\t289\t0.2\t1\t255 34\n",
      "115\t278\t0.2\t1\t238 40\n",
      "116\t212\t0.2\t1\t176 36\n",
      "117\t206\t0.2\t1\t169 37\n",
      "118\t172\t0.2\t1\t143 29\n",
      "119\t195\t0.2\t1\t166 29\n",
      "120\t140\t0.2\t1\t116 24\n",
      "121\t202\t0.2\t1\t168 34\n",
      "122\t138\t0.2\t1\t109 29\n",
      "123\t153\t0.2\t1\t107 46\n",
      "124\t119\t0.2\t1\t83 36\n",
      "125\t145\t0.2\t1\t103 42\n",
      "126\t121\t0.2\t1\t88 33\n",
      "127\t85\t0.2\t1\t60 25\n",
      "128\t81\t0.2\t1\t59 22\n",
      "129\t69\t0.2\t1\t44 25\n",
      "130\t62\t0.2\t1\t42 20\n",
      "131\t62\t0.2\t1\t38 24\n",
      "132\t38\t0.2\t1\t25 13\n",
      "133\t61\t0.2\t1\t55 6\n",
      "134\t40\t0.2\t1\t33 7\n",
      "135\t39\t0.2\t1\t35 4\n",
      "136\t22\t0.2\t1\t19 3\n",
      "137\t30\t0.2\t1\t22 8\n",
      "138\t17\t0.2\t1\t12 5\n",
      "139\t24\t0.2\t1\t17 7\n",
      "140\t27\t0.2\t1\t22 5\n",
      "141\t48\t0.2\t1\t44 4\n",
      "142\t28\t0.2\t1\t17 11\n",
      "143\t27\t0.2\t1\t22 5\n",
      "144\t18\t0.2\t1\t17 1\n",
      "145\t65\t0.2\t1\t63 2\n",
      "146\t74\t0.2\t1\t69 5\n",
      "147\t50\t0.2\t1\t47 3\n",
      "148\t99\t0.2\t1\t94 5\n",
      "149\t28\t0.2\t1\t23 5\n",
      "150\t76098\t0.2\t1\t50 76048\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/D.1.fastq.gz\n",
      "=============================================\n",
      "15648595 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/D.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/D.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to D.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/D.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/D.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 670.27 s (43 us/read; 1.40 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,648,595\n",
      "Reads with adapters:                 9,639,773 (61.6%)\n",
      "Reads written (passing filters):    15,648,595 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,347,289,250 bp\n",
      "Quality-trimmed:             202,065,728 bp (8.6%)\n",
      "Total written (filtered):  1,904,850,213 bp (81.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9639773 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.5%\n",
      "  C: 33.9%\n",
      "  G: 35.9%\n",
      "  T: 12.5%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2122146\t3912148.8\t0\t2122146\n",
      "2\t593117\t978037.2\t0\t593117\n",
      "3\t298513\t244509.3\t0\t298513\n",
      "4\t120336\t61127.3\t0\t120336\n",
      "5\t122028\t15281.8\t0\t122028\n",
      "6\t71853\t3820.5\t0\t71853\n",
      "7\t75953\t955.1\t0\t75953\n",
      "8\t101785\t238.8\t0\t101785\n",
      "9\t133969\t59.7\t0\t133754 215\n",
      "10\t136708\t14.9\t1\t130366 6342\n",
      "11\t72748\t3.7\t1\t65509 7239\n",
      "12\t72940\t0.9\t1\t67129 5811\n",
      "13\t97778\t0.2\t1\t91667 6111\n",
      "14\t97421\t0.2\t1\t92049 5372\n",
      "15\t86084\t0.2\t1\t81750 4334\n",
      "16\t108130\t0.2\t1\t100645 7485\n",
      "17\t124012\t0.2\t1\t116844 7168\n",
      "18\t134113\t0.2\t1\t127382 6731\n",
      "19\t101881\t0.2\t1\t96013 5868\n",
      "20\t137149\t0.2\t1\t131481 5668\n",
      "21\t63908\t0.2\t1\t58901 5007\n",
      "22\t80176\t0.2\t1\t76963 3213\n",
      "23\t85027\t0.2\t1\t81247 3780\n",
      "24\t121991\t0.2\t1\t116243 5748\n",
      "25\t89613\t0.2\t1\t85255 4358\n",
      "26\t121796\t0.2\t1\t115301 6495\n",
      "27\t190421\t0.2\t1\t179475 10946\n",
      "28\t103764\t0.2\t1\t98945 4819\n",
      "29\t119265\t0.2\t1\t108963 10302\n",
      "30\t94637\t0.2\t1\t89928 4709\n",
      "31\t168546\t0.2\t1\t161307 7239\n",
      "32\t90416\t0.2\t1\t85084 5332\n",
      "33\t195154\t0.2\t1\t187500 7654\n",
      "34\t45585\t0.2\t1\t42248 3337\n",
      "35\t104030\t0.2\t1\t99718 4312\n",
      "36\t186808\t0.2\t1\t180186 6622\n",
      "37\t75262\t0.2\t1\t72093 3169\n",
      "38\t104083\t0.2\t1\t101134 2949\n",
      "39\t93581\t0.2\t1\t90095 3486\n",
      "40\t138636\t0.2\t1\t133938 4698\n",
      "41\t77699\t0.2\t1\t74938 2761\n",
      "42\t94837\t0.2\t1\t92133 2704\n",
      "43\t110084\t0.2\t1\t106747 3337\n",
      "44\t67271\t0.2\t1\t64849 2422\n",
      "45\t156236\t0.2\t1\t151949 4287\n",
      "46\t87598\t0.2\t1\t83947 3651\n",
      "47\t127602\t0.2\t1\t124404 3198\n",
      "48\t65985\t0.2\t1\t62540 3445\n",
      "49\t129523\t0.2\t1\t125688 3835\n",
      "50\t43929\t0.2\t1\t41956 1973\n",
      "51\t94182\t0.2\t1\t91347 2835\n",
      "52\t56455\t0.2\t1\t54391 2064\n",
      "53\t76590\t0.2\t1\t74627 1963\n",
      "54\t106831\t0.2\t1\t104346 2485\n",
      "55\t70597\t0.2\t1\t68505 2092\n",
      "56\t72082\t0.2\t1\t70147 1935\n",
      "57\t108292\t0.2\t1\t105733 2559\n",
      "58\t89096\t0.2\t1\t86647 2449\n",
      "59\t44158\t0.2\t1\t42632 1526\n",
      "60\t86517\t0.2\t1\t84728 1789\n",
      "61\t42120\t0.2\t1\t40712 1408\n",
      "62\t47692\t0.2\t1\t46040 1652\n",
      "63\t72259\t0.2\t1\t70646 1613\n",
      "64\t56859\t0.2\t1\t55275 1584\n",
      "65\t25339\t0.2\t1\t24222 1117\n",
      "66\t57786\t0.2\t1\t56430 1356\n",
      "67\t16243\t0.2\t1\t15624 619\n",
      "68\t42122\t0.2\t1\t40674 1448\n",
      "69\t37388\t0.2\t1\t36339 1049\n",
      "70\t71097\t0.2\t1\t69488 1609\n",
      "71\t12529\t0.2\t1\t12045 484\n",
      "72\t22506\t0.2\t1\t21838 668\n",
      "73\t24411\t0.2\t1\t23712 699\n",
      "74\t25083\t0.2\t1\t24360 723\n",
      "75\t27250\t0.2\t1\t26460 790\n",
      "76\t36926\t0.2\t1\t35999 927\n",
      "77\t64348\t0.2\t1\t62830 1518\n",
      "78\t53583\t0.2\t1\t51760 1823\n",
      "79\t15338\t0.2\t1\t14302 1036\n",
      "80\t3242\t0.2\t1\t2881 361\n",
      "81\t393\t0.2\t1\t348 45\n",
      "82\t534\t0.2\t1\t501 33\n",
      "83\t1282\t0.2\t1\t1250 32\n",
      "84\t2354\t0.2\t1\t2310 44\n",
      "85\t3652\t0.2\t1\t3576 76\n",
      "86\t4637\t0.2\t1\t4545 92\n",
      "87\t4596\t0.2\t1\t4519 77\n",
      "88\t3593\t0.2\t1\t3520 73\n",
      "89\t2618\t0.2\t1\t2567 51\n",
      "90\t2157\t0.2\t1\t2118 39\n",
      "91\t1893\t0.2\t1\t1850 43\n",
      "92\t1512\t0.2\t1\t1465 47\n",
      "93\t1186\t0.2\t1\t1149 37\n",
      "94\t1030\t0.2\t1\t1005 25\n",
      "95\t880\t0.2\t1\t860 20\n",
      "96\t822\t0.2\t1\t794 28\n",
      "97\t696\t0.2\t1\t668 28\n",
      "98\t653\t0.2\t1\t635 18\n",
      "99\t536\t0.2\t1\t520 16\n",
      "100\t469\t0.2\t1\t454 15\n",
      "101\t385\t0.2\t1\t371 14\n",
      "102\t309\t0.2\t1\t294 15\n",
      "103\t270\t0.2\t1\t262 8\n",
      "104\t237\t0.2\t1\t225 12\n",
      "105\t188\t0.2\t1\t176 12\n",
      "106\t157\t0.2\t1\t143 14\n",
      "107\t151\t0.2\t1\t133 18\n",
      "108\t143\t0.2\t1\t135 8\n",
      "109\t107\t0.2\t1\t99 8\n",
      "110\t119\t0.2\t1\t111 8\n",
      "111\t88\t0.2\t1\t71 17\n",
      "112\t89\t0.2\t1\t74 15\n",
      "113\t81\t0.2\t1\t65 16\n",
      "114\t85\t0.2\t1\t70 15\n",
      "115\t65\t0.2\t1\t52 13\n",
      "116\t57\t0.2\t1\t43 14\n",
      "117\t45\t0.2\t1\t31 14\n",
      "118\t50\t0.2\t1\t37 13\n",
      "119\t67\t0.2\t1\t53 14\n",
      "120\t46\t0.2\t1\t32 14\n",
      "121\t63\t0.2\t1\t38 25\n",
      "122\t42\t0.2\t1\t26 16\n",
      "123\t64\t0.2\t1\t45 19\n",
      "124\t41\t0.2\t1\t26 15\n",
      "125\t42\t0.2\t1\t28 14\n",
      "126\t30\t0.2\t1\t17 13\n",
      "127\t30\t0.2\t1\t14 16\n",
      "128\t29\t0.2\t1\t15 14\n",
      "129\t21\t0.2\t1\t10 11\n",
      "130\t12\t0.2\t1\t8 4\n",
      "131\t14\t0.2\t1\t9 5\n",
      "132\t6\t0.2\t1\t3 3\n",
      "133\t14\t0.2\t1\t7 7\n",
      "134\t3\t0.2\t1\t2 1\n",
      "135\t6\t0.2\t1\t5 1\n",
      "136\t4\t0.2\t1\t4\n",
      "137\t5\t0.2\t1\t3 2\n",
      "138\t2\t0.2\t1\t0 2\n",
      "139\t4\t0.2\t1\t3 1\n",
      "140\t2\t0.2\t1\t1 1\n",
      "141\t12\t0.2\t1\t10 2\n",
      "142\t4\t0.2\t1\t3 1\n",
      "143\t2\t0.2\t1\t2\n",
      "144\t1\t0.2\t1\t1\n",
      "145\t7\t0.2\t1\t6 1\n",
      "146\t14\t0.2\t1\t9 5\n",
      "147\t6\t0.2\t1\t4 2\n",
      "148\t6\t0.2\t1\t5 1\n",
      "149\t3\t0.2\t1\t1 2\n",
      "150\t4\t0.2\t1\t3 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/D.2.fastq.gz\n",
      "=============================================\n",
      "15648595 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files D.1_trimmed.fq.gz and D.2_trimmed.fq.gz\n",
      "file_1: D.1_trimmed.fq.gz, file_2: D.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: D.1_trimmed.fq.gz and D.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to D.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to D.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to D.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to D.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 15648595\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 822275 (5.25%)\n",
      "\n",
      "Deleting both intermediate output files D.1_trimmed.fq.gz and D.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/D.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/D.1.fastq.gz ./data_raw/D.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/E.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t376351\tAGATCGGAAGAGC\t1000000\t37.64\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 376351). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/E.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/E.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/E.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to E.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/E.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/E.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 626.53 s (40 us/read; 1.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,805,620\n",
      "Reads with adapters:                10,220,028 (64.7%)\n",
      "Reads written (passing filters):    15,805,620 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,370,843,000 bp\n",
      "Quality-trimmed:              78,534,160 bp (3.3%)\n",
      "Total written (filtered):  2,005,407,040 bp (84.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10220028 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.2%\n",
      "  C: 33.3%\n",
      "  G: 30.1%\n",
      "  T: 17.8%\n",
      "  none/other: 1.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1906533\t3951405.0\t0\t1906533\n",
      "2\t629527\t987851.2\t0\t629527\n",
      "3\t257648\t246962.8\t0\t257648\n",
      "4\t130992\t61740.7\t0\t130992\n",
      "5\t92943\t15435.2\t0\t92943\n",
      "6\t77585\t3858.8\t0\t77585\n",
      "7\t78325\t964.7\t0\t78325\n",
      "8\t94365\t241.2\t0\t94365\n",
      "9\t123378\t60.3\t0\t123191 187\n",
      "10\t137563\t15.1\t1\t125964 11599\n",
      "11\t96156\t3.8\t1\t86145 10011\n",
      "12\t97788\t0.9\t1\t88007 9781\n",
      "13\t121630\t0.2\t1\t110706 10924\n",
      "14\t87073\t0.2\t1\t78403 8670\n",
      "15\t124301\t0.2\t1\t113485 10816\n",
      "16\t63837\t0.2\t1\t56724 7113\n",
      "17\t104249\t0.2\t1\t94583 9666\n",
      "18\t93815\t0.2\t1\t85831 7984\n",
      "19\t110916\t0.2\t1\t101361 9555\n",
      "20\t87521\t0.2\t1\t79831 7690\n",
      "21\t115797\t0.2\t1\t105680 10117\n",
      "22\t121493\t0.2\t1\t110619 10874\n",
      "23\t114795\t0.2\t1\t104386 10409\n",
      "24\t113355\t0.2\t1\t102254 11101\n",
      "25\t112031\t0.2\t1\t102095 9936\n",
      "26\t90365\t0.2\t1\t81983 8382\n",
      "27\t117298\t0.2\t1\t106732 10566\n",
      "28\t108324\t0.2\t1\t99195 9129\n",
      "29\t117436\t0.2\t1\t106828 10608\n",
      "30\t103935\t0.2\t1\t94829 9106\n",
      "31\t131462\t0.2\t1\t119837 11625\n",
      "32\t134407\t0.2\t1\t122722 11685\n",
      "33\t101926\t0.2\t1\t92200 9726\n",
      "34\t141259\t0.2\t1\t129421 11838\n",
      "35\t77879\t0.2\t1\t71075 6804\n",
      "36\t117160\t0.2\t1\t106249 10911\n",
      "37\t130675\t0.2\t1\t117853 12822\n",
      "38\t130051\t0.2\t1\t117540 12511\n",
      "39\t173990\t0.2\t1\t157036 16954\n",
      "40\t127717\t0.2\t1\t115783 11934\n",
      "41\t179954\t0.2\t1\t162879 17075\n",
      "42\t326649\t0.2\t1\t297425 29224\n",
      "43\t330900\t0.2\t1\t300723 30177\n",
      "44\t237983\t0.2\t1\t216128 21855\n",
      "45\t116777\t0.2\t1\t106735 10042\n",
      "46\t55703\t0.2\t1\t50642 5061\n",
      "47\t45701\t0.2\t1\t40473 5228\n",
      "48\t73201\t0.2\t1\t66827 6374\n",
      "49\t126724\t0.2\t1\t116344 10380\n",
      "50\t33891\t0.2\t1\t29943 3948\n",
      "51\t141159\t0.2\t1\t128869 12290\n",
      "52\t105885\t0.2\t1\t96854 9031\n",
      "53\t62147\t0.2\t1\t56649 5498\n",
      "54\t72787\t0.2\t1\t66155 6632\n",
      "55\t122073\t0.2\t1\t111733 10340\n",
      "56\t76601\t0.2\t1\t70152 6449\n",
      "57\t18644\t0.2\t1\t16871 1773\n",
      "58\t58669\t0.2\t1\t53554 5115\n",
      "59\t125690\t0.2\t1\t115007 10683\n",
      "60\t19408\t0.2\t1\t17546 1862\n",
      "61\t53866\t0.2\t1\t49286 4580\n",
      "62\t76331\t0.2\t1\t69920 6411\n",
      "63\t16488\t0.2\t1\t14981 1507\n",
      "64\t43964\t0.2\t1\t40098 3866\n",
      "65\t69632\t0.2\t1\t63513 6119\n",
      "66\t28314\t0.2\t1\t25763 2551\n",
      "67\t33814\t0.2\t1\t30857 2957\n",
      "68\t40243\t0.2\t1\t36712 3531\n",
      "69\t37424\t0.2\t1\t33824 3600\n",
      "70\t36560\t0.2\t1\t33073 3487\n",
      "71\t38804\t0.2\t1\t34085 4719\n",
      "72\t53858\t0.2\t1\t45382 8476\n",
      "73\t69157\t0.2\t1\t57582 11575\n",
      "74\t83759\t0.2\t1\t66222 17537\n",
      "75\t61287\t0.2\t1\t53255 8032\n",
      "76\t5434\t0.2\t1\t4735 699\n",
      "77\t290\t0.2\t1\t249 41\n",
      "78\t1617\t0.2\t1\t1470 147\n",
      "79\t4793\t0.2\t1\t4359 434\n",
      "80\t7937\t0.2\t1\t7202 735\n",
      "81\t9469\t0.2\t1\t8585 884\n",
      "82\t10058\t0.2\t1\t9143 915\n",
      "83\t10513\t0.2\t1\t9537 976\n",
      "84\t10291\t0.2\t1\t9315 976\n",
      "85\t9900\t0.2\t1\t9016 884\n",
      "86\t9549\t0.2\t1\t8669 880\n",
      "87\t9012\t0.2\t1\t8203 809\n",
      "88\t8201\t0.2\t1\t7432 769\n",
      "89\t7490\t0.2\t1\t6821 669\n",
      "90\t6965\t0.2\t1\t6338 627\n",
      "91\t6122\t0.2\t1\t5551 571\n",
      "92\t5558\t0.2\t1\t4997 561\n",
      "93\t4756\t0.2\t1\t4236 520\n",
      "94\t4222\t0.2\t1\t3833 389\n",
      "95\t3835\t0.2\t1\t3470 365\n",
      "96\t3601\t0.2\t1\t3262 339\n",
      "97\t3335\t0.2\t1\t2997 338\n",
      "98\t3358\t0.2\t1\t3036 322\n",
      "99\t2889\t0.2\t1\t2597 292\n",
      "100\t2526\t0.2\t1\t2289 237\n",
      "101\t2257\t0.2\t1\t2044 213\n",
      "102\t1909\t0.2\t1\t1723 186\n",
      "103\t1592\t0.2\t1\t1425 167\n",
      "104\t1357\t0.2\t1\t1204 153\n",
      "105\t1108\t0.2\t1\t991 117\n",
      "106\t1010\t0.2\t1\t893 117\n",
      "107\t1054\t0.2\t1\t926 128\n",
      "108\t914\t0.2\t1\t810 104\n",
      "109\t839\t0.2\t1\t757 82\n",
      "110\t695\t0.2\t1\t624 71\n",
      "111\t576\t0.2\t1\t509 67\n",
      "112\t532\t0.2\t1\t462 70\n",
      "113\t470\t0.2\t1\t403 67\n",
      "114\t369\t0.2\t1\t309 60\n",
      "115\t351\t0.2\t1\t301 50\n",
      "116\t313\t0.2\t1\t265 48\n",
      "117\t264\t0.2\t1\t217 47\n",
      "118\t269\t0.2\t1\t221 48\n",
      "119\t228\t0.2\t1\t191 37\n",
      "120\t173\t0.2\t1\t134 39\n",
      "121\t222\t0.2\t1\t171 51\n",
      "122\t143\t0.2\t1\t116 27\n",
      "123\t191\t0.2\t1\t142 49\n",
      "124\t124\t0.2\t1\t93 31\n",
      "125\t130\t0.2\t1\t90 40\n",
      "126\t102\t0.2\t1\t76 26\n",
      "127\t104\t0.2\t1\t74 30\n",
      "128\t103\t0.2\t1\t72 31\n",
      "129\t89\t0.2\t1\t52 37\n",
      "130\t53\t0.2\t1\t38 15\n",
      "131\t67\t0.2\t1\t44 23\n",
      "132\t36\t0.2\t1\t24 12\n",
      "133\t38\t0.2\t1\t32 6\n",
      "134\t45\t0.2\t1\t35 10\n",
      "135\t41\t0.2\t1\t38 3\n",
      "136\t26\t0.2\t1\t21 5\n",
      "137\t30\t0.2\t1\t27 3\n",
      "138\t15\t0.2\t1\t12 3\n",
      "139\t21\t0.2\t1\t16 5\n",
      "140\t47\t0.2\t1\t42 5\n",
      "141\t47\t0.2\t1\t39 8\n",
      "142\t40\t0.2\t1\t22 18\n",
      "143\t47\t0.2\t1\t35 12\n",
      "144\t13\t0.2\t1\t11 2\n",
      "145\t72\t0.2\t1\t69 3\n",
      "146\t116\t0.2\t1\t112 4\n",
      "147\t77\t0.2\t1\t67 10\n",
      "148\t169\t0.2\t1\t166 3\n",
      "149\t50\t0.2\t1\t36 14\n",
      "150\t78222\t0.2\t1\t36 78186\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/E.1.fastq.gz\n",
      "=============================================\n",
      "15805620 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/E.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/E.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to E.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/E.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/E.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 612.89 s (39 us/read; 1.55 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,805,620\n",
      "Reads with adapters:                 9,642,845 (61.0%)\n",
      "Reads written (passing filters):    15,805,620 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,370,843,000 bp\n",
      "Quality-trimmed:             256,382,339 bp (10.8%)\n",
      "Total written (filtered):  1,866,905,086 bp (78.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9642845 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.8%\n",
      "  C: 33.9%\n",
      "  G: 36.1%\n",
      "  T: 13.0%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2047188\t3951405.0\t0\t2047188\n",
      "2\t589289\t987851.2\t0\t589289\n",
      "3\t286936\t246962.8\t0\t286936\n",
      "4\t118478\t61740.7\t0\t118478\n",
      "5\t119352\t15435.2\t0\t119352\n",
      "6\t70281\t3858.8\t0\t70281\n",
      "7\t75395\t964.7\t0\t75395\n",
      "8\t100975\t241.2\t0\t100975\n",
      "9\t129685\t60.3\t0\t129460 225\n",
      "10\t137771\t15.1\t1\t130653 7118\n",
      "11\t70999\t3.8\t1\t62923 8076\n",
      "12\t73853\t0.9\t1\t67593 6260\n",
      "13\t94232\t0.2\t1\t87407 6825\n",
      "14\t96277\t0.2\t1\t90050 6227\n",
      "15\t84935\t0.2\t1\t79897 5038\n",
      "16\t105388\t0.2\t1\t97268 8120\n",
      "17\t122168\t0.2\t1\t114517 7651\n",
      "18\t132904\t0.2\t1\t125442 7462\n",
      "19\t110673\t0.2\t1\t103762 6911\n",
      "20\t130720\t0.2\t1\t124237 6483\n",
      "21\t60084\t0.2\t1\t55120 4964\n",
      "22\t78318\t0.2\t1\t74939 3379\n",
      "23\t81779\t0.2\t1\t77562 4217\n",
      "24\t122803\t0.2\t1\t115801 7002\n",
      "25\t89782\t0.2\t1\t84751 5031\n",
      "26\t117641\t0.2\t1\t110509 7132\n",
      "27\t194934\t0.2\t1\t182260 12674\n",
      "28\t113610\t0.2\t1\t107767 5843\n",
      "29\t117536\t0.2\t1\t106230 11306\n",
      "30\t113110\t0.2\t1\t107286 5824\n",
      "31\t147777\t0.2\t1\t140464 7313\n",
      "32\t100475\t0.2\t1\t95269 5206\n",
      "33\t123137\t0.2\t1\t116676 6461\n",
      "34\t185316\t0.2\t1\t177775 7541\n",
      "35\t95784\t0.2\t1\t90863 4921\n",
      "36\t152546\t0.2\t1\t146127 6419\n",
      "37\t96400\t0.2\t1\t92167 4233\n",
      "38\t71660\t0.2\t1\t67674 3986\n",
      "39\t90248\t0.2\t1\t86514 3734\n",
      "40\t63306\t0.2\t1\t59003 4303\n",
      "41\t136262\t0.2\t1\t131318 4944\n",
      "42\t66012\t0.2\t1\t63247 2765\n",
      "43\t127378\t0.2\t1\t122902 4476\n",
      "44\t60506\t0.2\t1\t57953 2553\n",
      "45\t169140\t0.2\t1\t163275 5865\n",
      "46\t87554\t0.2\t1\t83245 4309\n",
      "47\t118040\t0.2\t1\t114262 3778\n",
      "48\t66830\t0.2\t1\t62669 4161\n",
      "49\t136227\t0.2\t1\t131159 5068\n",
      "50\t41607\t0.2\t1\t39312 2295\n",
      "51\t100554\t0.2\t1\t96713 3841\n",
      "52\t53026\t0.2\t1\t50388 2638\n",
      "53\t80527\t0.2\t1\t77999 2528\n",
      "54\t110287\t0.2\t1\t107104 3183\n",
      "55\t72852\t0.2\t1\t70040 2812\n",
      "56\t76456\t0.2\t1\t73836 2620\n",
      "57\t121041\t0.2\t1\t117613 3428\n",
      "58\t95034\t0.2\t1\t91761 3273\n",
      "59\t44833\t0.2\t1\t42877 1956\n",
      "60\t99983\t0.2\t1\t97422 2561\n",
      "61\t44224\t0.2\t1\t42352 1872\n",
      "62\t47561\t0.2\t1\t45437 2124\n",
      "63\t82888\t0.2\t1\t80585 2303\n",
      "64\t61308\t0.2\t1\t59282 2026\n",
      "65\t26789\t0.2\t1\t25297 1492\n",
      "66\t62397\t0.2\t1\t60412 1985\n",
      "67\t17461\t0.2\t1\t16669 792\n",
      "68\t43245\t0.2\t1\t41343 1902\n",
      "69\t41889\t0.2\t1\t40379 1510\n",
      "70\t79943\t0.2\t1\t77478 2465\n",
      "71\t14484\t0.2\t1\t13797 687\n",
      "72\t25484\t0.2\t1\t24502 982\n",
      "73\t27453\t0.2\t1\t26437 1016\n",
      "74\t28320\t0.2\t1\t27279 1041\n",
      "75\t31399\t0.2\t1\t30266 1133\n",
      "76\t42047\t0.2\t1\t40695 1352\n",
      "77\t71025\t0.2\t1\t68784 2241\n",
      "78\t55654\t0.2\t1\t53100 2554\n",
      "79\t15880\t0.2\t1\t14513 1367\n",
      "80\t3392\t0.2\t1\t2865 527\n",
      "81\t424\t0.2\t1\t370 54\n",
      "82\t568\t0.2\t1\t540 28\n",
      "83\t1391\t0.2\t1\t1333 58\n",
      "84\t2617\t0.2\t1\t2546 71\n",
      "85\t4114\t0.2\t1\t4009 105\n",
      "86\t5079\t0.2\t1\t4945 134\n",
      "87\t5041\t0.2\t1\t4907 134\n",
      "88\t3983\t0.2\t1\t3876 107\n",
      "89\t2760\t0.2\t1\t2675 85\n",
      "90\t2413\t0.2\t1\t2343 70\n",
      "91\t1969\t0.2\t1\t1900 69\n",
      "92\t1646\t0.2\t1\t1583 63\n",
      "93\t1285\t0.2\t1\t1247 38\n",
      "94\t1067\t0.2\t1\t1034 33\n",
      "95\t939\t0.2\t1\t894 45\n",
      "96\t860\t0.2\t1\t826 34\n",
      "97\t753\t0.2\t1\t726 27\n",
      "98\t737\t0.2\t1\t703 34\n",
      "99\t581\t0.2\t1\t562 19\n",
      "100\t496\t0.2\t1\t468 28\n",
      "101\t411\t0.2\t1\t390 21\n",
      "102\t356\t0.2\t1\t337 19\n",
      "103\t279\t0.2\t1\t265 14\n",
      "104\t225\t0.2\t1\t206 19\n",
      "105\t184\t0.2\t1\t171 13\n",
      "106\t167\t0.2\t1\t159 8\n",
      "107\t162\t0.2\t1\t148 14\n",
      "108\t153\t0.2\t1\t145 8\n",
      "109\t135\t0.2\t1\t121 14\n",
      "110\t120\t0.2\t1\t108 12\n",
      "111\t100\t0.2\t1\t87 13\n",
      "112\t102\t0.2\t1\t86 16\n",
      "113\t74\t0.2\t1\t66 8\n",
      "114\t72\t0.2\t1\t62 10\n",
      "115\t74\t0.2\t1\t63 11\n",
      "116\t57\t0.2\t1\t40 17\n",
      "117\t64\t0.2\t1\t46 18\n",
      "118\t61\t0.2\t1\t51 10\n",
      "119\t44\t0.2\t1\t33 11\n",
      "120\t45\t0.2\t1\t30 15\n",
      "121\t61\t0.2\t1\t47 14\n",
      "122\t49\t0.2\t1\t34 15\n",
      "123\t46\t0.2\t1\t27 19\n",
      "124\t40\t0.2\t1\t23 17\n",
      "125\t47\t0.2\t1\t28 19\n",
      "126\t40\t0.2\t1\t20 20\n",
      "127\t27\t0.2\t1\t17 10\n",
      "128\t35\t0.2\t1\t15 20\n",
      "129\t26\t0.2\t1\t17 9\n",
      "130\t12\t0.2\t1\t5 7\n",
      "131\t13\t0.2\t1\t6 7\n",
      "132\t8\t0.2\t1\t6 2\n",
      "133\t7\t0.2\t1\t5 2\n",
      "134\t7\t0.2\t1\t4 3\n",
      "135\t6\t0.2\t1\t3 3\n",
      "136\t7\t0.2\t1\t3 4\n",
      "137\t10\t0.2\t1\t7 3\n",
      "138\t3\t0.2\t1\t0 3\n",
      "139\t4\t0.2\t1\t0 4\n",
      "140\t5\t0.2\t1\t3 2\n",
      "141\t3\t0.2\t1\t3\n",
      "142\t3\t0.2\t1\t3\n",
      "143\t5\t0.2\t1\t5\n",
      "144\t2\t0.2\t1\t1 1\n",
      "145\t6\t0.2\t1\t5 1\n",
      "146\t8\t0.2\t1\t8\n",
      "147\t7\t0.2\t1\t5 2\n",
      "148\t9\t0.2\t1\t8 1\n",
      "149\t1\t0.2\t1\t0 1\n",
      "150\t3\t0.2\t1\t2 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/E.2.fastq.gz\n",
      "=============================================\n",
      "15805620 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files E.1_trimmed.fq.gz and E.2_trimmed.fq.gz\n",
      "file_1: E.1_trimmed.fq.gz, file_2: E.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: E.1_trimmed.fq.gz and E.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to E.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to E.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to E.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to E.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 15805620\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1165290 (7.37%)\n",
      "\n",
      "Deleting both intermediate output files E.1_trimmed.fq.gz and E.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/E.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/E.1.fastq.gz ./data_raw/E.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/F.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t376796\tAGATCGGAAGAGC\t1000000\t37.68\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 376796). Second best hit was Nextera (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/F.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/F.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/F.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to F.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/F.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/F.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 871.80 s (40 us/read; 1.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,709,453\n",
      "Reads with adapters:                13,987,322 (64.4%)\n",
      "Reads written (passing filters):    21,709,453 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,256,417,950 bp\n",
      "Quality-trimmed:              96,130,982 bp (3.0%)\n",
      "Total written (filtered):  2,781,336,035 bp (85.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 13987322 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.9%\n",
      "  C: 36.7%\n",
      "  G: 28.3%\n",
      "  T: 18.5%\n",
      "  none/other: 0.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2618662\t5427363.2\t0\t2618662\n",
      "2\t876651\t1356840.8\t0\t876651\n",
      "3\t362988\t339210.2\t0\t362988\n",
      "4\t183146\t84802.6\t0\t183146\n",
      "5\t129820\t21200.6\t0\t129820\n",
      "6\t108159\t5300.2\t0\t108159\n",
      "7\t107301\t1325.0\t0\t107301\n",
      "8\t130643\t331.3\t0\t130643\n",
      "9\t170652\t82.8\t0\t170298 354\n",
      "10\t189482\t20.7\t1\t172354 17128\n",
      "11\t132681\t5.2\t1\t118297 14384\n",
      "12\t135502\t1.3\t1\t121508 13994\n",
      "13\t169626\t0.3\t1\t153517 16109\n",
      "14\t121822\t0.3\t1\t109066 12756\n",
      "15\t174542\t0.3\t1\t158587 15955\n",
      "16\t89272\t0.3\t1\t79080 10192\n",
      "17\t145914\t0.3\t1\t131983 13931\n",
      "18\t130005\t0.3\t1\t118493 11512\n",
      "19\t153270\t0.3\t1\t139648 13622\n",
      "20\t121297\t0.3\t1\t109857 11440\n",
      "21\t161504\t0.3\t1\t146717 14787\n",
      "22\t169790\t0.3\t1\t154079 15711\n",
      "23\t160737\t0.3\t1\t146359 14378\n",
      "24\t158671\t0.3\t1\t142671 16000\n",
      "25\t155636\t0.3\t1\t141654 13982\n",
      "26\t126636\t0.3\t1\t114715 11921\n",
      "27\t164044\t0.3\t1\t149234 14810\n",
      "28\t151102\t0.3\t1\t138013 13089\n",
      "29\t163622\t0.3\t1\t148549 15073\n",
      "30\t145466\t0.3\t1\t132292 13174\n",
      "31\t181748\t0.3\t1\t165428 16320\n",
      "32\t185025\t0.3\t1\t168616 16409\n",
      "33\t142374\t0.3\t1\t128727 13647\n",
      "34\t197436\t0.3\t1\t180324 17112\n",
      "35\t108682\t0.3\t1\t99169 9513\n",
      "36\t165176\t0.3\t1\t150300 14876\n",
      "37\t184078\t0.3\t1\t165919 18159\n",
      "38\t182140\t0.3\t1\t164752 17388\n",
      "39\t243432\t0.3\t1\t220096 23336\n",
      "40\t176203\t0.3\t1\t160131 16072\n",
      "41\t248102\t0.3\t1\t224919 23183\n",
      "42\t450989\t0.3\t1\t411737 39252\n",
      "43\t456730\t0.3\t1\t416969 39761\n",
      "44\t328948\t0.3\t1\t300644 28304\n",
      "45\t161653\t0.3\t1\t148655 12998\n",
      "46\t77016\t0.3\t1\t70232 6784\n",
      "47\t62147\t0.3\t1\t56397 5750\n",
      "48\t101801\t0.3\t1\t93214 8587\n",
      "49\t176915\t0.3\t1\t163240 13675\n",
      "50\t46267\t0.3\t1\t41372 4895\n",
      "51\t193092\t0.3\t1\t176956 16136\n",
      "52\t144309\t0.3\t1\t132261 12048\n",
      "53\t84449\t0.3\t1\t77267 7182\n",
      "54\t99332\t0.3\t1\t90984 8348\n",
      "55\t166423\t0.3\t1\t152913 13510\n",
      "56\t105280\t0.3\t1\t96847 8433\n",
      "57\t25646\t0.3\t1\t23290 2356\n",
      "58\t80594\t0.3\t1\t74011 6583\n",
      "59\t171973\t0.3\t1\t158321 13652\n",
      "60\t26161\t0.3\t1\t23671 2490\n",
      "61\t71794\t0.3\t1\t66035 5759\n",
      "62\t102238\t0.3\t1\t94037 8201\n",
      "63\t22186\t0.3\t1\t20239 1947\n",
      "64\t58345\t0.3\t1\t53708 4637\n",
      "65\t93828\t0.3\t1\t86320 7508\n",
      "66\t37505\t0.3\t1\t34330 3175\n",
      "67\t44928\t0.3\t1\t41306 3622\n",
      "68\t54567\t0.3\t1\t50123 4444\n",
      "69\t49376\t0.3\t1\t45195 4181\n",
      "70\t48837\t0.3\t1\t44765 4072\n",
      "71\t50375\t0.3\t1\t45616 4759\n",
      "72\t66024\t0.3\t1\t58802 7222\n",
      "73\t84766\t0.3\t1\t75319 9447\n",
      "74\t98319\t0.3\t1\t85585 12734\n",
      "75\t76293\t0.3\t1\t68666 7627\n",
      "76\t7048\t0.3\t1\t6220 828\n",
      "77\t337\t0.3\t1\t285 52\n",
      "78\t2001\t0.3\t1\t1837 164\n",
      "79\t6277\t0.3\t1\t5751 526\n",
      "80\t10168\t0.3\t1\t9281 887\n",
      "81\t12245\t0.3\t1\t11231 1014\n",
      "82\t12552\t0.3\t1\t11521 1031\n",
      "83\t13065\t0.3\t1\t12021 1044\n",
      "84\t12535\t0.3\t1\t11515 1020\n",
      "85\t12291\t0.3\t1\t11324 967\n",
      "86\t11893\t0.3\t1\t10935 958\n",
      "87\t11077\t0.3\t1\t10185 892\n",
      "88\t10270\t0.3\t1\t9513 757\n",
      "89\t9409\t0.3\t1\t8669 740\n",
      "90\t8355\t0.3\t1\t7721 634\n",
      "91\t7501\t0.3\t1\t6905 596\n",
      "92\t6582\t0.3\t1\t6057 525\n",
      "93\t5597\t0.3\t1\t5128 469\n",
      "94\t4978\t0.3\t1\t4587 391\n",
      "95\t4539\t0.3\t1\t4151 388\n",
      "96\t4233\t0.3\t1\t3895 338\n",
      "97\t3991\t0.3\t1\t3704 287\n",
      "98\t3889\t0.3\t1\t3553 336\n",
      "99\t3446\t0.3\t1\t3153 293\n",
      "100\t2918\t0.3\t1\t2682 236\n",
      "101\t2578\t0.3\t1\t2344 234\n",
      "102\t2104\t0.3\t1\t1916 188\n",
      "103\t1768\t0.3\t1\t1619 149\n",
      "104\t1562\t0.3\t1\t1416 146\n",
      "105\t1295\t0.3\t1\t1168 127\n",
      "106\t1220\t0.3\t1\t1114 106\n",
      "107\t1139\t0.3\t1\t1007 132\n",
      "108\t1030\t0.3\t1\t919 111\n",
      "109\t1031\t0.3\t1\t945 86\n",
      "110\t883\t0.3\t1\t791 92\n",
      "111\t780\t0.3\t1\t695 85\n",
      "112\t623\t0.3\t1\t549 74\n",
      "113\t565\t0.3\t1\t496 69\n",
      "114\t465\t0.3\t1\t401 64\n",
      "115\t452\t0.3\t1\t389 63\n",
      "116\t349\t0.3\t1\t304 45\n",
      "117\t336\t0.3\t1\t287 49\n",
      "118\t288\t0.3\t1\t239 49\n",
      "119\t315\t0.3\t1\t253 62\n",
      "120\t261\t0.3\t1\t212 49\n",
      "121\t255\t0.3\t1\t203 52\n",
      "122\t224\t0.3\t1\t169 55\n",
      "123\t248\t0.3\t1\t194 54\n",
      "124\t186\t0.3\t1\t134 52\n",
      "125\t215\t0.3\t1\t156 59\n",
      "126\t151\t0.3\t1\t96 55\n",
      "127\t158\t0.3\t1\t112 46\n",
      "128\t117\t0.3\t1\t83 34\n",
      "129\t139\t0.3\t1\t104 35\n",
      "130\t74\t0.3\t1\t47 27\n",
      "131\t87\t0.3\t1\t61 26\n",
      "132\t59\t0.3\t1\t45 14\n",
      "133\t67\t0.3\t1\t52 15\n",
      "134\t85\t0.3\t1\t72 13\n",
      "135\t64\t0.3\t1\t58 6\n",
      "136\t36\t0.3\t1\t31 5\n",
      "137\t38\t0.3\t1\t31 7\n",
      "138\t49\t0.3\t1\t41 8\n",
      "139\t40\t0.3\t1\t33 7\n",
      "140\t46\t0.3\t1\t37 9\n",
      "141\t68\t0.3\t1\t63 5\n",
      "142\t38\t0.3\t1\t24 14\n",
      "143\t66\t0.3\t1\t63 3\n",
      "144\t25\t0.3\t1\t24 1\n",
      "145\t128\t0.3\t1\t103 25\n",
      "146\t195\t0.3\t1\t186 9\n",
      "147\t92\t0.3\t1\t89 3\n",
      "148\t250\t0.3\t1\t241 9\n",
      "149\t67\t0.3\t1\t51 16\n",
      "150\t35669\t0.3\t1\t57 35612\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/F.1.fastq.gz\n",
      "=============================================\n",
      "21709453 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/F.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/F.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to F.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/F.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/F.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 843.32 s (39 us/read; 1.54 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,709,453\n",
      "Reads with adapters:                13,338,711 (61.4%)\n",
      "Reads written (passing filters):    21,709,453 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,256,417,950 bp\n",
      "Quality-trimmed:             333,168,591 bp (10.2%)\n",
      "Total written (filtered):  2,581,887,635 bp (79.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 13338711 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.8%\n",
      "  C: 37.8%\n",
      "  G: 33.5%\n",
      "  T: 12.7%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2870247\t5427363.2\t0\t2870247\n",
      "2\t809821\t1356840.8\t0\t809821\n",
      "3\t395016\t339210.2\t0\t395016\n",
      "4\t158551\t84802.6\t0\t158551\n",
      "5\t167528\t21200.6\t0\t167528\n",
      "6\t97408\t5300.2\t0\t97408\n",
      "7\t101915\t1325.0\t0\t101915\n",
      "8\t137380\t331.3\t0\t137380\n",
      "9\t179733\t82.8\t0\t179386 347\n",
      "10\t185732\t20.7\t1\t175556 10176\n",
      "11\t101778\t5.2\t1\t90569 11209\n",
      "12\t100208\t1.3\t1\t91326 8882\n",
      "13\t135382\t0.3\t1\t125803 9579\n",
      "14\t133727\t0.3\t1\t125282 8445\n",
      "15\t115793\t0.3\t1\t108943 6850\n",
      "16\t149755\t0.3\t1\t138350 11405\n",
      "17\t170764\t0.3\t1\t159829 10935\n",
      "18\t186172\t0.3\t1\t175471 10701\n",
      "19\t155508\t0.3\t1\t145989 9519\n",
      "20\t177188\t0.3\t1\t168355 8833\n",
      "21\t84540\t0.3\t1\t77446 7094\n",
      "22\t105430\t0.3\t1\t100756 4674\n",
      "23\t116834\t0.3\t1\t110740 6094\n",
      "24\t172055\t0.3\t1\t162860 9195\n",
      "25\t122737\t0.3\t1\t115678 7059\n",
      "26\t166701\t0.3\t1\t156732 9969\n",
      "27\t257541\t0.3\t1\t241247 16294\n",
      "28\t136179\t0.3\t1\t129010 7169\n",
      "29\t173665\t0.3\t1\t158092 15573\n",
      "30\t149339\t0.3\t1\t141656 7683\n",
      "31\t199985\t0.3\t1\t190372 9613\n",
      "32\t123915\t0.3\t1\t117596 6319\n",
      "33\t186307\t0.3\t1\t178134 8173\n",
      "34\t124191\t0.3\t1\t117571 6620\n",
      "35\t205886\t0.3\t1\t197439 8447\n",
      "36\t81124\t0.3\t1\t77070 4054\n",
      "37\t141313\t0.3\t1\t132837 8476\n",
      "38\t121913\t0.3\t1\t117053 4860\n",
      "39\t175722\t0.3\t1\t168745 6977\n",
      "40\t182390\t0.3\t1\t171405 10985\n",
      "41\t128917\t0.3\t1\t122387 6530\n",
      "42\t170486\t0.3\t1\t164674 5812\n",
      "43\t164118\t0.3\t1\t157814 6304\n",
      "44\t101035\t0.3\t1\t96737 4298\n",
      "45\t222793\t0.3\t1\t214885 7908\n",
      "46\t133060\t0.3\t1\t126208 6852\n",
      "47\t182248\t0.3\t1\t176712 5536\n",
      "48\t102879\t0.3\t1\t95990 6889\n",
      "49\t186655\t0.3\t1\t179640 7015\n",
      "50\t64710\t0.3\t1\t61020 3690\n",
      "51\t133879\t0.3\t1\t128551 5328\n",
      "52\t77250\t0.3\t1\t73594 3656\n",
      "53\t106557\t0.3\t1\t103290 3267\n",
      "54\t152435\t0.3\t1\t148130 4305\n",
      "55\t100974\t0.3\t1\t97172 3802\n",
      "56\t104084\t0.3\t1\t100656 3428\n",
      "57\t163308\t0.3\t1\t158723 4585\n",
      "58\t131805\t0.3\t1\t127247 4558\n",
      "59\t62502\t0.3\t1\t59821 2681\n",
      "60\t129311\t0.3\t1\t126143 3168\n",
      "61\t61705\t0.3\t1\t59149 2556\n",
      "62\t69882\t0.3\t1\t66818 3064\n",
      "63\t108829\t0.3\t1\t105876 2953\n",
      "64\t82392\t0.3\t1\t79499 2893\n",
      "65\t38493\t0.3\t1\t36425 2068\n",
      "66\t83812\t0.3\t1\t81333 2479\n",
      "67\t23625\t0.3\t1\t22493 1132\n",
      "68\t62288\t0.3\t1\t59718 2570\n",
      "69\t54723\t0.3\t1\t52814 1909\n",
      "70\t104059\t0.3\t1\t101106 2953\n",
      "71\t18885\t0.3\t1\t18043 842\n",
      "72\t33863\t0.3\t1\t32579 1284\n",
      "73\t37137\t0.3\t1\t35832 1305\n",
      "74\t37532\t0.3\t1\t36179 1353\n",
      "75\t40479\t0.3\t1\t39026 1453\n",
      "76\t54351\t0.3\t1\t52686 1665\n",
      "77\t92505\t0.3\t1\t89860 2645\n",
      "78\t75343\t0.3\t1\t72187 3156\n",
      "79\t21738\t0.3\t1\t20023 1715\n",
      "80\t4610\t0.3\t1\t3911 699\n",
      "81\t568\t0.3\t1\t489 79\n",
      "82\t671\t0.3\t1\t628 43\n",
      "83\t1658\t0.3\t1\t1600 58\n",
      "84\t3167\t0.3\t1\t3091 76\n",
      "85\t4906\t0.3\t1\t4798 108\n",
      "86\t6297\t0.3\t1\t6133 164\n",
      "87\t6434\t0.3\t1\t6294 140\n",
      "88\t5384\t0.3\t1\t5238 146\n",
      "89\t3918\t0.3\t1\t3818 100\n",
      "90\t3329\t0.3\t1\t3230 99\n",
      "91\t2699\t0.3\t1\t2636 63\n",
      "92\t2345\t0.3\t1\t2270 75\n",
      "93\t1871\t0.3\t1\t1826 45\n",
      "94\t1550\t0.3\t1\t1510 40\n",
      "95\t1357\t0.3\t1\t1320 37\n",
      "96\t1155\t0.3\t1\t1114 41\n",
      "97\t1122\t0.3\t1\t1067 55\n",
      "98\t992\t0.3\t1\t954 38\n",
      "99\t878\t0.3\t1\t844 34\n",
      "100\t688\t0.3\t1\t663 25\n",
      "101\t576\t0.3\t1\t546 30\n",
      "102\t495\t0.3\t1\t465 30\n",
      "103\t421\t0.3\t1\t399 22\n",
      "104\t345\t0.3\t1\t328 17\n",
      "105\t253\t0.3\t1\t231 22\n",
      "106\t262\t0.3\t1\t240 22\n",
      "107\t249\t0.3\t1\t222 27\n",
      "108\t231\t0.3\t1\t202 29\n",
      "109\t217\t0.3\t1\t202 15\n",
      "110\t163\t0.3\t1\t147 16\n",
      "111\t161\t0.3\t1\t142 19\n",
      "112\t117\t0.3\t1\t100 17\n",
      "113\t137\t0.3\t1\t119 18\n",
      "114\t106\t0.3\t1\t88 18\n",
      "115\t102\t0.3\t1\t83 19\n",
      "116\t85\t0.3\t1\t70 15\n",
      "117\t102\t0.3\t1\t81 21\n",
      "118\t79\t0.3\t1\t63 16\n",
      "119\t97\t0.3\t1\t72 25\n",
      "120\t86\t0.3\t1\t59 27\n",
      "121\t83\t0.3\t1\t60 23\n",
      "122\t72\t0.3\t1\t49 23\n",
      "123\t79\t0.3\t1\t41 38\n",
      "124\t64\t0.3\t1\t35 29\n",
      "125\t87\t0.3\t1\t47 40\n",
      "126\t47\t0.3\t1\t22 25\n",
      "127\t43\t0.3\t1\t22 21\n",
      "128\t43\t0.3\t1\t28 15\n",
      "129\t45\t0.3\t1\t23 22\n",
      "130\t20\t0.3\t1\t12 8\n",
      "131\t17\t0.3\t1\t10 7\n",
      "132\t21\t0.3\t1\t16 5\n",
      "133\t19\t0.3\t1\t11 8\n",
      "134\t14\t0.3\t1\t11 3\n",
      "135\t12\t0.3\t1\t7 5\n",
      "136\t5\t0.3\t1\t4 1\n",
      "137\t4\t0.3\t1\t4\n",
      "138\t9\t0.3\t1\t6 3\n",
      "139\t1\t0.3\t1\t1\n",
      "140\t10\t0.3\t1\t6 4\n",
      "141\t8\t0.3\t1\t7 1\n",
      "142\t6\t0.3\t1\t4 2\n",
      "143\t8\t0.3\t1\t6 2\n",
      "144\t5\t0.3\t1\t4 1\n",
      "145\t13\t0.3\t1\t8 5\n",
      "146\t29\t0.3\t1\t24 5\n",
      "147\t6\t0.3\t1\t5 1\n",
      "148\t23\t0.3\t1\t21 2\n",
      "149\t9\t0.3\t1\t6 3\n",
      "150\t6\t0.3\t1\t5 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/F.2.fastq.gz\n",
      "=============================================\n",
      "21709453 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files F.1_trimmed.fq.gz and F.2_trimmed.fq.gz\n",
      "file_1: F.1_trimmed.fq.gz, file_2: F.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: F.1_trimmed.fq.gz and F.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to F.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to F.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to F.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to F.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 21709453\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1451103 (6.68%)\n",
      "\n",
      "Deleting both intermediate output files F.1_trimmed.fq.gz and F.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/F.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/F.1.fastq.gz ./data_raw/F.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/G.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t374529\tAGATCGGAAGAGC\t1000000\t37.45\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 374529). Second best hit was Nextera (count: 1)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/G.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/G.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/G.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to G.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/G.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/G.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 655.06 s (40 us/read; 1.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,509,329\n",
      "Reads with adapters:                10,739,700 (65.1%)\n",
      "Reads written (passing filters):    16,509,329 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,476,399,350 bp\n",
      "Quality-trimmed:              80,245,440 bp (3.2%)\n",
      "Total written (filtered):  2,115,075,312 bp (85.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10739700 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.2%\n",
      "  C: 32.0%\n",
      "  G: 30.7%\n",
      "  T: 17.4%\n",
      "  none/other: 1.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2071870\t4127332.2\t0\t2071870\n",
      "2\t672277\t1031833.1\t0\t672277\n",
      "3\t282515\t257958.3\t0\t282515\n",
      "4\t147475\t64489.6\t0\t147475\n",
      "5\t106467\t16122.4\t0\t106467\n",
      "6\t91077\t4030.6\t0\t91077\n",
      "7\t92752\t1007.6\t0\t92752\n",
      "8\t112502\t251.9\t0\t112502\n",
      "9\t144065\t63.0\t0\t143898 167\n",
      "10\t158105\t15.7\t1\t146543 11562\n",
      "11\t109338\t3.9\t1\t99296 10042\n",
      "12\t113858\t1.0\t1\t104006 9852\n",
      "13\t142328\t0.2\t1\t131500 10828\n",
      "14\t103209\t0.2\t1\t94258 8951\n",
      "15\t145320\t0.2\t1\t134123 11197\n",
      "16\t74381\t0.2\t1\t67222 7159\n",
      "17\t122038\t0.2\t1\t112643 9395\n",
      "18\t108386\t0.2\t1\t100640 7746\n",
      "19\t128545\t0.2\t1\t119367 9178\n",
      "20\t98121\t0.2\t1\t90624 7497\n",
      "21\t132548\t0.2\t1\t122691 9857\n",
      "22\t137902\t0.2\t1\t127416 10486\n",
      "23\t132145\t0.2\t1\t122079 10066\n",
      "24\t129365\t0.2\t1\t118803 10562\n",
      "25\t125886\t0.2\t1\t116514 9372\n",
      "26\t103473\t0.2\t1\t95248 8225\n",
      "27\t135525\t0.2\t1\t125667 9858\n",
      "28\t123396\t0.2\t1\t114833 8563\n",
      "29\t133308\t0.2\t1\t123182 10126\n",
      "30\t118713\t0.2\t1\t110029 8684\n",
      "31\t146670\t0.2\t1\t135585 11085\n",
      "32\t148074\t0.2\t1\t137291 10783\n",
      "33\t113957\t0.2\t1\t104864 9093\n",
      "34\t157961\t0.2\t1\t147079 10882\n",
      "35\t89075\t0.2\t1\t82691 6384\n",
      "36\t129806\t0.2\t1\t119516 10290\n",
      "37\t141962\t0.2\t1\t130462 11500\n",
      "38\t140056\t0.2\t1\t128524 11532\n",
      "39\t184083\t0.2\t1\t168998 15085\n",
      "40\t132052\t0.2\t1\t121848 10204\n",
      "41\t180901\t0.2\t1\t166793 14108\n",
      "42\t329950\t0.2\t1\t305557 24393\n",
      "43\t319294\t0.2\t1\t295170 24124\n",
      "44\t235737\t0.2\t1\t217650 18087\n",
      "45\t108639\t0.2\t1\t100912 7727\n",
      "46\t56539\t0.2\t1\t52079 4460\n",
      "47\t46137\t0.2\t1\t41655 4482\n",
      "48\t73703\t0.2\t1\t68371 5332\n",
      "49\t129570\t0.2\t1\t120825 8745\n",
      "50\t30100\t0.2\t1\t27016 3084\n",
      "51\t134635\t0.2\t1\t124897 9738\n",
      "52\t100523\t0.2\t1\t93535 6988\n",
      "53\t58256\t0.2\t1\t53975 4281\n",
      "54\t66400\t0.2\t1\t61406 4994\n",
      "55\t112562\t0.2\t1\t104704 7858\n",
      "56\t70776\t0.2\t1\t65757 5019\n",
      "57\t15846\t0.2\t1\t14596 1250\n",
      "58\t53508\t0.2\t1\t49777 3731\n",
      "59\t113428\t0.2\t1\t105745 7683\n",
      "60\t15850\t0.2\t1\t14586 1264\n",
      "61\t46968\t0.2\t1\t43604 3364\n",
      "62\t65093\t0.2\t1\t60453 4640\n",
      "63\t13260\t0.2\t1\t12237 1023\n",
      "64\t36087\t0.2\t1\t33544 2543\n",
      "65\t57190\t0.2\t1\t53203 3987\n",
      "66\t22286\t0.2\t1\t20626 1660\n",
      "67\t26678\t0.2\t1\t24861 1817\n",
      "68\t32060\t0.2\t1\t29764 2296\n",
      "69\t29271\t0.2\t1\t27025 2246\n",
      "70\t28910\t0.2\t1\t26761 2149\n",
      "71\t30176\t0.2\t1\t26895 3281\n",
      "72\t41036\t0.2\t1\t34744 6292\n",
      "73\t53483\t0.2\t1\t44420 9063\n",
      "74\t65156\t0.2\t1\t50485 14671\n",
      "75\t46075\t0.2\t1\t40435 5640\n",
      "76\t3874\t0.2\t1\t3416 458\n",
      "77\t182\t0.2\t1\t155 27\n",
      "78\t1196\t0.2\t1\t1097 99\n",
      "79\t3782\t0.2\t1\t3517 265\n",
      "80\t6046\t0.2\t1\t5607 439\n",
      "81\t7127\t0.2\t1\t6539 588\n",
      "82\t7764\t0.2\t1\t7193 571\n",
      "83\t7757\t0.2\t1\t7240 517\n",
      "84\t7829\t0.2\t1\t7225 604\n",
      "85\t7481\t0.2\t1\t6885 596\n",
      "86\t7206\t0.2\t1\t6701 505\n",
      "87\t6863\t0.2\t1\t6375 488\n",
      "88\t6239\t0.2\t1\t5792 447\n",
      "89\t5845\t0.2\t1\t5442 403\n",
      "90\t5296\t0.2\t1\t4890 406\n",
      "91\t4836\t0.2\t1\t4467 369\n",
      "92\t4347\t0.2\t1\t3971 376\n",
      "93\t3712\t0.2\t1\t3417 295\n",
      "94\t3352\t0.2\t1\t3101 251\n",
      "95\t3045\t0.2\t1\t2789 256\n",
      "96\t2958\t0.2\t1\t2739 219\n",
      "97\t2726\t0.2\t1\t2540 186\n",
      "98\t2686\t0.2\t1\t2472 214\n",
      "99\t2458\t0.2\t1\t2279 179\n",
      "100\t2246\t0.2\t1\t2070 176\n",
      "101\t2032\t0.2\t1\t1855 177\n",
      "102\t1623\t0.2\t1\t1499 124\n",
      "103\t1362\t0.2\t1\t1252 110\n",
      "104\t1089\t0.2\t1\t1002 87\n",
      "105\t953\t0.2\t1\t853 100\n",
      "106\t929\t0.2\t1\t838 91\n",
      "107\t905\t0.2\t1\t828 77\n",
      "108\t790\t0.2\t1\t703 87\n",
      "109\t738\t0.2\t1\t674 64\n",
      "110\t626\t0.2\t1\t566 60\n",
      "111\t530\t0.2\t1\t479 51\n",
      "112\t467\t0.2\t1\t419 48\n",
      "113\t449\t0.2\t1\t393 56\n",
      "114\t337\t0.2\t1\t302 35\n",
      "115\t328\t0.2\t1\t287 41\n",
      "116\t244\t0.2\t1\t215 29\n",
      "117\t213\t0.2\t1\t194 19\n",
      "118\t208\t0.2\t1\t176 32\n",
      "119\t210\t0.2\t1\t175 35\n",
      "120\t168\t0.2\t1\t136 32\n",
      "121\t165\t0.2\t1\t128 37\n",
      "122\t119\t0.2\t1\t94 25\n",
      "123\t113\t0.2\t1\t89 24\n",
      "124\t95\t0.2\t1\t65 30\n",
      "125\t111\t0.2\t1\t84 27\n",
      "126\t56\t0.2\t1\t40 16\n",
      "127\t57\t0.2\t1\t39 18\n",
      "128\t49\t0.2\t1\t36 13\n",
      "129\t62\t0.2\t1\t42 20\n",
      "130\t55\t0.2\t1\t36 19\n",
      "131\t49\t0.2\t1\t24 25\n",
      "132\t31\t0.2\t1\t22 9\n",
      "133\t42\t0.2\t1\t34 8\n",
      "134\t41\t0.2\t1\t36 5\n",
      "135\t31\t0.2\t1\t26 5\n",
      "136\t20\t0.2\t1\t16 4\n",
      "137\t30\t0.2\t1\t21 9\n",
      "138\t21\t0.2\t1\t17 4\n",
      "139\t27\t0.2\t1\t21 6\n",
      "140\t35\t0.2\t1\t32 3\n",
      "141\t41\t0.2\t1\t39 2\n",
      "142\t30\t0.2\t1\t26 4\n",
      "143\t25\t0.2\t1\t20 5\n",
      "144\t19\t0.2\t1\t16 3\n",
      "145\t38\t0.2\t1\t34 4\n",
      "146\t57\t0.2\t1\t53 4\n",
      "147\t42\t0.2\t1\t40 2\n",
      "148\t100\t0.2\t1\t93 7\n",
      "149\t35\t0.2\t1\t24 11\n",
      "150\t80381\t0.2\t1\t29 80352\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/G.1.fastq.gz\n",
      "=============================================\n",
      "16509329 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/G.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/G.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to G.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/G.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/G.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 646.62 s (39 us/read; 1.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,509,329\n",
      "Reads with adapters:                10,198,697 (61.8%)\n",
      "Reads written (passing filters):    16,509,329 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,476,399,350 bp\n",
      "Quality-trimmed:             246,746,747 bp (10.0%)\n",
      "Total written (filtered):  1,982,736,446 bp (80.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10198697 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.2%\n",
      "  C: 32.5%\n",
      "  G: 36.1%\n",
      "  T: 13.0%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2251770\t4127332.2\t0\t2251770\n",
      "2\t569552\t1031833.1\t0\t569552\n",
      "3\t333502\t257958.3\t0\t333502\n",
      "4\t140762\t64489.6\t0\t140762\n",
      "5\t140358\t16122.4\t0\t140358\n",
      "6\t84117\t4030.6\t0\t84117\n",
      "7\t87730\t1007.6\t0\t87730\n",
      "8\t111598\t251.9\t0\t111598\n",
      "9\t150561\t63.0\t0\t150398 163\n",
      "10\t137998\t15.7\t1\t131748 6250\n",
      "11\t93907\t3.9\t1\t85559 8348\n",
      "12\t81077\t1.0\t1\t74660 6417\n",
      "13\t117391\t0.2\t1\t110672 6719\n",
      "14\t106906\t0.2\t1\t101512 5394\n",
      "15\t96511\t0.2\t1\t91727 4784\n",
      "16\t127511\t0.2\t1\t119513 7998\n",
      "17\t136744\t0.2\t1\t129530 7214\n",
      "18\t161179\t0.2\t1\t153644 7535\n",
      "19\t114644\t0.2\t1\t108692 5952\n",
      "20\t164919\t0.2\t1\t157968 6951\n",
      "21\t73398\t0.2\t1\t68463 4935\n",
      "22\t74805\t0.2\t1\t72126 2679\n",
      "23\t95192\t0.2\t1\t91464 3728\n",
      "24\t125063\t0.2\t1\t119988 5075\n",
      "25\t121346\t0.2\t1\t115973 5373\n",
      "26\t124930\t0.2\t1\t119074 5856\n",
      "27\t196867\t0.2\t1\t187077 9790\n",
      "28\t71454\t0.2\t1\t68084 3370\n",
      "29\t165341\t0.2\t1\t154631 10710\n",
      "30\t100376\t0.2\t1\t96270 4106\n",
      "31\t168485\t0.2\t1\t162439 6046\n",
      "32\t70505\t0.2\t1\t67171 3334\n",
      "33\t117662\t0.2\t1\t112643 5019\n",
      "34\t230647\t0.2\t1\t223195 7452\n",
      "35\t55504\t0.2\t1\t52333 3171\n",
      "36\t147924\t0.2\t1\t141792 6132\n",
      "37\t48645\t0.2\t1\t46545 2100\n",
      "38\t134195\t0.2\t1\t126542 7653\n",
      "39\t129527\t0.2\t1\t124021 5506\n",
      "40\t248281\t0.2\t1\t240548 7733\n",
      "41\t67717\t0.2\t1\t63574 4143\n",
      "42\t52778\t0.2\t1\t50525 2253\n",
      "43\t152493\t0.2\t1\t148301 4192\n",
      "44\t44361\t0.2\t1\t42275 2086\n",
      "45\t189530\t0.2\t1\t184586 4944\n",
      "46\t109618\t0.2\t1\t105457 4161\n",
      "47\t77937\t0.2\t1\t75707 2230\n",
      "48\t99048\t0.2\t1\t94137 4911\n",
      "49\t156407\t0.2\t1\t151501 4906\n",
      "50\t36919\t0.2\t1\t34882 2037\n",
      "51\t113554\t0.2\t1\t110055 3499\n",
      "52\t45845\t0.2\t1\t44027 1818\n",
      "53\t79185\t0.2\t1\t77315 1870\n",
      "54\t87554\t0.2\t1\t85548 2006\n",
      "55\t85732\t0.2\t1\t83294 2438\n",
      "56\t65555\t0.2\t1\t63849 1706\n",
      "57\t76279\t0.2\t1\t74271 2008\n",
      "58\t102121\t0.2\t1\t99449 2672\n",
      "59\t38819\t0.2\t1\t37419 1400\n",
      "60\t73198\t0.2\t1\t71602 1596\n",
      "61\t41994\t0.2\t1\t40634 1360\n",
      "62\t46117\t0.2\t1\t44604 1513\n",
      "63\t58563\t0.2\t1\t57237 1326\n",
      "64\t59078\t0.2\t1\t57592 1486\n",
      "65\t24356\t0.2\t1\t23310 1046\n",
      "66\t55790\t0.2\t1\t54498 1292\n",
      "67\t11422\t0.2\t1\t10880 542\n",
      "68\t40836\t0.2\t1\t39494 1342\n",
      "69\t34378\t0.2\t1\t33328 1050\n",
      "70\t68204\t0.2\t1\t66610 1594\n",
      "71\t11961\t0.2\t1\t11551 410\n",
      "72\t20428\t0.2\t1\t19782 646\n",
      "73\t21386\t0.2\t1\t20765 621\n",
      "74\t21556\t0.2\t1\t20895 661\n",
      "75\t23240\t0.2\t1\t22571 669\n",
      "76\t30812\t0.2\t1\t29979 833\n",
      "77\t55483\t0.2\t1\t54159 1324\n",
      "78\t52961\t0.2\t1\t51125 1836\n",
      "79\t17424\t0.2\t1\t16244 1180\n",
      "80\t4363\t0.2\t1\t3791 572\n",
      "81\t370\t0.2\t1\t309 61\n",
      "82\t375\t0.2\t1\t358 17\n",
      "83\t788\t0.2\t1\t761 27\n",
      "84\t1523\t0.2\t1\t1493 30\n",
      "85\t2403\t0.2\t1\t2365 38\n",
      "86\t3316\t0.2\t1\t3258 58\n",
      "87\t3687\t0.2\t1\t3616 71\n",
      "88\t3318\t0.2\t1\t3252 66\n",
      "89\t2502\t0.2\t1\t2445 57\n",
      "90\t2112\t0.2\t1\t2061 51\n",
      "91\t1854\t0.2\t1\t1794 60\n",
      "92\t1637\t0.2\t1\t1610 27\n",
      "93\t1380\t0.2\t1\t1338 42\n",
      "94\t1139\t0.2\t1\t1106 33\n",
      "95\t950\t0.2\t1\t906 44\n",
      "96\t924\t0.2\t1\t895 29\n",
      "97\t856\t0.2\t1\t826 30\n",
      "98\t799\t0.2\t1\t775 24\n",
      "99\t666\t0.2\t1\t642 24\n",
      "100\t621\t0.2\t1\t601 20\n",
      "101\t511\t0.2\t1\t489 22\n",
      "102\t406\t0.2\t1\t387 19\n",
      "103\t307\t0.2\t1\t290 17\n",
      "104\t268\t0.2\t1\t259 9\n",
      "105\t196\t0.2\t1\t186 10\n",
      "106\t221\t0.2\t1\t205 16\n",
      "107\t156\t0.2\t1\t143 13\n",
      "108\t160\t0.2\t1\t153 7\n",
      "109\t167\t0.2\t1\t153 14\n",
      "110\t140\t0.2\t1\t127 13\n",
      "111\t104\t0.2\t1\t91 13\n",
      "112\t86\t0.2\t1\t81 5\n",
      "113\t79\t0.2\t1\t68 11\n",
      "114\t81\t0.2\t1\t69 12\n",
      "115\t64\t0.2\t1\t58 6\n",
      "116\t53\t0.2\t1\t42 11\n",
      "117\t51\t0.2\t1\t43 8\n",
      "118\t53\t0.2\t1\t45 8\n",
      "119\t60\t0.2\t1\t45 15\n",
      "120\t44\t0.2\t1\t37 7\n",
      "121\t47\t0.2\t1\t32 15\n",
      "122\t27\t0.2\t1\t17 10\n",
      "123\t29\t0.2\t1\t23 6\n",
      "124\t37\t0.2\t1\t30 7\n",
      "125\t32\t0.2\t1\t24 8\n",
      "126\t26\t0.2\t1\t23 3\n",
      "127\t27\t0.2\t1\t13 14\n",
      "128\t19\t0.2\t1\t11 8\n",
      "129\t12\t0.2\t1\t7 5\n",
      "130\t8\t0.2\t1\t6 2\n",
      "131\t14\t0.2\t1\t8 6\n",
      "132\t9\t0.2\t1\t6 3\n",
      "133\t10\t0.2\t1\t9 1\n",
      "134\t5\t0.2\t1\t3 2\n",
      "135\t5\t0.2\t1\t2 3\n",
      "136\t4\t0.2\t1\t3 1\n",
      "137\t5\t0.2\t1\t3 2\n",
      "138\t4\t0.2\t1\t3 1\n",
      "139\t2\t0.2\t1\t2\n",
      "140\t6\t0.2\t1\t4 2\n",
      "141\t4\t0.2\t1\t3 1\n",
      "142\t3\t0.2\t1\t2 1\n",
      "143\t3\t0.2\t1\t1 2\n",
      "144\t2\t0.2\t1\t0 2\n",
      "145\t4\t0.2\t1\t3 1\n",
      "146\t13\t0.2\t1\t11 2\n",
      "147\t8\t0.2\t1\t7 1\n",
      "148\t12\t0.2\t1\t10 2\n",
      "149\t5\t0.2\t1\t2 3\n",
      "150\t2\t0.2\t1\t2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/G.2.fastq.gz\n",
      "=============================================\n",
      "16509329 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files G.1_trimmed.fq.gz and G.2_trimmed.fq.gz\n",
      "file_1: G.1_trimmed.fq.gz, file_2: G.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: G.1_trimmed.fq.gz and G.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to G.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to G.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to G.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to G.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16509329\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1084101 (6.57%)\n",
      "\n",
      "Deleting both intermediate output files G.1_trimmed.fq.gz and G.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/G.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/G.1.fastq.gz ./data_raw/G.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/H.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t371397\tAGATCGGAAGAGC\t1000000\t37.14\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 371397). Second best hit was Nextera (count: 1)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/H.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/H.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/H.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to H.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/H.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/H.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 626.22 s (40 us/read; 1.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,816,920\n",
      "Reads with adapters:                10,158,763 (64.2%)\n",
      "Reads written (passing filters):    15,816,920 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,372,538,000 bp\n",
      "Quality-trimmed:              79,761,885 bp (3.4%)\n",
      "Total written (filtered):  2,021,355,088 bp (85.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10158763 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.9%\n",
      "  C: 32.7%\n",
      "  G: 30.8%\n",
      "  T: 17.0%\n",
      "  none/other: 1.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1935150\t3954230.0\t0\t1935150\n",
      "2\t652037\t988557.5\t0\t652037\n",
      "3\t273574\t247139.4\t0\t273574\n",
      "4\t140588\t61784.8\t0\t140588\n",
      "5\t100070\t15446.2\t0\t100070\n",
      "6\t83921\t3861.6\t0\t83921\n",
      "7\t83421\t965.4\t0\t83421\n",
      "8\t100903\t241.3\t0\t100903\n",
      "9\t131930\t60.3\t0\t131714 216\n",
      "10\t144652\t15.1\t1\t134415 10237\n",
      "11\t101352\t3.8\t1\t92331 9021\n",
      "12\t103894\t0.9\t1\t94752 9142\n",
      "13\t127580\t0.2\t1\t117911 9669\n",
      "14\t92448\t0.2\t1\t84782 7666\n",
      "15\t131256\t0.2\t1\t121860 9396\n",
      "16\t67552\t0.2\t1\t61177 6375\n",
      "17\t110843\t0.2\t1\t102383 8460\n",
      "18\t98943\t0.2\t1\t92151 6792\n",
      "19\t116649\t0.2\t1\t108293 8356\n",
      "20\t91930\t0.2\t1\t85260 6670\n",
      "21\t122404\t0.2\t1\t113570 8834\n",
      "22\t128344\t0.2\t1\t118690 9654\n",
      "23\t121181\t0.2\t1\t112242 8939\n",
      "24\t118769\t0.2\t1\t109311 9458\n",
      "25\t116689\t0.2\t1\t108313 8376\n",
      "26\t95511\t0.2\t1\t88137 7374\n",
      "27\t123640\t0.2\t1\t114704 8936\n",
      "28\t113388\t0.2\t1\t105786 7602\n",
      "29\t123763\t0.2\t1\t114714 9049\n",
      "30\t109380\t0.2\t1\t101502 7878\n",
      "31\t137698\t0.2\t1\t127571 10127\n",
      "32\t138331\t0.2\t1\t128512 9819\n",
      "33\t105369\t0.2\t1\t97172 8197\n",
      "34\t145351\t0.2\t1\t135380 9971\n",
      "35\t80189\t0.2\t1\t74528 5661\n",
      "36\t120535\t0.2\t1\t111642 8893\n",
      "37\t134021\t0.2\t1\t123374 10647\n",
      "38\t133789\t0.2\t1\t123350 10439\n",
      "39\t176701\t0.2\t1\t162736 13965\n",
      "40\t128590\t0.2\t1\t118997 9593\n",
      "41\t176056\t0.2\t1\t162838 13218\n",
      "42\t313184\t0.2\t1\t290936 22248\n",
      "43\t305997\t0.2\t1\t283870 22127\n",
      "44\t222840\t0.2\t1\t206529 16311\n",
      "45\t107336\t0.2\t1\t100184 7152\n",
      "46\t54810\t0.2\t1\t50916 3894\n",
      "47\t45087\t0.2\t1\t40794 4293\n",
      "48\t70314\t0.2\t1\t65675 4639\n",
      "49\t122752\t0.2\t1\t115288 7464\n",
      "50\t32819\t0.2\t1\t29591 3228\n",
      "51\t132685\t0.2\t1\t123650 9035\n",
      "52\t98290\t0.2\t1\t91740 6550\n",
      "53\t58723\t0.2\t1\t54594 4129\n",
      "54\t66035\t0.2\t1\t61449 4586\n",
      "55\t109299\t0.2\t1\t102033 7266\n",
      "56\t68166\t0.2\t1\t63878 4288\n",
      "57\t16954\t0.2\t1\t15677 1277\n",
      "58\t52368\t0.2\t1\t49069 3299\n",
      "59\t110105\t0.2\t1\t103176 6929\n",
      "60\t16788\t0.2\t1\t15485 1303\n",
      "61\t45773\t0.2\t1\t42845 2928\n",
      "62\t64222\t0.2\t1\t60111 4111\n",
      "63\t13855\t0.2\t1\t12845 1010\n",
      "64\t36222\t0.2\t1\t33892 2330\n",
      "65\t57650\t0.2\t1\t54026 3624\n",
      "66\t22961\t0.2\t1\t21342 1619\n",
      "67\t27204\t0.2\t1\t25392 1812\n",
      "68\t32386\t0.2\t1\t30241 2145\n",
      "69\t29001\t0.2\t1\t26948 2053\n",
      "70\t28771\t0.2\t1\t26637 2134\n",
      "71\t30857\t0.2\t1\t27312 3545\n",
      "72\t42409\t0.2\t1\t35257 7152\n",
      "73\t55547\t0.2\t1\t45383 10164\n",
      "74\t67932\t0.2\t1\t51063 16869\n",
      "75\t46803\t0.2\t1\t40696 6107\n",
      "76\t4181\t0.2\t1\t3755 426\n",
      "77\t219\t0.2\t1\t187 32\n",
      "78\t1105\t0.2\t1\t1023 82\n",
      "79\t3612\t0.2\t1\t3366 246\n",
      "80\t5779\t0.2\t1\t5422 357\n",
      "81\t6932\t0.2\t1\t6477 455\n",
      "82\t7631\t0.2\t1\t7137 494\n",
      "83\t7497\t0.2\t1\t6979 518\n",
      "84\t7412\t0.2\t1\t6864 548\n",
      "85\t7112\t0.2\t1\t6651 461\n",
      "86\t6892\t0.2\t1\t6422 470\n",
      "87\t6533\t0.2\t1\t6115 418\n",
      "88\t5846\t0.2\t1\t5401 445\n",
      "89\t5477\t0.2\t1\t5119 358\n",
      "90\t5033\t0.2\t1\t4692 341\n",
      "91\t4384\t0.2\t1\t4113 271\n",
      "92\t3926\t0.2\t1\t3634 292\n",
      "93\t3399\t0.2\t1\t3181 218\n",
      "94\t2868\t0.2\t1\t2664 204\n",
      "95\t2670\t0.2\t1\t2472 198\n",
      "96\t2538\t0.2\t1\t2341 197\n",
      "97\t2359\t0.2\t1\t2214 145\n",
      "98\t2220\t0.2\t1\t2071 149\n",
      "99\t2046\t0.2\t1\t1920 126\n",
      "100\t1825\t0.2\t1\t1695 130\n",
      "101\t1612\t0.2\t1\t1495 117\n",
      "102\t1259\t0.2\t1\t1157 102\n",
      "103\t1080\t0.2\t1\t1003 77\n",
      "104\t882\t0.2\t1\t813 69\n",
      "105\t794\t0.2\t1\t736 58\n",
      "106\t650\t0.2\t1\t586 64\n",
      "107\t670\t0.2\t1\t586 84\n",
      "108\t585\t0.2\t1\t516 69\n",
      "109\t571\t0.2\t1\t525 46\n",
      "110\t467\t0.2\t1\t422 45\n",
      "111\t378\t0.2\t1\t346 32\n",
      "112\t324\t0.2\t1\t273 51\n",
      "113\t334\t0.2\t1\t293 41\n",
      "114\t246\t0.2\t1\t201 45\n",
      "115\t251\t0.2\t1\t208 43\n",
      "116\t185\t0.2\t1\t150 35\n",
      "117\t186\t0.2\t1\t155 31\n",
      "118\t169\t0.2\t1\t141 28\n",
      "119\t158\t0.2\t1\t127 31\n",
      "120\t132\t0.2\t1\t99 33\n",
      "121\t145\t0.2\t1\t102 43\n",
      "122\t112\t0.2\t1\t83 29\n",
      "123\t143\t0.2\t1\t108 35\n",
      "124\t89\t0.2\t1\t69 20\n",
      "125\t129\t0.2\t1\t78 51\n",
      "126\t84\t0.2\t1\t49 35\n",
      "127\t99\t0.2\t1\t76 23\n",
      "128\t61\t0.2\t1\t37 24\n",
      "129\t68\t0.2\t1\t48 20\n",
      "130\t46\t0.2\t1\t32 14\n",
      "131\t56\t0.2\t1\t34 22\n",
      "132\t33\t0.2\t1\t25 8\n",
      "133\t35\t0.2\t1\t29 6\n",
      "134\t44\t0.2\t1\t38 6\n",
      "135\t33\t0.2\t1\t28 5\n",
      "136\t24\t0.2\t1\t17 7\n",
      "137\t26\t0.2\t1\t23 3\n",
      "138\t28\t0.2\t1\t22 6\n",
      "139\t23\t0.2\t1\t18 5\n",
      "140\t18\t0.2\t1\t12 6\n",
      "141\t29\t0.2\t1\t28 1\n",
      "142\t38\t0.2\t1\t20 18\n",
      "143\t21\t0.2\t1\t20 1\n",
      "144\t13\t0.2\t1\t12 1\n",
      "145\t74\t0.2\t1\t66 8\n",
      "146\t71\t0.2\t1\t71\n",
      "147\t56\t0.2\t1\t53 3\n",
      "148\t116\t0.2\t1\t111 5\n",
      "149\t42\t0.2\t1\t37 5\n",
      "150\t92041\t0.2\t1\t25 92016\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/H.1.fastq.gz\n",
      "=============================================\n",
      "15816920 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/H.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/H.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to H.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/H.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/H.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 623.06 s (39 us/read; 1.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,816,920\n",
      "Reads with adapters:                 9,733,749 (61.5%)\n",
      "Reads written (passing filters):    15,816,920 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,372,538,000 bp\n",
      "Quality-trimmed:             240,255,539 bp (10.1%)\n",
      "Total written (filtered):  1,907,564,981 bp (80.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9733749 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.0%\n",
      "  C: 33.2%\n",
      "  G: 36.2%\n",
      "  T: 12.3%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2139221\t3954230.0\t0\t2139221\n",
      "2\t607711\t988557.5\t0\t607711\n",
      "3\t302497\t247139.4\t0\t302497\n",
      "4\t122862\t61784.8\t0\t122862\n",
      "5\t129348\t15446.2\t0\t129348\n",
      "6\t76025\t3861.6\t0\t76025\n",
      "7\t79764\t965.4\t0\t79764\n",
      "8\t107718\t241.3\t0\t107718\n",
      "9\t139400\t60.3\t0\t139065 335\n",
      "10\t145403\t15.1\t1\t138058 7345\n",
      "11\t76632\t3.8\t1\t68557 8075\n",
      "12\t77238\t0.9\t1\t70629 6609\n",
      "13\t102700\t0.2\t1\t95694 7006\n",
      "14\t103382\t0.2\t1\t97408 5974\n",
      "15\t89259\t0.2\t1\t84075 5184\n",
      "16\t116185\t0.2\t1\t107638 8547\n",
      "17\t133461\t0.2\t1\t125042 8419\n",
      "18\t144399\t0.2\t1\t136511 7888\n",
      "19\t121763\t0.2\t1\t114438 7325\n",
      "20\t135453\t0.2\t1\t129044 6409\n",
      "21\t65026\t0.2\t1\t59670 5356\n",
      "22\t83194\t0.2\t1\t79640 3554\n",
      "23\t89811\t0.2\t1\t85297 4514\n",
      "24\t136394\t0.2\t1\t129319 7075\n",
      "25\t91787\t0.2\t1\t86781 5006\n",
      "26\t130497\t0.2\t1\t123002 7495\n",
      "27\t199342\t0.2\t1\t187390 11952\n",
      "28\t116171\t0.2\t1\t110370 5801\n",
      "29\t126149\t0.2\t1\t114984 11165\n",
      "30\t116974\t0.2\t1\t111144 5830\n",
      "31\t144456\t0.2\t1\t137924 6532\n",
      "32\t87922\t0.2\t1\t83022 4900\n",
      "33\t108923\t0.2\t1\t103408 5515\n",
      "34\t236788\t0.2\t1\t228006 8782\n",
      "35\t71978\t0.2\t1\t68377 3601\n",
      "36\t144391\t0.2\t1\t136708 7683\n",
      "37\t173413\t0.2\t1\t168186 5227\n",
      "38\t58174\t0.2\t1\t53211 4963\n",
      "39\t86134\t0.2\t1\t82489 3645\n",
      "40\t114913\t0.2\t1\t109842 5071\n",
      "41\t163262\t0.2\t1\t156684 6578\n",
      "42\t176571\t0.2\t1\t171217 5354\n",
      "43\t133993\t0.2\t1\t128895 5098\n",
      "44\t461709\t0.2\t1\t452722 8987\n",
      "45\t147851\t0.2\t1\t142549 5302\n",
      "46\t95528\t0.2\t1\t90847 4681\n",
      "47\t109130\t0.2\t1\t105851 3279\n",
      "48\t47094\t0.2\t1\t43822 3272\n",
      "49\t85407\t0.2\t1\t82225 3182\n",
      "50\t30766\t0.2\t1\t29259 1507\n",
      "51\t49818\t0.2\t1\t47881 1937\n",
      "52\t25806\t0.2\t1\t24587 1219\n",
      "53\t35885\t0.2\t1\t34722 1163\n",
      "54\t76124\t0.2\t1\t74196 1928\n",
      "55\t33839\t0.2\t1\t32581 1258\n",
      "56\t44245\t0.2\t1\t42903 1342\n",
      "57\t115145\t0.2\t1\t112566 2579\n",
      "58\t50413\t0.2\t1\t48723 1690\n",
      "59\t20015\t0.2\t1\t19138 877\n",
      "60\t58851\t0.2\t1\t57515 1336\n",
      "61\t22077\t0.2\t1\t21197 880\n",
      "62\t32042\t0.2\t1\t30767 1275\n",
      "63\t55458\t0.2\t1\t54131 1327\n",
      "64\t23882\t0.2\t1\t23023 859\n",
      "65\t12521\t0.2\t1\t11872 649\n",
      "66\t27315\t0.2\t1\t26475 840\n",
      "67\t7301\t0.2\t1\t6966 335\n",
      "68\t19428\t0.2\t1\t18591 837\n",
      "69\t15735\t0.2\t1\t15264 471\n",
      "70\t36597\t0.2\t1\t35673 924\n",
      "71\t5132\t0.2\t1\t4891 241\n",
      "72\t9998\t0.2\t1\t9606 392\n",
      "73\t12405\t0.2\t1\t11971 434\n",
      "74\t13682\t0.2\t1\t13240 442\n",
      "75\t15150\t0.2\t1\t14652 498\n",
      "76\t18884\t0.2\t1\t18309 575\n",
      "77\t33122\t0.2\t1\t32242 880\n",
      "78\t36713\t0.2\t1\t35454 1259\n",
      "79\t13021\t0.2\t1\t12176 845\n",
      "80\t2756\t0.2\t1\t2454 302\n",
      "81\t306\t0.2\t1\t263 43\n",
      "82\t343\t0.2\t1\t324 19\n",
      "83\t714\t0.2\t1\t692 22\n",
      "84\t1467\t0.2\t1\t1431 36\n",
      "85\t2264\t0.2\t1\t2215 49\n",
      "86\t2978\t0.2\t1\t2934 44\n",
      "87\t3028\t0.2\t1\t2962 66\n",
      "88\t2563\t0.2\t1\t2508 55\n",
      "89\t1745\t0.2\t1\t1699 46\n",
      "90\t1539\t0.2\t1\t1494 45\n",
      "91\t1237\t0.2\t1\t1210 27\n",
      "92\t1077\t0.2\t1\t1059 18\n",
      "93\t875\t0.2\t1\t855 20\n",
      "94\t688\t0.2\t1\t668 20\n",
      "95\t632\t0.2\t1\t614 18\n",
      "96\t561\t0.2\t1\t537 24\n",
      "97\t511\t0.2\t1\t500 11\n",
      "98\t473\t0.2\t1\t464 9\n",
      "99\t420\t0.2\t1\t403 17\n",
      "100\t353\t0.2\t1\t336 17\n",
      "101\t299\t0.2\t1\t291 8\n",
      "102\t206\t0.2\t1\t198 8\n",
      "103\t170\t0.2\t1\t161 9\n",
      "104\t160\t0.2\t1\t149 11\n",
      "105\t148\t0.2\t1\t134 14\n",
      "106\t116\t0.2\t1\t111 5\n",
      "107\t117\t0.2\t1\t104 13\n",
      "108\t100\t0.2\t1\t94 6\n",
      "109\t100\t0.2\t1\t85 15\n",
      "110\t87\t0.2\t1\t78 9\n",
      "111\t66\t0.2\t1\t58 8\n",
      "112\t68\t0.2\t1\t65 3\n",
      "113\t62\t0.2\t1\t53 9\n",
      "114\t56\t0.2\t1\t45 11\n",
      "115\t49\t0.2\t1\t36 13\n",
      "116\t37\t0.2\t1\t28 9\n",
      "117\t41\t0.2\t1\t32 9\n",
      "118\t48\t0.2\t1\t29 19\n",
      "119\t44\t0.2\t1\t27 17\n",
      "120\t41\t0.2\t1\t30 11\n",
      "121\t46\t0.2\t1\t35 11\n",
      "122\t30\t0.2\t1\t21 9\n",
      "123\t57\t0.2\t1\t34 23\n",
      "124\t26\t0.2\t1\t17 9\n",
      "125\t44\t0.2\t1\t24 20\n",
      "126\t38\t0.2\t1\t19 19\n",
      "127\t32\t0.2\t1\t17 15\n",
      "128\t18\t0.2\t1\t12 6\n",
      "129\t10\t0.2\t1\t5 5\n",
      "130\t14\t0.2\t1\t11 3\n",
      "131\t9\t0.2\t1\t8 1\n",
      "132\t5\t0.2\t1\t3 2\n",
      "133\t11\t0.2\t1\t7 4\n",
      "134\t9\t0.2\t1\t7 2\n",
      "135\t6\t0.2\t1\t4 2\n",
      "136\t4\t0.2\t1\t1 3\n",
      "137\t2\t0.2\t1\t1 1\n",
      "138\t9\t0.2\t1\t7 2\n",
      "139\t6\t0.2\t1\t4 2\n",
      "141\t2\t0.2\t1\t0 2\n",
      "142\t3\t0.2\t1\t1 2\n",
      "143\t2\t0.2\t1\t2\n",
      "144\t1\t0.2\t1\t1\n",
      "145\t7\t0.2\t1\t6 1\n",
      "146\t16\t0.2\t1\t15 1\n",
      "147\t6\t0.2\t1\t6\n",
      "148\t10\t0.2\t1\t8 2\n",
      "149\t5\t0.2\t1\t5\n",
      "150\t3\t0.2\t1\t1 2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/H.2.fastq.gz\n",
      "=============================================\n",
      "15816920 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files H.1_trimmed.fq.gz and H.2_trimmed.fq.gz\n",
      "file_1: H.1_trimmed.fq.gz, file_2: H.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: H.1_trimmed.fq.gz and H.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to H.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to H.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to H.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to H.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 15816920\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 977396 (6.18%)\n",
      "\n",
      "Deleting both intermediate output files H.1_trimmed.fq.gz and H.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/H.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/H.1.fastq.gz ./data_raw/H.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/I.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t403869\tAGATCGGAAGAGC\t1000000\t40.39\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 403869). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/I.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/I.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/I.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to I.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/I.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/I.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 851.47 s (40 us/read; 1.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,348,992\n",
      "Reads with adapters:                14,084,620 (66.0%)\n",
      "Reads written (passing filters):    21,348,992 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,202,348,800 bp\n",
      "Quality-trimmed:              79,598,046 bp (2.5%)\n",
      "Total written (filtered):  2,718,905,223 bp (84.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 14084620 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.3%\n",
      "  C: 32.1%\n",
      "  G: 34.0%\n",
      "  T: 17.2%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2470427\t5337248.0\t0\t2470427\n",
      "2\t849096\t1334312.0\t0\t849096\n",
      "3\t355963\t333578.0\t0\t355963\n",
      "4\t184745\t83394.5\t0\t184745\n",
      "5\t133042\t20848.6\t0\t133042\n",
      "6\t109925\t5212.2\t0\t109925\n",
      "7\t109312\t1303.0\t0\t109312\n",
      "8\t132984\t325.8\t0\t132984\n",
      "9\t170722\t81.4\t0\t170500 222\n",
      "10\t188750\t20.4\t1\t175864 12886\n",
      "11\t130899\t5.1\t1\t119539 11360\n",
      "12\t132972\t1.3\t1\t122444 10528\n",
      "13\t167325\t0.3\t1\t155444 11881\n",
      "14\t121306\t0.3\t1\t111556 9750\n",
      "15\t172684\t0.3\t1\t160624 12060\n",
      "16\t92330\t0.3\t1\t83925 8405\n",
      "17\t146338\t0.3\t1\t135342 10996\n",
      "18\t129393\t0.3\t1\t120875 8518\n",
      "19\t152851\t0.3\t1\t142557 10294\n",
      "20\t120704\t0.3\t1\t112206 8498\n",
      "21\t159765\t0.3\t1\t148532 11233\n",
      "22\t162322\t0.3\t1\t151020 11302\n",
      "23\t161608\t0.3\t1\t150102 11506\n",
      "24\t165868\t0.3\t1\t152547 13321\n",
      "25\t158606\t0.3\t1\t147638 10968\n",
      "26\t145104\t0.3\t1\t134273 10831\n",
      "27\t175926\t0.3\t1\t163625 12301\n",
      "28\t137913\t0.3\t1\t128817 9096\n",
      "29\t197221\t0.3\t1\t183091 14130\n",
      "30\t168136\t0.3\t1\t156635 11501\n",
      "31\t222481\t0.3\t1\t207049 15432\n",
      "32\t133685\t0.3\t1\t124332 9353\n",
      "33\t179323\t0.3\t1\t165054 14269\n",
      "34\t212635\t0.3\t1\t198196 14439\n",
      "35\t108339\t0.3\t1\t100844 7495\n",
      "36\t145389\t0.3\t1\t135198 10191\n",
      "37\t155422\t0.3\t1\t143903 11519\n",
      "38\t142446\t0.3\t1\t131903 10543\n",
      "39\t171681\t0.3\t1\t158793 12888\n",
      "40\t240299\t0.3\t1\t222397 17902\n",
      "41\t328668\t0.3\t1\t306136 22532\n",
      "42\t409303\t0.3\t1\t382629 26674\n",
      "43\t209251\t0.3\t1\t195285 13966\n",
      "44\t75974\t0.3\t1\t70962 5012\n",
      "45\t23568\t0.3\t1\t21939 1629\n",
      "46\t45166\t0.3\t1\t42103 3063\n",
      "47\t89005\t0.3\t1\t83423 5582\n",
      "48\t122802\t0.3\t1\t115309 7493\n",
      "49\t226946\t0.3\t1\t213494 13452\n",
      "50\t65340\t0.3\t1\t60548 4792\n",
      "51\t183059\t0.3\t1\t170615 12444\n",
      "52\t197442\t0.3\t1\t185198 12244\n",
      "53\t142590\t0.3\t1\t133552 9038\n",
      "54\t116185\t0.3\t1\t108743 7442\n",
      "55\t179366\t0.3\t1\t168362 11004\n",
      "56\t150216\t0.3\t1\t141333 8883\n",
      "57\t56670\t0.3\t1\t52905 3765\n",
      "58\t125093\t0.3\t1\t117526 7567\n",
      "59\t208846\t0.3\t1\t196080 12766\n",
      "60\t51005\t0.3\t1\t47683 3322\n",
      "61\t107327\t0.3\t1\t100643 6684\n",
      "62\t152502\t0.3\t1\t143126 9376\n",
      "63\t44540\t0.3\t1\t41660 2880\n",
      "64\t95840\t0.3\t1\t89813 6027\n",
      "65\t122101\t0.3\t1\t114537 7564\n",
      "66\t66579\t0.3\t1\t62342 4237\n",
      "67\t82125\t0.3\t1\t76959 5166\n",
      "68\t81308\t0.3\t1\t76309 4999\n",
      "69\t72562\t0.3\t1\t67854 4708\n",
      "70\t70157\t0.3\t1\t65531 4626\n",
      "71\t73916\t0.3\t1\t68153 5763\n",
      "72\t126346\t0.3\t1\t116905 9441\n",
      "73\t107781\t0.3\t1\t99730 8051\n",
      "74\t112352\t0.3\t1\t103046 9306\n",
      "75\t85293\t0.3\t1\t79320 5973\n",
      "76\t6948\t0.3\t1\t6267 681\n",
      "77\t462\t0.3\t1\t413 49\n",
      "78\t2620\t0.3\t1\t2460 160\n",
      "79\t7997\t0.3\t1\t7525 472\n",
      "80\t12209\t0.3\t1\t11460 749\n",
      "81\t14869\t0.3\t1\t13905 964\n",
      "82\t15857\t0.3\t1\t14908 949\n",
      "83\t16563\t0.3\t1\t15535 1028\n",
      "84\t15747\t0.3\t1\t14748 999\n",
      "85\t14953\t0.3\t1\t14038 915\n",
      "86\t14219\t0.3\t1\t13301 918\n",
      "87\t13232\t0.3\t1\t12355 877\n",
      "88\t11812\t0.3\t1\t11105 707\n",
      "89\t10902\t0.3\t1\t10234 668\n",
      "90\t9695\t0.3\t1\t9108 587\n",
      "91\t8604\t0.3\t1\t8052 552\n",
      "92\t7635\t0.3\t1\t7205 430\n",
      "93\t6262\t0.3\t1\t5877 385\n",
      "94\t5620\t0.3\t1\t5296 324\n",
      "95\t4960\t0.3\t1\t4636 324\n",
      "96\t4490\t0.3\t1\t4205 285\n",
      "97\t4323\t0.3\t1\t4053 270\n",
      "98\t4079\t0.3\t1\t3806 273\n",
      "99\t3603\t0.3\t1\t3362 241\n",
      "100\t3096\t0.3\t1\t2909 187\n",
      "101\t2747\t0.3\t1\t2585 162\n",
      "102\t2261\t0.3\t1\t2082 179\n",
      "103\t1960\t0.3\t1\t1825 135\n",
      "104\t1566\t0.3\t1\t1449 117\n",
      "105\t1437\t0.3\t1\t1329 108\n",
      "106\t1207\t0.3\t1\t1113 94\n",
      "107\t1115\t0.3\t1\t1021 94\n",
      "108\t1114\t0.3\t1\t1009 105\n",
      "109\t935\t0.3\t1\t870 65\n",
      "110\t843\t0.3\t1\t771 72\n",
      "111\t716\t0.3\t1\t655 61\n",
      "112\t599\t0.3\t1\t554 45\n",
      "113\t543\t0.3\t1\t489 54\n",
      "114\t467\t0.3\t1\t409 58\n",
      "115\t412\t0.3\t1\t359 53\n",
      "116\t381\t0.3\t1\t321 60\n",
      "117\t367\t0.3\t1\t320 47\n",
      "118\t297\t0.3\t1\t255 42\n",
      "119\t303\t0.3\t1\t251 52\n",
      "120\t250\t0.3\t1\t202 48\n",
      "121\t227\t0.3\t1\t175 52\n",
      "122\t232\t0.3\t1\t176 56\n",
      "123\t190\t0.3\t1\t139 51\n",
      "124\t151\t0.3\t1\t101 50\n",
      "125\t224\t0.3\t1\t171 53\n",
      "126\t156\t0.3\t1\t110 46\n",
      "127\t160\t0.3\t1\t114 46\n",
      "128\t126\t0.3\t1\t90 36\n",
      "129\t140\t0.3\t1\t107 33\n",
      "130\t75\t0.3\t1\t56 19\n",
      "131\t120\t0.3\t1\t85 35\n",
      "132\t63\t0.3\t1\t47 16\n",
      "133\t87\t0.3\t1\t75 12\n",
      "134\t95\t0.3\t1\t87 8\n",
      "135\t84\t0.3\t1\t82 2\n",
      "136\t69\t0.3\t1\t50 19\n",
      "137\t85\t0.3\t1\t80 5\n",
      "138\t64\t0.3\t1\t56 8\n",
      "139\t44\t0.3\t1\t39 5\n",
      "140\t38\t0.3\t1\t35 3\n",
      "141\t104\t0.3\t1\t102 2\n",
      "142\t61\t0.3\t1\t51 10\n",
      "143\t79\t0.3\t1\t74 5\n",
      "144\t46\t0.3\t1\t42 4\n",
      "145\t196\t0.3\t1\t192 4\n",
      "146\t291\t0.3\t1\t283 8\n",
      "147\t146\t0.3\t1\t142 4\n",
      "148\t599\t0.3\t1\t591 8\n",
      "149\t121\t0.3\t1\t98 23\n",
      "150\t17109\t0.3\t1\t61 17048\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/I.1.fastq.gz\n",
      "=============================================\n",
      "21348992 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/I.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/I.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to I.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/I.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/I.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 860.12 s (40 us/read; 1.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,348,992\n",
      "Reads with adapters:                13,517,793 (63.3%)\n",
      "Reads written (passing filters):    21,348,992 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,202,348,800 bp\n",
      "Quality-trimmed:             264,379,378 bp (8.3%)\n",
      "Total written (filtered):  2,575,273,672 bp (80.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 13517793 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.6%\n",
      "  C: 32.8%\n",
      "  G: 39.2%\n",
      "  T: 12.2%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2611373\t5337248.0\t0\t2611373\n",
      "2\t789303\t1334312.0\t0\t789303\n",
      "3\t369086\t333578.0\t0\t369086\n",
      "4\t167020\t83394.5\t0\t167020\n",
      "5\t168806\t20848.6\t0\t168806\n",
      "6\t99598\t5212.2\t0\t99598\n",
      "7\t108310\t1303.0\t0\t108310\n",
      "8\t145863\t325.8\t0\t145863\n",
      "9\t180479\t81.4\t0\t180230 249\n",
      "10\t191907\t20.4\t1\t182830 9077\n",
      "11\t94363\t5.1\t1\t84310 10053\n",
      "12\t104094\t1.3\t1\t96230 7864\n",
      "13\t127496\t0.3\t1\t119471 8025\n",
      "14\t133771\t0.3\t1\t125948 7823\n",
      "15\t127452\t0.3\t1\t120859 6593\n",
      "16\t150392\t0.3\t1\t140464 9928\n",
      "17\t172591\t0.3\t1\t162763 9828\n",
      "18\t191326\t0.3\t1\t181559 9767\n",
      "19\t157332\t0.3\t1\t148159 9173\n",
      "20\t187548\t0.3\t1\t179240 8308\n",
      "21\t80595\t0.3\t1\t74398 6197\n",
      "22\t118735\t0.3\t1\t114045 4690\n",
      "23\t117418\t0.3\t1\t111747 5671\n",
      "24\t185694\t0.3\t1\t176328 9366\n",
      "25\t119929\t0.3\t1\t113736 6193\n",
      "26\t170472\t0.3\t1\t161527 8945\n",
      "27\t249781\t0.3\t1\t235610 14171\n",
      "28\t159282\t0.3\t1\t152380 6902\n",
      "29\t143369\t0.3\t1\t131639 11730\n",
      "30\t152173\t0.3\t1\t145460 6713\n",
      "31\t201575\t0.3\t1\t193283 8292\n",
      "32\t112665\t0.3\t1\t106184 6481\n",
      "33\t233095\t0.3\t1\t223913 9182\n",
      "34\t174439\t0.3\t1\t163242 11197\n",
      "35\t140732\t0.3\t1\t133988 6744\n",
      "36\t204730\t0.3\t1\t198263 6467\n",
      "37\t184945\t0.3\t1\t178040 6905\n",
      "38\t127161\t0.3\t1\t121629 5532\n",
      "39\t182106\t0.3\t1\t175223 6883\n",
      "40\t109335\t0.3\t1\t104861 4474\n",
      "41\t158651\t0.3\t1\t153818 4833\n",
      "42\t195004\t0.3\t1\t189668 5336\n",
      "43\t138789\t0.3\t1\t134157 4632\n",
      "44\t121800\t0.3\t1\t117404 4396\n",
      "45\t240645\t0.3\t1\t233665 6980\n",
      "46\t121068\t0.3\t1\t115789 5279\n",
      "47\t215037\t0.3\t1\t209551 5486\n",
      "48\t85855\t0.3\t1\t80955 4900\n",
      "49\t200323\t0.3\t1\t194566 5757\n",
      "50\t60892\t0.3\t1\t58124 2768\n",
      "51\t136440\t0.3\t1\t132241 4199\n",
      "52\t76377\t0.3\t1\t73357 3020\n",
      "53\t122713\t0.3\t1\t119380 3333\n",
      "54\t168741\t0.3\t1\t164709 4032\n",
      "55\t98177\t0.3\t1\t95078 3099\n",
      "56\t123017\t0.3\t1\t119604 3413\n",
      "57\t201414\t0.3\t1\t196733 4681\n",
      "58\t132700\t0.3\t1\t128857 3843\n",
      "59\t64039\t0.3\t1\t61666 2373\n",
      "60\t153518\t0.3\t1\t150374 3144\n",
      "61\t63626\t0.3\t1\t61402 2224\n",
      "62\t61684\t0.3\t1\t59362 2322\n",
      "63\t126935\t0.3\t1\t124061 2874\n",
      "64\t92069\t0.3\t1\t89500 2569\n",
      "65\t36930\t0.3\t1\t35326 1604\n",
      "66\t87964\t0.3\t1\t85826 2138\n",
      "67\t26494\t0.3\t1\t25534 960\n",
      "68\t59379\t0.3\t1\t57339 2040\n",
      "69\t64420\t0.3\t1\t62632 1788\n",
      "70\t122656\t0.3\t1\t119825 2831\n",
      "71\t21305\t0.3\t1\t20488 817\n",
      "72\t37922\t0.3\t1\t36767 1155\n",
      "73\t41017\t0.3\t1\t39809 1208\n",
      "74\t42366\t0.3\t1\t41142 1224\n",
      "75\t46328\t0.3\t1\t45001 1327\n",
      "76\t62707\t0.3\t1\t61136 1571\n",
      "77\t104439\t0.3\t1\t101971 2468\n",
      "78\t77285\t0.3\t1\t74461 2824\n",
      "79\t20214\t0.3\t1\t18824 1390\n",
      "80\t3894\t0.3\t1\t3351 543\n",
      "81\t545\t0.3\t1\t488 57\n",
      "82\t873\t0.3\t1\t829 44\n",
      "83\t2257\t0.3\t1\t2206 51\n",
      "84\t4238\t0.3\t1\t4145 93\n",
      "85\t6463\t0.3\t1\t6345 118\n",
      "86\t7811\t0.3\t1\t7657 154\n",
      "87\t7100\t0.3\t1\t6943 157\n",
      "88\t5205\t0.3\t1\t5091 114\n",
      "89\t3516\t0.3\t1\t3425 91\n",
      "90\t2807\t0.3\t1\t2737 70\n",
      "91\t2477\t0.3\t1\t2406 71\n",
      "92\t2000\t0.3\t1\t1954 46\n",
      "93\t1505\t0.3\t1\t1467 38\n",
      "94\t1229\t0.3\t1\t1196 33\n",
      "95\t1067\t0.3\t1\t1030 37\n",
      "96\t935\t0.3\t1\t915 20\n",
      "97\t813\t0.3\t1\t775 38\n",
      "98\t710\t0.3\t1\t675 35\n",
      "99\t666\t0.3\t1\t642 24\n",
      "100\t430\t0.3\t1\t416 14\n",
      "101\t464\t0.3\t1\t436 28\n",
      "102\t378\t0.3\t1\t362 16\n",
      "103\t291\t0.3\t1\t268 23\n",
      "104\t222\t0.3\t1\t203 19\n",
      "105\t220\t0.3\t1\t203 17\n",
      "106\t167\t0.3\t1\t149 18\n",
      "107\t180\t0.3\t1\t162 18\n",
      "108\t158\t0.3\t1\t146 12\n",
      "109\t136\t0.3\t1\t124 12\n",
      "110\t115\t0.3\t1\t98 17\n",
      "111\t100\t0.3\t1\t76 24\n",
      "112\t110\t0.3\t1\t97 13\n",
      "113\t105\t0.3\t1\t83 22\n",
      "114\t87\t0.3\t1\t60 27\n",
      "115\t79\t0.3\t1\t61 18\n",
      "116\t72\t0.3\t1\t56 16\n",
      "117\t89\t0.3\t1\t65 24\n",
      "118\t69\t0.3\t1\t47 22\n",
      "119\t80\t0.3\t1\t57 23\n",
      "120\t73\t0.3\t1\t48 25\n",
      "121\t74\t0.3\t1\t44 30\n",
      "122\t65\t0.3\t1\t50 15\n",
      "123\t70\t0.3\t1\t38 32\n",
      "124\t57\t0.3\t1\t36 21\n",
      "125\t65\t0.3\t1\t43 22\n",
      "126\t56\t0.3\t1\t37 19\n",
      "127\t38\t0.3\t1\t19 19\n",
      "128\t33\t0.3\t1\t24 9\n",
      "129\t42\t0.3\t1\t23 19\n",
      "130\t18\t0.3\t1\t15 3\n",
      "131\t21\t0.3\t1\t14 7\n",
      "132\t13\t0.3\t1\t9 4\n",
      "133\t11\t0.3\t1\t7 4\n",
      "134\t14\t0.3\t1\t10 4\n",
      "135\t9\t0.3\t1\t4 5\n",
      "136\t5\t0.3\t1\t4 1\n",
      "137\t11\t0.3\t1\t8 3\n",
      "138\t8\t0.3\t1\t3 5\n",
      "139\t8\t0.3\t1\t6 2\n",
      "140\t7\t0.3\t1\t4 3\n",
      "141\t11\t0.3\t1\t8 3\n",
      "142\t7\t0.3\t1\t3 4\n",
      "143\t13\t0.3\t1\t7 6\n",
      "144\t7\t0.3\t1\t4 3\n",
      "145\t13\t0.3\t1\t9 4\n",
      "146\t29\t0.3\t1\t26 3\n",
      "147\t14\t0.3\t1\t11 3\n",
      "148\t46\t0.3\t1\t41 5\n",
      "149\t4\t0.3\t1\t3 1\n",
      "150\t7\t0.3\t1\t5 2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/I.2.fastq.gz\n",
      "=============================================\n",
      "21348992 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files I.1_trimmed.fq.gz and I.2_trimmed.fq.gz\n",
      "file_1: I.1_trimmed.fq.gz, file_2: I.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: I.1_trimmed.fq.gz and I.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to I.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to I.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to I.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to I.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 21348992\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1017726 (4.77%)\n",
      "\n",
      "Deleting both intermediate output files I.1_trimmed.fq.gz and I.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/I.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/I.1.fastq.gz ./data_raw/I.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/J.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t355259\tAGATCGGAAGAGC\t1000000\t35.53\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 355259). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/J.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/J.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/J.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to J.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/J.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/J.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 714.65 s (40 us/read; 1.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,779,840\n",
      "Reads with adapters:                11,116,646 (62.5%)\n",
      "Reads written (passing filters):    17,779,840 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,666,976,000 bp\n",
      "Quality-trimmed:              62,108,635 bp (2.3%)\n",
      "Total written (filtered):  2,319,678,049 bp (87.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11116646 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.2%\n",
      "  C: 32.8%\n",
      "  G: 31.5%\n",
      "  T: 16.8%\n",
      "  none/other: 0.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2244550\t4444960.0\t0\t2244550\n",
      "2\t749500\t1111240.0\t0\t749500\n",
      "3\t307803\t277810.0\t0\t307803\n",
      "4\t158662\t69452.5\t0\t158662\n",
      "5\t114886\t17363.1\t0\t114886\n",
      "6\t94608\t4340.8\t0\t94608\n",
      "7\t95223\t1085.2\t0\t95223\n",
      "8\t114828\t271.3\t0\t114828\n",
      "9\t148277\t67.8\t0\t148109 168\n",
      "10\t162764\t17.0\t1\t151468 11296\n",
      "11\t111983\t4.2\t1\t102303 9680\n",
      "12\t113473\t1.1\t1\t104783 8690\n",
      "13\t143839\t0.3\t1\t133301 10538\n",
      "14\t103949\t0.3\t1\t95476 8473\n",
      "15\t147521\t0.3\t1\t136904 10617\n",
      "16\t77441\t0.3\t1\t70360 7081\n",
      "17\t123403\t0.3\t1\t113884 9519\n",
      "18\t109965\t0.3\t1\t102518 7447\n",
      "19\t128977\t0.3\t1\t120033 8944\n",
      "20\t101357\t0.3\t1\t94147 7210\n",
      "21\t134980\t0.3\t1\t125464 9516\n",
      "22\t135831\t0.3\t1\t126199 9632\n",
      "23\t134775\t0.3\t1\t125004 9771\n",
      "24\t137467\t0.3\t1\t126451 11016\n",
      "25\t131474\t0.3\t1\t121923 9551\n",
      "26\t118981\t0.3\t1\t110068 8913\n",
      "27\t144400\t0.3\t1\t134124 10276\n",
      "28\t113524\t0.3\t1\t105957 7567\n",
      "29\t159985\t0.3\t1\t148337 11648\n",
      "30\t136910\t0.3\t1\t127461 9449\n",
      "31\t180912\t0.3\t1\t168063 12849\n",
      "32\t108120\t0.3\t1\t100325 7795\n",
      "33\t141291\t0.3\t1\t130051 11240\n",
      "34\t167708\t0.3\t1\t156044 11664\n",
      "35\t85335\t0.3\t1\t79318 6017\n",
      "36\t113292\t0.3\t1\t105288 8004\n",
      "37\t118634\t0.3\t1\t110140 8494\n",
      "38\t109509\t0.3\t1\t101321 8188\n",
      "39\t130736\t0.3\t1\t121058 9678\n",
      "40\t179388\t0.3\t1\t165903 13485\n",
      "41\t242817\t0.3\t1\t225647 17170\n",
      "42\t305589\t0.3\t1\t284624 20965\n",
      "43\t155765\t0.3\t1\t144926 10839\n",
      "44\t57169\t0.3\t1\t53344 3825\n",
      "45\t17138\t0.3\t1\t15909 1229\n",
      "46\t33267\t0.3\t1\t30986 2281\n",
      "47\t64538\t0.3\t1\t60314 4224\n",
      "48\t88087\t0.3\t1\t82322 5765\n",
      "49\t160543\t0.3\t1\t150654 9889\n",
      "50\t45018\t0.3\t1\t41720 3298\n",
      "51\t125940\t0.3\t1\t117051 8889\n",
      "52\t136247\t0.3\t1\t127410 8837\n",
      "53\t97272\t0.3\t1\t90839 6433\n",
      "54\t78617\t0.3\t1\t73305 5312\n",
      "55\t120168\t0.3\t1\t112389 7779\n",
      "56\t99400\t0.3\t1\t93112 6288\n",
      "57\t37045\t0.3\t1\t34606 2439\n",
      "58\t80678\t0.3\t1\t75596 5082\n",
      "59\t131410\t0.3\t1\t123094 8316\n",
      "60\t31838\t0.3\t1\t29645 2193\n",
      "61\t66342\t0.3\t1\t62156 4186\n",
      "62\t93883\t0.3\t1\t87773 6110\n",
      "63\t27302\t0.3\t1\t25475 1827\n",
      "64\t58434\t0.3\t1\t54375 4059\n",
      "65\t73065\t0.3\t1\t68141 4924\n",
      "66\t39278\t0.3\t1\t36575 2703\n",
      "67\t48228\t0.3\t1\t45226 3002\n",
      "68\t46368\t0.3\t1\t43375 2993\n",
      "69\t40309\t0.3\t1\t37599 2710\n",
      "70\t38040\t0.3\t1\t35359 2681\n",
      "71\t41447\t0.3\t1\t36909 4538\n",
      "72\t68732\t0.3\t1\t61954 6778\n",
      "73\t59172\t0.3\t1\t53273 5899\n",
      "74\t62939\t0.3\t1\t54831 8108\n",
      "75\t46512\t0.3\t1\t42622 3890\n",
      "76\t3705\t0.3\t1\t3307 398\n",
      "77\t236\t0.3\t1\t206 30\n",
      "78\t1401\t0.3\t1\t1313 88\n",
      "79\t4030\t0.3\t1\t3771 259\n",
      "80\t6469\t0.3\t1\t6064 405\n",
      "81\t7618\t0.3\t1\t7148 470\n",
      "82\t8138\t0.3\t1\t7636 502\n",
      "83\t8425\t0.3\t1\t7873 552\n",
      "84\t8098\t0.3\t1\t7546 552\n",
      "85\t7681\t0.3\t1\t7199 482\n",
      "86\t7438\t0.3\t1\t6933 505\n",
      "87\t6997\t0.3\t1\t6536 461\n",
      "88\t6445\t0.3\t1\t6074 371\n",
      "89\t5824\t0.3\t1\t5480 344\n",
      "90\t5184\t0.3\t1\t4861 323\n",
      "91\t4683\t0.3\t1\t4382 301\n",
      "92\t4056\t0.3\t1\t3801 255\n",
      "93\t3348\t0.3\t1\t3128 220\n",
      "94\t2888\t0.3\t1\t2718 170\n",
      "95\t2670\t0.3\t1\t2486 184\n",
      "96\t2456\t0.3\t1\t2288 168\n",
      "97\t2366\t0.3\t1\t2238 128\n",
      "98\t2185\t0.3\t1\t2043 142\n",
      "99\t1945\t0.3\t1\t1816 129\n",
      "100\t1674\t0.3\t1\t1572 102\n",
      "101\t1463\t0.3\t1\t1368 95\n",
      "102\t1291\t0.3\t1\t1191 100\n",
      "103\t988\t0.3\t1\t919 69\n",
      "104\t826\t0.3\t1\t762 64\n",
      "105\t706\t0.3\t1\t655 51\n",
      "106\t622\t0.3\t1\t562 60\n",
      "107\t595\t0.3\t1\t533 62\n",
      "108\t546\t0.3\t1\t481 65\n",
      "109\t513\t0.3\t1\t470 43\n",
      "110\t406\t0.3\t1\t368 38\n",
      "111\t369\t0.3\t1\t333 36\n",
      "112\t291\t0.3\t1\t259 32\n",
      "113\t281\t0.3\t1\t255 26\n",
      "114\t240\t0.3\t1\t211 29\n",
      "115\t199\t0.3\t1\t173 26\n",
      "116\t165\t0.3\t1\t134 31\n",
      "117\t136\t0.3\t1\t123 13\n",
      "118\t151\t0.3\t1\t130 21\n",
      "119\t157\t0.3\t1\t139 18\n",
      "120\t122\t0.3\t1\t96 26\n",
      "121\t113\t0.3\t1\t88 25\n",
      "122\t105\t0.3\t1\t79 26\n",
      "123\t82\t0.3\t1\t57 25\n",
      "124\t74\t0.3\t1\t50 24\n",
      "125\t102\t0.3\t1\t78 24\n",
      "126\t81\t0.3\t1\t56 25\n",
      "127\t77\t0.3\t1\t55 22\n",
      "128\t57\t0.3\t1\t43 14\n",
      "129\t53\t0.3\t1\t33 20\n",
      "130\t39\t0.3\t1\t19 20\n",
      "131\t50\t0.3\t1\t30 20\n",
      "132\t37\t0.3\t1\t29 8\n",
      "133\t33\t0.3\t1\t27 6\n",
      "134\t36\t0.3\t1\t32 4\n",
      "135\t32\t0.3\t1\t28 4\n",
      "136\t26\t0.3\t1\t21 5\n",
      "137\t34\t0.3\t1\t30 4\n",
      "138\t24\t0.3\t1\t22 2\n",
      "139\t23\t0.3\t1\t20 3\n",
      "140\t28\t0.3\t1\t22 6\n",
      "141\t50\t0.3\t1\t45 5\n",
      "142\t29\t0.3\t1\t13 16\n",
      "143\t36\t0.3\t1\t32 4\n",
      "144\t22\t0.3\t1\t20 2\n",
      "145\t69\t0.3\t1\t66 3\n",
      "146\t117\t0.3\t1\t114 3\n",
      "147\t71\t0.3\t1\t69 2\n",
      "148\t215\t0.3\t1\t210 5\n",
      "149\t62\t0.3\t1\t49 13\n",
      "150\t33984\t0.3\t1\t23 33961\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/J.1.fastq.gz\n",
      "=============================================\n",
      "17779840 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/J.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/J.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to J.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/J.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/J.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 715.52 s (40 us/read; 1.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,779,840\n",
      "Reads with adapters:                10,697,345 (60.2%)\n",
      "Reads written (passing filters):    17,779,840 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,666,976,000 bp\n",
      "Quality-trimmed:             223,733,185 bp (8.4%)\n",
      "Total written (filtered):  2,190,860,830 bp (82.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10697345 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.7%\n",
      "  C: 32.7%\n",
      "  G: 36.9%\n",
      "  T: 12.5%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2390839\t4444960.0\t0\t2390839\n",
      "2\t698503\t1111240.0\t0\t698503\n",
      "3\t324680\t277810.0\t0\t324680\n",
      "4\t150807\t69452.5\t0\t150807\n",
      "5\t147009\t17363.1\t0\t147009\n",
      "6\t85793\t4340.8\t0\t85793\n",
      "7\t95322\t1085.2\t0\t95322\n",
      "8\t125392\t271.3\t0\t125392\n",
      "9\t155188\t67.8\t0\t154971 217\n",
      "10\t160992\t17.0\t1\t153808 7184\n",
      "11\t81095\t4.2\t1\t73483 7612\n",
      "12\t87876\t1.1\t1\t81619 6257\n",
      "13\t108834\t0.3\t1\t102471 6363\n",
      "14\t113024\t0.3\t1\t107008 6016\n",
      "15\t107379\t0.3\t1\t102219 5160\n",
      "16\t127131\t0.3\t1\t119228 7903\n",
      "17\t145373\t0.3\t1\t137699 7674\n",
      "18\t162505\t0.3\t1\t154488 8017\n",
      "19\t135812\t0.3\t1\t128521 7291\n",
      "20\t161065\t0.3\t1\t154119 6946\n",
      "21\t65056\t0.3\t1\t60339 4717\n",
      "22\t94252\t0.3\t1\t90673 3579\n",
      "23\t96053\t0.3\t1\t91864 4189\n",
      "24\t149509\t0.3\t1\t142951 6558\n",
      "25\t108337\t0.3\t1\t103204 5133\n",
      "26\t137409\t0.3\t1\t130638 6771\n",
      "27\t189715\t0.3\t1\t179629 10086\n",
      "28\t107306\t0.3\t1\t102751 4555\n",
      "29\t130335\t0.3\t1\t120900 9435\n",
      "30\t133910\t0.3\t1\t128671 5239\n",
      "31\t158241\t0.3\t1\t151860 6381\n",
      "32\t92530\t0.3\t1\t88283 4247\n",
      "33\t181327\t0.3\t1\t174602 6725\n",
      "34\t78266\t0.3\t1\t73173 5093\n",
      "35\t98969\t0.3\t1\t95377 3592\n",
      "36\t189957\t0.3\t1\t184095 5862\n",
      "37\t86156\t0.3\t1\t81150 5006\n",
      "38\t184396\t0.3\t1\t178372 6024\n",
      "39\t79735\t0.3\t1\t74476 5259\n",
      "40\t88226\t0.3\t1\t85136 3090\n",
      "41\t130972\t0.3\t1\t126494 4478\n",
      "42\t161277\t0.3\t1\t156456 4821\n",
      "43\t119213\t0.3\t1\t115186 4027\n",
      "44\t111547\t0.3\t1\t107426 4121\n",
      "45\t188101\t0.3\t1\t182312 5789\n",
      "46\t88644\t0.3\t1\t84687 3957\n",
      "47\t143239\t0.3\t1\t139366 3873\n",
      "48\t53458\t0.3\t1\t50016 3442\n",
      "49\t142290\t0.3\t1\t137923 4367\n",
      "50\t37148\t0.3\t1\t35351 1797\n",
      "51\t100692\t0.3\t1\t97406 3286\n",
      "52\t42287\t0.3\t1\t40433 1854\n",
      "53\t81682\t0.3\t1\t79427 2255\n",
      "54\t113301\t0.3\t1\t110642 2659\n",
      "55\t65683\t0.3\t1\t63415 2268\n",
      "56\t77568\t0.3\t1\t75485 2083\n",
      "57\t135480\t0.3\t1\t132418 3062\n",
      "58\t84047\t0.3\t1\t81541 2506\n",
      "59\t35943\t0.3\t1\t34600 1343\n",
      "60\t100704\t0.3\t1\t98714 1990\n",
      "61\t33460\t0.3\t1\t32210 1250\n",
      "62\t34160\t0.3\t1\t32836 1324\n",
      "63\t77305\t0.3\t1\t75594 1711\n",
      "64\t51153\t0.3\t1\t49611 1542\n",
      "65\t18917\t0.3\t1\t18047 870\n",
      "66\t52321\t0.3\t1\t50997 1324\n",
      "67\t13033\t0.3\t1\t12534 499\n",
      "68\t30618\t0.3\t1\t29457 1161\n",
      "69\t34168\t0.3\t1\t33208 960\n",
      "70\t68195\t0.3\t1\t66648 1547\n",
      "71\t10466\t0.3\t1\t10044 422\n",
      "72\t19156\t0.3\t1\t18540 616\n",
      "73\t20932\t0.3\t1\t20240 692\n",
      "74\t21257\t0.3\t1\t20586 671\n",
      "75\t23540\t0.3\t1\t22876 664\n",
      "76\t32337\t0.3\t1\t31600 737\n",
      "77\t51716\t0.3\t1\t50384 1332\n",
      "78\t36886\t0.3\t1\t35471 1415\n",
      "79\t9586\t0.3\t1\t8852 734\n",
      "80\t1825\t0.3\t1\t1598 227\n",
      "81\t268\t0.3\t1\t228 40\n",
      "82\t363\t0.3\t1\t342 21\n",
      "83\t966\t0.3\t1\t937 29\n",
      "84\t1866\t0.3\t1\t1828 38\n",
      "85\t2884\t0.3\t1\t2818 66\n",
      "86\t3748\t0.3\t1\t3670 78\n",
      "87\t3554\t0.3\t1\t3487 67\n",
      "88\t2802\t0.3\t1\t2738 64\n",
      "89\t1848\t0.3\t1\t1802 46\n",
      "90\t1577\t0.3\t1\t1537 40\n",
      "91\t1359\t0.3\t1\t1318 41\n",
      "92\t1117\t0.3\t1\t1083 34\n",
      "93\t880\t0.3\t1\t851 29\n",
      "94\t707\t0.3\t1\t687 20\n",
      "95\t617\t0.3\t1\t599 18\n",
      "96\t505\t0.3\t1\t487 18\n",
      "97\t459\t0.3\t1\t442 17\n",
      "98\t433\t0.3\t1\t415 18\n",
      "99\t354\t0.3\t1\t336 18\n",
      "100\t307\t0.3\t1\t297 10\n",
      "101\t252\t0.3\t1\t242 10\n",
      "102\t199\t0.3\t1\t189 10\n",
      "103\t154\t0.3\t1\t142 12\n",
      "104\t142\t0.3\t1\t131 11\n",
      "105\t134\t0.3\t1\t123 11\n",
      "106\t99\t0.3\t1\t88 11\n",
      "107\t93\t0.3\t1\t82 11\n",
      "108\t86\t0.3\t1\t79 7\n",
      "109\t85\t0.3\t1\t69 16\n",
      "110\t64\t0.3\t1\t57 7\n",
      "111\t64\t0.3\t1\t54 10\n",
      "112\t57\t0.3\t1\t38 19\n",
      "113\t52\t0.3\t1\t42 10\n",
      "114\t53\t0.3\t1\t43 10\n",
      "115\t39\t0.3\t1\t28 11\n",
      "116\t39\t0.3\t1\t31 8\n",
      "117\t43\t0.3\t1\t31 12\n",
      "118\t38\t0.3\t1\t29 9\n",
      "119\t42\t0.3\t1\t31 11\n",
      "120\t36\t0.3\t1\t22 14\n",
      "121\t32\t0.3\t1\t20 12\n",
      "122\t43\t0.3\t1\t33 10\n",
      "123\t38\t0.3\t1\t21 17\n",
      "124\t25\t0.3\t1\t18 7\n",
      "125\t25\t0.3\t1\t16 9\n",
      "126\t24\t0.3\t1\t17 7\n",
      "127\t30\t0.3\t1\t22 8\n",
      "128\t16\t0.3\t1\t10 6\n",
      "129\t13\t0.3\t1\t8 5\n",
      "130\t8\t0.3\t1\t2 6\n",
      "131\t7\t0.3\t1\t3 4\n",
      "132\t8\t0.3\t1\t3 5\n",
      "133\t9\t0.3\t1\t2 7\n",
      "134\t4\t0.3\t1\t3 1\n",
      "135\t3\t0.3\t1\t1 2\n",
      "136\t5\t0.3\t1\t3 2\n",
      "137\t8\t0.3\t1\t4 4\n",
      "138\t3\t0.3\t1\t3\n",
      "139\t1\t0.3\t1\t1\n",
      "140\t3\t0.3\t1\t1 2\n",
      "141\t4\t0.3\t1\t4\n",
      "143\t5\t0.3\t1\t3 2\n",
      "144\t2\t0.3\t1\t1 1\n",
      "145\t8\t0.3\t1\t8\n",
      "146\t7\t0.3\t1\t6 1\n",
      "147\t2\t0.3\t1\t2\n",
      "148\t11\t0.3\t1\t11\n",
      "149\t4\t0.3\t1\t2 2\n",
      "150\t1\t0.3\t1\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/J.2.fastq.gz\n",
      "=============================================\n",
      "17779840 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files J.1_trimmed.fq.gz and J.2_trimmed.fq.gz\n",
      "file_1: J.1_trimmed.fq.gz, file_2: J.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: J.1_trimmed.fq.gz and J.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to J.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to J.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to J.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to J.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 17779840\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 924352 (5.20%)\n",
      "\n",
      "Deleting both intermediate output files J.1_trimmed.fq.gz and J.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/J.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/J.1.fastq.gz ./data_raw/J.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/K.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t409983\tAGATCGGAAGAGC\t1000000\t41.00\n",
      "Nextera\t3\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 409983). Second best hit was Nextera (count: 3)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/K.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/K.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/K.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to K.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/K.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/K.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 660.82 s (40 us/read; 1.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,432,126\n",
      "Reads with adapters:                10,993,042 (66.9%)\n",
      "Reads written (passing filters):    16,432,126 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,464,818,900 bp\n",
      "Quality-trimmed:              66,099,077 bp (2.7%)\n",
      "Total written (filtered):  2,079,971,406 bp (84.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10993042 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.4%\n",
      "  C: 32.6%\n",
      "  G: 33.0%\n",
      "  T: 17.5%\n",
      "  none/other: 0.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1865428\t4108031.5\t0\t1865428\n",
      "2\t628658\t1027007.9\t0\t628658\n",
      "3\t266995\t256752.0\t0\t266995\n",
      "4\t142023\t64188.0\t0\t142023\n",
      "5\t102563\t16047.0\t0\t102563\n",
      "6\t84814\t4011.7\t0\t84814\n",
      "7\t85751\t1002.9\t0\t85751\n",
      "8\t103548\t250.7\t0\t103548\n",
      "9\t133914\t62.7\t0\t133759 155\n",
      "10\t150249\t15.7\t1\t137763 12486\n",
      "11\t103582\t3.9\t1\t93225 10357\n",
      "12\t105820\t1.0\t1\t95841 9979\n",
      "13\t133016\t0.2\t1\t121433 11583\n",
      "14\t96824\t0.2\t1\t87212 9612\n",
      "15\t137454\t0.2\t1\t125555 11899\n",
      "16\t73073\t0.2\t1\t65262 7811\n",
      "17\t116088\t0.2\t1\t105842 10246\n",
      "18\t104465\t0.2\t1\t95973 8492\n",
      "19\t123345\t0.2\t1\t113372 9973\n",
      "20\t96494\t0.2\t1\t88195 8299\n",
      "21\t127988\t0.2\t1\t117189 10799\n",
      "22\t129702\t0.2\t1\t118662 11040\n",
      "23\t129018\t0.2\t1\t118034 10984\n",
      "24\t131873\t0.2\t1\t119029 12844\n",
      "25\t125614\t0.2\t1\t114985 10629\n",
      "26\t115476\t0.2\t1\t105214 10262\n",
      "27\t140574\t0.2\t1\t128863 11711\n",
      "28\t110205\t0.2\t1\t101396 8809\n",
      "29\t158064\t0.2\t1\t144620 13444\n",
      "30\t134425\t0.2\t1\t123233 11192\n",
      "31\t178189\t0.2\t1\t163308 14881\n",
      "32\t107803\t0.2\t1\t98666 9137\n",
      "33\t143656\t0.2\t1\t130066 13590\n",
      "34\t167492\t0.2\t1\t153660 13832\n",
      "35\t86716\t0.2\t1\t79352 7364\n",
      "36\t115539\t0.2\t1\t105479 10060\n",
      "37\t123571\t0.2\t1\t112657 10914\n",
      "38\t113604\t0.2\t1\t103491 10113\n",
      "39\t136446\t0.2\t1\t124616 11830\n",
      "40\t191052\t0.2\t1\t174380 16672\n",
      "41\t260627\t0.2\t1\t239071 21556\n",
      "42\t326994\t0.2\t1\t300725 26269\n",
      "43\t165616\t0.2\t1\t152354 13262\n",
      "44\t61247\t0.2\t1\t56346 4901\n",
      "45\t18419\t0.2\t1\t16855 1564\n",
      "46\t35662\t0.2\t1\t32830 2832\n",
      "47\t69877\t0.2\t1\t64666 5211\n",
      "48\t97364\t0.2\t1\t90008 7356\n",
      "49\t178337\t0.2\t1\t165439 12898\n",
      "50\t50608\t0.2\t1\t46243 4365\n",
      "51\t144046\t0.2\t1\t132243 11803\n",
      "52\t156818\t0.2\t1\t144589 12229\n",
      "53\t112960\t0.2\t1\t104213 8747\n",
      "54\t91123\t0.2\t1\t83875 7248\n",
      "55\t141414\t0.2\t1\t130683 10731\n",
      "56\t118569\t0.2\t1\t109575 8994\n",
      "57\t44154\t0.2\t1\t40606 3548\n",
      "58\t98033\t0.2\t1\t90754 7279\n",
      "59\t162518\t0.2\t1\t150191 12327\n",
      "60\t39409\t0.2\t1\t36137 3272\n",
      "61\t83015\t0.2\t1\t76798 6217\n",
      "62\t118889\t0.2\t1\t109851 9038\n",
      "63\t34421\t0.2\t1\t31714 2707\n",
      "64\t74615\t0.2\t1\t68927 5688\n",
      "65\t95514\t0.2\t1\t88184 7330\n",
      "66\t51385\t0.2\t1\t47402 3983\n",
      "67\t63556\t0.2\t1\t58663 4893\n",
      "68\t62771\t0.2\t1\t57938 4833\n",
      "69\t55386\t0.2\t1\t50962 4424\n",
      "70\t52751\t0.2\t1\t48522 4229\n",
      "71\t56524\t0.2\t1\t51442 5082\n",
      "72\t97339\t0.2\t1\t88821 8518\n",
      "73\t83394\t0.2\t1\t75922 7472\n",
      "74\t86884\t0.2\t1\t78418 8466\n",
      "75\t67377\t0.2\t1\t61799 5578\n",
      "76\t5627\t0.2\t1\t4957 670\n",
      "77\t353\t0.2\t1\t313 40\n",
      "78\t2053\t0.2\t1\t1875 178\n",
      "79\t6250\t0.2\t1\t5801 449\n",
      "80\t9646\t0.2\t1\t8958 688\n",
      "81\t11422\t0.2\t1\t10548 874\n",
      "82\t12376\t0.2\t1\t11462 914\n",
      "83\t13063\t0.2\t1\t12092 971\n",
      "84\t12435\t0.2\t1\t11447 988\n",
      "85\t11937\t0.2\t1\t11031 906\n",
      "86\t11716\t0.2\t1\t10856 860\n",
      "87\t10495\t0.2\t1\t9691 804\n",
      "88\t9705\t0.2\t1\t8978 727\n",
      "89\t8730\t0.2\t1\t8061 669\n",
      "90\t8096\t0.2\t1\t7487 609\n",
      "91\t7173\t0.2\t1\t6663 510\n",
      "92\t6204\t0.2\t1\t5744 460\n",
      "93\t5258\t0.2\t1\t4836 422\n",
      "94\t4745\t0.2\t1\t4441 304\n",
      "95\t4257\t0.2\t1\t3942 315\n",
      "96\t3959\t0.2\t1\t3653 306\n",
      "97\t3864\t0.2\t1\t3572 292\n",
      "98\t3654\t0.2\t1\t3397 257\n",
      "99\t3273\t0.2\t1\t3028 245\n",
      "100\t2799\t0.2\t1\t2597 202\n",
      "101\t2568\t0.2\t1\t2368 200\n",
      "102\t2202\t0.2\t1\t2038 164\n",
      "103\t1815\t0.2\t1\t1665 150\n",
      "104\t1506\t0.2\t1\t1382 124\n",
      "105\t1355\t0.2\t1\t1242 113\n",
      "106\t1218\t0.2\t1\t1113 105\n",
      "107\t1107\t0.2\t1\t986 121\n",
      "108\t1097\t0.2\t1\t1007 90\n",
      "109\t1036\t0.2\t1\t959 77\n",
      "110\t887\t0.2\t1\t821 66\n",
      "111\t717\t0.2\t1\t648 69\n",
      "112\t614\t0.2\t1\t549 65\n",
      "113\t533\t0.2\t1\t471 62\n",
      "114\t451\t0.2\t1\t404 47\n",
      "115\t413\t0.2\t1\t363 50\n",
      "116\t350\t0.2\t1\t305 45\n",
      "117\t314\t0.2\t1\t267 47\n",
      "118\t279\t0.2\t1\t237 42\n",
      "119\t260\t0.2\t1\t222 38\n",
      "120\t183\t0.2\t1\t154 29\n",
      "121\t158\t0.2\t1\t132 26\n",
      "122\t178\t0.2\t1\t147 31\n",
      "123\t137\t0.2\t1\t111 26\n",
      "124\t114\t0.2\t1\t89 25\n",
      "125\t128\t0.2\t1\t92 36\n",
      "126\t90\t0.2\t1\t51 39\n",
      "127\t96\t0.2\t1\t73 23\n",
      "128\t88\t0.2\t1\t68 20\n",
      "129\t75\t0.2\t1\t58 17\n",
      "130\t59\t0.2\t1\t44 15\n",
      "131\t71\t0.2\t1\t51 20\n",
      "132\t30\t0.2\t1\t19 11\n",
      "133\t48\t0.2\t1\t41 7\n",
      "134\t49\t0.2\t1\t42 7\n",
      "135\t39\t0.2\t1\t34 5\n",
      "136\t38\t0.2\t1\t31 7\n",
      "137\t37\t0.2\t1\t30 7\n",
      "138\t20\t0.2\t1\t15 5\n",
      "139\t22\t0.2\t1\t20 2\n",
      "140\t27\t0.2\t1\t24 3\n",
      "141\t78\t0.2\t1\t73 5\n",
      "142\t39\t0.2\t1\t29 10\n",
      "143\t33\t0.2\t1\t30 3\n",
      "144\t25\t0.2\t1\t22 3\n",
      "145\t115\t0.2\t1\t113 2\n",
      "146\t180\t0.2\t1\t179 1\n",
      "147\t96\t0.2\t1\t94 2\n",
      "148\t312\t0.2\t1\t303 9\n",
      "149\t64\t0.2\t1\t48 16\n",
      "150\t13874\t0.2\t1\t40 13834\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/K.1.fastq.gz\n",
      "=============================================\n",
      "16432126 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/K.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/K.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to K.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/K.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/K.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 656.05 s (40 us/read; 1.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,432,126\n",
      "Reads with adapters:                10,431,436 (63.5%)\n",
      "Reads written (passing filters):    16,432,126 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,464,818,900 bp\n",
      "Quality-trimmed:             240,506,403 bp (9.8%)\n",
      "Total written (filtered):  1,944,750,278 bp (78.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10431436 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.4%\n",
      "  C: 33.5%\n",
      "  G: 38.2%\n",
      "  T: 12.7%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1961313\t4108031.5\t0\t1961313\n",
      "2\t579376\t1027007.9\t0\t579376\n",
      "3\t280428\t256752.0\t0\t280428\n",
      "4\t131520\t64188.0\t0\t131520\n",
      "5\t132386\t16047.0\t0\t132386\n",
      "6\t77065\t4011.7\t0\t77065\n",
      "7\t85539\t1002.9\t0\t85539\n",
      "8\t111580\t250.7\t0\t111580\n",
      "9\t141503\t62.7\t0\t141343 160\n",
      "10\t146545\t15.7\t1\t139134 7411\n",
      "11\t76714\t3.9\t1\t68765 7949\n",
      "12\t80832\t1.0\t1\t74371 6461\n",
      "13\t100474\t0.2\t1\t93982 6492\n",
      "14\t103878\t0.2\t1\t97431 6447\n",
      "15\t98603\t0.2\t1\t93190 5413\n",
      "16\t119696\t0.2\t1\t111662 8034\n",
      "17\t135777\t0.2\t1\t127823 7954\n",
      "18\t154359\t0.2\t1\t146063 8296\n",
      "19\t126370\t0.2\t1\t118990 7380\n",
      "20\t154108\t0.2\t1\t146495 7613\n",
      "21\t63251\t0.2\t1\t58416 4835\n",
      "22\t88111\t0.2\t1\t84383 3728\n",
      "23\t91559\t0.2\t1\t87073 4486\n",
      "24\t142008\t0.2\t1\t134660 7348\n",
      "25\t103908\t0.2\t1\t98373 5535\n",
      "26\t132733\t0.2\t1\t125484 7249\n",
      "27\t192113\t0.2\t1\t181055 11058\n",
      "28\t105880\t0.2\t1\t100770 5110\n",
      "29\t134040\t0.2\t1\t123478 10562\n",
      "30\t128804\t0.2\t1\t123133 5671\n",
      "31\t161100\t0.2\t1\t153886 7214\n",
      "32\t88710\t0.2\t1\t83948 4762\n",
      "33\t180832\t0.2\t1\t173672 7160\n",
      "34\t111602\t0.2\t1\t103487 8115\n",
      "35\t143979\t0.2\t1\t138362 5617\n",
      "36\t332978\t0.2\t1\t322178 10800\n",
      "37\t100976\t0.2\t1\t95811 5165\n",
      "38\t138027\t0.2\t1\t132472 5555\n",
      "39\t68021\t0.2\t1\t63471 4550\n",
      "40\t176245\t0.2\t1\t170012 6233\n",
      "41\t76355\t0.2\t1\t71666 4689\n",
      "42\t61871\t0.2\t1\t59248 2623\n",
      "43\t145309\t0.2\t1\t140632 4677\n",
      "44\t73586\t0.2\t1\t70720 2866\n",
      "45\t207674\t0.2\t1\t201265 6409\n",
      "46\t104366\t0.2\t1\t99677 4689\n",
      "47\t119698\t0.2\t1\t116071 3627\n",
      "48\t69503\t0.2\t1\t65319 4184\n",
      "49\t160459\t0.2\t1\t155280 5179\n",
      "50\t39505\t0.2\t1\t37462 2043\n",
      "51\t118335\t0.2\t1\t114401 3934\n",
      "52\t44030\t0.2\t1\t41947 2083\n",
      "53\t84978\t0.2\t1\t82570 2408\n",
      "54\t126125\t0.2\t1\t122954 3171\n",
      "55\t81907\t0.2\t1\t79014 2893\n",
      "56\t84344\t0.2\t1\t81856 2488\n",
      "57\t149590\t0.2\t1\t146116 3474\n",
      "58\t107021\t0.2\t1\t103693 3328\n",
      "59\t40796\t0.2\t1\t39151 1645\n",
      "60\t118689\t0.2\t1\t116179 2510\n",
      "61\t40808\t0.2\t1\t39295 1513\n",
      "62\t50110\t0.2\t1\t48202 1908\n",
      "63\t98074\t0.2\t1\t95829 2245\n",
      "64\t62709\t0.2\t1\t60624 2085\n",
      "65\t23877\t0.2\t1\t22653 1224\n",
      "66\t70143\t0.2\t1\t68179 1964\n",
      "67\t15010\t0.2\t1\t14367 643\n",
      "68\t44606\t0.2\t1\t42801 1805\n",
      "69\t44852\t0.2\t1\t43428 1424\n",
      "70\t99824\t0.2\t1\t97264 2560\n",
      "71\t13664\t0.2\t1\t13073 591\n",
      "72\t25187\t0.2\t1\t24324 863\n",
      "73\t28219\t0.2\t1\t27274 945\n",
      "74\t29741\t0.2\t1\t28798 943\n",
      "75\t33053\t0.2\t1\t32053 1000\n",
      "76\t45083\t0.2\t1\t43766 1317\n",
      "77\t78181\t0.2\t1\t76195 1986\n",
      "78\t64860\t0.2\t1\t62336 2524\n",
      "79\t18526\t0.2\t1\t17215 1311\n",
      "80\t3831\t0.2\t1\t3336 495\n",
      "81\t447\t0.2\t1\t381 66\n",
      "82\t523\t0.2\t1\t485 38\n",
      "83\t1400\t0.2\t1\t1358 42\n",
      "84\t2738\t0.2\t1\t2659 79\n",
      "85\t4456\t0.2\t1\t4363 93\n",
      "86\t5651\t0.2\t1\t5524 127\n",
      "87\t5514\t0.2\t1\t5373 141\n",
      "88\t4296\t0.2\t1\t4185 111\n",
      "89\t2868\t0.2\t1\t2792 76\n",
      "90\t2414\t0.2\t1\t2361 53\n",
      "91\t2016\t0.2\t1\t1969 47\n",
      "92\t1678\t0.2\t1\t1629 49\n",
      "93\t1296\t0.2\t1\t1259 37\n",
      "94\t1105\t0.2\t1\t1075 30\n",
      "95\t902\t0.2\t1\t868 34\n",
      "96\t815\t0.2\t1\t783 32\n",
      "97\t727\t0.2\t1\t700 27\n",
      "98\t697\t0.2\t1\t672 25\n",
      "99\t563\t0.2\t1\t549 14\n",
      "100\t458\t0.2\t1\t430 28\n",
      "101\t432\t0.2\t1\t413 19\n",
      "102\t347\t0.2\t1\t337 10\n",
      "103\t307\t0.2\t1\t287 20\n",
      "104\t246\t0.2\t1\t231 15\n",
      "105\t191\t0.2\t1\t179 12\n",
      "106\t173\t0.2\t1\t161 12\n",
      "107\t166\t0.2\t1\t149 17\n",
      "108\t145\t0.2\t1\t131 14\n",
      "109\t133\t0.2\t1\t126 7\n",
      "110\t116\t0.2\t1\t102 14\n",
      "111\t113\t0.2\t1\t102 11\n",
      "112\t73\t0.2\t1\t67 6\n",
      "113\t88\t0.2\t1\t68 20\n",
      "114\t92\t0.2\t1\t76 16\n",
      "115\t73\t0.2\t1\t61 12\n",
      "116\t54\t0.2\t1\t46 8\n",
      "117\t59\t0.2\t1\t42 17\n",
      "118\t58\t0.2\t1\t42 16\n",
      "119\t52\t0.2\t1\t31 21\n",
      "120\t43\t0.2\t1\t32 11\n",
      "121\t44\t0.2\t1\t33 11\n",
      "122\t35\t0.2\t1\t27 8\n",
      "123\t32\t0.2\t1\t20 12\n",
      "124\t34\t0.2\t1\t20 14\n",
      "125\t37\t0.2\t1\t22 15\n",
      "126\t32\t0.2\t1\t18 14\n",
      "127\t25\t0.2\t1\t15 10\n",
      "128\t27\t0.2\t1\t20 7\n",
      "129\t27\t0.2\t1\t18 9\n",
      "130\t19\t0.2\t1\t9 10\n",
      "131\t17\t0.2\t1\t11 6\n",
      "132\t12\t0.2\t1\t8 4\n",
      "133\t12\t0.2\t1\t6 6\n",
      "134\t10\t0.2\t1\t6 4\n",
      "135\t5\t0.2\t1\t4 1\n",
      "136\t4\t0.2\t1\t4\n",
      "137\t9\t0.2\t1\t6 3\n",
      "138\t2\t0.2\t1\t2\n",
      "139\t6\t0.2\t1\t3 3\n",
      "140\t6\t0.2\t1\t3 3\n",
      "141\t6\t0.2\t1\t4 2\n",
      "142\t2\t0.2\t1\t1 1\n",
      "143\t2\t0.2\t1\t1 1\n",
      "144\t2\t0.2\t1\t1 1\n",
      "145\t9\t0.2\t1\t6 3\n",
      "146\t18\t0.2\t1\t14 4\n",
      "147\t11\t0.2\t1\t8 3\n",
      "148\t20\t0.2\t1\t17 3\n",
      "149\t1\t0.2\t1\t0 1\n",
      "150\t3\t0.2\t1\t2 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/K.2.fastq.gz\n",
      "=============================================\n",
      "16432126 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files K.1_trimmed.fq.gz and K.2_trimmed.fq.gz\n",
      "file_1: K.1_trimmed.fq.gz, file_2: K.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: K.1_trimmed.fq.gz and K.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to K.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to K.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to K.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to K.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16432126\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1012129 (6.16%)\n",
      "\n",
      "Deleting both intermediate output files K.1_trimmed.fq.gz and K.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/K.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/K.1.fastq.gz ./data_raw/K.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/L.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t381399\tAGATCGGAAGAGC\t1000000\t38.14\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 381399). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/L.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/L.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/L.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to L.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/L.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/L.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 674.83 s (40 us/read; 1.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,849,948\n",
      "Reads with adapters:                10,808,761 (64.1%)\n",
      "Reads written (passing filters):    16,849,948 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,527,492,200 bp\n",
      "Quality-trimmed:              58,856,572 bp (2.3%)\n",
      "Total written (filtered):  2,187,924,857 bp (86.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10808761 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.7%\n",
      "  C: 30.2%\n",
      "  G: 35.4%\n",
      "  T: 16.4%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2067735\t4212487.0\t0\t2067735\n",
      "2\t697211\t1053121.8\t0\t697211\n",
      "3\t301533\t263280.4\t0\t301533\n",
      "4\t156102\t65820.1\t0\t156102\n",
      "5\t112070\t16455.0\t0\t112070\n",
      "6\t93467\t4113.8\t0\t93467\n",
      "7\t93559\t1028.4\t0\t93559\n",
      "8\t113211\t257.1\t0\t113211\n",
      "9\t145622\t64.3\t0\t145445 177\n",
      "10\t155546\t16.1\t1\t147879 7667\n",
      "11\t110805\t4.0\t1\t102892 7913\n",
      "12\t111145\t1.0\t1\t104200 6945\n",
      "13\t138115\t0.3\t1\t130827 7288\n",
      "14\t101769\t0.3\t1\t95362 6407\n",
      "15\t143549\t0.3\t1\t136077 7472\n",
      "16\t76920\t0.3\t1\t71017 5903\n",
      "17\t122546\t0.3\t1\t115482 7064\n",
      "18\t107325\t0.3\t1\t102110 5215\n",
      "19\t127754\t0.3\t1\t121179 6575\n",
      "20\t99339\t0.3\t1\t94078 5261\n",
      "21\t133858\t0.3\t1\t126925 6933\n",
      "22\t132838\t0.3\t1\t125900 6938\n",
      "23\t133511\t0.3\t1\t126395 7116\n",
      "24\t135015\t0.3\t1\t126484 8531\n",
      "25\t129824\t0.3\t1\t122778 7046\n",
      "26\t119476\t0.3\t1\t112449 7027\n",
      "27\t144744\t0.3\t1\t137183 7561\n",
      "28\t112010\t0.3\t1\t106548 5462\n",
      "29\t161379\t0.3\t1\t152486 8893\n",
      "30\t137307\t0.3\t1\t130362 6945\n",
      "31\t179676\t0.3\t1\t170416 9260\n",
      "32\t106277\t0.3\t1\t100740 5537\n",
      "33\t140517\t0.3\t1\t132095 8422\n",
      "34\t167345\t0.3\t1\t158928 8417\n",
      "35\t85152\t0.3\t1\t80881 4271\n",
      "36\t115185\t0.3\t1\t108842 6343\n",
      "37\t118771\t0.3\t1\t112222 6549\n",
      "38\t111792\t0.3\t1\t105405 6387\n",
      "39\t132909\t0.3\t1\t125283 7626\n",
      "40\t181698\t0.3\t1\t171394 10304\n",
      "41\t245129\t0.3\t1\t232626 12503\n",
      "42\t302247\t0.3\t1\t288261 13986\n",
      "43\t158427\t0.3\t1\t150587 7840\n",
      "44\t58139\t0.3\t1\t55291 2848\n",
      "45\t17527\t0.3\t1\t16583 944\n",
      "46\t33952\t0.3\t1\t32303 1649\n",
      "47\t65963\t0.3\t1\t62903 3060\n",
      "48\t88901\t0.3\t1\t84813 4088\n",
      "49\t161662\t0.3\t1\t154798 6864\n",
      "50\t45984\t0.3\t1\t43292 2692\n",
      "51\t126947\t0.3\t1\t120713 6234\n",
      "52\t135891\t0.3\t1\t129806 6085\n",
      "53\t97867\t0.3\t1\t93241 4626\n",
      "54\t79113\t0.3\t1\t75228 3885\n",
      "55\t118459\t0.3\t1\t113215 5244\n",
      "56\t99306\t0.3\t1\t95058 4248\n",
      "57\t36505\t0.3\t1\t34691 1814\n",
      "58\t80186\t0.3\t1\t76614 3572\n",
      "59\t131279\t0.3\t1\t125697 5582\n",
      "60\t31366\t0.3\t1\t29741 1625\n",
      "61\t66518\t0.3\t1\t63562 2956\n",
      "62\t92139\t0.3\t1\t88174 3965\n",
      "63\t26896\t0.3\t1\t25564 1332\n",
      "64\t57475\t0.3\t1\t54960 2515\n",
      "65\t71253\t0.3\t1\t68097 3156\n",
      "66\t37871\t0.3\t1\t36118 1753\n",
      "67\t47158\t0.3\t1\t45062 2096\n",
      "68\t45593\t0.3\t1\t43575 2018\n",
      "69\t39589\t0.3\t1\t37738 1851\n",
      "70\t37944\t0.3\t1\t36175 1769\n",
      "71\t40383\t0.3\t1\t37807 2576\n",
      "72\t67577\t0.3\t1\t63784 3793\n",
      "73\t58075\t0.3\t1\t54782 3293\n",
      "74\t60085\t0.3\t1\t55980 4105\n",
      "75\t46076\t0.3\t1\t43667 2409\n",
      "76\t3765\t0.3\t1\t3437 328\n",
      "77\t240\t0.3\t1\t221 19\n",
      "78\t1395\t0.3\t1\t1327 68\n",
      "79\t4110\t0.3\t1\t3909 201\n",
      "80\t6429\t0.3\t1\t6157 272\n",
      "81\t7725\t0.3\t1\t7394 331\n",
      "82\t8340\t0.3\t1\t7971 369\n",
      "83\t8694\t0.3\t1\t8286 408\n",
      "84\t8472\t0.3\t1\t8106 366\n",
      "85\t7886\t0.3\t1\t7559 327\n",
      "86\t7756\t0.3\t1\t7413 343\n",
      "87\t7004\t0.3\t1\t6716 288\n",
      "88\t6576\t0.3\t1\t6313 263\n",
      "89\t5952\t0.3\t1\t5686 266\n",
      "90\t5160\t0.3\t1\t4925 235\n",
      "91\t4778\t0.3\t1\t4565 213\n",
      "92\t4070\t0.3\t1\t3914 156\n",
      "93\t3562\t0.3\t1\t3395 167\n",
      "94\t3197\t0.3\t1\t3056 141\n",
      "95\t2916\t0.3\t1\t2786 130\n",
      "96\t2824\t0.3\t1\t2693 131\n",
      "97\t2616\t0.3\t1\t2491 125\n",
      "98\t2390\t0.3\t1\t2275 115\n",
      "99\t2133\t0.3\t1\t2048 85\n",
      "100\t1872\t0.3\t1\t1788 84\n",
      "101\t1624\t0.3\t1\t1563 61\n",
      "102\t1408\t0.3\t1\t1335 73\n",
      "103\t1180\t0.3\t1\t1118 62\n",
      "104\t1044\t0.3\t1\t984 60\n",
      "105\t912\t0.3\t1\t863 49\n",
      "106\t811\t0.3\t1\t746 65\n",
      "107\t800\t0.3\t1\t726 74\n",
      "108\t699\t0.3\t1\t648 51\n",
      "109\t609\t0.3\t1\t562 47\n",
      "110\t518\t0.3\t1\t468 50\n",
      "111\t461\t0.3\t1\t422 39\n",
      "112\t407\t0.3\t1\t360 47\n",
      "113\t359\t0.3\t1\t317 42\n",
      "114\t276\t0.3\t1\t243 33\n",
      "115\t275\t0.3\t1\t241 34\n",
      "116\t236\t0.3\t1\t199 37\n",
      "117\t191\t0.3\t1\t165 26\n",
      "118\t206\t0.3\t1\t171 35\n",
      "119\t196\t0.3\t1\t161 35\n",
      "120\t141\t0.3\t1\t114 27\n",
      "121\t186\t0.3\t1\t145 41\n",
      "122\t129\t0.3\t1\t98 31\n",
      "123\t136\t0.3\t1\t105 31\n",
      "124\t110\t0.3\t1\t77 33\n",
      "125\t127\t0.3\t1\t97 30\n",
      "126\t96\t0.3\t1\t57 39\n",
      "127\t80\t0.3\t1\t57 23\n",
      "128\t86\t0.3\t1\t65 21\n",
      "129\t85\t0.3\t1\t60 25\n",
      "130\t52\t0.3\t1\t36 16\n",
      "131\t62\t0.3\t1\t38 24\n",
      "132\t29\t0.3\t1\t26 3\n",
      "133\t47\t0.3\t1\t33 14\n",
      "134\t37\t0.3\t1\t35 2\n",
      "135\t35\t0.3\t1\t35\n",
      "136\t31\t0.3\t1\t25 6\n",
      "137\t26\t0.3\t1\t21 5\n",
      "138\t19\t0.3\t1\t17 2\n",
      "139\t13\t0.3\t1\t11 2\n",
      "140\t19\t0.3\t1\t13 6\n",
      "141\t60\t0.3\t1\t55 5\n",
      "142\t20\t0.3\t1\t9 11\n",
      "143\t33\t0.3\t1\t32 1\n",
      "144\t17\t0.3\t1\t17\n",
      "145\t91\t0.3\t1\t89 2\n",
      "146\t81\t0.3\t1\t79 2\n",
      "147\t72\t0.3\t1\t71 1\n",
      "148\t241\t0.3\t1\t238 3\n",
      "149\t37\t0.3\t1\t28 9\n",
      "150\t10400\t0.3\t1\t14 10386\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/L.1.fastq.gz\n",
      "=============================================\n",
      "16849948 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/L.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/L.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to L.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/L.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/L.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 678.08 s (40 us/read; 1.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,849,948\n",
      "Reads with adapters:                10,596,868 (62.9%)\n",
      "Reads written (passing filters):    16,849,948 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,527,492,200 bp\n",
      "Quality-trimmed:             165,132,908 bp (6.5%)\n",
      "Total written (filtered):  2,104,844,738 bp (83.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10596868 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.2%\n",
      "  C: 29.9%\n",
      "  G: 40.6%\n",
      "  T: 12.2%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2226729\t4212487.0\t0\t2226729\n",
      "2\t676704\t1053121.8\t0\t676704\n",
      "3\t315821\t263280.4\t0\t315821\n",
      "4\t143655\t65820.1\t0\t143655\n",
      "5\t141075\t16455.0\t0\t141075\n",
      "6\t85409\t4113.8\t0\t85409\n",
      "7\t93532\t1028.4\t0\t93532\n",
      "8\t126846\t257.1\t0\t126846\n",
      "9\t152959\t64.3\t0\t152736 223\n",
      "10\t167065\t16.1\t1\t160236 6829\n",
      "11\t75958\t4.0\t1\t68511 7447\n",
      "12\t89336\t1.0\t1\t83286 6050\n",
      "13\t105786\t0.3\t1\t99671 6115\n",
      "14\t115227\t0.3\t1\t109164 6063\n",
      "15\t108547\t0.3\t1\t103297 5250\n",
      "16\t122631\t0.3\t1\t115139 7492\n",
      "17\t146738\t0.3\t1\t139127 7611\n",
      "18\t156783\t0.3\t1\t149263 7520\n",
      "19\t128139\t0.3\t1\t120921 7218\n",
      "20\t160907\t0.3\t1\t154531 6376\n",
      "21\t66833\t0.3\t1\t61897 4936\n",
      "22\t102960\t0.3\t1\t99227 3733\n",
      "23\t99324\t0.3\t1\t94838 4486\n",
      "24\t156557\t0.3\t1\t149250 7307\n",
      "25\t96480\t0.3\t1\t91675 4805\n",
      "26\t142662\t0.3\t1\t135693 6969\n",
      "27\t201570\t0.3\t1\t190914 10656\n",
      "28\t136988\t0.3\t1\t131492 5496\n",
      "29\t110435\t0.3\t1\t101552 8883\n",
      "30\t117795\t0.3\t1\t112691 5104\n",
      "31\t190333\t0.3\t1\t182742 7591\n",
      "32\t121054\t0.3\t1\t114612 6442\n",
      "33\t218908\t0.3\t1\t210918 7990\n",
      "34\t52775\t0.3\t1\t49414 3361\n",
      "35\t119098\t0.3\t1\t114536 4562\n",
      "36\t209486\t0.3\t1\t202511 6975\n",
      "37\t85557\t0.3\t1\t82208 3349\n",
      "38\t146257\t0.3\t1\t142468 3789\n",
      "39\t100059\t0.3\t1\t96501 3558\n",
      "40\t139984\t0.3\t1\t135467 4517\n",
      "41\t88753\t0.3\t1\t86074 2679\n",
      "42\t114047\t0.3\t1\t111052 2995\n",
      "43\t113127\t0.3\t1\t109988 3139\n",
      "44\t86320\t0.3\t1\t83555 2765\n",
      "45\t168481\t0.3\t1\t164121 4360\n",
      "46\t85671\t0.3\t1\t82262 3409\n",
      "47\t151609\t0.3\t1\t148142 3467\n",
      "48\t56198\t0.3\t1\t53354 2844\n",
      "49\t140783\t0.3\t1\t137037 3746\n",
      "50\t43352\t0.3\t1\t41456 1896\n",
      "51\t104524\t0.3\t1\t101713 2811\n",
      "52\t56922\t0.3\t1\t54929 1993\n",
      "53\t85759\t0.3\t1\t83740 2019\n",
      "54\t116756\t0.3\t1\t114350 2406\n",
      "55\t70489\t0.3\t1\t68478 2011\n",
      "56\t78290\t0.3\t1\t76293 1997\n",
      "57\t125413\t0.3\t1\t122779 2634\n",
      "58\t87524\t0.3\t1\t85349 2175\n",
      "59\t43615\t0.3\t1\t42198 1417\n",
      "60\t99878\t0.3\t1\t97997 1881\n",
      "61\t37724\t0.3\t1\t36534 1190\n",
      "62\t40159\t0.3\t1\t38828 1331\n",
      "63\t76490\t0.3\t1\t74952 1538\n",
      "64\t52521\t0.3\t1\t51265 1256\n",
      "65\t21040\t0.3\t1\t20191 849\n",
      "66\t55495\t0.3\t1\t54321 1174\n",
      "67\t16064\t0.3\t1\t15470 594\n",
      "68\t35444\t0.3\t1\t34298 1146\n",
      "69\t37370\t0.3\t1\t36459 911\n",
      "70\t70120\t0.3\t1\t68833 1287\n",
      "71\t11901\t0.3\t1\t11482 419\n",
      "72\t20675\t0.3\t1\t20097 578\n",
      "73\t22723\t0.3\t1\t22129 594\n",
      "74\t22769\t0.3\t1\t22211 558\n",
      "75\t25057\t0.3\t1\t24424 633\n",
      "76\t34061\t0.3\t1\t33292 769\n",
      "77\t57093\t0.3\t1\t55840 1253\n",
      "78\t42732\t0.3\t1\t41355 1377\n",
      "79\t10787\t0.3\t1\t10111 676\n",
      "80\t2097\t0.3\t1\t1841 256\n",
      "81\t323\t0.3\t1\t294 29\n",
      "82\t491\t0.3\t1\t472 19\n",
      "83\t1267\t0.3\t1\t1236 31\n",
      "84\t2456\t0.3\t1\t2407 49\n",
      "85\t3619\t0.3\t1\t3569 50\n",
      "86\t4531\t0.3\t1\t4462 69\n",
      "87\t4012\t0.3\t1\t3952 60\n",
      "88\t2901\t0.3\t1\t2854 47\n",
      "89\t1850\t0.3\t1\t1812 38\n",
      "90\t1444\t0.3\t1\t1418 26\n",
      "91\t1308\t0.3\t1\t1276 32\n",
      "92\t1055\t0.3\t1\t1032 23\n",
      "93\t815\t0.3\t1\t790 25\n",
      "94\t654\t0.3\t1\t632 22\n",
      "95\t634\t0.3\t1\t615 19\n",
      "96\t528\t0.3\t1\t518 10\n",
      "97\t476\t0.3\t1\t466 10\n",
      "98\t430\t0.3\t1\t420 10\n",
      "99\t342\t0.3\t1\t334 8\n",
      "100\t315\t0.3\t1\t302 13\n",
      "101\t302\t0.3\t1\t286 16\n",
      "102\t214\t0.3\t1\t204 10\n",
      "103\t197\t0.3\t1\t189 8\n",
      "104\t165\t0.3\t1\t154 11\n",
      "105\t140\t0.3\t1\t127 13\n",
      "106\t124\t0.3\t1\t116 8\n",
      "107\t120\t0.3\t1\t111 9\n",
      "108\t105\t0.3\t1\t96 9\n",
      "109\t90\t0.3\t1\t78 12\n",
      "110\t84\t0.3\t1\t72 12\n",
      "111\t66\t0.3\t1\t60 6\n",
      "112\t78\t0.3\t1\t61 17\n",
      "113\t80\t0.3\t1\t61 19\n",
      "114\t57\t0.3\t1\t41 16\n",
      "115\t58\t0.3\t1\t45 13\n",
      "116\t55\t0.3\t1\t41 14\n",
      "117\t62\t0.3\t1\t41 21\n",
      "118\t52\t0.3\t1\t32 20\n",
      "119\t70\t0.3\t1\t44 26\n",
      "120\t43\t0.3\t1\t30 13\n",
      "121\t59\t0.3\t1\t44 15\n",
      "122\t42\t0.3\t1\t26 16\n",
      "123\t59\t0.3\t1\t31 28\n",
      "124\t34\t0.3\t1\t18 16\n",
      "125\t38\t0.3\t1\t20 18\n",
      "126\t21\t0.3\t1\t12 9\n",
      "127\t36\t0.3\t1\t15 21\n",
      "128\t19\t0.3\t1\t12 7\n",
      "129\t21\t0.3\t1\t12 9\n",
      "130\t6\t0.3\t1\t4 2\n",
      "131\t15\t0.3\t1\t6 9\n",
      "132\t10\t0.3\t1\t4 6\n",
      "133\t8\t0.3\t1\t6 2\n",
      "134\t8\t0.3\t1\t6 2\n",
      "135\t8\t0.3\t1\t6 2\n",
      "136\t4\t0.3\t1\t3 1\n",
      "137\t1\t0.3\t1\t0 1\n",
      "138\t7\t0.3\t1\t4 3\n",
      "139\t1\t0.3\t1\t1\n",
      "140\t4\t0.3\t1\t4\n",
      "141\t2\t0.3\t1\t1 1\n",
      "142\t3\t0.3\t1\t2 1\n",
      "143\t7\t0.3\t1\t5 2\n",
      "145\t6\t0.3\t1\t5 1\n",
      "146\t12\t0.3\t1\t11 1\n",
      "147\t11\t0.3\t1\t8 3\n",
      "148\t13\t0.3\t1\t12 1\n",
      "149\t4\t0.3\t1\t3 1\n",
      "150\t1\t0.3\t1\t0 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/L.2.fastq.gz\n",
      "=============================================\n",
      "16849948 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files L.1_trimmed.fq.gz and L.2_trimmed.fq.gz\n",
      "file_1: L.1_trimmed.fq.gz, file_2: L.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: L.1_trimmed.fq.gz and L.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to L.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to L.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to L.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to L.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16849948\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 547673 (3.25%)\n",
      "\n",
      "Deleting both intermediate output files L.1_trimmed.fq.gz and L.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/L.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/L.1.fastq.gz ./data_raw/L.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/M.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t345311\tAGATCGGAAGAGC\t1000000\t34.53\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 345311). Second best hit was Nextera (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/M.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/M.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/M.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to M.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/M.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/M.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 643.98 s (41 us/read; 1.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,841,289\n",
      "Reads with adapters:                10,002,840 (63.1%)\n",
      "Reads written (passing filters):    15,841,289 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,376,193,350 bp\n",
      "Quality-trimmed:              57,722,264 bp (2.4%)\n",
      "Total written (filtered):  2,063,643,699 bp (86.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10002840 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.8%\n",
      "  C: 35.5%\n",
      "  G: 28.5%\n",
      "  T: 18.8%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2017082\t3960322.2\t0\t2017082\n",
      "2\t661537\t990080.6\t0\t661537\n",
      "3\t271684\t247520.1\t0\t271684\n",
      "4\t139675\t61880.0\t0\t139675\n",
      "5\t99341\t15470.0\t0\t99341\n",
      "6\t83210\t3867.5\t0\t83210\n",
      "7\t81316\t966.9\t0\t81316\n",
      "8\t99388\t241.7\t0\t99388\n",
      "9\t129295\t60.4\t0\t129140 155\n",
      "10\t145503\t15.1\t1\t128833 16670\n",
      "11\t104817\t3.8\t1\t91187 13630\n",
      "12\t104238\t0.9\t1\t91673 12565\n",
      "13\t131484\t0.2\t1\t116220 15264\n",
      "14\t96206\t0.2\t1\t83748 12458\n",
      "15\t133495\t0.2\t1\t117853 15642\n",
      "16\t71994\t0.2\t1\t62216 9778\n",
      "17\t112921\t0.2\t1\t99382 13539\n",
      "18\t99788\t0.2\t1\t88480 11308\n",
      "19\t119259\t0.2\t1\t105775 13484\n",
      "20\t91898\t0.2\t1\t81190 10708\n",
      "21\t123744\t0.2\t1\t109504 14240\n",
      "22\t122288\t0.2\t1\t108218 14070\n",
      "23\t122849\t0.2\t1\t108743 14106\n",
      "24\t128259\t0.2\t1\t111882 16377\n",
      "25\t120195\t0.2\t1\t106540 13655\n",
      "26\t109049\t0.2\t1\t95849 13200\n",
      "27\t131576\t0.2\t1\t116303 15273\n",
      "28\t102161\t0.2\t1\t90737 11424\n",
      "29\t146476\t0.2\t1\t129212 17264\n",
      "30\t123103\t0.2\t1\t108958 14145\n",
      "31\t164591\t0.2\t1\t145824 18767\n",
      "32\t98212\t0.2\t1\t86513 11699\n",
      "33\t129745\t0.2\t1\t113223 16522\n",
      "34\t153824\t0.2\t1\t136716 17108\n",
      "35\t77560\t0.2\t1\t68461 9099\n",
      "36\t105134\t0.2\t1\t92548 12586\n",
      "37\t107663\t0.2\t1\t95332 12331\n",
      "38\t99940\t0.2\t1\t88029 11911\n",
      "39\t120455\t0.2\t1\t106228 14227\n",
      "40\t164983\t0.2\t1\t144973 20010\n",
      "41\t222926\t0.2\t1\t197437 25489\n",
      "42\t277804\t0.2\t1\t246344 31460\n",
      "43\t142581\t0.2\t1\t126590 15991\n",
      "44\t52684\t0.2\t1\t46928 5756\n",
      "45\t15469\t0.2\t1\t13636 1833\n",
      "46\t30537\t0.2\t1\t27197 3340\n",
      "47\t58284\t0.2\t1\t51891 6393\n",
      "48\t79579\t0.2\t1\t70838 8741\n",
      "49\t143548\t0.2\t1\t127955 15593\n",
      "50\t41003\t0.2\t1\t36223 4780\n",
      "51\t115591\t0.2\t1\t102174 13417\n",
      "52\t123112\t0.2\t1\t109750 13362\n",
      "53\t88384\t0.2\t1\t78527 9857\n",
      "54\t71460\t0.2\t1\t63375 8085\n",
      "55\t107125\t0.2\t1\t95102 12023\n",
      "56\t90841\t0.2\t1\t80919 9922\n",
      "57\t33224\t0.2\t1\t29492 3732\n",
      "58\t72479\t0.2\t1\t64439 8040\n",
      "59\t117937\t0.2\t1\t104930 13007\n",
      "60\t28264\t0.2\t1\t24963 3301\n",
      "61\t59627\t0.2\t1\t53200 6427\n",
      "62\t83499\t0.2\t1\t74233 9266\n",
      "63\t23726\t0.2\t1\t21127 2599\n",
      "64\t51709\t0.2\t1\t46005 5704\n",
      "65\t64641\t0.2\t1\t57477 7164\n",
      "66\t34845\t0.2\t1\t30858 3987\n",
      "67\t43126\t0.2\t1\t38459 4667\n",
      "68\t41033\t0.2\t1\t36614 4419\n",
      "69\t36089\t0.2\t1\t31826 4263\n",
      "70\t34465\t0.2\t1\t30527 3938\n",
      "71\t35820\t0.2\t1\t31353 4467\n",
      "72\t60602\t0.2\t1\t53153 7449\n",
      "73\t51427\t0.2\t1\t45040 6387\n",
      "74\t53697\t0.2\t1\t46649 7048\n",
      "75\t41811\t0.2\t1\t36816 4995\n",
      "76\t3649\t0.2\t1\t3099 550\n",
      "77\t232\t0.2\t1\t186 46\n",
      "78\t1236\t0.2\t1\t1113 123\n",
      "79\t3708\t0.2\t1\t3288 420\n",
      "80\t5775\t0.2\t1\t5132 643\n",
      "81\t6905\t0.2\t1\t6122 783\n",
      "82\t7290\t0.2\t1\t6466 824\n",
      "83\t7669\t0.2\t1\t6762 907\n",
      "84\t7496\t0.2\t1\t6624 872\n",
      "85\t7121\t0.2\t1\t6304 817\n",
      "86\t6888\t0.2\t1\t6129 759\n",
      "87\t6336\t0.2\t1\t5631 705\n",
      "88\t5653\t0.2\t1\t5077 576\n",
      "89\t5300\t0.2\t1\t4727 573\n",
      "90\t4639\t0.2\t1\t4146 493\n",
      "91\t4153\t0.2\t1\t3701 452\n",
      "92\t3536\t0.2\t1\t3144 392\n",
      "93\t3110\t0.2\t1\t2742 368\n",
      "94\t2844\t0.2\t1\t2536 308\n",
      "95\t2522\t0.2\t1\t2242 280\n",
      "96\t2365\t0.2\t1\t2096 269\n",
      "97\t2281\t0.2\t1\t2041 240\n",
      "98\t2142\t0.2\t1\t1914 228\n",
      "99\t1935\t0.2\t1\t1713 222\n",
      "100\t1747\t0.2\t1\t1581 166\n",
      "101\t1487\t0.2\t1\t1316 171\n",
      "102\t1264\t0.2\t1\t1113 151\n",
      "103\t989\t0.2\t1\t847 142\n",
      "104\t905\t0.2\t1\t798 107\n",
      "105\t749\t0.2\t1\t658 91\n",
      "106\t689\t0.2\t1\t602 87\n",
      "107\t690\t0.2\t1\t605 85\n",
      "108\t640\t0.2\t1\t569 71\n",
      "109\t571\t0.2\t1\t495 76\n",
      "110\t486\t0.2\t1\t418 68\n",
      "111\t405\t0.2\t1\t353 52\n",
      "112\t348\t0.2\t1\t303 45\n",
      "113\t349\t0.2\t1\t301 48\n",
      "114\t289\t0.2\t1\t257 32\n",
      "115\t233\t0.2\t1\t199 34\n",
      "116\t210\t0.2\t1\t172 38\n",
      "117\t215\t0.2\t1\t174 41\n",
      "118\t189\t0.2\t1\t162 27\n",
      "119\t185\t0.2\t1\t155 30\n",
      "120\t135\t0.2\t1\t110 25\n",
      "121\t140\t0.2\t1\t111 29\n",
      "122\t120\t0.2\t1\t98 22\n",
      "123\t105\t0.2\t1\t87 18\n",
      "124\t93\t0.2\t1\t74 19\n",
      "125\t115\t0.2\t1\t86 29\n",
      "126\t84\t0.2\t1\t59 25\n",
      "127\t70\t0.2\t1\t47 23\n",
      "128\t60\t0.2\t1\t46 14\n",
      "129\t46\t0.2\t1\t30 16\n",
      "130\t43\t0.2\t1\t29 14\n",
      "131\t47\t0.2\t1\t30 17\n",
      "132\t21\t0.2\t1\t16 5\n",
      "133\t47\t0.2\t1\t35 12\n",
      "134\t46\t0.2\t1\t41 5\n",
      "135\t48\t0.2\t1\t44 4\n",
      "136\t36\t0.2\t1\t31 5\n",
      "137\t36\t0.2\t1\t35 1\n",
      "138\t46\t0.2\t1\t41 5\n",
      "139\t33\t0.2\t1\t24 9\n",
      "140\t54\t0.2\t1\t52 2\n",
      "141\t67\t0.2\t1\t64 3\n",
      "142\t38\t0.2\t1\t30 8\n",
      "143\t43\t0.2\t1\t40 3\n",
      "144\t19\t0.2\t1\t18 1\n",
      "145\t78\t0.2\t1\t76 2\n",
      "146\t174\t0.2\t1\t168 6\n",
      "147\t81\t0.2\t1\t73 8\n",
      "148\t194\t0.2\t1\t190 4\n",
      "149\t62\t0.2\t1\t42 20\n",
      "150\t8347\t0.2\t1\t51 8296\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/M.1.fastq.gz\n",
      "=============================================\n",
      "15841289 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/M.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/M.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to M.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/M.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/M.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 629.20 s (40 us/read; 1.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,841,289\n",
      "Reads with adapters:                 9,283,304 (58.6%)\n",
      "Reads written (passing filters):    15,841,289 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,376,193,350 bp\n",
      "Quality-trimmed:             279,872,350 bp (11.8%)\n",
      "Total written (filtered):  1,877,531,156 bp (79.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9283304 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.7%\n",
      "  C: 37.1%\n",
      "  G: 33.1%\n",
      "  T: 13.7%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2064786\t3960322.2\t0\t2064786\n",
      "2\t610852\t990080.6\t0\t610852\n",
      "3\t274732\t247520.1\t0\t274732\n",
      "4\t130733\t61880.0\t0\t130733\n",
      "5\t125456\t15470.0\t0\t125456\n",
      "6\t75040\t3867.5\t0\t75040\n",
      "7\t81560\t966.9\t0\t81560\n",
      "8\t109585\t241.7\t0\t109585\n",
      "9\t133715\t60.4\t0\t133542 173\n",
      "10\t146020\t15.1\t1\t137843 8177\n",
      "11\t68787\t3.8\t1\t61052 7735\n",
      "12\t80239\t0.9\t1\t73539 6700\n",
      "13\t93729\t0.2\t1\t86956 6773\n",
      "14\t101130\t0.2\t1\t94130 7000\n",
      "15\t94670\t0.2\t1\t88910 5760\n",
      "16\t107804\t0.2\t1\t99595 8209\n",
      "17\t127501\t0.2\t1\t119212 8289\n",
      "18\t138211\t0.2\t1\t129727 8484\n",
      "19\t115670\t0.2\t1\t107844 7826\n",
      "20\t137619\t0.2\t1\t129848 7771\n",
      "21\t57421\t0.2\t1\t52396 5025\n",
      "22\t85935\t0.2\t1\t81637 4298\n",
      "23\t83184\t0.2\t1\t78394 4790\n",
      "24\t129986\t0.2\t1\t121924 8062\n",
      "25\t88866\t0.2\t1\t82946 5920\n",
      "26\t120048\t0.2\t1\t112290 7758\n",
      "27\t185172\t0.2\t1\t172546 12626\n",
      "28\t120804\t0.2\t1\t114553 6251\n",
      "29\t101116\t0.2\t1\t91299 9817\n",
      "30\t117920\t0.2\t1\t111406 6514\n",
      "31\t146789\t0.2\t1\t138876 7913\n",
      "32\t105371\t0.2\t1\t99555 5816\n",
      "33\t122352\t0.2\t1\t115444 6908\n",
      "34\t191124\t0.2\t1\t182713 8411\n",
      "35\t91014\t0.2\t1\t85848 5166\n",
      "36\t140279\t0.2\t1\t133340 6939\n",
      "37\t97761\t0.2\t1\t93209 4552\n",
      "38\t64921\t0.2\t1\t60986 3935\n",
      "39\t94755\t0.2\t1\t90084 4671\n",
      "40\t51515\t0.2\t1\t47662 3853\n",
      "41\t134625\t0.2\t1\t129107 5518\n",
      "42\t65841\t0.2\t1\t62833 3008\n",
      "43\t116879\t0.2\t1\t111815 5064\n",
      "44\t60553\t0.2\t1\t57746 2807\n",
      "45\t158516\t0.2\t1\t152026 6490\n",
      "46\t75474\t0.2\t1\t71252 4222\n",
      "47\t109394\t0.2\t1\t105493 3901\n",
      "48\t52461\t0.2\t1\t48653 3808\n",
      "49\t122258\t0.2\t1\t117099 5159\n",
      "50\t33873\t0.2\t1\t31764 2109\n",
      "51\t92666\t0.2\t1\t88717 3949\n",
      "52\t45384\t0.2\t1\t42735 2649\n",
      "53\t74008\t0.2\t1\t71138 2870\n",
      "54\t95778\t0.2\t1\t92429 3349\n",
      "55\t61646\t0.2\t1\t58827 2819\n",
      "56\t66589\t0.2\t1\t64020 2569\n",
      "57\t101509\t0.2\t1\t98124 3385\n",
      "58\t75558\t0.2\t1\t72253 3305\n",
      "59\t35890\t0.2\t1\t33995 1895\n",
      "60\t83093\t0.2\t1\t80624 2469\n",
      "61\t32080\t0.2\t1\t30458 1622\n",
      "62\t33319\t0.2\t1\t31477 1842\n",
      "63\t62667\t0.2\t1\t60580 2087\n",
      "64\t45389\t0.2\t1\t43412 1977\n",
      "65\t17828\t0.2\t1\t16743 1085\n",
      "66\t47636\t0.2\t1\t45829 1807\n",
      "67\t13125\t0.2\t1\t12397 728\n",
      "68\t30157\t0.2\t1\t28615 1542\n",
      "69\t31331\t0.2\t1\t29963 1368\n",
      "70\t59837\t0.2\t1\t57614 2223\n",
      "71\t9984\t0.2\t1\t9453 531\n",
      "72\t17215\t0.2\t1\t16437 778\n",
      "73\t18289\t0.2\t1\t17490 799\n",
      "74\t18710\t0.2\t1\t17933 777\n",
      "75\t20859\t0.2\t1\t20018 841\n",
      "76\t27968\t0.2\t1\t26948 1020\n",
      "77\t47203\t0.2\t1\t45509 1694\n",
      "78\t36233\t0.2\t1\t34263 1970\n",
      "79\t9653\t0.2\t1\t8742 911\n",
      "80\t2063\t0.2\t1\t1763 300\n",
      "81\t260\t0.2\t1\t221 39\n",
      "82\t361\t0.2\t1\t334 27\n",
      "83\t928\t0.2\t1\t892 36\n",
      "84\t1852\t0.2\t1\t1786 66\n",
      "85\t2755\t0.2\t1\t2655 100\n",
      "86\t3428\t0.2\t1\t3307 121\n",
      "87\t3240\t0.2\t1\t3123 117\n",
      "88\t2414\t0.2\t1\t2314 100\n",
      "89\t1606\t0.2\t1\t1548 58\n",
      "90\t1284\t0.2\t1\t1234 50\n",
      "91\t1055\t0.2\t1\t1016 39\n",
      "92\t850\t0.2\t1\t811 39\n",
      "93\t642\t0.2\t1\t611 31\n",
      "94\t597\t0.2\t1\t568 29\n",
      "95\t466\t0.2\t1\t445 21\n",
      "96\t432\t0.2\t1\t417 15\n",
      "97\t407\t0.2\t1\t396 11\n",
      "98\t340\t0.2\t1\t322 18\n",
      "99\t342\t0.2\t1\t331 11\n",
      "100\t278\t0.2\t1\t261 17\n",
      "101\t227\t0.2\t1\t206 21\n",
      "102\t168\t0.2\t1\t155 13\n",
      "103\t126\t0.2\t1\t119 7\n",
      "104\t142\t0.2\t1\t128 14\n",
      "105\t116\t0.2\t1\t104 12\n",
      "106\t87\t0.2\t1\t76 11\n",
      "107\t99\t0.2\t1\t92 7\n",
      "108\t97\t0.2\t1\t87 10\n",
      "109\t68\t0.2\t1\t53 15\n",
      "110\t87\t0.2\t1\t75 12\n",
      "111\t54\t0.2\t1\t48 6\n",
      "112\t55\t0.2\t1\t48 7\n",
      "113\t59\t0.2\t1\t47 12\n",
      "114\t49\t0.2\t1\t40 9\n",
      "115\t46\t0.2\t1\t39 7\n",
      "116\t42\t0.2\t1\t31 11\n",
      "117\t50\t0.2\t1\t41 9\n",
      "118\t40\t0.2\t1\t24 16\n",
      "119\t34\t0.2\t1\t28 6\n",
      "120\t40\t0.2\t1\t29 11\n",
      "121\t37\t0.2\t1\t24 13\n",
      "122\t28\t0.2\t1\t21 7\n",
      "123\t32\t0.2\t1\t23 9\n",
      "124\t35\t0.2\t1\t20 15\n",
      "125\t33\t0.2\t1\t16 17\n",
      "126\t16\t0.2\t1\t12 4\n",
      "127\t23\t0.2\t1\t16 7\n",
      "128\t16\t0.2\t1\t8 8\n",
      "129\t6\t0.2\t1\t1 5\n",
      "130\t8\t0.2\t1\t7 1\n",
      "131\t7\t0.2\t1\t3 4\n",
      "132\t4\t0.2\t1\t4\n",
      "133\t10\t0.2\t1\t7 3\n",
      "134\t14\t0.2\t1\t7 7\n",
      "135\t8\t0.2\t1\t4 4\n",
      "136\t8\t0.2\t1\t6 2\n",
      "137\t3\t0.2\t1\t1 2\n",
      "138\t4\t0.2\t1\t1 3\n",
      "139\t3\t0.2\t1\t3\n",
      "140\t5\t0.2\t1\t3 2\n",
      "141\t6\t0.2\t1\t4 2\n",
      "142\t4\t0.2\t1\t2 2\n",
      "143\t3\t0.2\t1\t2 1\n",
      "144\t2\t0.2\t1\t1 1\n",
      "145\t4\t0.2\t1\t4\n",
      "146\t20\t0.2\t1\t18 2\n",
      "147\t7\t0.2\t1\t4 3\n",
      "148\t11\t0.2\t1\t11\n",
      "149\t6\t0.2\t1\t3 3\n",
      "150\t5\t0.2\t1\t4 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/M.2.fastq.gz\n",
      "=============================================\n",
      "15841289 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files M.1_trimmed.fq.gz and M.2_trimmed.fq.gz\n",
      "file_1: M.1_trimmed.fq.gz, file_2: M.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: M.1_trimmed.fq.gz and M.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to M.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to M.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to M.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to M.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 15841289\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1328251 (8.38%)\n",
      "\n",
      "Deleting both intermediate output files M.1_trimmed.fq.gz and M.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/M.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/M.1.fastq.gz ./data_raw/M.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/N.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t400810\tAGATCGGAAGAGC\t1000000\t40.08\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 400810). Second best hit was Nextera (count: 1)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/N.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/N.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/N.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to N.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/N.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/N.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 757.61 s (41 us/read; 1.47 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,548,306\n",
      "Reads with adapters:                12,331,999 (66.5%)\n",
      "Reads written (passing filters):    18,548,306 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,782,245,900 bp\n",
      "Quality-trimmed:              73,507,805 bp (2.6%)\n",
      "Total written (filtered):  2,347,918,757 bp (84.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 12331999 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.3%\n",
      "  C: 35.3%\n",
      "  G: 29.6%\n",
      "  T: 18.3%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2137377\t4637076.5\t0\t2137377\n",
      "2\t717738\t1159269.1\t0\t717738\n",
      "3\t303527\t289817.3\t0\t303527\n",
      "4\t155761\t72454.3\t0\t155761\n",
      "5\t109399\t18113.6\t0\t109399\n",
      "6\t90212\t4528.4\t0\t90212\n",
      "7\t89935\t1132.1\t0\t89935\n",
      "8\t109976\t283.0\t0\t109976\n",
      "9\t143628\t70.8\t0\t143429 199\n",
      "10\t160477\t17.7\t1\t145180 15297\n",
      "11\t113371\t4.4\t1\t100542 12829\n",
      "12\t113921\t1.1\t1\t101936 11985\n",
      "13\t142967\t0.3\t1\t129140 13827\n",
      "14\t104678\t0.3\t1\t93178 11500\n",
      "15\t148537\t0.3\t1\t134214 14323\n",
      "16\t79743\t0.3\t1\t70218 9525\n",
      "17\t125319\t0.3\t1\t112730 12589\n",
      "18\t111381\t0.3\t1\t100869 10512\n",
      "19\t132630\t0.3\t1\t120564 12066\n",
      "20\t104567\t0.3\t1\t94586 9981\n",
      "21\t139597\t0.3\t1\t125917 13680\n",
      "22\t141247\t0.3\t1\t127860 13387\n",
      "23\t141011\t0.3\t1\t127503 13508\n",
      "24\t146195\t0.3\t1\t130430 15765\n",
      "25\t138516\t0.3\t1\t125198 13318\n",
      "26\t125969\t0.3\t1\t113236 12733\n",
      "27\t152987\t0.3\t1\t138172 14815\n",
      "28\t119606\t0.3\t1\t108542 11064\n",
      "29\t172431\t0.3\t1\t155676 16755\n",
      "30\t146919\t0.3\t1\t133402 13517\n",
      "31\t195451\t0.3\t1\t176832 18619\n",
      "32\t118410\t0.3\t1\t107021 11389\n",
      "33\t159008\t0.3\t1\t142299 16709\n",
      "34\t186522\t0.3\t1\t168959 17563\n",
      "35\t96122\t0.3\t1\t87105 9017\n",
      "36\t129147\t0.3\t1\t116716 12431\n",
      "37\t136670\t0.3\t1\t122939 13731\n",
      "38\t125319\t0.3\t1\t112785 12534\n",
      "39\t152205\t0.3\t1\t137141 15064\n",
      "40\t214540\t0.3\t1\t192761 21779\n",
      "41\t294121\t0.3\t1\t266586 27535\n",
      "42\t366875\t0.3\t1\t333479 33396\n",
      "43\t188207\t0.3\t1\t170870 17337\n",
      "44\t68469\t0.3\t1\t62231 6238\n",
      "45\t21123\t0.3\t1\t19086 2037\n",
      "46\t40735\t0.3\t1\t36997 3738\n",
      "47\t78178\t0.3\t1\t71242 6936\n",
      "48\t108204\t0.3\t1\t98657 9547\n",
      "49\t200769\t0.3\t1\t183842 16927\n",
      "50\t57873\t0.3\t1\t52077 5796\n",
      "51\t163370\t0.3\t1\t147934 15436\n",
      "52\t176239\t0.3\t1\t160317 15922\n",
      "53\t128530\t0.3\t1\t116960 11570\n",
      "54\t104666\t0.3\t1\t95108 9558\n",
      "55\t160598\t0.3\t1\t146339 14259\n",
      "56\t135710\t0.3\t1\t123520 12190\n",
      "57\t50254\t0.3\t1\t45630 4624\n",
      "58\t111470\t0.3\t1\t101742 9728\n",
      "59\t185583\t0.3\t1\t169203 16380\n",
      "60\t44652\t0.3\t1\t40467 4185\n",
      "61\t94335\t0.3\t1\t86087 8248\n",
      "62\t134981\t0.3\t1\t122987 11994\n",
      "63\t40099\t0.3\t1\t36504 3595\n",
      "64\t86077\t0.3\t1\t78362 7715\n",
      "65\t110402\t0.3\t1\t100580 9822\n",
      "66\t59613\t0.3\t1\t54152 5461\n",
      "67\t74080\t0.3\t1\t67575 6505\n",
      "68\t72638\t0.3\t1\t66146 6492\n",
      "69\t64297\t0.3\t1\t58278 6019\n",
      "70\t61203\t0.3\t1\t55522 5681\n",
      "71\t66138\t0.3\t1\t59022 7116\n",
      "72\t114523\t0.3\t1\t102678 11845\n",
      "73\t98237\t0.3\t1\t87983 10254\n",
      "74\t102798\t0.3\t1\t91148 11650\n",
      "75\t79164\t0.3\t1\t71339 7825\n",
      "76\t6894\t0.3\t1\t5957 937\n",
      "77\t426\t0.3\t1\t365 61\n",
      "78\t2435\t0.3\t1\t2216 219\n",
      "79\t7276\t0.3\t1\t6606 670\n",
      "80\t11194\t0.3\t1\t10177 1017\n",
      "81\t13614\t0.3\t1\t12393 1221\n",
      "82\t14572\t0.3\t1\t13287 1285\n",
      "83\t14911\t0.3\t1\t13569 1342\n",
      "84\t14656\t0.3\t1\t13313 1343\n",
      "85\t14013\t0.3\t1\t12756 1257\n",
      "86\t13406\t0.3\t1\t12241 1165\n",
      "87\t12911\t0.3\t1\t11754 1157\n",
      "88\t11666\t0.3\t1\t10665 1001\n",
      "89\t10727\t0.3\t1\t9775 952\n",
      "90\t9414\t0.3\t1\t8582 832\n",
      "91\t8493\t0.3\t1\t7800 693\n",
      "92\t7478\t0.3\t1\t6849 629\n",
      "93\t6434\t0.3\t1\t5880 554\n",
      "94\t5505\t0.3\t1\t5020 485\n",
      "95\t4971\t0.3\t1\t4534 437\n",
      "96\t4662\t0.3\t1\t4227 435\n",
      "97\t4395\t0.3\t1\t4031 364\n",
      "98\t4109\t0.3\t1\t3743 366\n",
      "99\t3744\t0.3\t1\t3406 338\n",
      "100\t3189\t0.3\t1\t2927 262\n",
      "101\t2842\t0.3\t1\t2599 243\n",
      "102\t2466\t0.3\t1\t2240 226\n",
      "103\t1936\t0.3\t1\t1759 177\n",
      "104\t1639\t0.3\t1\t1475 164\n",
      "105\t1376\t0.3\t1\t1252 124\n",
      "106\t1261\t0.3\t1\t1117 144\n",
      "107\t1266\t0.3\t1\t1120 146\n",
      "108\t1108\t0.3\t1\t984 124\n",
      "109\t978\t0.3\t1\t883 95\n",
      "110\t869\t0.3\t1\t781 88\n",
      "111\t743\t0.3\t1\t656 87\n",
      "112\t604\t0.3\t1\t528 76\n",
      "113\t528\t0.3\t1\t472 56\n",
      "114\t448\t0.3\t1\t382 66\n",
      "115\t399\t0.3\t1\t350 49\n",
      "116\t360\t0.3\t1\t300 60\n",
      "117\t383\t0.3\t1\t309 74\n",
      "118\t257\t0.3\t1\t213 44\n",
      "119\t296\t0.3\t1\t243 53\n",
      "120\t217\t0.3\t1\t187 30\n",
      "121\t233\t0.3\t1\t185 48\n",
      "122\t172\t0.3\t1\t124 48\n",
      "123\t193\t0.3\t1\t151 42\n",
      "124\t156\t0.3\t1\t122 34\n",
      "125\t198\t0.3\t1\t138 60\n",
      "126\t124\t0.3\t1\t83 41\n",
      "127\t142\t0.3\t1\t109 33\n",
      "128\t115\t0.3\t1\t83 32\n",
      "129\t124\t0.3\t1\t87 37\n",
      "130\t70\t0.3\t1\t48 22\n",
      "131\t102\t0.3\t1\t68 34\n",
      "132\t52\t0.3\t1\t40 12\n",
      "133\t63\t0.3\t1\t47 16\n",
      "134\t59\t0.3\t1\t54 5\n",
      "135\t55\t0.3\t1\t45 10\n",
      "136\t40\t0.3\t1\t27 13\n",
      "137\t46\t0.3\t1\t38 8\n",
      "138\t39\t0.3\t1\t35 4\n",
      "139\t40\t0.3\t1\t31 9\n",
      "140\t31\t0.3\t1\t23 8\n",
      "141\t62\t0.3\t1\t59 3\n",
      "142\t40\t0.3\t1\t29 11\n",
      "143\t42\t0.3\t1\t39 3\n",
      "144\t23\t0.3\t1\t20 3\n",
      "145\t165\t0.3\t1\t138 27\n",
      "146\t234\t0.3\t1\t224 10\n",
      "147\t133\t0.3\t1\t128 5\n",
      "148\t309\t0.3\t1\t307 2\n",
      "149\t117\t0.3\t1\t93 24\n",
      "150\t20522\t0.3\t1\t79 20443\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/N.1.fastq.gz\n",
      "=============================================\n",
      "18548306 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/N.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/N.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to N.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/N.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/N.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 733.69 s (40 us/read; 1.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,548,306\n",
      "Reads with adapters:                11,559,234 (62.3%)\n",
      "Reads written (passing filters):    18,548,306 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,782,245,900 bp\n",
      "Quality-trimmed:             292,523,291 bp (10.5%)\n",
      "Total written (filtered):  2,175,206,190 bp (78.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11559234 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.9%\n",
      "  C: 37.1%\n",
      "  G: 34.4%\n",
      "  T: 13.3%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2222614\t4637076.5\t0\t2222614\n",
      "2\t669690\t1159269.1\t0\t669690\n",
      "3\t298389\t289817.3\t0\t298389\n",
      "4\t137035\t72454.3\t0\t137035\n",
      "5\t136844\t18113.6\t0\t136844\n",
      "6\t81513\t4528.4\t0\t81513\n",
      "7\t90374\t1132.1\t0\t90374\n",
      "8\t121542\t283.0\t0\t121542\n",
      "9\t148091\t70.8\t0\t147835 256\n",
      "10\t164016\t17.7\t1\t154867 9149\n",
      "11\t78106\t4.4\t1\t68921 9185\n",
      "12\t91431\t1.1\t1\t83720 7711\n",
      "13\t105862\t0.3\t1\t98369 7493\n",
      "14\t116072\t0.3\t1\t108045 8027\n",
      "15\t106955\t0.3\t1\t100620 6335\n",
      "16\t122819\t0.3\t1\t113370 9449\n",
      "17\t148543\t0.3\t1\t138942 9601\n",
      "18\t160047\t0.3\t1\t150691 9356\n",
      "19\t146918\t0.3\t1\t137276 9642\n",
      "20\t153902\t0.3\t1\t145581 8321\n",
      "21\t65201\t0.3\t1\t59490 5711\n",
      "22\t100001\t0.3\t1\t95353 4648\n",
      "23\t100495\t0.3\t1\t94883 5612\n",
      "24\t171776\t0.3\t1\t161494 10282\n",
      "25\t99135\t0.3\t1\t92944 6191\n",
      "26\t148356\t0.3\t1\t139286 9070\n",
      "27\t206101\t0.3\t1\t192807 13294\n",
      "28\t137768\t0.3\t1\t130776 6992\n",
      "29\t123703\t0.3\t1\t112033 11670\n",
      "30\t147298\t0.3\t1\t140030 7268\n",
      "31\t169785\t0.3\t1\t161276 8509\n",
      "32\t129038\t0.3\t1\t122317 6721\n",
      "33\t157726\t0.3\t1\t150365 7361\n",
      "34\t138781\t0.3\t1\t131397 7384\n",
      "35\t179842\t0.3\t1\t171972 7870\n",
      "36\t75460\t0.3\t1\t71459 4001\n",
      "37\t107948\t0.3\t1\t101095 6853\n",
      "38\t123288\t0.3\t1\t118252 5036\n",
      "39\t169835\t0.3\t1\t162955 6880\n",
      "40\t140729\t0.3\t1\t131846 8883\n",
      "41\t108900\t0.3\t1\t103148 5752\n",
      "42\t168299\t0.3\t1\t162452 5847\n",
      "43\t140833\t0.3\t1\t135049 5784\n",
      "44\t108213\t0.3\t1\t103716 4497\n",
      "45\t203378\t0.3\t1\t195638 7740\n",
      "46\t106622\t0.3\t1\t100762 5860\n",
      "47\t182544\t0.3\t1\t176633 5911\n",
      "48\t72750\t0.3\t1\t67417 5333\n",
      "49\t171688\t0.3\t1\t165054 6634\n",
      "50\t52932\t0.3\t1\t49959 2973\n",
      "51\t128618\t0.3\t1\t123387 5231\n",
      "52\t67281\t0.3\t1\t63754 3527\n",
      "53\t106430\t0.3\t1\t102848 3582\n",
      "54\t149753\t0.3\t1\t145118 4635\n",
      "55\t89204\t0.3\t1\t85376 3828\n",
      "56\t104715\t0.3\t1\t101008 3707\n",
      "57\t179650\t0.3\t1\t174401 5249\n",
      "58\t119579\t0.3\t1\t115058 4521\n",
      "59\t57594\t0.3\t1\t54910 2684\n",
      "60\t144520\t0.3\t1\t140760 3760\n",
      "61\t51765\t0.3\t1\t49332 2433\n",
      "62\t54648\t0.3\t1\t52061 2587\n",
      "63\t116573\t0.3\t1\t113332 3241\n",
      "64\t74769\t0.3\t1\t71893 2876\n",
      "65\t30973\t0.3\t1\t29251 1722\n",
      "66\t82168\t0.3\t1\t79481 2687\n",
      "67\t23546\t0.3\t1\t22392 1154\n",
      "68\t52784\t0.3\t1\t50440 2344\n",
      "69\t56556\t0.3\t1\t54441 2115\n",
      "70\t108699\t0.3\t1\t105319 3380\n",
      "71\t18368\t0.3\t1\t17495 873\n",
      "72\t32187\t0.3\t1\t30824 1363\n",
      "73\t35226\t0.3\t1\t33921 1305\n",
      "74\t36350\t0.3\t1\t34954 1396\n",
      "75\t40674\t0.3\t1\t39275 1399\n",
      "76\t55482\t0.3\t1\t53721 1761\n",
      "77\t91547\t0.3\t1\t88653 2894\n",
      "78\t67219\t0.3\t1\t64023 3196\n",
      "79\t17015\t0.3\t1\t15499 1516\n",
      "80\t3401\t0.3\t1\t2853 548\n",
      "81\t475\t0.3\t1\t419 56\n",
      "82\t710\t0.3\t1\t668 42\n",
      "83\t1877\t0.3\t1\t1808 69\n",
      "84\t3705\t0.3\t1\t3584 121\n",
      "85\t5646\t0.3\t1\t5502 144\n",
      "86\t7129\t0.3\t1\t6964 165\n",
      "87\t6657\t0.3\t1\t6469 188\n",
      "88\t4993\t0.3\t1\t4825 168\n",
      "89\t3297\t0.3\t1\t3199 98\n",
      "90\t2710\t0.3\t1\t2627 83\n",
      "91\t2334\t0.3\t1\t2247 87\n",
      "92\t1834\t0.3\t1\t1766 68\n",
      "93\t1478\t0.3\t1\t1406 72\n",
      "94\t1237\t0.3\t1\t1189 48\n",
      "95\t1014\t0.3\t1\t966 48\n",
      "96\t902\t0.3\t1\t865 37\n",
      "97\t808\t0.3\t1\t780 28\n",
      "98\t746\t0.3\t1\t717 29\n",
      "99\t675\t0.3\t1\t634 41\n",
      "100\t558\t0.3\t1\t538 20\n",
      "101\t478\t0.3\t1\t455 23\n",
      "102\t408\t0.3\t1\t382 26\n",
      "103\t310\t0.3\t1\t293 17\n",
      "104\t234\t0.3\t1\t221 13\n",
      "105\t222\t0.3\t1\t201 21\n",
      "106\t186\t0.3\t1\t168 18\n",
      "107\t165\t0.3\t1\t149 16\n",
      "108\t174\t0.3\t1\t161 13\n",
      "109\t154\t0.3\t1\t139 15\n",
      "110\t122\t0.3\t1\t110 12\n",
      "111\t140\t0.3\t1\t120 20\n",
      "112\t89\t0.3\t1\t67 22\n",
      "113\t112\t0.3\t1\t92 20\n",
      "114\t74\t0.3\t1\t59 15\n",
      "115\t68\t0.3\t1\t54 14\n",
      "116\t60\t0.3\t1\t42 18\n",
      "117\t87\t0.3\t1\t71 16\n",
      "118\t57\t0.3\t1\t39 18\n",
      "119\t80\t0.3\t1\t58 22\n",
      "120\t50\t0.3\t1\t35 15\n",
      "121\t87\t0.3\t1\t57 30\n",
      "122\t77\t0.3\t1\t44 33\n",
      "123\t65\t0.3\t1\t39 26\n",
      "124\t56\t0.3\t1\t33 23\n",
      "125\t56\t0.3\t1\t35 21\n",
      "126\t38\t0.3\t1\t23 15\n",
      "127\t50\t0.3\t1\t31 19\n",
      "128\t27\t0.3\t1\t15 12\n",
      "129\t27\t0.3\t1\t17 10\n",
      "130\t20\t0.3\t1\t11 9\n",
      "131\t23\t0.3\t1\t18 5\n",
      "132\t11\t0.3\t1\t7 4\n",
      "133\t16\t0.3\t1\t11 5\n",
      "134\t11\t0.3\t1\t9 2\n",
      "135\t7\t0.3\t1\t4 3\n",
      "136\t2\t0.3\t1\t1 1\n",
      "137\t6\t0.3\t1\t4 2\n",
      "138\t6\t0.3\t1\t2 4\n",
      "139\t3\t0.3\t1\t2 1\n",
      "140\t5\t0.3\t1\t0 5\n",
      "141\t9\t0.3\t1\t7 2\n",
      "142\t6\t0.3\t1\t3 3\n",
      "143\t6\t0.3\t1\t6\n",
      "144\t3\t0.3\t1\t1 2\n",
      "145\t11\t0.3\t1\t8 3\n",
      "146\t22\t0.3\t1\t20 2\n",
      "147\t13\t0.3\t1\t10 3\n",
      "148\t21\t0.3\t1\t19 2\n",
      "149\t8\t0.3\t1\t6 2\n",
      "150\t5\t0.3\t1\t3 2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/N.2.fastq.gz\n",
      "=============================================\n",
      "18548306 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files N.1_trimmed.fq.gz and N.2_trimmed.fq.gz\n",
      "file_1: N.1_trimmed.fq.gz, file_2: N.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: N.1_trimmed.fq.gz and N.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to N.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to N.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to N.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to N.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 18548306\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1269884 (6.85%)\n",
      "\n",
      "Deleting both intermediate output files N.1_trimmed.fq.gz and N.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/N.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/N.1.fastq.gz ./data_raw/N.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/O.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t394544\tAGATCGGAAGAGC\t1000000\t39.45\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 394544). Second best hit was Nextera (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/O.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/O.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/O.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to O.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/O.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/O.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 657.50 s (40 us/read; 1.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,285,961\n",
      "Reads with adapters:                10,805,267 (66.3%)\n",
      "Reads written (passing filters):    16,285,961 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,442,894,150 bp\n",
      "Quality-trimmed:              69,006,645 bp (2.8%)\n",
      "Total written (filtered):  2,060,503,858 bp (84.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10805267 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.1%\n",
      "  C: 34.7%\n",
      "  G: 29.4%\n",
      "  T: 18.0%\n",
      "  none/other: 0.8%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1963000\t4071490.2\t0\t1963000\n",
      "2\t620154\t1017872.6\t0\t620154\n",
      "3\t267317\t254468.1\t0\t267317\n",
      "4\t137316\t63617.0\t0\t137316\n",
      "5\t97831\t15904.3\t0\t97831\n",
      "6\t81131\t3976.1\t0\t81131\n",
      "7\t80452\t994.0\t0\t80452\n",
      "8\t98129\t248.5\t0\t98129\n",
      "9\t127006\t62.1\t0\t126875 131\n",
      "10\t141666\t15.5\t1\t128470 13196\n",
      "11\t99286\t3.9\t1\t88430 10856\n",
      "12\t100775\t1.0\t1\t90632 10143\n",
      "13\t126888\t0.2\t1\t114675 12213\n",
      "14\t92174\t0.2\t1\t82437 9737\n",
      "15\t130233\t0.2\t1\t118075 12158\n",
      "16\t69885\t0.2\t1\t61900 7985\n",
      "17\t109631\t0.2\t1\t98997 10634\n",
      "18\t99036\t0.2\t1\t90234 8802\n",
      "19\t116843\t0.2\t1\t106302 10541\n",
      "20\t90090\t0.2\t1\t81568 8522\n",
      "21\t121583\t0.2\t1\t110330 11253\n",
      "22\t122353\t0.2\t1\t110912 11441\n",
      "23\t122117\t0.2\t1\t110570 11547\n",
      "24\t125744\t0.2\t1\t112558 13186\n",
      "25\t117560\t0.2\t1\t106541 11019\n",
      "26\t107799\t0.2\t1\t97184 10615\n",
      "27\t130266\t0.2\t1\t118029 12237\n",
      "28\t103114\t0.2\t1\t93991 9123\n",
      "29\t147737\t0.2\t1\t133716 14021\n",
      "30\t125503\t0.2\t1\t113941 11562\n",
      "31\t168398\t0.2\t1\t152611 15787\n",
      "32\t100872\t0.2\t1\t91426 9446\n",
      "33\t135968\t0.2\t1\t121937 14031\n",
      "34\t158123\t0.2\t1\t143679 14444\n",
      "35\t81517\t0.2\t1\t74020 7497\n",
      "36\t107964\t0.2\t1\t97785 10179\n",
      "37\t116282\t0.2\t1\t105002 11280\n",
      "38\t105806\t0.2\t1\t95401 10405\n",
      "39\t129531\t0.2\t1\t116790 12741\n",
      "40\t182165\t0.2\t1\t164056 18109\n",
      "41\t246961\t0.2\t1\t223817 23144\n",
      "42\t311464\t0.2\t1\t283417 28047\n",
      "43\t158256\t0.2\t1\t143923 14333\n",
      "44\t57880\t0.2\t1\t52643 5237\n",
      "45\t16852\t0.2\t1\t15277 1575\n",
      "46\t34020\t0.2\t1\t30938 3082\n",
      "47\t66427\t0.2\t1\t60602 5825\n",
      "48\t91993\t0.2\t1\t83829 8164\n",
      "49\t171324\t0.2\t1\t156834 14490\n",
      "50\t47612\t0.2\t1\t42897 4715\n",
      "51\t139209\t0.2\t1\t126074 13135\n",
      "52\t151201\t0.2\t1\t137942 13259\n",
      "53\t109869\t0.2\t1\t99960 9909\n",
      "54\t89966\t0.2\t1\t81827 8139\n",
      "55\t139467\t0.2\t1\t127249 12218\n",
      "56\t118897\t0.2\t1\t108467 10430\n",
      "57\t43491\t0.2\t1\t39516 3975\n",
      "58\t97131\t0.2\t1\t88587 8544\n",
      "59\t163154\t0.2\t1\t148849 14305\n",
      "60\t38846\t0.2\t1\t35244 3602\n",
      "61\t84785\t0.2\t1\t77339 7446\n",
      "62\t123180\t0.2\t1\t112273 10907\n",
      "63\t35598\t0.2\t1\t32371 3227\n",
      "64\t78527\t0.2\t1\t71342 7185\n",
      "65\t99496\t0.2\t1\t90653 8843\n",
      "66\t55116\t0.2\t1\t50072 5044\n",
      "67\t66407\t0.2\t1\t60463 5944\n",
      "68\t65435\t0.2\t1\t59526 5909\n",
      "69\t57792\t0.2\t1\t52377 5415\n",
      "70\t55914\t0.2\t1\t50685 5229\n",
      "71\t60089\t0.2\t1\t53161 6928\n",
      "72\t102519\t0.2\t1\t91223 11296\n",
      "73\t88294\t0.2\t1\t78522 9772\n",
      "74\t92793\t0.2\t1\t81510 11283\n",
      "75\t70384\t0.2\t1\t63145 7239\n",
      "76\t5589\t0.2\t1\t4798 791\n",
      "77\t367\t0.2\t1\t317 50\n",
      "78\t2243\t0.2\t1\t2030 213\n",
      "79\t6685\t0.2\t1\t6033 652\n",
      "80\t10103\t0.2\t1\t9169 934\n",
      "81\t12277\t0.2\t1\t11079 1198\n",
      "82\t12912\t0.2\t1\t11720 1192\n",
      "83\t13223\t0.2\t1\t11973 1250\n",
      "84\t12745\t0.2\t1\t11520 1225\n",
      "85\t12077\t0.2\t1\t10973 1104\n",
      "86\t11633\t0.2\t1\t10518 1115\n",
      "87\t10687\t0.2\t1\t9651 1036\n",
      "88\t9490\t0.2\t1\t8592 898\n",
      "89\t8494\t0.2\t1\t7655 839\n",
      "90\t7690\t0.2\t1\t6968 722\n",
      "91\t6677\t0.2\t1\t6040 637\n",
      "92\t5633\t0.2\t1\t5112 521\n",
      "93\t4874\t0.2\t1\t4348 526\n",
      "94\t4184\t0.2\t1\t3773 411\n",
      "95\t3784\t0.2\t1\t3406 378\n",
      "96\t3469\t0.2\t1\t3150 319\n",
      "97\t3334\t0.2\t1\t3012 322\n",
      "98\t2940\t0.2\t1\t2672 268\n",
      "99\t2660\t0.2\t1\t2388 272\n",
      "100\t2225\t0.2\t1\t1999 226\n",
      "101\t1933\t0.2\t1\t1748 185\n",
      "102\t1619\t0.2\t1\t1409 210\n",
      "103\t1312\t0.2\t1\t1139 173\n",
      "104\t1069\t0.2\t1\t959 110\n",
      "105\t948\t0.2\t1\t814 134\n",
      "106\t825\t0.2\t1\t726 99\n",
      "107\t813\t0.2\t1\t703 110\n",
      "108\t737\t0.2\t1\t636 101\n",
      "109\t642\t0.2\t1\t572 70\n",
      "110\t551\t0.2\t1\t484 67\n",
      "111\t460\t0.2\t1\t395 65\n",
      "112\t387\t0.2\t1\t349 38\n",
      "113\t376\t0.2\t1\t328 48\n",
      "114\t294\t0.2\t1\t250 44\n",
      "115\t274\t0.2\t1\t230 44\n",
      "116\t248\t0.2\t1\t216 32\n",
      "117\t182\t0.2\t1\t153 29\n",
      "118\t185\t0.2\t1\t148 37\n",
      "119\t187\t0.2\t1\t156 31\n",
      "120\t178\t0.2\t1\t144 34\n",
      "121\t181\t0.2\t1\t145 36\n",
      "122\t143\t0.2\t1\t108 35\n",
      "123\t144\t0.2\t1\t108 36\n",
      "124\t113\t0.2\t1\t84 29\n",
      "125\t125\t0.2\t1\t88 37\n",
      "126\t107\t0.2\t1\t59 48\n",
      "127\t117\t0.2\t1\t80 37\n",
      "128\t77\t0.2\t1\t54 23\n",
      "129\t96\t0.2\t1\t71 25\n",
      "130\t55\t0.2\t1\t37 18\n",
      "131\t60\t0.2\t1\t39 21\n",
      "132\t44\t0.2\t1\t27 17\n",
      "133\t45\t0.2\t1\t37 8\n",
      "134\t56\t0.2\t1\t47 9\n",
      "135\t49\t0.2\t1\t45 4\n",
      "136\t26\t0.2\t1\t18 8\n",
      "137\t35\t0.2\t1\t32 3\n",
      "138\t29\t0.2\t1\t23 6\n",
      "139\t39\t0.2\t1\t29 10\n",
      "140\t32\t0.2\t1\t27 5\n",
      "141\t84\t0.2\t1\t81 3\n",
      "142\t30\t0.2\t1\t24 6\n",
      "143\t40\t0.2\t1\t38 2\n",
      "144\t16\t0.2\t1\t14 2\n",
      "145\t83\t0.2\t1\t80 3\n",
      "146\t154\t0.2\t1\t150 4\n",
      "147\t84\t0.2\t1\t80 4\n",
      "148\t260\t0.2\t1\t255 5\n",
      "149\t64\t0.2\t1\t48 16\n",
      "150\t24070\t0.2\t1\t38 24032\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/O.1.fastq.gz\n",
      "=============================================\n",
      "16285961 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/O.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/O.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to O.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/O.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/O.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 644.72 s (40 us/read; 1.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,285,961\n",
      "Reads with adapters:                10,112,432 (62.1%)\n",
      "Reads written (passing filters):    16,285,961 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,442,894,150 bp\n",
      "Quality-trimmed:             260,798,033 bp (10.7%)\n",
      "Total written (filtered):  1,907,908,100 bp (78.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10112432 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.4%\n",
      "  C: 35.4%\n",
      "  G: 34.8%\n",
      "  T: 13.1%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2022002\t4071490.2\t0\t2022002\n",
      "2\t557160\t1017872.6\t0\t557160\n",
      "3\t280027\t254468.1\t0\t280027\n",
      "4\t128117\t63617.0\t0\t128117\n",
      "5\t124188\t15904.3\t0\t124188\n",
      "6\t74823\t3976.1\t0\t74823\n",
      "7\t80384\t994.0\t0\t80384\n",
      "8\t103113\t248.5\t0\t103113\n",
      "9\t130732\t62.1\t0\t130599 133\n",
      "10\t131687\t15.5\t1\t124794 6893\n",
      "11\t75265\t3.9\t1\t67735 7530\n",
      "12\t77274\t1.0\t1\t71027 6247\n",
      "13\t95589\t0.2\t1\t89545 6044\n",
      "14\t97018\t0.2\t1\t91150 5868\n",
      "15\t92783\t0.2\t1\t87578 5205\n",
      "16\t110847\t0.2\t1\t103272 7575\n",
      "17\t123909\t0.2\t1\t116889 7020\n",
      "18\t145387\t0.2\t1\t137582 7805\n",
      "19\t117710\t0.2\t1\t110646 7064\n",
      "20\t150201\t0.2\t1\t142804 7397\n",
      "21\t59356\t0.2\t1\t54711 4645\n",
      "22\t77502\t0.2\t1\t74224 3278\n",
      "23\t85078\t0.2\t1\t80834 4244\n",
      "24\t131345\t0.2\t1\t124486 6859\n",
      "25\t101774\t0.2\t1\t96420 5354\n",
      "26\t118597\t0.2\t1\t112149 6448\n",
      "27\t169153\t0.2\t1\t159598 9555\n",
      "28\t84278\t0.2\t1\t80224 4054\n",
      "29\t126463\t0.2\t1\t117167 9296\n",
      "30\t110582\t0.2\t1\t105665 4917\n",
      "31\t152516\t0.2\t1\t146137 6379\n",
      "32\t92121\t0.2\t1\t87660 4461\n",
      "33\t105934\t0.2\t1\t100444 5490\n",
      "34\t219938\t0.2\t1\t211028 8910\n",
      "35\t52514\t0.2\t1\t49062 3452\n",
      "36\t129573\t0.2\t1\t123273 6300\n",
      "37\t59795\t0.2\t1\t57224 2571\n",
      "38\t106890\t0.2\t1\t99750 7140\n",
      "39\t129228\t0.2\t1\t123109 6119\n",
      "40\t254194\t0.2\t1\t244574 9620\n",
      "41\t63336\t0.2\t1\t59117 4219\n",
      "42\t60516\t0.2\t1\t57860 2656\n",
      "43\t149650\t0.2\t1\t144495 5155\n",
      "44\t53918\t0.2\t1\t51294 2624\n",
      "45\t198490\t0.2\t1\t191804 6686\n",
      "46\t103839\t0.2\t1\t99095 4744\n",
      "47\t97111\t0.2\t1\t93882 3229\n",
      "48\t80560\t0.2\t1\t75579 4981\n",
      "49\t168722\t0.2\t1\t162581 6141\n",
      "50\t37287\t0.2\t1\t35017 2270\n",
      "51\t133461\t0.2\t1\t128384 5077\n",
      "52\t51530\t0.2\t1\t48916 2614\n",
      "53\t100029\t0.2\t1\t97057 2972\n",
      "54\t113372\t0.2\t1\t110251 3121\n",
      "55\t97778\t0.2\t1\t94263 3515\n",
      "56\t87962\t0.2\t1\t85147 2815\n",
      "57\t116890\t0.2\t1\t113465 3425\n",
      "58\t122476\t0.2\t1\t118309 4167\n",
      "59\t50035\t0.2\t1\t47789 2246\n",
      "60\t116390\t0.2\t1\t113507 2883\n",
      "61\t48512\t0.2\t1\t46522 1990\n",
      "62\t51646\t0.2\t1\t49407 2239\n",
      "63\t90607\t0.2\t1\t88260 2347\n",
      "64\t80822\t0.2\t1\t78025 2797\n",
      "65\t28104\t0.2\t1\t26682 1422\n",
      "66\t82865\t0.2\t1\t80330 2535\n",
      "67\t18514\t0.2\t1\t17570 944\n",
      "68\t52878\t0.2\t1\t50702 2176\n",
      "69\t54361\t0.2\t1\t52409 1952\n",
      "70\t111180\t0.2\t1\t108023 3157\n",
      "71\t17748\t0.2\t1\t16930 818\n",
      "72\t30469\t0.2\t1\t29332 1137\n",
      "73\t32008\t0.2\t1\t30810 1198\n",
      "74\t32664\t0.2\t1\t31509 1155\n",
      "75\t35645\t0.2\t1\t34429 1216\n",
      "76\t46960\t0.2\t1\t45455 1505\n",
      "77\t79030\t0.2\t1\t76790 2240\n",
      "78\t65218\t0.2\t1\t62421 2797\n",
      "79\t19577\t0.2\t1\t18013 1564\n",
      "80\t4236\t0.2\t1\t3647 589\n",
      "81\t442\t0.2\t1\t377 65\n",
      "82\t544\t0.2\t1\t518 26\n",
      "83\t1290\t0.2\t1\t1236 54\n",
      "84\t2592\t0.2\t1\t2518 74\n",
      "85\t4208\t0.2\t1\t4083 125\n",
      "86\t5441\t0.2\t1\t5307 134\n",
      "87\t5333\t0.2\t1\t5181 152\n",
      "88\t4219\t0.2\t1\t4100 119\n",
      "89\t2744\t0.2\t1\t2644 100\n",
      "90\t2303\t0.2\t1\t2222 81\n",
      "91\t1880\t0.2\t1\t1825 55\n",
      "92\t1522\t0.2\t1\t1467 55\n",
      "93\t1176\t0.2\t1\t1139 37\n",
      "94\t961\t0.2\t1\t932 29\n",
      "95\t835\t0.2\t1\t790 45\n",
      "96\t684\t0.2\t1\t656 28\n",
      "97\t691\t0.2\t1\t665 26\n",
      "98\t573\t0.2\t1\t549 24\n",
      "99\t465\t0.2\t1\t446 19\n",
      "100\t382\t0.2\t1\t360 22\n",
      "101\t310\t0.2\t1\t288 22\n",
      "102\t288\t0.2\t1\t257 31\n",
      "103\t215\t0.2\t1\t203 12\n",
      "104\t173\t0.2\t1\t157 16\n",
      "105\t146\t0.2\t1\t134 12\n",
      "106\t113\t0.2\t1\t104 9\n",
      "107\t126\t0.2\t1\t112 14\n",
      "108\t104\t0.2\t1\t95 9\n",
      "109\t93\t0.2\t1\t80 13\n",
      "110\t95\t0.2\t1\t83 12\n",
      "111\t70\t0.2\t1\t62 8\n",
      "112\t65\t0.2\t1\t59 6\n",
      "113\t74\t0.2\t1\t66 8\n",
      "114\t68\t0.2\t1\t56 12\n",
      "115\t45\t0.2\t1\t39 6\n",
      "116\t51\t0.2\t1\t40 11\n",
      "117\t48\t0.2\t1\t35 13\n",
      "118\t42\t0.2\t1\t33 9\n",
      "119\t49\t0.2\t1\t33 16\n",
      "120\t55\t0.2\t1\t42 13\n",
      "121\t43\t0.2\t1\t28 15\n",
      "122\t34\t0.2\t1\t23 11\n",
      "123\t43\t0.2\t1\t28 15\n",
      "124\t48\t0.2\t1\t37 11\n",
      "125\t38\t0.2\t1\t27 11\n",
      "126\t47\t0.2\t1\t28 19\n",
      "127\t27\t0.2\t1\t14 13\n",
      "128\t16\t0.2\t1\t12 4\n",
      "129\t24\t0.2\t1\t15 9\n",
      "130\t13\t0.2\t1\t6 7\n",
      "131\t10\t0.2\t1\t8 2\n",
      "132\t8\t0.2\t1\t3 5\n",
      "133\t13\t0.2\t1\t7 6\n",
      "134\t6\t0.2\t1\t3 3\n",
      "135\t8\t0.2\t1\t5 3\n",
      "136\t5\t0.2\t1\t1 4\n",
      "137\t6\t0.2\t1\t4 2\n",
      "138\t2\t0.2\t1\t2\n",
      "139\t2\t0.2\t1\t2\n",
      "140\t4\t0.2\t1\t1 3\n",
      "141\t10\t0.2\t1\t4 6\n",
      "142\t3\t0.2\t1\t2 1\n",
      "143\t9\t0.2\t1\t6 3\n",
      "144\t1\t0.2\t1\t0 1\n",
      "145\t7\t0.2\t1\t5 2\n",
      "146\t17\t0.2\t1\t14 3\n",
      "147\t14\t0.2\t1\t9 5\n",
      "148\t12\t0.2\t1\t9 3\n",
      "149\t1\t0.2\t1\t1\n",
      "150\t3\t0.2\t1\t2 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/O.2.fastq.gz\n",
      "=============================================\n",
      "16285961 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files O.1_trimmed.fq.gz and O.2_trimmed.fq.gz\n",
      "file_1: O.1_trimmed.fq.gz, file_2: O.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: O.1_trimmed.fq.gz and O.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to O.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to O.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to O.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to O.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16285961\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1166109 (7.16%)\n",
      "\n",
      "Deleting both intermediate output files O.1_trimmed.fq.gz and O.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/O.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/O.1.fastq.gz ./data_raw/O.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/P.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t391437\tAGATCGGAAGAGC\t1000000\t39.14\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 391437). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/P.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/P.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/P.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to P.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/P.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/P.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 596.47 s (41 us/read; 1.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,726,854\n",
      "Reads with adapters:                 9,602,110 (65.2%)\n",
      "Reads written (passing filters):    14,726,854 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,209,028,100 bp\n",
      "Quality-trimmed:              53,941,433 bp (2.4%)\n",
      "Total written (filtered):  1,890,909,989 bp (85.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9602110 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.1%\n",
      "  C: 33.6%\n",
      "  G: 31.6%\n",
      "  T: 17.2%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1752664\t3681713.5\t0\t1752664\n",
      "2\t586319\t920428.4\t0\t586319\n",
      "3\t246717\t230107.1\t0\t246717\n",
      "4\t128118\t57526.8\t0\t128118\n",
      "5\t92128\t14381.7\t0\t92128\n",
      "6\t77091\t3595.4\t0\t77091\n",
      "7\t76204\t898.9\t0\t76204\n",
      "8\t93868\t224.7\t0\t93868\n",
      "9\t123189\t56.2\t0\t123060 129\n",
      "10\t133038\t14.0\t1\t123318 9720\n",
      "11\t95828\t3.5\t1\t87013 8815\n",
      "12\t94775\t0.9\t1\t86942 7833\n",
      "13\t118073\t0.2\t1\t109250 8823\n",
      "14\t86994\t0.2\t1\t79436 7558\n",
      "15\t120927\t0.2\t1\t111739 9188\n",
      "16\t65756\t0.2\t1\t59330 6426\n",
      "17\t104050\t0.2\t1\t95690 8360\n",
      "18\t91073\t0.2\t1\t84787 6286\n",
      "19\t109644\t0.2\t1\t101885 7759\n",
      "20\t84747\t0.2\t1\t78428 6319\n",
      "21\t115201\t0.2\t1\t106689 8512\n",
      "22\t115191\t0.2\t1\t106756 8435\n",
      "23\t115156\t0.2\t1\t106388 8768\n",
      "24\t118254\t0.2\t1\t108277 9977\n",
      "25\t111674\t0.2\t1\t103399 8275\n",
      "26\t103053\t0.2\t1\t94798 8255\n",
      "27\t124147\t0.2\t1\t115005 9142\n",
      "28\t95611\t0.2\t1\t89033 6578\n",
      "29\t139588\t0.2\t1\t129254 10334\n",
      "30\t119128\t0.2\t1\t110790 8338\n",
      "31\t158115\t0.2\t1\t146420 11695\n",
      "32\t93591\t0.2\t1\t86624 6967\n",
      "33\t124890\t0.2\t1\t114601 10289\n",
      "34\t147456\t0.2\t1\t136973 10483\n",
      "35\t74664\t0.2\t1\t69201 5463\n",
      "36\t102003\t0.2\t1\t94330 7673\n",
      "37\t104745\t0.2\t1\t96861 7884\n",
      "38\t98179\t0.2\t1\t90642 7537\n",
      "39\t118542\t0.2\t1\t109524 9018\n",
      "40\t165804\t0.2\t1\t152851 12953\n",
      "41\t222893\t0.2\t1\t206834 16059\n",
      "42\t275321\t0.2\t1\t256533 18788\n",
      "43\t146013\t0.2\t1\t135809 10204\n",
      "44\t53862\t0.2\t1\t50139 3723\n",
      "45\t16266\t0.2\t1\t15071 1195\n",
      "46\t31457\t0.2\t1\t29372 2085\n",
      "47\t60245\t0.2\t1\t56246 3999\n",
      "48\t81427\t0.2\t1\t75972 5455\n",
      "49\t148858\t0.2\t1\t139605 9253\n",
      "50\t43798\t0.2\t1\t40404 3394\n",
      "51\t121483\t0.2\t1\t112954 8529\n",
      "52\t130943\t0.2\t1\t122314 8629\n",
      "53\t95473\t0.2\t1\t89237 6236\n",
      "54\t76561\t0.2\t1\t71403 5158\n",
      "55\t114754\t0.2\t1\t107342 7412\n",
      "56\t97481\t0.2\t1\t91093 6388\n",
      "57\t36668\t0.2\t1\t34175 2493\n",
      "58\t78961\t0.2\t1\t73912 5049\n",
      "59\t129691\t0.2\t1\t121347 8344\n",
      "60\t31725\t0.2\t1\t29487 2238\n",
      "61\t66835\t0.2\t1\t62600 4235\n",
      "62\t94387\t0.2\t1\t88361 6026\n",
      "63\t27747\t0.2\t1\t25842 1905\n",
      "64\t59666\t0.2\t1\t55797 3869\n",
      "65\t74764\t0.2\t1\t69963 4801\n",
      "66\t39809\t0.2\t1\t37248 2561\n",
      "67\t50245\t0.2\t1\t47082 3163\n",
      "68\t48202\t0.2\t1\t45102 3100\n",
      "69\t41900\t0.2\t1\t39042 2858\n",
      "70\t39448\t0.2\t1\t36820 2628\n",
      "71\t42084\t0.2\t1\t38491 3593\n",
      "72\t71052\t0.2\t1\t65140 5912\n",
      "73\t61562\t0.2\t1\t56460 5102\n",
      "74\t64467\t0.2\t1\t58151 6316\n",
      "75\t49560\t0.2\t1\t45802 3758\n",
      "76\t4156\t0.2\t1\t3713 443\n",
      "77\t259\t0.2\t1\t232 27\n",
      "78\t1473\t0.2\t1\t1379 94\n",
      "79\t4412\t0.2\t1\t4130 282\n",
      "80\t6654\t0.2\t1\t6260 394\n",
      "81\t8020\t0.2\t1\t7500 520\n",
      "82\t8567\t0.2\t1\t8033 534\n",
      "83\t8730\t0.2\t1\t8162 568\n",
      "84\t8560\t0.2\t1\t8012 548\n",
      "85\t8104\t0.2\t1\t7577 527\n",
      "86\t7816\t0.2\t1\t7316 500\n",
      "87\t7201\t0.2\t1\t6763 438\n",
      "88\t6713\t0.2\t1\t6305 408\n",
      "89\t6023\t0.2\t1\t5618 405\n",
      "90\t5400\t0.2\t1\t5043 357\n",
      "91\t4695\t0.2\t1\t4396 299\n",
      "92\t4166\t0.2\t1\t3896 270\n",
      "93\t3463\t0.2\t1\t3230 233\n",
      "94\t3195\t0.2\t1\t2993 202\n",
      "95\t2746\t0.2\t1\t2552 194\n",
      "96\t2551\t0.2\t1\t2368 183\n",
      "97\t2504\t0.2\t1\t2356 148\n",
      "98\t2265\t0.2\t1\t2120 145\n",
      "99\t2018\t0.2\t1\t1877 141\n",
      "100\t1836\t0.2\t1\t1689 147\n",
      "101\t1530\t0.2\t1\t1440 90\n",
      "102\t1337\t0.2\t1\t1233 104\n",
      "103\t1101\t0.2\t1\t1026 75\n",
      "104\t939\t0.2\t1\t850 89\n",
      "105\t841\t0.2\t1\t770 71\n",
      "106\t713\t0.2\t1\t636 77\n",
      "107\t742\t0.2\t1\t672 70\n",
      "108\t607\t0.2\t1\t542 65\n",
      "109\t618\t0.2\t1\t570 48\n",
      "110\t495\t0.2\t1\t443 52\n",
      "111\t385\t0.2\t1\t347 38\n",
      "112\t359\t0.2\t1\t318 41\n",
      "113\t330\t0.2\t1\t303 27\n",
      "114\t259\t0.2\t1\t223 36\n",
      "115\t246\t0.2\t1\t210 36\n",
      "116\t196\t0.2\t1\t160 36\n",
      "117\t208\t0.2\t1\t168 40\n",
      "118\t181\t0.2\t1\t151 30\n",
      "119\t170\t0.2\t1\t127 43\n",
      "120\t158\t0.2\t1\t121 37\n",
      "121\t167\t0.2\t1\t137 30\n",
      "122\t151\t0.2\t1\t113 38\n",
      "123\t140\t0.2\t1\t107 33\n",
      "124\t104\t0.2\t1\t68 36\n",
      "125\t144\t0.2\t1\t114 30\n",
      "126\t92\t0.2\t1\t64 28\n",
      "127\t96\t0.2\t1\t65 31\n",
      "128\t72\t0.2\t1\t49 23\n",
      "129\t75\t0.2\t1\t56 19\n",
      "130\t51\t0.2\t1\t37 14\n",
      "131\t79\t0.2\t1\t58 21\n",
      "132\t41\t0.2\t1\t35 6\n",
      "133\t52\t0.2\t1\t42 10\n",
      "134\t47\t0.2\t1\t38 9\n",
      "135\t23\t0.2\t1\t19 4\n",
      "136\t24\t0.2\t1\t20 4\n",
      "137\t16\t0.2\t1\t15 1\n",
      "138\t21\t0.2\t1\t18 3\n",
      "139\t15\t0.2\t1\t13 2\n",
      "140\t31\t0.2\t1\t27 4\n",
      "141\t49\t0.2\t1\t42 7\n",
      "142\t26\t0.2\t1\t15 11\n",
      "143\t29\t0.2\t1\t26 3\n",
      "144\t15\t0.2\t1\t15\n",
      "145\t65\t0.2\t1\t63 2\n",
      "146\t85\t0.2\t1\t81 4\n",
      "147\t56\t0.2\t1\t54 2\n",
      "148\t134\t0.2\t1\t133 1\n",
      "149\t41\t0.2\t1\t32 9\n",
      "150\t15424\t0.2\t1\t18 15406\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/P.1.fastq.gz\n",
      "=============================================\n",
      "14726854 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/P.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/P.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to P.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/P.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/P.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 589.64 s (40 us/read; 1.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,726,854\n",
      "Reads with adapters:                 9,231,597 (62.7%)\n",
      "Reads written (passing filters):    14,726,854 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,209,028,100 bp\n",
      "Quality-trimmed:             201,088,176 bp (9.1%)\n",
      "Total written (filtered):  1,784,665,484 bp (80.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9231597 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.3%\n",
      "  C: 33.7%\n",
      "  G: 37.0%\n",
      "  T: 12.7%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1861831\t3681713.5\t0\t1861831\n",
      "2\t571252\t920428.4\t0\t571252\n",
      "3\t256380\t230107.1\t0\t256380\n",
      "4\t117136\t57526.8\t0\t117136\n",
      "5\t116050\t14381.7\t0\t116050\n",
      "6\t69833\t3595.4\t0\t69833\n",
      "7\t77821\t898.9\t0\t77821\n",
      "8\t106572\t224.7\t0\t106572\n",
      "9\t126642\t56.2\t0\t126472 170\n",
      "10\t141546\t14.0\t1\t134737 6809\n",
      "11\t63052\t3.5\t1\t56157 6895\n",
      "12\t77080\t0.9\t1\t71064 6016\n",
      "13\t88727\t0.2\t1\t82671 6056\n",
      "14\t97717\t0.2\t1\t91392 6325\n",
      "15\t91085\t0.2\t1\t86024 5061\n",
      "16\t103522\t0.2\t1\t95941 7581\n",
      "17\t126456\t0.2\t1\t118754 7702\n",
      "18\t134456\t0.2\t1\t126792 7664\n",
      "19\t127295\t0.2\t1\t119325 7970\n",
      "20\t126279\t0.2\t1\t119997 6282\n",
      "21\t55180\t0.2\t1\t50491 4689\n",
      "22\t86338\t0.2\t1\t82328 4010\n",
      "23\t84750\t0.2\t1\t80184 4566\n",
      "24\t148723\t0.2\t1\t140335 8388\n",
      "25\t79586\t0.2\t1\t74827 4759\n",
      "26\t125761\t0.2\t1\t118443 7318\n",
      "27\t169427\t0.2\t1\t158857 10570\n",
      "28\t125969\t0.2\t1\t119894 6075\n",
      "29\t94973\t0.2\t1\t86111 8862\n",
      "30\t125458\t0.2\t1\t119546 5912\n",
      "31\t131365\t0.2\t1\t125210 6155\n",
      "32\t110359\t0.2\t1\t104597 5762\n",
      "33\t100294\t0.2\t1\t94825 5469\n",
      "34\t235810\t0.2\t1\t226779 9031\n",
      "35\t73800\t0.2\t1\t70038 3762\n",
      "36\t119522\t0.2\t1\t112731 6791\n",
      "37\t182928\t0.2\t1\t177019 5909\n",
      "38\t46209\t0.2\t1\t42019 4190\n",
      "39\t105827\t0.2\t1\t101779 4048\n",
      "40\t105218\t0.2\t1\t100530 4688\n",
      "41\t137293\t0.2\t1\t131473 5820\n",
      "42\t187448\t0.2\t1\t181567 5881\n",
      "43\t125101\t0.2\t1\t120192 4909\n",
      "44\t505418\t0.2\t1\t494899 10519\n",
      "45\t138833\t0.2\t1\t133382 5451\n",
      "46\t79803\t0.2\t1\t75626 4177\n",
      "47\t114329\t0.2\t1\t110704 3625\n",
      "48\t35183\t0.2\t1\t32368 2815\n",
      "49\t86735\t0.2\t1\t83607 3128\n",
      "50\t28470\t0.2\t1\t27062 1408\n",
      "51\t50505\t0.2\t1\t48579 1926\n",
      "52\t23924\t0.2\t1\t22729 1195\n",
      "53\t38641\t0.2\t1\t37374 1267\n",
      "54\t85393\t0.2\t1\t83136 2257\n",
      "55\t32464\t0.2\t1\t31112 1352\n",
      "56\t50714\t0.2\t1\t49132 1582\n",
      "57\t140857\t0.2\t1\t137603 3254\n",
      "58\t47527\t0.2\t1\t45731 1796\n",
      "59\t21064\t0.2\t1\t20142 922\n",
      "60\t72576\t0.2\t1\t70915 1661\n",
      "61\t20321\t0.2\t1\t19456 865\n",
      "62\t28380\t0.2\t1\t27154 1226\n",
      "63\t66221\t0.2\t1\t64610 1611\n",
      "64\t23815\t0.2\t1\t22851 964\n",
      "65\t11073\t0.2\t1\t10463 610\n",
      "66\t28830\t0.2\t1\t27925 905\n",
      "67\t7959\t0.2\t1\t7581 378\n",
      "68\t17262\t0.2\t1\t16456 806\n",
      "69\t17589\t0.2\t1\t16988 601\n",
      "70\t40855\t0.2\t1\t39663 1192\n",
      "71\t5244\t0.2\t1\t4984 260\n",
      "72\t10027\t0.2\t1\t9648 379\n",
      "73\t12364\t0.2\t1\t11923 441\n",
      "74\t13778\t0.2\t1\t13311 467\n",
      "75\t15483\t0.2\t1\t14935 548\n",
      "76\t19705\t0.2\t1\t19118 587\n",
      "77\t33362\t0.2\t1\t32446 916\n",
      "78\t33262\t0.2\t1\t32013 1249\n",
      "79\t10212\t0.2\t1\t9552 660\n",
      "80\t1968\t0.2\t1\t1712 256\n",
      "81\t248\t0.2\t1\t215 33\n",
      "82\t303\t0.2\t1\t281 22\n",
      "83\t783\t0.2\t1\t750 33\n",
      "84\t1649\t0.2\t1\t1618 31\n",
      "85\t2374\t0.2\t1\t2317 57\n",
      "86\t3094\t0.2\t1\t3019 75\n",
      "87\t3223\t0.2\t1\t3148 75\n",
      "88\t2323\t0.2\t1\t2251 72\n",
      "89\t1329\t0.2\t1\t1292 37\n",
      "90\t1101\t0.2\t1\t1071 30\n",
      "91\t965\t0.2\t1\t935 30\n",
      "92\t733\t0.2\t1\t706 27\n",
      "93\t596\t0.2\t1\t576 20\n",
      "94\t563\t0.2\t1\t546 17\n",
      "95\t440\t0.2\t1\t426 14\n",
      "96\t429\t0.2\t1\t417 12\n",
      "97\t394\t0.2\t1\t376 18\n",
      "98\t305\t0.2\t1\t294 11\n",
      "99\t277\t0.2\t1\t267 10\n",
      "100\t220\t0.2\t1\t208 12\n",
      "101\t197\t0.2\t1\t185 12\n",
      "102\t162\t0.2\t1\t152 10\n",
      "103\t137\t0.2\t1\t124 13\n",
      "104\t130\t0.2\t1\t120 10\n",
      "105\t102\t0.2\t1\t93 9\n",
      "106\t92\t0.2\t1\t85 7\n",
      "107\t115\t0.2\t1\t102 13\n",
      "108\t68\t0.2\t1\t63 5\n",
      "109\t91\t0.2\t1\t82 9\n",
      "110\t62\t0.2\t1\t54 8\n",
      "111\t63\t0.2\t1\t52 11\n",
      "112\t59\t0.2\t1\t45 14\n",
      "113\t59\t0.2\t1\t50 9\n",
      "114\t58\t0.2\t1\t41 17\n",
      "115\t65\t0.2\t1\t46 19\n",
      "116\t41\t0.2\t1\t29 12\n",
      "117\t51\t0.2\t1\t36 15\n",
      "118\t39\t0.2\t1\t32 7\n",
      "119\t34\t0.2\t1\t18 16\n",
      "120\t41\t0.2\t1\t33 8\n",
      "121\t45\t0.2\t1\t33 12\n",
      "122\t36\t0.2\t1\t24 12\n",
      "123\t41\t0.2\t1\t26 15\n",
      "124\t36\t0.2\t1\t20 16\n",
      "125\t34\t0.2\t1\t20 14\n",
      "126\t24\t0.2\t1\t17 7\n",
      "127\t23\t0.2\t1\t11 12\n",
      "128\t16\t0.2\t1\t7 9\n",
      "129\t18\t0.2\t1\t11 7\n",
      "130\t8\t0.2\t1\t5 3\n",
      "131\t9\t0.2\t1\t5 4\n",
      "132\t10\t0.2\t1\t8 2\n",
      "133\t6\t0.2\t1\t6\n",
      "134\t3\t0.2\t1\t0 3\n",
      "135\t3\t0.2\t1\t2 1\n",
      "136\t6\t0.2\t1\t1 5\n",
      "137\t1\t0.2\t1\t1\n",
      "138\t1\t0.2\t1\t0 1\n",
      "139\t3\t0.2\t1\t1 2\n",
      "140\t2\t0.2\t1\t1 1\n",
      "141\t6\t0.2\t1\t5 1\n",
      "142\t1\t0.2\t1\t1\n",
      "143\t4\t0.2\t1\t4\n",
      "144\t3\t0.2\t1\t1 2\n",
      "145\t2\t0.2\t1\t1 1\n",
      "146\t9\t0.2\t1\t7 2\n",
      "147\t2\t0.2\t1\t2\n",
      "148\t9\t0.2\t1\t8 1\n",
      "149\t3\t0.2\t1\t2 1\n",
      "150\t1\t0.2\t1\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/P.2.fastq.gz\n",
      "=============================================\n",
      "14726854 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files P.1_trimmed.fq.gz and P.2_trimmed.fq.gz\n",
      "file_1: P.1_trimmed.fq.gz, file_2: P.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: P.1_trimmed.fq.gz and P.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to P.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to P.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to P.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to P.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 14726854\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 754580 (5.12%)\n",
      "\n",
      "Deleting both intermediate output files P.1_trimmed.fq.gz and P.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/P.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/P.1.fastq.gz ./data_raw/P.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/Q.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t359241\tAGATCGGAAGAGC\t1000000\t35.92\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 359241). Second best hit was Nextera (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/Q.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/Q.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/Q.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to Q.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/Q.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/Q.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 740.75 s (40 us/read; 1.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,460,544\n",
      "Reads with adapters:                11,589,191 (62.8%)\n",
      "Reads written (passing filters):    18,460,544 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,769,081,600 bp\n",
      "Quality-trimmed:              71,250,745 bp (2.6%)\n",
      "Total written (filtered):  2,414,374,001 bp (87.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11589191 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.5%\n",
      "  C: 31.7%\n",
      "  G: 33.8%\n",
      "  T: 16.7%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2307678\t4615136.0\t0\t2307678\n",
      "2\t795704\t1153784.0\t0\t795704\n",
      "3\t337811\t288446.0\t0\t337811\n",
      "4\t173263\t72111.5\t0\t173263\n",
      "5\t122585\t18027.9\t0\t122585\n",
      "6\t102489\t4507.0\t0\t102489\n",
      "7\t102167\t1126.7\t0\t102167\n",
      "8\t123748\t281.7\t0\t123748\n",
      "9\t160623\t70.4\t0\t160442 181\n",
      "10\t176431\t17.6\t1\t165349 11082\n",
      "11\t121475\t4.4\t1\t111451 10024\n",
      "12\t123676\t1.1\t1\t114252 9424\n",
      "13\t154560\t0.3\t1\t144308 10252\n",
      "14\t110894\t0.3\t1\t102456 8438\n",
      "15\t159013\t0.3\t1\t148687 10326\n",
      "16\t81898\t0.3\t1\t74793 7105\n",
      "17\t133051\t0.3\t1\t123620 9431\n",
      "18\t116882\t0.3\t1\t109643 7239\n",
      "19\t143945\t0.3\t1\t134922 9023\n",
      "20\t109618\t0.3\t1\t102468 7150\n",
      "21\t145788\t0.3\t1\t136094 9694\n",
      "22\t154608\t0.3\t1\t144258 10350\n",
      "23\t144043\t0.3\t1\t134484 9559\n",
      "24\t138518\t0.3\t1\t128345 10173\n",
      "25\t136659\t0.3\t1\t127991 8668\n",
      "26\t112745\t0.3\t1\t104937 7808\n",
      "27\t145201\t0.3\t1\t135689 9512\n",
      "28\t134951\t0.3\t1\t126811 8140\n",
      "29\t142985\t0.3\t1\t133451 9534\n",
      "30\t133350\t0.3\t1\t124805 8545\n",
      "31\t144716\t0.3\t1\t135284 9432\n",
      "32\t157687\t0.3\t1\t147969 9718\n",
      "33\t123752\t0.3\t1\t115515 8237\n",
      "34\t124385\t0.3\t1\t116892 7493\n",
      "35\t151536\t0.3\t1\t141391 10145\n",
      "36\t139914\t0.3\t1\t130636 9278\n",
      "37\t167839\t0.3\t1\t157189 10650\n",
      "38\t111423\t0.3\t1\t104249 7174\n",
      "39\t181057\t0.3\t1\t168787 12270\n",
      "40\t162967\t0.3\t1\t152222 10745\n",
      "41\t226852\t0.3\t1\t211977 14875\n",
      "42\t318627\t0.3\t1\t297299 21328\n",
      "43\t282380\t0.3\t1\t262092 20288\n",
      "44\t299243\t0.3\t1\t280875 18368\n",
      "45\t41090\t0.3\t1\t38294 2796\n",
      "46\t58085\t0.3\t1\t54427 3658\n",
      "47\t61259\t0.3\t1\t57525 3734\n",
      "48\t82731\t0.3\t1\t77721 5010\n",
      "49\t141176\t0.3\t1\t133307 7869\n",
      "50\t41174\t0.3\t1\t38181 2993\n",
      "51\t137031\t0.3\t1\t128435 8596\n",
      "52\t107578\t0.3\t1\t101259 6319\n",
      "53\t68639\t0.3\t1\t64478 4161\n",
      "54\t77711\t0.3\t1\t73011 4700\n",
      "55\t112836\t0.3\t1\t106337 6499\n",
      "56\t77950\t0.3\t1\t73515 4435\n",
      "57\t21534\t0.3\t1\t20122 1412\n",
      "58\t58687\t0.3\t1\t55098 3589\n",
      "59\t116353\t0.3\t1\t109415 6938\n",
      "60\t20780\t0.3\t1\t19446 1334\n",
      "61\t49648\t0.3\t1\t46838 2810\n",
      "62\t68076\t0.3\t1\t64065 4011\n",
      "63\t16981\t0.3\t1\t15903 1078\n",
      "64\t39301\t0.3\t1\t36908 2393\n",
      "65\t59138\t0.3\t1\t55641 3497\n",
      "66\t25188\t0.3\t1\t23708 1480\n",
      "67\t29991\t0.3\t1\t28199 1792\n",
      "68\t34090\t0.3\t1\t32113 1977\n",
      "69\t30628\t0.3\t1\t28710 1918\n",
      "70\t29562\t0.3\t1\t27691 1871\n",
      "71\t30614\t0.3\t1\t28502 2112\n",
      "72\t41197\t0.3\t1\t37985 3212\n",
      "73\t49620\t0.3\t1\t45833 3787\n",
      "74\t53268\t0.3\t1\t48638 4630\n",
      "75\t40919\t0.3\t1\t37956 2963\n",
      "76\t3814\t0.3\t1\t3447 367\n",
      "77\t194\t0.3\t1\t175 19\n",
      "78\t1085\t0.3\t1\t1026 59\n",
      "79\t3297\t0.3\t1\t3104 193\n",
      "80\t5489\t0.3\t1\t5155 334\n",
      "81\t6504\t0.3\t1\t6093 411\n",
      "82\t7059\t0.3\t1\t6624 435\n",
      "83\t7153\t0.3\t1\t6735 418\n",
      "84\t6874\t0.3\t1\t6458 416\n",
      "85\t6589\t0.3\t1\t6160 429\n",
      "86\t6514\t0.3\t1\t6133 381\n",
      "87\t6013\t0.3\t1\t5646 367\n",
      "88\t5369\t0.3\t1\t5036 333\n",
      "89\t5102\t0.3\t1\t4762 340\n",
      "90\t4383\t0.3\t1\t4116 267\n",
      "91\t4048\t0.3\t1\t3838 210\n",
      "92\t3484\t0.3\t1\t3274 210\n",
      "93\t3161\t0.3\t1\t2937 224\n",
      "94\t2771\t0.3\t1\t2585 186\n",
      "95\t2521\t0.3\t1\t2374 147\n",
      "96\t2388\t0.3\t1\t2215 173\n",
      "97\t2327\t0.3\t1\t2197 130\n",
      "98\t2183\t0.3\t1\t2025 158\n",
      "99\t1955\t0.3\t1\t1824 131\n",
      "100\t1738\t0.3\t1\t1637 101\n",
      "101\t1521\t0.3\t1\t1408 113\n",
      "102\t1390\t0.3\t1\t1298 92\n",
      "103\t1038\t0.3\t1\t957 81\n",
      "104\t973\t0.3\t1\t913 60\n",
      "105\t779\t0.3\t1\t719 60\n",
      "106\t788\t0.3\t1\t716 72\n",
      "107\t780\t0.3\t1\t703 77\n",
      "108\t664\t0.3\t1\t607 57\n",
      "109\t605\t0.3\t1\t566 39\n",
      "110\t513\t0.3\t1\t470 43\n",
      "111\t434\t0.3\t1\t396 38\n",
      "112\t379\t0.3\t1\t347 32\n",
      "113\t369\t0.3\t1\t328 41\n",
      "114\t292\t0.3\t1\t260 32\n",
      "115\t278\t0.3\t1\t245 33\n",
      "116\t244\t0.3\t1\t209 35\n",
      "117\t213\t0.3\t1\t170 43\n",
      "118\t224\t0.3\t1\t189 35\n",
      "119\t228\t0.3\t1\t197 31\n",
      "120\t165\t0.3\t1\t131 34\n",
      "121\t195\t0.3\t1\t147 48\n",
      "122\t144\t0.3\t1\t115 29\n",
      "123\t154\t0.3\t1\t118 36\n",
      "124\t126\t0.3\t1\t99 27\n",
      "125\t123\t0.3\t1\t85 38\n",
      "126\t101\t0.3\t1\t61 40\n",
      "127\t114\t0.3\t1\t77 37\n",
      "128\t71\t0.3\t1\t48 23\n",
      "129\t97\t0.3\t1\t72 25\n",
      "130\t68\t0.3\t1\t49 19\n",
      "131\t73\t0.3\t1\t47 26\n",
      "132\t41\t0.3\t1\t36 5\n",
      "133\t44\t0.3\t1\t35 9\n",
      "134\t54\t0.3\t1\t46 8\n",
      "135\t47\t0.3\t1\t46 1\n",
      "136\t34\t0.3\t1\t30 4\n",
      "137\t38\t0.3\t1\t33 5\n",
      "138\t40\t0.3\t1\t34 6\n",
      "139\t34\t0.3\t1\t31 3\n",
      "140\t26\t0.3\t1\t23 3\n",
      "141\t62\t0.3\t1\t56 6\n",
      "142\t38\t0.3\t1\t30 8\n",
      "143\t36\t0.3\t1\t35 1\n",
      "144\t16\t0.3\t1\t15 1\n",
      "145\t83\t0.3\t1\t81 2\n",
      "146\t128\t0.3\t1\t128\n",
      "147\t107\t0.3\t1\t103 4\n",
      "148\t316\t0.3\t1\t309 7\n",
      "149\t53\t0.3\t1\t47 6\n",
      "150\t11264\t0.3\t1\t26 11238\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/Q.1.fastq.gz\n",
      "=============================================\n",
      "18460544 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/Q.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/Q.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to Q.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/Q.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/Q.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 738.23 s (40 us/read; 1.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,460,544\n",
      "Reads with adapters:                11,257,587 (61.0%)\n",
      "Reads written (passing filters):    18,460,544 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,769,081,600 bp\n",
      "Quality-trimmed:             219,201,424 bp (7.9%)\n",
      "Total written (filtered):  2,289,679,758 bp (82.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11257587 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.6%\n",
      "  C: 31.9%\n",
      "  G: 38.1%\n",
      "  T: 12.2%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2535041\t4615136.0\t0\t2535041\n",
      "2\t734109\t1153784.0\t0\t734109\n",
      "3\t360836\t288446.0\t0\t360836\n",
      "4\t152719\t72111.5\t0\t152719\n",
      "5\t158987\t18027.9\t0\t158987\n",
      "6\t92066\t4507.0\t0\t92066\n",
      "7\t98036\t1126.7\t0\t98036\n",
      "8\t131603\t281.7\t0\t131603\n",
      "9\t171337\t70.4\t0\t171080 257\n",
      "10\t176777\t17.6\t1\t168319 8458\n",
      "11\t92001\t4.4\t1\t82684 9317\n",
      "12\t91335\t1.1\t1\t83934 7401\n",
      "13\t123761\t0.3\t1\t116165 7596\n",
      "14\t121529\t0.3\t1\t115014 6515\n",
      "15\t110260\t0.3\t1\t104720 5540\n",
      "16\t139647\t0.3\t1\t130456 9191\n",
      "17\t156597\t0.3\t1\t147810 8787\n",
      "18\t172052\t0.3\t1\t163544 8508\n",
      "19\t133561\t0.3\t1\t126330 7231\n",
      "20\t166143\t0.3\t1\t159316 6827\n",
      "21\t77092\t0.3\t1\t71428 5664\n",
      "22\t102377\t0.3\t1\t98511 3866\n",
      "23\t106884\t0.3\t1\t102179 4705\n",
      "24\t151328\t0.3\t1\t144318 7010\n",
      "25\t111321\t0.3\t1\t105827 5494\n",
      "26\t147542\t0.3\t1\t139883 7659\n",
      "27\t227601\t0.3\t1\t214937 12664\n",
      "28\t129497\t0.3\t1\t123809 5688\n",
      "29\t135744\t0.3\t1\t124696 11048\n",
      "30\t123165\t0.3\t1\t117706 5459\n",
      "31\t175400\t0.3\t1\t168264 7136\n",
      "32\t83008\t0.3\t1\t78224 4784\n",
      "33\t186919\t0.3\t1\t179988 6931\n",
      "34\t157164\t0.3\t1\t147593 9571\n",
      "35\t122889\t0.3\t1\t117010 5879\n",
      "36\t159570\t0.3\t1\t154425 5145\n",
      "37\t145929\t0.3\t1\t140465 5464\n",
      "38\t107735\t0.3\t1\t103255 4480\n",
      "39\t160436\t0.3\t1\t154486 5950\n",
      "40\t83709\t0.3\t1\t80289 3420\n",
      "41\t123713\t0.3\t1\t120180 3533\n",
      "42\t147409\t0.3\t1\t143482 3927\n",
      "43\t112377\t0.3\t1\t108837 3540\n",
      "44\t82622\t0.3\t1\t79642 2980\n",
      "45\t181796\t0.3\t1\t176804 4992\n",
      "46\t96626\t0.3\t1\t92624 4002\n",
      "47\t154281\t0.3\t1\t150426 3855\n",
      "48\t72306\t0.3\t1\t68435 3871\n",
      "49\t141698\t0.3\t1\t137500 4198\n",
      "50\t47608\t0.3\t1\t45525 2083\n",
      "51\t91679\t0.3\t1\t88855 2824\n",
      "52\t56553\t0.3\t1\t54379 2174\n",
      "53\t79964\t0.3\t1\t77853 2111\n",
      "54\t112213\t0.3\t1\t109624 2589\n",
      "55\t66363\t0.3\t1\t64292 2071\n",
      "56\t76966\t0.3\t1\t74848 2118\n",
      "57\t116900\t0.3\t1\t114191 2709\n",
      "58\t82883\t0.3\t1\t80591 2292\n",
      "59\t40593\t0.3\t1\t39114 1479\n",
      "60\t82996\t0.3\t1\t81311 1685\n",
      "61\t41132\t0.3\t1\t39827 1305\n",
      "62\t41213\t0.3\t1\t39812 1401\n",
      "63\t68996\t0.3\t1\t67436 1560\n",
      "64\t52953\t0.3\t1\t51578 1375\n",
      "65\t23139\t0.3\t1\t22144 995\n",
      "66\t47153\t0.3\t1\t46013 1140\n",
      "67\t13921\t0.3\t1\t13450 471\n",
      "68\t34033\t0.3\t1\t32911 1122\n",
      "69\t32481\t0.3\t1\t31541 940\n",
      "70\t58844\t0.3\t1\t57537 1307\n",
      "71\t11065\t0.3\t1\t10647 418\n",
      "72\t19623\t0.3\t1\t19039 584\n",
      "73\t21284\t0.3\t1\t20682 602\n",
      "74\t21625\t0.3\t1\t20983 642\n",
      "75\t22692\t0.3\t1\t22082 610\n",
      "76\t30439\t0.3\t1\t29638 801\n",
      "77\t51492\t0.3\t1\t50297 1195\n",
      "78\t40245\t0.3\t1\t38722 1523\n",
      "79\t11550\t0.3\t1\t10760 790\n",
      "80\t2372\t0.3\t1\t2055 317\n",
      "81\t312\t0.3\t1\t271 41\n",
      "82\t423\t0.3\t1\t396 27\n",
      "83\t1038\t0.3\t1\t1012 26\n",
      "84\t1924\t0.3\t1\t1883 41\n",
      "85\t2766\t0.3\t1\t2706 60\n",
      "86\t3676\t0.3\t1\t3600 76\n",
      "87\t3614\t0.3\t1\t3541 73\n",
      "88\t2603\t0.3\t1\t2540 63\n",
      "89\t1930\t0.3\t1\t1892 38\n",
      "90\t1641\t0.3\t1\t1602 39\n",
      "91\t1346\t0.3\t1\t1309 37\n",
      "92\t1147\t0.3\t1\t1117 30\n",
      "93\t919\t0.3\t1\t893 26\n",
      "94\t776\t0.3\t1\t751 25\n",
      "95\t696\t0.3\t1\t669 27\n",
      "96\t567\t0.3\t1\t553 14\n",
      "97\t536\t0.3\t1\t520 16\n",
      "98\t497\t0.3\t1\t480 17\n",
      "99\t450\t0.3\t1\t434 16\n",
      "100\t389\t0.3\t1\t372 17\n",
      "101\t311\t0.3\t1\t300 11\n",
      "102\t265\t0.3\t1\t249 16\n",
      "103\t192\t0.3\t1\t179 13\n",
      "104\t185\t0.3\t1\t180 5\n",
      "105\t144\t0.3\t1\t132 12\n",
      "106\t121\t0.3\t1\t114 7\n",
      "107\t121\t0.3\t1\t109 12\n",
      "108\t112\t0.3\t1\t109 3\n",
      "109\t99\t0.3\t1\t86 13\n",
      "110\t95\t0.3\t1\t86 9\n",
      "111\t91\t0.3\t1\t78 13\n",
      "112\t66\t0.3\t1\t59 7\n",
      "113\t76\t0.3\t1\t58 18\n",
      "114\t57\t0.3\t1\t40 17\n",
      "115\t59\t0.3\t1\t49 10\n",
      "116\t58\t0.3\t1\t45 13\n",
      "117\t68\t0.3\t1\t46 22\n",
      "118\t51\t0.3\t1\t36 15\n",
      "119\t64\t0.3\t1\t39 25\n",
      "120\t48\t0.3\t1\t35 13\n",
      "121\t65\t0.3\t1\t50 15\n",
      "122\t47\t0.3\t1\t30 17\n",
      "123\t50\t0.3\t1\t28 22\n",
      "124\t38\t0.3\t1\t28 10\n",
      "125\t37\t0.3\t1\t23 14\n",
      "126\t38\t0.3\t1\t23 15\n",
      "127\t26\t0.3\t1\t14 12\n",
      "128\t19\t0.3\t1\t12 7\n",
      "129\t21\t0.3\t1\t12 9\n",
      "130\t13\t0.3\t1\t4 9\n",
      "131\t16\t0.3\t1\t10 6\n",
      "132\t5\t0.3\t1\t2 3\n",
      "133\t15\t0.3\t1\t8 7\n",
      "134\t8\t0.3\t1\t6 2\n",
      "135\t11\t0.3\t1\t8 3\n",
      "136\t9\t0.3\t1\t7 2\n",
      "137\t12\t0.3\t1\t8 4\n",
      "138\t5\t0.3\t1\t3 2\n",
      "139\t7\t0.3\t1\t3 4\n",
      "140\t5\t0.3\t1\t1 4\n",
      "141\t10\t0.3\t1\t4 6\n",
      "142\t5\t0.3\t1\t4 1\n",
      "143\t6\t0.3\t1\t3 3\n",
      "144\t1\t0.3\t1\t0 1\n",
      "145\t5\t0.3\t1\t3 2\n",
      "146\t17\t0.3\t1\t16 1\n",
      "147\t16\t0.3\t1\t10 6\n",
      "148\t29\t0.3\t1\t25 4\n",
      "149\t9\t0.3\t1\t6 3\n",
      "150\t4\t0.3\t1\t3 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/Q.2.fastq.gz\n",
      "=============================================\n",
      "18460544 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files Q.1_trimmed.fq.gz and Q.2_trimmed.fq.gz\n",
      "file_1: Q.1_trimmed.fq.gz, file_2: Q.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: Q.1_trimmed.fq.gz and Q.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to Q.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to Q.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to Q.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to Q.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 18460544\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 816193 (4.42%)\n",
      "\n",
      "Deleting both intermediate output files Q.1_trimmed.fq.gz and Q.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/Q.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/Q.1.fastq.gz ./data_raw/Q.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/R.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t365448\tAGATCGGAAGAGC\t1000000\t36.54\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 365448). Second best hit was Nextera (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/R.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/R.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/R.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to R.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/R.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/R.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 692.85 s (41 us/read; 1.46 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,891,058\n",
      "Reads with adapters:                10,768,798 (63.8%)\n",
      "Reads written (passing filters):    16,891,058 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,533,658,700 bp\n",
      "Quality-trimmed:              66,484,873 bp (2.6%)\n",
      "Total written (filtered):  2,188,612,644 bp (86.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10768798 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.7%\n",
      "  C: 35.7%\n",
      "  G: 29.2%\n",
      "  T: 18.0%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2069543\t4222764.5\t0\t2069543\n",
      "2\t712988\t1055691.1\t0\t712988\n",
      "3\t292467\t263922.8\t0\t292467\n",
      "4\t151536\t65980.7\t0\t151536\n",
      "5\t105896\t16495.2\t0\t105896\n",
      "6\t88246\t4123.8\t0\t88246\n",
      "7\t88714\t1030.9\t0\t88714\n",
      "8\t107976\t257.7\t0\t107976\n",
      "9\t138915\t64.4\t0\t138759 156\n",
      "10\t157450\t16.1\t1\t144311 13139\n",
      "11\t105635\t4.0\t1\t95072 10563\n",
      "12\t109565\t1.0\t1\t99060 10505\n",
      "13\t137516\t0.3\t1\t125435 12081\n",
      "14\t96970\t0.3\t1\t87605 9365\n",
      "15\t141923\t0.3\t1\t129662 12261\n",
      "16\t71711\t0.3\t1\t64203 7508\n",
      "17\t116886\t0.3\t1\t106511 10375\n",
      "18\t103673\t0.3\t1\t95260 8413\n",
      "19\t128173\t0.3\t1\t117585 10588\n",
      "20\t99268\t0.3\t1\t90841 8427\n",
      "21\t130585\t0.3\t1\t119337 11248\n",
      "22\t138945\t0.3\t1\t126613 12332\n",
      "23\t128106\t0.3\t1\t117207 10899\n",
      "24\t125175\t0.3\t1\t113603 11572\n",
      "25\t121894\t0.3\t1\t111673 10221\n",
      "26\t100391\t0.3\t1\t91571 8820\n",
      "27\t130805\t0.3\t1\t119397 11408\n",
      "28\t121553\t0.3\t1\t111545 10008\n",
      "29\t127835\t0.3\t1\t116560 11275\n",
      "30\t121212\t0.3\t1\t110731 10481\n",
      "31\t131456\t0.3\t1\t120459 10997\n",
      "32\t144255\t0.3\t1\t132274 11981\n",
      "33\t111556\t0.3\t1\t101590 9966\n",
      "34\t114196\t0.3\t1\t105128 9068\n",
      "35\t137865\t0.3\t1\t125704 12161\n",
      "36\t129631\t0.3\t1\t118174 11457\n",
      "37\t152915\t0.3\t1\t140011 12904\n",
      "38\t102368\t0.3\t1\t93947 8421\n",
      "39\t167681\t0.3\t1\t152920 14761\n",
      "40\t157033\t0.3\t1\t143118 13915\n",
      "41\t218911\t0.3\t1\t199732 19179\n",
      "42\t312575\t0.3\t1\t285059 27516\n",
      "43\t274674\t0.3\t1\t248524 26150\n",
      "44\t288142\t0.3\t1\t264130 24012\n",
      "45\t39456\t0.3\t1\t36101 3355\n",
      "46\t55128\t0.3\t1\t50505 4623\n",
      "47\t59009\t0.3\t1\t54098 4911\n",
      "48\t81613\t0.3\t1\t74994 6619\n",
      "49\t138546\t0.3\t1\t127482 11064\n",
      "50\t40687\t0.3\t1\t36814 3873\n",
      "51\t135608\t0.3\t1\t124102 11506\n",
      "52\t109732\t0.3\t1\t100845 8887\n",
      "53\t69211\t0.3\t1\t63492 5719\n",
      "54\t80209\t0.3\t1\t73682 6527\n",
      "55\t118102\t0.3\t1\t108647 9455\n",
      "56\t80865\t0.3\t1\t74380 6485\n",
      "57\t22306\t0.3\t1\t20417 1889\n",
      "58\t61405\t0.3\t1\t56396 5009\n",
      "59\t122551\t0.3\t1\t112645 9906\n",
      "60\t22498\t0.3\t1\t20586 1912\n",
      "61\t52704\t0.3\t1\t48515 4189\n",
      "62\t74077\t0.3\t1\t68005 6072\n",
      "63\t18559\t0.3\t1\t17008 1551\n",
      "64\t43356\t0.3\t1\t39880 3476\n",
      "65\t66536\t0.3\t1\t61125 5411\n",
      "66\t28447\t0.3\t1\t26067 2380\n",
      "67\t33396\t0.3\t1\t30711 2685\n",
      "68\t38330\t0.3\t1\t35268 3062\n",
      "69\t35031\t0.3\t1\t31975 3056\n",
      "70\t33764\t0.3\t1\t30858 2906\n",
      "71\t34539\t0.3\t1\t31356 3183\n",
      "72\t47099\t0.3\t1\t42081 5018\n",
      "73\t56543\t0.3\t1\t50631 5912\n",
      "74\t61943\t0.3\t1\t54609 7334\n",
      "75\t47072\t0.3\t1\t42552 4520\n",
      "76\t4353\t0.3\t1\t3883 470\n",
      "77\t243\t0.3\t1\t215 28\n",
      "78\t1270\t0.3\t1\t1157 113\n",
      "79\t3900\t0.3\t1\t3562 338\n",
      "80\t6208\t0.3\t1\t5673 535\n",
      "81\t7263\t0.3\t1\t6627 636\n",
      "82\t7809\t0.3\t1\t7154 655\n",
      "83\t8236\t0.3\t1\t7486 750\n",
      "84\t7992\t0.3\t1\t7282 710\n",
      "85\t7548\t0.3\t1\t6920 628\n",
      "86\t7160\t0.3\t1\t6568 592\n",
      "87\t6971\t0.3\t1\t6356 615\n",
      "88\t6345\t0.3\t1\t5850 495\n",
      "89\t5625\t0.3\t1\t5171 454\n",
      "90\t4915\t0.3\t1\t4484 431\n",
      "91\t4446\t0.3\t1\t4063 383\n",
      "92\t3895\t0.3\t1\t3548 347\n",
      "93\t3382\t0.3\t1\t3074 308\n",
      "94\t2946\t0.3\t1\t2650 296\n",
      "95\t2684\t0.3\t1\t2433 251\n",
      "96\t2472\t0.3\t1\t2261 211\n",
      "97\t2379\t0.3\t1\t2181 198\n",
      "98\t2307\t0.3\t1\t2103 204\n",
      "99\t1930\t0.3\t1\t1775 155\n",
      "100\t1754\t0.3\t1\t1604 150\n",
      "101\t1495\t0.3\t1\t1350 145\n",
      "102\t1267\t0.3\t1\t1132 135\n",
      "103\t1033\t0.3\t1\t920 113\n",
      "104\t855\t0.3\t1\t761 94\n",
      "105\t758\t0.3\t1\t684 74\n",
      "106\t655\t0.3\t1\t585 70\n",
      "107\t720\t0.3\t1\t618 102\n",
      "108\t616\t0.3\t1\t533 83\n",
      "109\t588\t0.3\t1\t537 51\n",
      "110\t503\t0.3\t1\t452 51\n",
      "111\t452\t0.3\t1\t377 75\n",
      "112\t381\t0.3\t1\t335 46\n",
      "113\t307\t0.3\t1\t280 27\n",
      "114\t254\t0.3\t1\t222 32\n",
      "115\t264\t0.3\t1\t237 27\n",
      "116\t241\t0.3\t1\t203 38\n",
      "117\t206\t0.3\t1\t181 25\n",
      "118\t211\t0.3\t1\t171 40\n",
      "119\t196\t0.3\t1\t161 35\n",
      "120\t154\t0.3\t1\t126 28\n",
      "121\t159\t0.3\t1\t132 27\n",
      "122\t149\t0.3\t1\t113 36\n",
      "123\t150\t0.3\t1\t109 41\n",
      "124\t115\t0.3\t1\t88 27\n",
      "125\t120\t0.3\t1\t92 28\n",
      "126\t95\t0.3\t1\t68 27\n",
      "127\t119\t0.3\t1\t84 35\n",
      "128\t78\t0.3\t1\t55 23\n",
      "129\t65\t0.3\t1\t46 19\n",
      "130\t57\t0.3\t1\t41 16\n",
      "131\t72\t0.3\t1\t53 19\n",
      "132\t40\t0.3\t1\t31 9\n",
      "133\t47\t0.3\t1\t41 6\n",
      "134\t58\t0.3\t1\t51 7\n",
      "135\t58\t0.3\t1\t51 7\n",
      "136\t43\t0.3\t1\t33 10\n",
      "137\t31\t0.3\t1\t24 7\n",
      "138\t43\t0.3\t1\t40 3\n",
      "139\t31\t0.3\t1\t27 4\n",
      "140\t23\t0.3\t1\t17 6\n",
      "141\t74\t0.3\t1\t69 5\n",
      "142\t34\t0.3\t1\t19 15\n",
      "143\t44\t0.3\t1\t41 3\n",
      "144\t29\t0.3\t1\t24 5\n",
      "145\t104\t0.3\t1\t100 4\n",
      "146\t142\t0.3\t1\t139 3\n",
      "147\t91\t0.3\t1\t85 6\n",
      "148\t228\t0.3\t1\t226 2\n",
      "149\t66\t0.3\t1\t50 16\n",
      "150\t18112\t0.3\t1\t53 18059\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/R.1.fastq.gz\n",
      "=============================================\n",
      "16891058 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/R.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/R.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to R.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/R.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/R.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 676.30 s (40 us/read; 1.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,891,058\n",
      "Reads with adapters:                10,209,783 (60.4%)\n",
      "Reads written (passing filters):    16,891,058 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,533,658,700 bp\n",
      "Quality-trimmed:             246,678,171 bp (9.7%)\n",
      "Total written (filtered):  2,037,995,355 bp (80.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10209783 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.3%\n",
      "  C: 37.3%\n",
      "  G: 33.3%\n",
      "  T: 12.9%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2229579\t4222764.5\t0\t2229579\n",
      "2\t631367\t1055691.1\t0\t631367\n",
      "3\t307999\t263922.8\t0\t307999\n",
      "4\t131053\t65980.7\t0\t131053\n",
      "5\t139797\t16495.2\t0\t139797\n",
      "6\t79611\t4123.8\t0\t79611\n",
      "7\t84075\t1030.9\t0\t84075\n",
      "8\t111784\t257.7\t0\t111784\n",
      "9\t148881\t64.4\t0\t148694 187\n",
      "10\t146523\t16.1\t1\t138862 7661\n",
      "11\t83624\t4.0\t1\t74952 8672\n",
      "12\t78475\t1.0\t1\t71765 6710\n",
      "13\t109195\t0.3\t1\t101946 7249\n",
      "14\t104936\t0.3\t1\t98788 6148\n",
      "15\t94350\t0.3\t1\t89152 5198\n",
      "16\t123670\t0.3\t1\t114848 8822\n",
      "17\t134686\t0.3\t1\t126662 8024\n",
      "18\t153171\t0.3\t1\t144787 8384\n",
      "19\t119384\t0.3\t1\t112413 6971\n",
      "20\t150978\t0.3\t1\t143710 7268\n",
      "21\t68895\t0.3\t1\t63656 5239\n",
      "22\t82663\t0.3\t1\t79223 3440\n",
      "23\t92258\t0.3\t1\t87926 4332\n",
      "24\t127942\t0.3\t1\t121699 6243\n",
      "25\t109398\t0.3\t1\t103773 5625\n",
      "26\t128564\t0.3\t1\t121364 7200\n",
      "27\t192887\t0.3\t1\t181521 11366\n",
      "28\t90301\t0.3\t1\t85783 4518\n",
      "29\t147376\t0.3\t1\t135432 11944\n",
      "30\t116895\t0.3\t1\t111598 5297\n",
      "31\t154712\t0.3\t1\t147692 7020\n",
      "32\t79319\t0.3\t1\t75121 4198\n",
      "33\t171497\t0.3\t1\t164568 6929\n",
      "34\t85984\t0.3\t1\t79979 6005\n",
      "35\t90881\t0.3\t1\t87206 3675\n",
      "36\t174101\t0.3\t1\t168156 5945\n",
      "37\t100902\t0.3\t1\t94712 6190\n",
      "38\t163811\t0.3\t1\t157765 6046\n",
      "39\t90475\t0.3\t1\t83692 6783\n",
      "40\t81343\t0.3\t1\t78151 3192\n",
      "41\t128791\t0.3\t1\t123955 4836\n",
      "42\t149034\t0.3\t1\t143948 5086\n",
      "43\t122373\t0.3\t1\t117887 4486\n",
      "44\t92916\t0.3\t1\t88804 4112\n",
      "45\t184946\t0.3\t1\t178597 6349\n",
      "46\t96681\t0.3\t1\t91843 4838\n",
      "47\t136825\t0.3\t1\t132817 4008\n",
      "48\t67911\t0.3\t1\t63575 4336\n",
      "49\t142313\t0.3\t1\t137308 5005\n",
      "50\t42554\t0.3\t1\t40307 2247\n",
      "51\t97483\t0.3\t1\t93800 3683\n",
      "52\t46703\t0.3\t1\t44519 2184\n",
      "53\t77358\t0.3\t1\t75001 2357\n",
      "54\t109733\t0.3\t1\t106744 2989\n",
      "55\t69089\t0.3\t1\t66641 2448\n",
      "56\t74242\t0.3\t1\t71932 2310\n",
      "57\t122139\t0.3\t1\t118878 3261\n",
      "58\t88807\t0.3\t1\t85887 2920\n",
      "59\t37863\t0.3\t1\t36192 1671\n",
      "60\t89254\t0.3\t1\t87189 2065\n",
      "61\t40033\t0.3\t1\t38461 1572\n",
      "62\t42559\t0.3\t1\t40708 1851\n",
      "63\t74267\t0.3\t1\t72376 1891\n",
      "64\t55071\t0.3\t1\t53345 1726\n",
      "65\t24220\t0.3\t1\t23004 1216\n",
      "66\t53348\t0.3\t1\t51748 1600\n",
      "67\t13531\t0.3\t1\t12960 571\n",
      "68\t36625\t0.3\t1\t35166 1459\n",
      "69\t34011\t0.3\t1\t32894 1117\n",
      "70\t65183\t0.3\t1\t63311 1872\n",
      "71\t11286\t0.3\t1\t10772 514\n",
      "72\t20520\t0.3\t1\t19789 731\n",
      "73\t22664\t0.3\t1\t21875 789\n",
      "74\t23073\t0.3\t1\t22277 796\n",
      "75\t25092\t0.3\t1\t24332 760\n",
      "76\t32944\t0.3\t1\t31965 979\n",
      "77\t54020\t0.3\t1\t52452 1568\n",
      "78\t43336\t0.3\t1\t41443 1893\n",
      "79\t12632\t0.3\t1\t11606 1026\n",
      "80\t2732\t0.3\t1\t2378 354\n",
      "81\t314\t0.3\t1\t281 33\n",
      "82\t387\t0.3\t1\t364 23\n",
      "83\t944\t0.3\t1\t909 35\n",
      "84\t1728\t0.3\t1\t1687 41\n",
      "85\t2734\t0.3\t1\t2665 69\n",
      "86\t3397\t0.3\t1\t3323 74\n",
      "87\t3607\t0.3\t1\t3500 107\n",
      "88\t3076\t0.3\t1\t2986 90\n",
      "89\t2317\t0.3\t1\t2258 59\n",
      "90\t1913\t0.3\t1\t1862 51\n",
      "91\t1660\t0.3\t1\t1612 48\n",
      "92\t1444\t0.3\t1\t1404 40\n",
      "93\t1094\t0.3\t1\t1059 35\n",
      "94\t886\t0.3\t1\t853 33\n",
      "95\t807\t0.3\t1\t776 31\n",
      "96\t746\t0.3\t1\t715 31\n",
      "97\t686\t0.3\t1\t658 28\n",
      "98\t612\t0.3\t1\t590 22\n",
      "99\t502\t0.3\t1\t479 23\n",
      "100\t457\t0.3\t1\t433 24\n",
      "101\t343\t0.3\t1\t320 23\n",
      "102\t301\t0.3\t1\t282 19\n",
      "103\t201\t0.3\t1\t186 15\n",
      "104\t198\t0.3\t1\t183 15\n",
      "105\t142\t0.3\t1\t135 7\n",
      "106\t135\t0.3\t1\t121 14\n",
      "107\t138\t0.3\t1\t128 10\n",
      "108\t123\t0.3\t1\t107 16\n",
      "109\t121\t0.3\t1\t106 15\n",
      "110\t90\t0.3\t1\t86 4\n",
      "111\t75\t0.3\t1\t63 12\n",
      "112\t87\t0.3\t1\t77 10\n",
      "113\t72\t0.3\t1\t65 7\n",
      "114\t58\t0.3\t1\t48 10\n",
      "115\t70\t0.3\t1\t59 11\n",
      "116\t62\t0.3\t1\t47 15\n",
      "117\t61\t0.3\t1\t49 12\n",
      "118\t54\t0.3\t1\t43 11\n",
      "119\t65\t0.3\t1\t51 14\n",
      "120\t48\t0.3\t1\t34 14\n",
      "121\t56\t0.3\t1\t41 15\n",
      "122\t43\t0.3\t1\t29 14\n",
      "123\t45\t0.3\t1\t31 14\n",
      "124\t41\t0.3\t1\t29 12\n",
      "125\t42\t0.3\t1\t24 18\n",
      "126\t29\t0.3\t1\t15 14\n",
      "127\t36\t0.3\t1\t24 12\n",
      "128\t23\t0.3\t1\t15 8\n",
      "129\t17\t0.3\t1\t9 8\n",
      "130\t19\t0.3\t1\t12 7\n",
      "131\t9\t0.3\t1\t5 4\n",
      "132\t13\t0.3\t1\t7 6\n",
      "133\t12\t0.3\t1\t10 2\n",
      "134\t8\t0.3\t1\t6 2\n",
      "135\t7\t0.3\t1\t5 2\n",
      "136\t9\t0.3\t1\t6 3\n",
      "137\t6\t0.3\t1\t5 1\n",
      "138\t4\t0.3\t1\t4\n",
      "139\t2\t0.3\t1\t1 1\n",
      "140\t2\t0.3\t1\t0 2\n",
      "141\t10\t0.3\t1\t8 2\n",
      "142\t3\t0.3\t1\t1 2\n",
      "143\t8\t0.3\t1\t6 2\n",
      "144\t4\t0.3\t1\t3 1\n",
      "145\t16\t0.3\t1\t13 3\n",
      "146\t13\t0.3\t1\t10 3\n",
      "147\t12\t0.3\t1\t8 4\n",
      "148\t27\t0.3\t1\t22 5\n",
      "149\t2\t0.3\t1\t0 2\n",
      "150\t6\t0.3\t1\t6\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/R.2.fastq.gz\n",
      "=============================================\n",
      "16891058 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files R.1_trimmed.fq.gz and R.2_trimmed.fq.gz\n",
      "file_1: R.1_trimmed.fq.gz, file_2: R.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: R.1_trimmed.fq.gz and R.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to R.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to R.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to R.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to R.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16891058\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1035570 (6.13%)\n",
      "\n",
      "Deleting both intermediate output files R.1_trimmed.fq.gz and R.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/R.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/R.1.fastq.gz ./data_raw/R.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/S.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t403094\tAGATCGGAAGAGC\t1000000\t40.31\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 403094). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/S.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/S.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/S.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to S.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/S.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/S.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 642.75 s (41 us/read; 1.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,824,914\n",
      "Reads with adapters:                10,477,802 (66.2%)\n",
      "Reads written (passing filters):    15,824,914 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,373,737,100 bp\n",
      "Quality-trimmed:              67,927,388 bp (2.9%)\n",
      "Total written (filtered):  2,011,141,568 bp (84.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10477802 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.9%\n",
      "  C: 33.4%\n",
      "  G: 32.0%\n",
      "  T: 17.2%\n",
      "  none/other: 0.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1820362\t3956228.5\t0\t1820362\n",
      "2\t629322\t989057.1\t0\t629322\n",
      "3\t266430\t247264.3\t0\t266430\n",
      "4\t139177\t61816.1\t0\t139177\n",
      "5\t97671\t15454.0\t0\t97671\n",
      "6\t81638\t3863.5\t0\t81638\n",
      "7\t81810\t965.9\t0\t81810\n",
      "8\t99374\t241.5\t0\t99374\n",
      "9\t127628\t60.4\t0\t127474 154\n",
      "10\t143383\t15.1\t1\t132282 11101\n",
      "11\t96453\t3.8\t1\t87118 9335\n",
      "12\t100589\t0.9\t1\t91483 9106\n",
      "13\t126277\t0.2\t1\t115982 10295\n",
      "14\t90155\t0.2\t1\t81974 8181\n",
      "15\t131109\t0.2\t1\t120570 10539\n",
      "16\t67419\t0.2\t1\t60631 6788\n",
      "17\t109190\t0.2\t1\t100118 9072\n",
      "18\t96788\t0.2\t1\t89421 7367\n",
      "19\t118505\t0.2\t1\t109367 9138\n",
      "20\t91660\t0.2\t1\t84405 7255\n",
      "21\t120180\t0.2\t1\t110586 9594\n",
      "22\t129932\t0.2\t1\t119464 10468\n",
      "23\t119939\t0.2\t1\t110421 9518\n",
      "24\t116684\t0.2\t1\t106400 10284\n",
      "25\t115249\t0.2\t1\t106275 8974\n",
      "26\t95796\t0.2\t1\t88102 7694\n",
      "27\t123940\t0.2\t1\t114123 9817\n",
      "28\t115418\t0.2\t1\t106823 8595\n",
      "29\t121838\t0.2\t1\t112162 9676\n",
      "30\t114199\t0.2\t1\t105102 9097\n",
      "31\t123912\t0.2\t1\t114447 9465\n",
      "32\t136602\t0.2\t1\t126137 10465\n",
      "33\t106580\t0.2\t1\t97886 8694\n",
      "34\t109862\t0.2\t1\t101523 8339\n",
      "35\t133214\t0.2\t1\t122283 10931\n",
      "36\t125663\t0.2\t1\t115401 10262\n",
      "37\t150334\t0.2\t1\t138632 11702\n",
      "38\t101694\t0.2\t1\t93793 7901\n",
      "39\t165442\t0.2\t1\t151901 13541\n",
      "40\t158122\t0.2\t1\t145499 12623\n",
      "41\t224492\t0.2\t1\t206621 17871\n",
      "42\t321963\t0.2\t1\t295978 25985\n",
      "43\t285877\t0.2\t1\t261463 24414\n",
      "44\t302281\t0.2\t1\t279930 22351\n",
      "45\t41796\t0.2\t1\t38405 3391\n",
      "46\t57941\t0.2\t1\t53519 4422\n",
      "47\t59994\t0.2\t1\t55460 4534\n",
      "48\t84500\t0.2\t1\t78278 6222\n",
      "49\t144958\t0.2\t1\t135102 9856\n",
      "50\t43001\t0.2\t1\t39324 3677\n",
      "51\t143715\t0.2\t1\t132803 10912\n",
      "52\t115636\t0.2\t1\t107220 8416\n",
      "53\t72840\t0.2\t1\t67354 5486\n",
      "54\t85582\t0.2\t1\t79303 6279\n",
      "55\t128109\t0.2\t1\t118888 9221\n",
      "56\t88515\t0.2\t1\t82304 6211\n",
      "57\t24616\t0.2\t1\t22750 1866\n",
      "58\t68710\t0.2\t1\t64006 4704\n",
      "59\t140081\t0.2\t1\t130550 9531\n",
      "60\t24855\t0.2\t1\t22972 1883\n",
      "61\t61053\t0.2\t1\t56763 4290\n",
      "62\t85746\t0.2\t1\t79783 5963\n",
      "63\t21362\t0.2\t1\t19787 1575\n",
      "64\t50620\t0.2\t1\t46995 3625\n",
      "65\t77894\t0.2\t1\t72541 5353\n",
      "66\t34044\t0.2\t1\t31532 2512\n",
      "67\t40650\t0.2\t1\t37825 2825\n",
      "68\t48226\t0.2\t1\t44871 3355\n",
      "69\t43399\t0.2\t1\t40124 3275\n",
      "70\t42400\t0.2\t1\t39263 3137\n",
      "71\t44196\t0.2\t1\t40618 3578\n",
      "72\t60634\t0.2\t1\t55014 5620\n",
      "73\t74028\t0.2\t1\t67276 6752\n",
      "74\t80595\t0.2\t1\t72328 8267\n",
      "75\t61323\t0.2\t1\t56188 5135\n",
      "76\t5538\t0.2\t1\t4940 598\n",
      "77\t314\t0.2\t1\t280 34\n",
      "78\t1685\t0.2\t1\t1561 124\n",
      "79\t5276\t0.2\t1\t4927 349\n",
      "80\t8685\t0.2\t1\t8052 633\n",
      "81\t10363\t0.2\t1\t9610 753\n",
      "82\t11088\t0.2\t1\t10299 789\n",
      "83\t11005\t0.2\t1\t10267 738\n",
      "84\t10739\t0.2\t1\t9966 773\n",
      "85\t10104\t0.2\t1\t9424 680\n",
      "86\t10051\t0.2\t1\t9356 695\n",
      "87\t9274\t0.2\t1\t8613 661\n",
      "88\t8448\t0.2\t1\t7837 611\n",
      "89\t7836\t0.2\t1\t7331 505\n",
      "90\t6930\t0.2\t1\t6503 427\n",
      "91\t6232\t0.2\t1\t5853 379\n",
      "92\t5607\t0.2\t1\t5229 378\n",
      "93\t4632\t0.2\t1\t4314 318\n",
      "94\t3963\t0.2\t1\t3688 275\n",
      "95\t3624\t0.2\t1\t3355 269\n",
      "96\t3369\t0.2\t1\t3149 220\n",
      "97\t3241\t0.2\t1\t3021 220\n",
      "98\t3032\t0.2\t1\t2811 221\n",
      "99\t2695\t0.2\t1\t2497 198\n",
      "100\t2402\t0.2\t1\t2229 173\n",
      "101\t2089\t0.2\t1\t1945 144\n",
      "102\t1666\t0.2\t1\t1548 118\n",
      "103\t1361\t0.2\t1\t1242 119\n",
      "104\t1117\t0.2\t1\t1036 81\n",
      "105\t922\t0.2\t1\t845 77\n",
      "106\t859\t0.2\t1\t781 78\n",
      "107\t819\t0.2\t1\t740 79\n",
      "108\t779\t0.2\t1\t726 53\n",
      "109\t704\t0.2\t1\t651 53\n",
      "110\t602\t0.2\t1\t556 46\n",
      "111\t482\t0.2\t1\t443 39\n",
      "112\t376\t0.2\t1\t345 31\n",
      "113\t395\t0.2\t1\t352 43\n",
      "114\t311\t0.2\t1\t265 46\n",
      "115\t330\t0.2\t1\t288 42\n",
      "116\t226\t0.2\t1\t198 28\n",
      "117\t214\t0.2\t1\t189 25\n",
      "118\t182\t0.2\t1\t159 23\n",
      "119\t156\t0.2\t1\t140 16\n",
      "120\t142\t0.2\t1\t122 20\n",
      "121\t170\t0.2\t1\t137 33\n",
      "122\t122\t0.2\t1\t100 22\n",
      "123\t132\t0.2\t1\t98 34\n",
      "124\t106\t0.2\t1\t77 29\n",
      "125\t114\t0.2\t1\t89 25\n",
      "126\t108\t0.2\t1\t73 35\n",
      "127\t92\t0.2\t1\t67 25\n",
      "128\t77\t0.2\t1\t54 23\n",
      "129\t72\t0.2\t1\t51 21\n",
      "130\t58\t0.2\t1\t36 22\n",
      "131\t62\t0.2\t1\t40 22\n",
      "132\t34\t0.2\t1\t28 6\n",
      "133\t48\t0.2\t1\t39 9\n",
      "134\t46\t0.2\t1\t41 5\n",
      "135\t48\t0.2\t1\t41 7\n",
      "136\t33\t0.2\t1\t25 8\n",
      "137\t34\t0.2\t1\t26 8\n",
      "138\t39\t0.2\t1\t36 3\n",
      "139\t33\t0.2\t1\t29 4\n",
      "140\t20\t0.2\t1\t17 3\n",
      "141\t61\t0.2\t1\t58 3\n",
      "142\t29\t0.2\t1\t20 9\n",
      "143\t46\t0.2\t1\t39 7\n",
      "144\t17\t0.2\t1\t15 2\n",
      "145\t95\t0.2\t1\t90 5\n",
      "146\t160\t0.2\t1\t158 2\n",
      "147\t72\t0.2\t1\t68 4\n",
      "148\t247\t0.2\t1\t247\n",
      "149\t73\t0.2\t1\t65 8\n",
      "150\t20517\t0.2\t1\t29 20488\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/S.1.fastq.gz\n",
      "=============================================\n",
      "15824914 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/S.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/S.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to S.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/S.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/S.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 631.45 s (40 us/read; 1.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,824,914\n",
      "Reads with adapters:                 9,964,182 (63.0%)\n",
      "Reads written (passing filters):    15,824,914 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,373,737,100 bp\n",
      "Quality-trimmed:             227,959,418 bp (9.6%)\n",
      "Total written (filtered):  1,882,366,732 bp (79.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9964182 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.3%\n",
      "  C: 34.7%\n",
      "  G: 36.0%\n",
      "  T: 12.9%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1983588\t3956228.5\t0\t1983588\n",
      "2\t555891\t989057.1\t0\t555891\n",
      "3\t280205\t247264.3\t0\t280205\n",
      "4\t121904\t61816.1\t0\t121904\n",
      "5\t128270\t15454.0\t0\t128270\n",
      "6\t74038\t3863.5\t0\t74038\n",
      "7\t77903\t965.9\t0\t77903\n",
      "8\t101551\t241.5\t0\t101551\n",
      "9\t135537\t60.4\t0\t135367 170\n",
      "10\t134154\t15.1\t1\t127067 7087\n",
      "11\t78580\t3.8\t1\t70188 8392\n",
      "12\t72762\t0.9\t1\t66335 6427\n",
      "13\t101018\t0.2\t1\t94281 6737\n",
      "14\t98127\t0.2\t1\t92206 5921\n",
      "15\t88108\t0.2\t1\t83155 4953\n",
      "16\t116498\t0.2\t1\t108277 8221\n",
      "17\t126202\t0.2\t1\t118778 7424\n",
      "18\t143755\t0.2\t1\t135959 7796\n",
      "19\t110128\t0.2\t1\t103640 6488\n",
      "20\t142340\t0.2\t1\t135589 6751\n",
      "21\t64475\t0.2\t1\t59478 4997\n",
      "22\t77946\t0.2\t1\t74903 3043\n",
      "23\t86901\t0.2\t1\t82807 4094\n",
      "24\t122443\t0.2\t1\t116351 6092\n",
      "25\t104490\t0.2\t1\t99221 5269\n",
      "26\t123269\t0.2\t1\t116398 6871\n",
      "27\t191906\t0.2\t1\t180283 11623\n",
      "28\t88411\t0.2\t1\t83943 4468\n",
      "29\t144598\t0.2\t1\t133008 11590\n",
      "30\t109146\t0.2\t1\t104013 5133\n",
      "31\t151724\t0.2\t1\t144961 6763\n",
      "32\t73557\t0.2\t1\t69247 4310\n",
      "33\t162873\t0.2\t1\t156333 6540\n",
      "34\t117361\t0.2\t1\t108783 8578\n",
      "35\t128163\t0.2\t1\t123194 4969\n",
      "36\t287763\t0.2\t1\t277969 9794\n",
      "37\t107691\t0.2\t1\t102388 5303\n",
      "38\t124578\t0.2\t1\t119596 4982\n",
      "39\t76927\t0.2\t1\t71771 5156\n",
      "40\t155949\t0.2\t1\t150569 5380\n",
      "41\t80699\t0.2\t1\t75738 4961\n",
      "42\t54113\t0.2\t1\t51746 2367\n",
      "43\t137254\t0.2\t1\t132834 4420\n",
      "44\t62345\t0.2\t1\t59788 2557\n",
      "45\t191876\t0.2\t1\t185949 5927\n",
      "46\t108840\t0.2\t1\t104085 4755\n",
      "47\t109031\t0.2\t1\t105933 3098\n",
      "48\t81731\t0.2\t1\t77059 4672\n",
      "49\t152164\t0.2\t1\t147116 5048\n",
      "50\t41930\t0.2\t1\t39825 2105\n",
      "51\t106669\t0.2\t1\t103122 3547\n",
      "52\t44062\t0.2\t1\t42083 1979\n",
      "53\t73739\t0.2\t1\t71804 1935\n",
      "54\t112389\t0.2\t1\t109645 2744\n",
      "55\t78869\t0.2\t1\t76236 2633\n",
      "56\t75438\t0.2\t1\t73267 2171\n",
      "57\t128379\t0.2\t1\t125391 2988\n",
      "58\t106423\t0.2\t1\t103206 3217\n",
      "59\t39860\t0.2\t1\t38282 1578\n",
      "60\t100105\t0.2\t1\t97993 2112\n",
      "61\t43773\t0.2\t1\t42242 1531\n",
      "62\t55404\t0.2\t1\t53453 1951\n",
      "63\t87109\t0.2\t1\t85147 1962\n",
      "64\t61995\t0.2\t1\t60123 1872\n",
      "65\t27447\t0.2\t1\t26140 1307\n",
      "66\t65570\t0.2\t1\t63805 1765\n",
      "67\t14369\t0.2\t1\t13755 614\n",
      "68\t48466\t0.2\t1\t46614 1852\n",
      "69\t41184\t0.2\t1\t39910 1274\n",
      "70\t88509\t0.2\t1\t86296 2213\n",
      "71\t13494\t0.2\t1\t12947 547\n",
      "72\t24555\t0.2\t1\t23705 850\n",
      "73\t28239\t0.2\t1\t27341 898\n",
      "74\t29147\t0.2\t1\t28258 889\n",
      "75\t31474\t0.2\t1\t30544 930\n",
      "76\t41416\t0.2\t1\t40327 1089\n",
      "77\t72366\t0.2\t1\t70593 1773\n",
      "78\t64702\t0.2\t1\t62515 2187\n",
      "79\t20447\t0.2\t1\t19140 1307\n",
      "80\t4566\t0.2\t1\t4055 511\n",
      "81\t439\t0.2\t1\t383 56\n",
      "82\t501\t0.2\t1\t475 26\n",
      "83\t1174\t0.2\t1\t1148 26\n",
      "84\t2267\t0.2\t1\t2217 50\n",
      "85\t3657\t0.2\t1\t3590 67\n",
      "86\t4803\t0.2\t1\t4695 108\n",
      "87\t5081\t0.2\t1\t4983 98\n",
      "88\t4360\t0.2\t1\t4247 113\n",
      "89\t3114\t0.2\t1\t3049 65\n",
      "90\t2660\t0.2\t1\t2594 66\n",
      "91\t2363\t0.2\t1\t2300 63\n",
      "92\t1968\t0.2\t1\t1917 51\n",
      "93\t1535\t0.2\t1\t1489 46\n",
      "94\t1217\t0.2\t1\t1184 33\n",
      "95\t1044\t0.2\t1\t1005 39\n",
      "96\t929\t0.2\t1\t909 20\n",
      "97\t868\t0.2\t1\t832 36\n",
      "98\t735\t0.2\t1\t703 32\n",
      "99\t610\t0.2\t1\t588 22\n",
      "100\t583\t0.2\t1\t556 27\n",
      "101\t472\t0.2\t1\t451 21\n",
      "102\t355\t0.2\t1\t345 10\n",
      "103\t276\t0.2\t1\t262 14\n",
      "104\t230\t0.2\t1\t219 11\n",
      "105\t188\t0.2\t1\t172 16\n",
      "106\t159\t0.2\t1\t153 6\n",
      "107\t160\t0.2\t1\t151 9\n",
      "108\t154\t0.2\t1\t144 10\n",
      "109\t119\t0.2\t1\t111 8\n",
      "110\t116\t0.2\t1\t102 14\n",
      "111\t120\t0.2\t1\t102 18\n",
      "112\t72\t0.2\t1\t62 10\n",
      "113\t90\t0.2\t1\t78 12\n",
      "114\t81\t0.2\t1\t70 11\n",
      "115\t57\t0.2\t1\t44 13\n",
      "116\t50\t0.2\t1\t40 10\n",
      "117\t50\t0.2\t1\t41 9\n",
      "118\t36\t0.2\t1\t25 11\n",
      "119\t47\t0.2\t1\t34 13\n",
      "120\t49\t0.2\t1\t39 10\n",
      "121\t54\t0.2\t1\t35 19\n",
      "122\t46\t0.2\t1\t29 17\n",
      "123\t43\t0.2\t1\t30 13\n",
      "124\t30\t0.2\t1\t20 10\n",
      "125\t44\t0.2\t1\t25 19\n",
      "126\t34\t0.2\t1\t24 10\n",
      "127\t34\t0.2\t1\t24 10\n",
      "128\t31\t0.2\t1\t22 9\n",
      "129\t19\t0.2\t1\t12 7\n",
      "130\t12\t0.2\t1\t6 6\n",
      "131\t11\t0.2\t1\t6 5\n",
      "132\t4\t0.2\t1\t4\n",
      "133\t11\t0.2\t1\t6 5\n",
      "134\t9\t0.2\t1\t5 4\n",
      "135\t5\t0.2\t1\t2 3\n",
      "136\t7\t0.2\t1\t5 2\n",
      "137\t5\t0.2\t1\t4 1\n",
      "138\t6\t0.2\t1\t4 2\n",
      "139\t7\t0.2\t1\t3 4\n",
      "140\t8\t0.2\t1\t3 5\n",
      "141\t13\t0.2\t1\t11 2\n",
      "142\t5\t0.2\t1\t2 3\n",
      "143\t6\t0.2\t1\t5 1\n",
      "144\t4\t0.2\t1\t3 1\n",
      "145\t11\t0.2\t1\t11\n",
      "146\t19\t0.2\t1\t18 1\n",
      "147\t8\t0.2\t1\t6 2\n",
      "148\t19\t0.2\t1\t17 2\n",
      "149\t8\t0.2\t1\t8\n",
      "150\t2\t0.2\t1\t2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/S.2.fastq.gz\n",
      "=============================================\n",
      "15824914 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files S.1_trimmed.fq.gz and S.2_trimmed.fq.gz\n",
      "file_1: S.1_trimmed.fq.gz, file_2: S.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: S.1_trimmed.fq.gz and S.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to S.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to S.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to S.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to S.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 15824914\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 912940 (5.77%)\n",
      "\n",
      "Deleting both intermediate output files S.1_trimmed.fq.gz and S.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/S.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/S.1.fastq.gz ./data_raw/S.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/T.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t364140\tAGATCGGAAGAGC\t1000000\t36.41\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 364140). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/T.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/T.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/T.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to T.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/T.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/T.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 728.82 s (40 us/read; 1.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,037,170\n",
      "Reads with adapters:                11,434,822 (63.4%)\n",
      "Reads written (passing filters):    18,037,170 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,705,575,500 bp\n",
      "Quality-trimmed:              64,686,087 bp (2.4%)\n",
      "Total written (filtered):  2,351,066,627 bp (86.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11434822 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.4%\n",
      "  C: 33.6%\n",
      "  G: 31.9%\n",
      "  T: 17.3%\n",
      "  none/other: 0.8%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2243101\t4509292.5\t0\t2243101\n",
      "2\t761706\t1127323.1\t0\t761706\n",
      "3\t317272\t281830.8\t0\t317272\n",
      "4\t166474\t70457.7\t0\t166474\n",
      "5\t120529\t17614.4\t0\t120529\n",
      "6\t100638\t4403.6\t0\t100638\n",
      "7\t99995\t1100.9\t0\t99995\n",
      "8\t121593\t275.2\t0\t121593\n",
      "9\t156692\t68.8\t0\t156531 161\n",
      "10\t170432\t17.2\t1\t159913 10519\n",
      "11\t119853\t4.3\t1\t110284 9569\n",
      "12\t120812\t1.1\t1\t111861 8951\n",
      "13\t150637\t0.3\t1\t140795 9842\n",
      "14\t109310\t0.3\t1\t101227 8083\n",
      "15\t155565\t0.3\t1\t145646 9919\n",
      "16\t80493\t0.3\t1\t73573 6920\n",
      "17\t130615\t0.3\t1\t121645 8970\n",
      "18\t114213\t0.3\t1\t107353 6860\n",
      "19\t141191\t0.3\t1\t132255 8936\n",
      "20\t107899\t0.3\t1\t100854 7045\n",
      "21\t142880\t0.3\t1\t133703 9177\n",
      "22\t150043\t0.3\t1\t140228 9815\n",
      "23\t140767\t0.3\t1\t131668 9099\n",
      "24\t137286\t0.3\t1\t127620 9666\n",
      "25\t136477\t0.3\t1\t127872 8605\n",
      "26\t111812\t0.3\t1\t104449 7363\n",
      "27\t143006\t0.3\t1\t133878 9128\n",
      "28\t131946\t0.3\t1\t124153 7793\n",
      "29\t140743\t0.3\t1\t131959 8784\n",
      "30\t130598\t0.3\t1\t122436 8162\n",
      "31\t142176\t0.3\t1\t133326 8850\n",
      "32\t155648\t0.3\t1\t146368 9280\n",
      "33\t120887\t0.3\t1\t112874 8013\n",
      "34\t123123\t0.3\t1\t116018 7105\n",
      "35\t150621\t0.3\t1\t140660 9961\n",
      "36\t139302\t0.3\t1\t130254 9048\n",
      "37\t166661\t0.3\t1\t156360 10301\n",
      "38\t109429\t0.3\t1\t102799 6630\n",
      "39\t178871\t0.3\t1\t167242 11629\n",
      "40\t157545\t0.3\t1\t147685 9860\n",
      "41\t222481\t0.3\t1\t208245 14236\n",
      "42\t315518\t0.3\t1\t295109 20409\n",
      "43\t283293\t0.3\t1\t262592 20701\n",
      "44\t300841\t0.3\t1\t283212 17629\n",
      "45\t41142\t0.3\t1\t38328 2814\n",
      "46\t58336\t0.3\t1\t54855 3481\n",
      "47\t61652\t0.3\t1\t57841 3811\n",
      "48\t83456\t0.3\t1\t78798 4658\n",
      "49\t140908\t0.3\t1\t133323 7585\n",
      "50\t41456\t0.3\t1\t38440 3016\n",
      "51\t135207\t0.3\t1\t127169 8038\n",
      "52\t106058\t0.3\t1\t100075 5983\n",
      "53\t68892\t0.3\t1\t64986 3906\n",
      "54\t78014\t0.3\t1\t73586 4428\n",
      "55\t111595\t0.3\t1\t105563 6032\n",
      "56\t78472\t0.3\t1\t74206 4266\n",
      "57\t21478\t0.3\t1\t20148 1330\n",
      "58\t58985\t0.3\t1\t55705 3280\n",
      "59\t116812\t0.3\t1\t110590 6222\n",
      "60\t20761\t0.3\t1\t19523 1238\n",
      "61\t49620\t0.3\t1\t46929 2691\n",
      "62\t67076\t0.3\t1\t63445 3631\n",
      "63\t16940\t0.3\t1\t15944 996\n",
      "64\t38788\t0.3\t1\t36660 2128\n",
      "65\t59109\t0.3\t1\t55953 3156\n",
      "66\t24839\t0.3\t1\t23438 1401\n",
      "67\t30499\t0.3\t1\t28818 1681\n",
      "68\t34513\t0.3\t1\t32649 1864\n",
      "69\t30889\t0.3\t1\t29149 1740\n",
      "70\t29641\t0.3\t1\t27903 1738\n",
      "71\t31535\t0.3\t1\t28551 2984\n",
      "72\t43834\t0.3\t1\t38056 5778\n",
      "73\t52689\t0.3\t1\t45561 7128\n",
      "74\t59217\t0.3\t1\t48595 10622\n",
      "75\t42642\t0.3\t1\t38610 4032\n",
      "76\t3845\t0.3\t1\t3487 358\n",
      "77\t193\t0.3\t1\t171 22\n",
      "78\t1106\t0.3\t1\t1042 64\n",
      "79\t3444\t0.3\t1\t3245 199\n",
      "80\t5451\t0.3\t1\t5134 317\n",
      "81\t6483\t0.3\t1\t6130 353\n",
      "82\t6793\t0.3\t1\t6421 372\n",
      "83\t7112\t0.3\t1\t6698 414\n",
      "84\t6847\t0.3\t1\t6461 386\n",
      "85\t6456\t0.3\t1\t6109 347\n",
      "86\t6428\t0.3\t1\t6074 354\n",
      "87\t5843\t0.3\t1\t5506 337\n",
      "88\t5373\t0.3\t1\t5074 299\n",
      "89\t4862\t0.3\t1\t4597 265\n",
      "90\t4506\t0.3\t1\t4242 264\n",
      "91\t4029\t0.3\t1\t3819 210\n",
      "92\t3579\t0.3\t1\t3393 186\n",
      "93\t3067\t0.3\t1\t2907 160\n",
      "94\t2849\t0.3\t1\t2670 179\n",
      "95\t2566\t0.3\t1\t2397 169\n",
      "96\t2419\t0.3\t1\t2271 148\n",
      "97\t2325\t0.3\t1\t2172 153\n",
      "98\t2197\t0.3\t1\t2071 126\n",
      "99\t1874\t0.3\t1\t1776 98\n",
      "100\t1759\t0.3\t1\t1654 105\n",
      "101\t1486\t0.3\t1\t1383 103\n",
      "102\t1316\t0.3\t1\t1243 73\n",
      "103\t1076\t0.3\t1\t1007 69\n",
      "104\t909\t0.3\t1\t850 59\n",
      "105\t847\t0.3\t1\t792 55\n",
      "106\t791\t0.3\t1\t720 71\n",
      "107\t692\t0.3\t1\t630 62\n",
      "108\t685\t0.3\t1\t624 61\n",
      "109\t604\t0.3\t1\t572 32\n",
      "110\t567\t0.3\t1\t525 42\n",
      "111\t448\t0.3\t1\t395 53\n",
      "112\t413\t0.3\t1\t369 44\n",
      "113\t362\t0.3\t1\t322 40\n",
      "114\t298\t0.3\t1\t262 36\n",
      "115\t302\t0.3\t1\t266 36\n",
      "116\t256\t0.3\t1\t222 34\n",
      "117\t238\t0.3\t1\t208 30\n",
      "118\t195\t0.3\t1\t158 37\n",
      "119\t208\t0.3\t1\t172 36\n",
      "120\t137\t0.3\t1\t106 31\n",
      "121\t160\t0.3\t1\t137 23\n",
      "122\t138\t0.3\t1\t108 30\n",
      "123\t147\t0.3\t1\t109 38\n",
      "124\t133\t0.3\t1\t98 35\n",
      "125\t140\t0.3\t1\t104 36\n",
      "126\t93\t0.3\t1\t64 29\n",
      "127\t98\t0.3\t1\t72 26\n",
      "128\t70\t0.3\t1\t52 18\n",
      "129\t80\t0.3\t1\t59 21\n",
      "130\t59\t0.3\t1\t39 20\n",
      "131\t71\t0.3\t1\t37 34\n",
      "132\t50\t0.3\t1\t39 11\n",
      "133\t51\t0.3\t1\t37 14\n",
      "134\t51\t0.3\t1\t43 8\n",
      "135\t57\t0.3\t1\t51 6\n",
      "136\t41\t0.3\t1\t30 11\n",
      "137\t44\t0.3\t1\t36 8\n",
      "138\t39\t0.3\t1\t32 7\n",
      "139\t27\t0.3\t1\t22 5\n",
      "140\t55\t0.3\t1\t44 11\n",
      "141\t80\t0.3\t1\t73 7\n",
      "142\t40\t0.3\t1\t29 11\n",
      "143\t38\t0.3\t1\t34 4\n",
      "144\t35\t0.3\t1\t32 3\n",
      "145\t229\t0.3\t1\t163 66\n",
      "146\t207\t0.3\t1\t202 5\n",
      "147\t118\t0.3\t1\t110 8\n",
      "148\t303\t0.3\t1\t299 4\n",
      "149\t77\t0.3\t1\t62 15\n",
      "150\t62826\t0.3\t1\t55 62771\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/T.1.fastq.gz\n",
      "=============================================\n",
      "18037170 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/T.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/T.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to T.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/T.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/T.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 717.58 s (40 us/read; 1.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,037,170\n",
      "Reads with adapters:                11,087,743 (61.5%)\n",
      "Reads written (passing filters):    18,037,170 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,705,575,500 bp\n",
      "Quality-trimmed:             212,592,651 bp (7.9%)\n",
      "Total written (filtered):  2,233,767,958 bp (82.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11087743 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.4%\n",
      "  C: 33.8%\n",
      "  G: 37.2%\n",
      "  T: 12.5%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2465411\t4509292.5\t0\t2465411\n",
      "2\t712736\t1127323.1\t0\t712736\n",
      "3\t354811\t281830.8\t0\t354811\n",
      "4\t149582\t70457.7\t0\t149582\n",
      "5\t156726\t17614.4\t0\t156726\n",
      "6\t90693\t4403.6\t0\t90693\n",
      "7\t95608\t1100.9\t0\t95608\n",
      "8\t130679\t275.2\t0\t130679\n",
      "9\t167072\t68.8\t0\t166847 225\n",
      "10\t173675\t17.2\t1\t165577 8098\n",
      "11\t89114\t4.3\t1\t80038 9076\n",
      "12\t90369\t1.1\t1\t83295 7074\n",
      "13\t120887\t0.3\t1\t113391 7496\n",
      "14\t120882\t0.3\t1\t114132 6750\n",
      "15\t108011\t0.3\t1\t102551 5460\n",
      "16\t135098\t0.3\t1\t125978 9120\n",
      "17\t153636\t0.3\t1\t144953 8683\n",
      "18\t166392\t0.3\t1\t158107 8285\n",
      "19\t123565\t0.3\t1\t116595 6970\n",
      "20\t167922\t0.3\t1\t161109 6813\n",
      "21\t78031\t0.3\t1\t72151 5880\n",
      "22\t102868\t0.3\t1\t98888 3980\n",
      "23\t105223\t0.3\t1\t100449 4774\n",
      "24\t148531\t0.3\t1\t141728 6803\n",
      "25\t111403\t0.3\t1\t105970 5433\n",
      "26\t148062\t0.3\t1\t140246 7816\n",
      "27\t225524\t0.3\t1\t212418 13106\n",
      "28\t128547\t0.3\t1\t122683 5864\n",
      "29\t136959\t0.3\t1\t125429 11530\n",
      "30\t114569\t0.3\t1\t108952 5617\n",
      "31\t199351\t0.3\t1\t190800 8551\n",
      "32\t107533\t0.3\t1\t101470 6063\n",
      "33\t227684\t0.3\t1\t218929 8755\n",
      "34\t55523\t0.3\t1\t51621 3902\n",
      "35\t123463\t0.3\t1\t118284 5179\n",
      "36\t218016\t0.3\t1\t210359 7657\n",
      "37\t92457\t0.3\t1\t88641 3816\n",
      "38\t121712\t0.3\t1\t118170 3542\n",
      "39\t110765\t0.3\t1\t106726 4039\n",
      "40\t159321\t0.3\t1\t153875 5446\n",
      "41\t89817\t0.3\t1\t86795 3022\n",
      "42\t109953\t0.3\t1\t106927 3026\n",
      "43\t122916\t0.3\t1\t119297 3619\n",
      "44\t75378\t0.3\t1\t72787 2591\n",
      "45\t172333\t0.3\t1\t167729 4604\n",
      "46\t96662\t0.3\t1\t92804 3858\n",
      "47\t143740\t0.3\t1\t140248 3492\n",
      "48\t71774\t0.3\t1\t68064 3710\n",
      "49\t140912\t0.3\t1\t136858 4054\n",
      "50\t48871\t0.3\t1\t46768 2103\n",
      "51\t100549\t0.3\t1\t97400 3149\n",
      "52\t62422\t0.3\t1\t60110 2312\n",
      "53\t81372\t0.3\t1\t79292 2080\n",
      "54\t110834\t0.3\t1\t108270 2564\n",
      "55\t72458\t0.3\t1\t70346 2112\n",
      "56\t74520\t0.3\t1\t72588 1932\n",
      "57\t108992\t0.3\t1\t106392 2600\n",
      "58\t88006\t0.3\t1\t85621 2385\n",
      "59\t44555\t0.3\t1\t43081 1474\n",
      "60\t84044\t0.3\t1\t82375 1669\n",
      "61\t41018\t0.3\t1\t39701 1317\n",
      "62\t45117\t0.3\t1\t43567 1550\n",
      "63\t66981\t0.3\t1\t65544 1437\n",
      "64\t52462\t0.3\t1\t51072 1390\n",
      "65\t23912\t0.3\t1\t22892 1020\n",
      "66\t52316\t0.3\t1\t51095 1221\n",
      "67\t15137\t0.3\t1\t14613 524\n",
      "68\t37710\t0.3\t1\t36482 1228\n",
      "69\t34026\t0.3\t1\t33077 949\n",
      "70\t61895\t0.3\t1\t60581 1314\n",
      "71\t11317\t0.3\t1\t10913 404\n",
      "72\t20014\t0.3\t1\t19379 635\n",
      "73\t21361\t0.3\t1\t20752 609\n",
      "74\t21559\t0.3\t1\t20935 624\n",
      "75\t22650\t0.3\t1\t22039 611\n",
      "76\t30052\t0.3\t1\t29271 781\n",
      "77\t52504\t0.3\t1\t51218 1286\n",
      "78\t43125\t0.3\t1\t41708 1417\n",
      "79\t12555\t0.3\t1\t11740 815\n",
      "80\t2755\t0.3\t1\t2420 335\n",
      "81\t359\t0.3\t1\t307 52\n",
      "82\t414\t0.3\t1\t392 22\n",
      "83\t1077\t0.3\t1\t1056 21\n",
      "84\t1927\t0.3\t1\t1887 40\n",
      "85\t2963\t0.3\t1\t2903 60\n",
      "86\t3679\t0.3\t1\t3614 65\n",
      "87\t3630\t0.3\t1\t3550 80\n",
      "88\t2820\t0.3\t1\t2766 54\n",
      "89\t1905\t0.3\t1\t1866 39\n",
      "90\t1648\t0.3\t1\t1606 42\n",
      "91\t1415\t0.3\t1\t1380 35\n",
      "92\t1146\t0.3\t1\t1120 26\n",
      "93\t900\t0.3\t1\t867 33\n",
      "94\t826\t0.3\t1\t813 13\n",
      "95\t699\t0.3\t1\t673 26\n",
      "96\t582\t0.3\t1\t562 20\n",
      "97\t562\t0.3\t1\t542 20\n",
      "98\t490\t0.3\t1\t470 20\n",
      "99\t423\t0.3\t1\t407 16\n",
      "100\t359\t0.3\t1\t343 16\n",
      "101\t306\t0.3\t1\t298 8\n",
      "102\t248\t0.3\t1\t235 13\n",
      "103\t201\t0.3\t1\t185 16\n",
      "104\t179\t0.3\t1\t167 12\n",
      "105\t160\t0.3\t1\t152 8\n",
      "106\t142\t0.3\t1\t131 11\n",
      "107\t128\t0.3\t1\t115 13\n",
      "108\t117\t0.3\t1\t107 10\n",
      "109\t120\t0.3\t1\t114 6\n",
      "110\t119\t0.3\t1\t102 17\n",
      "111\t81\t0.3\t1\t69 12\n",
      "112\t77\t0.3\t1\t66 11\n",
      "113\t74\t0.3\t1\t64 10\n",
      "114\t56\t0.3\t1\t49 7\n",
      "115\t80\t0.3\t1\t63 17\n",
      "116\t54\t0.3\t1\t39 15\n",
      "117\t74\t0.3\t1\t54 20\n",
      "118\t47\t0.3\t1\t29 18\n",
      "119\t52\t0.3\t1\t35 17\n",
      "120\t55\t0.3\t1\t37 18\n",
      "121\t45\t0.3\t1\t30 15\n",
      "122\t54\t0.3\t1\t35 19\n",
      "123\t56\t0.3\t1\t39 17\n",
      "124\t42\t0.3\t1\t29 13\n",
      "125\t48\t0.3\t1\t25 23\n",
      "126\t41\t0.3\t1\t33 8\n",
      "127\t25\t0.3\t1\t17 8\n",
      "128\t18\t0.3\t1\t12 6\n",
      "129\t23\t0.3\t1\t17 6\n",
      "130\t9\t0.3\t1\t5 4\n",
      "131\t15\t0.3\t1\t8 7\n",
      "132\t15\t0.3\t1\t9 6\n",
      "133\t13\t0.3\t1\t7 6\n",
      "134\t9\t0.3\t1\t4 5\n",
      "135\t11\t0.3\t1\t9 2\n",
      "136\t7\t0.3\t1\t5 2\n",
      "137\t4\t0.3\t1\t2 2\n",
      "138\t4\t0.3\t1\t2 2\n",
      "139\t2\t0.3\t1\t0 2\n",
      "140\t3\t0.3\t1\t2 1\n",
      "141\t9\t0.3\t1\t7 2\n",
      "142\t2\t0.3\t1\t2\n",
      "143\t7\t0.3\t1\t4 3\n",
      "144\t5\t0.3\t1\t4 1\n",
      "145\t25\t0.3\t1\t20 5\n",
      "146\t28\t0.3\t1\t25 3\n",
      "147\t5\t0.3\t1\t4 1\n",
      "148\t22\t0.3\t1\t19 3\n",
      "149\t6\t0.3\t1\t3 3\n",
      "150\t11\t0.3\t1\t10 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/T.2.fastq.gz\n",
      "=============================================\n",
      "18037170 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files T.1_trimmed.fq.gz and T.2_trimmed.fq.gz\n",
      "file_1: T.1_trimmed.fq.gz, file_2: T.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: T.1_trimmed.fq.gz and T.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to T.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to T.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to T.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to T.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 18037170\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 834954 (4.63%)\n",
      "\n",
      "Deleting both intermediate output files T.1_trimmed.fq.gz and T.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/T.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/T.1.fastq.gz ./data_raw/T.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/U.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t358276\tAGATCGGAAGAGC\t1000000\t35.83\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 358276). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/U.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/U.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/U.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to U.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/U.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/U.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 638.43 s (41 us/read; 1.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,702,314\n",
      "Reads with adapters:                10,047,857 (64.0%)\n",
      "Reads written (passing filters):    15,702,314 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,355,347,100 bp\n",
      "Quality-trimmed:              67,689,191 bp (2.9%)\n",
      "Total written (filtered):  2,026,885,585 bp (86.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10047857 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.7%\n",
      "  C: 36.0%\n",
      "  G: 28.6%\n",
      "  T: 18.3%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1918962\t3925578.5\t0\t1918962\n",
      "2\t645192\t981394.6\t0\t645192\n",
      "3\t267071\t245348.7\t0\t267071\n",
      "4\t139354\t61337.2\t0\t139354\n",
      "5\t97807\t15334.3\t0\t97807\n",
      "6\t80452\t3833.6\t0\t80452\n",
      "7\t81375\t958.4\t0\t81375\n",
      "8\t98069\t239.6\t0\t98069\n",
      "9\t128620\t59.9\t0\t128461 159\n",
      "10\t148754\t15.0\t1\t131388 17366\n",
      "11\t100961\t3.7\t1\t87737 13224\n",
      "12\t102927\t0.9\t1\t90161 12766\n",
      "13\t131763\t0.2\t1\t116252 15511\n",
      "14\t92531\t0.2\t1\t80735 11796\n",
      "15\t134585\t0.2\t1\t118684 15901\n",
      "16\t68996\t0.2\t1\t59639 9357\n",
      "17\t110589\t0.2\t1\t97237 13352\n",
      "18\t98161\t0.2\t1\t87066 11095\n",
      "19\t120324\t0.2\t1\t106607 13717\n",
      "20\t92955\t0.2\t1\t82120 10835\n",
      "21\t122887\t0.2\t1\t108711 14176\n",
      "22\t131105\t0.2\t1\t115609 15496\n",
      "23\t120800\t0.2\t1\t106635 14165\n",
      "24\t118499\t0.2\t1\t103890 14609\n",
      "25\t116244\t0.2\t1\t102992 13252\n",
      "26\t94452\t0.2\t1\t83286 11166\n",
      "27\t122834\t0.2\t1\t108398 14436\n",
      "28\t114653\t0.2\t1\t101883 12770\n",
      "29\t120602\t0.2\t1\t106574 14028\n",
      "30\t112821\t0.2\t1\t99672 13149\n",
      "31\t123138\t0.2\t1\t109130 14008\n",
      "32\t134945\t0.2\t1\t119657 15288\n",
      "33\t105130\t0.2\t1\t92542 12588\n",
      "34\t106727\t0.2\t1\t95063 11664\n",
      "35\t130255\t0.2\t1\t114929 15326\n",
      "36\t121795\t0.2\t1\t107352 14443\n",
      "37\t144308\t0.2\t1\t127735 16573\n",
      "38\t96261\t0.2\t1\t85259 11002\n",
      "39\t157832\t0.2\t1\t139046 18786\n",
      "40\t146203\t0.2\t1\t129126 17077\n",
      "41\t204210\t0.2\t1\t180115 24095\n",
      "42\t293422\t0.2\t1\t258423 34999\n",
      "43\t262030\t0.2\t1\t229774 32256\n",
      "44\t270455\t0.2\t1\t240147 30308\n",
      "45\t37210\t0.2\t1\t32864 4346\n",
      "46\t51980\t0.2\t1\t46164 5816\n",
      "47\t54187\t0.2\t1\t48118 6069\n",
      "48\t75390\t0.2\t1\t67016 8374\n",
      "49\t127971\t0.2\t1\t114438 13533\n",
      "50\t38427\t0.2\t1\t33819 4608\n",
      "51\t126556\t0.2\t1\t112216 14340\n",
      "52\t101242\t0.2\t1\t89845 11397\n",
      "53\t64367\t0.2\t1\t57160 7207\n",
      "54\t74253\t0.2\t1\t66002 8251\n",
      "55\t109218\t0.2\t1\t96906 12312\n",
      "56\t74957\t0.2\t1\t66676 8281\n",
      "57\t21078\t0.2\t1\t18652 2426\n",
      "58\t56917\t0.2\t1\t50730 6187\n",
      "59\t115089\t0.2\t1\t102705 12384\n",
      "60\t20867\t0.2\t1\t18497 2370\n",
      "61\t49202\t0.2\t1\t43835 5367\n",
      "62\t68698\t0.2\t1\t61236 7462\n",
      "63\t17166\t0.2\t1\t15291 1875\n",
      "64\t40689\t0.2\t1\t36171 4518\n",
      "65\t61930\t0.2\t1\t55171 6759\n",
      "66\t26582\t0.2\t1\t23500 3082\n",
      "67\t31223\t0.2\t1\t27774 3449\n",
      "68\t36441\t0.2\t1\t32453 3988\n",
      "69\t33000\t0.2\t1\t29072 3928\n",
      "70\t31789\t0.2\t1\t28209 3580\n",
      "71\t32613\t0.2\t1\t28818 3795\n",
      "72\t45466\t0.2\t1\t39464 6002\n",
      "73\t53801\t0.2\t1\t46943 6858\n",
      "74\t58400\t0.2\t1\t50449 7951\n",
      "75\t45223\t0.2\t1\t39810 5413\n",
      "76\t4485\t0.2\t1\t3816 669\n",
      "77\t222\t0.2\t1\t180 42\n",
      "78\t1289\t0.2\t1\t1151 138\n",
      "79\t3612\t0.2\t1\t3222 390\n",
      "80\t5977\t0.2\t1\t5308 669\n",
      "81\t7063\t0.2\t1\t6285 778\n",
      "82\t7721\t0.2\t1\t6868 853\n",
      "83\t7885\t0.2\t1\t7000 885\n",
      "84\t7680\t0.2\t1\t6786 894\n",
      "85\t7310\t0.2\t1\t6478 832\n",
      "86\t6908\t0.2\t1\t6099 809\n",
      "87\t6467\t0.2\t1\t5766 701\n",
      "88\t5884\t0.2\t1\t5201 683\n",
      "89\t5306\t0.2\t1\t4680 626\n",
      "90\t4843\t0.2\t1\t4307 536\n",
      "91\t4198\t0.2\t1\t3712 486\n",
      "92\t3816\t0.2\t1\t3403 413\n",
      "93\t3150\t0.2\t1\t2794 356\n",
      "94\t2868\t0.2\t1\t2527 341\n",
      "95\t2567\t0.2\t1\t2278 289\n",
      "96\t2468\t0.2\t1\t2189 279\n",
      "97\t2335\t0.2\t1\t2091 244\n",
      "98\t2227\t0.2\t1\t1967 260\n",
      "99\t1944\t0.2\t1\t1721 223\n",
      "100\t1649\t0.2\t1\t1455 194\n",
      "101\t1545\t0.2\t1\t1368 177\n",
      "102\t1244\t0.2\t1\t1088 156\n",
      "103\t1053\t0.2\t1\t925 128\n",
      "104\t926\t0.2\t1\t827 99\n",
      "105\t865\t0.2\t1\t763 102\n",
      "106\t743\t0.2\t1\t638 105\n",
      "107\t668\t0.2\t1\t572 96\n",
      "108\t631\t0.2\t1\t542 89\n",
      "109\t617\t0.2\t1\t536 81\n",
      "110\t515\t0.2\t1\t448 67\n",
      "111\t465\t0.2\t1\t393 72\n",
      "112\t388\t0.2\t1\t333 55\n",
      "113\t379\t0.2\t1\t314 65\n",
      "114\t292\t0.2\t1\t251 41\n",
      "115\t262\t0.2\t1\t227 35\n",
      "116\t240\t0.2\t1\t203 37\n",
      "117\t218\t0.2\t1\t176 42\n",
      "118\t197\t0.2\t1\t162 35\n",
      "119\t205\t0.2\t1\t169 36\n",
      "120\t166\t0.2\t1\t131 35\n",
      "121\t160\t0.2\t1\t131 29\n",
      "122\t135\t0.2\t1\t105 30\n",
      "123\t146\t0.2\t1\t102 44\n",
      "124\t100\t0.2\t1\t74 26\n",
      "125\t100\t0.2\t1\t76 24\n",
      "126\t79\t0.2\t1\t60 19\n",
      "127\t104\t0.2\t1\t76 28\n",
      "128\t64\t0.2\t1\t43 21\n",
      "129\t76\t0.2\t1\t58 18\n",
      "130\t49\t0.2\t1\t26 23\n",
      "131\t57\t0.2\t1\t40 17\n",
      "132\t35\t0.2\t1\t30 5\n",
      "133\t42\t0.2\t1\t36 6\n",
      "134\t58\t0.2\t1\t50 8\n",
      "135\t57\t0.2\t1\t55 2\n",
      "136\t28\t0.2\t1\t17 11\n",
      "137\t31\t0.2\t1\t27 4\n",
      "138\t29\t0.2\t1\t23 6\n",
      "139\t32\t0.2\t1\t24 8\n",
      "140\t21\t0.2\t1\t16 5\n",
      "141\t62\t0.2\t1\t58 4\n",
      "142\t35\t0.2\t1\t27 8\n",
      "143\t29\t0.2\t1\t26 3\n",
      "144\t17\t0.2\t1\t15 2\n",
      "145\t92\t0.2\t1\t91 1\n",
      "146\t159\t0.2\t1\t157 2\n",
      "147\t77\t0.2\t1\t75 2\n",
      "148\t181\t0.2\t1\t174 7\n",
      "149\t62\t0.2\t1\t51 11\n",
      "150\t11259\t0.2\t1\t42 11217\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/U.1.fastq.gz\n",
      "=============================================\n",
      "15702314 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/U.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/U.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to U.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/U.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/U.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 613.25 s (39 us/read; 1.54 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,702,314\n",
      "Reads with adapters:                 9,282,994 (59.1%)\n",
      "Reads written (passing filters):    15,702,314 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,355,347,100 bp\n",
      "Quality-trimmed:             289,801,282 bp (12.3%)\n",
      "Total written (filtered):  1,839,083,826 bp (78.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9282994 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.7%\n",
      "  C: 38.3%\n",
      "  G: 32.5%\n",
      "  T: 13.2%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1988730\t3925578.5\t0\t1988730\n",
      "2\t593370\t981394.6\t0\t593370\n",
      "3\t265202\t245348.7\t0\t265202\n",
      "4\t118083\t61337.2\t0\t118083\n",
      "5\t123917\t15334.3\t0\t123917\n",
      "6\t72045\t3833.6\t0\t72045\n",
      "7\t78258\t958.4\t0\t78258\n",
      "8\t105398\t239.6\t0\t105398\n",
      "9\t134268\t59.9\t0\t134059 209\n",
      "10\t145436\t15.0\t1\t136535 8901\n",
      "11\t69336\t3.7\t1\t61010 8326\n",
      "12\t76847\t0.9\t1\t69667 7180\n",
      "13\t95744\t0.2\t1\t88227 7517\n",
      "14\t100972\t0.2\t1\t93354 7618\n",
      "15\t89685\t0.2\t1\t83830 5855\n",
      "16\t109008\t0.2\t1\t100076 8932\n",
      "17\t127106\t0.2\t1\t118048 9058\n",
      "18\t136797\t0.2\t1\t127838 8959\n",
      "19\t113026\t0.2\t1\t105052 7974\n",
      "20\t131478\t0.2\t1\t123668 7810\n",
      "21\t59513\t0.2\t1\t54050 5463\n",
      "22\t84144\t0.2\t1\t79628 4516\n",
      "23\t83909\t0.2\t1\t78922 4987\n",
      "24\t128442\t0.2\t1\t119989 8453\n",
      "25\t90060\t0.2\t1\t83858 6202\n",
      "26\t120987\t0.2\t1\t112526 8461\n",
      "27\t190398\t0.2\t1\t176365 14033\n",
      "28\t120547\t0.2\t1\t113442 7105\n",
      "29\t108720\t0.2\t1\t97599 11121\n",
      "30\t116655\t0.2\t1\t109704 6951\n",
      "31\t143093\t0.2\t1\t134897 8196\n",
      "32\t104822\t0.2\t1\t98492 6330\n",
      "33\t118199\t0.2\t1\t110926 7273\n",
      "34\t190371\t0.2\t1\t181141 9230\n",
      "35\t94754\t0.2\t1\t89034 5720\n",
      "36\t147877\t0.2\t1\t140189 7688\n",
      "37\t94632\t0.2\t1\t89785 4847\n",
      "38\t71610\t0.2\t1\t67048 4562\n",
      "39\t89426\t0.2\t1\t84944 4482\n",
      "40\t58683\t0.2\t1\t54045 4638\n",
      "41\t132325\t0.2\t1\t126503 5822\n",
      "42\t65891\t0.2\t1\t62613 3278\n",
      "43\t118543\t0.2\t1\t113201 5342\n",
      "44\t62173\t0.2\t1\t59111 3062\n",
      "45\t157219\t0.2\t1\t150473 6746\n",
      "46\t79705\t0.2\t1\t75086 4619\n",
      "47\t116372\t0.2\t1\t112055 4317\n",
      "48\t59034\t0.2\t1\t54594 4440\n",
      "49\t124014\t0.2\t1\t118487 5527\n",
      "50\t38486\t0.2\t1\t35971 2515\n",
      "51\t92047\t0.2\t1\t87697 4350\n",
      "52\t49776\t0.2\t1\t46848 2928\n",
      "53\t74156\t0.2\t1\t71188 2968\n",
      "54\t100218\t0.2\t1\t96482 3736\n",
      "55\t63747\t0.2\t1\t60591 3156\n",
      "56\t69211\t0.2\t1\t66385 2826\n",
      "57\t109909\t0.2\t1\t106016 3893\n",
      "58\t80899\t0.2\t1\t77282 3617\n",
      "59\t39441\t0.2\t1\t37306 2135\n",
      "60\t86915\t0.2\t1\t84117 2798\n",
      "61\t36568\t0.2\t1\t34642 1926\n",
      "62\t38516\t0.2\t1\t36369 2147\n",
      "63\t70188\t0.2\t1\t67755 2433\n",
      "64\t50117\t0.2\t1\t47875 2242\n",
      "65\t21736\t0.2\t1\t20356 1380\n",
      "66\t50296\t0.2\t1\t48268 2028\n",
      "67\t14890\t0.2\t1\t14077 813\n",
      "68\t33883\t0.2\t1\t32062 1821\n",
      "69\t34046\t0.2\t1\t32441 1605\n",
      "70\t62399\t0.2\t1\t60097 2302\n",
      "71\t11404\t0.2\t1\t10790 614\n",
      "72\t19741\t0.2\t1\t18783 958\n",
      "73\t21033\t0.2\t1\t20107 926\n",
      "74\t21443\t0.2\t1\t20473 970\n",
      "75\t23367\t0.2\t1\t22342 1025\n",
      "76\t30576\t0.2\t1\t29370 1206\n",
      "77\t50133\t0.2\t1\t48266 1867\n",
      "78\t39645\t0.2\t1\t37301 2344\n",
      "79\t11147\t0.2\t1\t10061 1086\n",
      "80\t2414\t0.2\t1\t2045 369\n",
      "81\t334\t0.2\t1\t283 51\n",
      "82\t429\t0.2\t1\t396 33\n",
      "83\t1001\t0.2\t1\t952 49\n",
      "84\t1913\t0.2\t1\t1849 64\n",
      "85\t2919\t0.2\t1\t2816 103\n",
      "86\t3577\t0.2\t1\t3455 122\n",
      "87\t3345\t0.2\t1\t3231 114\n",
      "88\t2654\t0.2\t1\t2554 100\n",
      "89\t1741\t0.2\t1\t1684 57\n",
      "90\t1459\t0.2\t1\t1388 71\n",
      "91\t1137\t0.2\t1\t1088 49\n",
      "92\t1019\t0.2\t1\t976 43\n",
      "93\t777\t0.2\t1\t746 31\n",
      "94\t630\t0.2\t1\t601 29\n",
      "95\t574\t0.2\t1\t538 36\n",
      "96\t517\t0.2\t1\t490 27\n",
      "97\t437\t0.2\t1\t419 18\n",
      "98\t414\t0.2\t1\t389 25\n",
      "99\t355\t0.2\t1\t337 18\n",
      "100\t284\t0.2\t1\t271 13\n",
      "101\t275\t0.2\t1\t255 20\n",
      "102\t187\t0.2\t1\t171 16\n",
      "103\t148\t0.2\t1\t141 7\n",
      "104\t132\t0.2\t1\t121 11\n",
      "105\t140\t0.2\t1\t129 11\n",
      "106\t121\t0.2\t1\t108 13\n",
      "107\t115\t0.2\t1\t97 18\n",
      "108\t91\t0.2\t1\t87 4\n",
      "109\t98\t0.2\t1\t86 12\n",
      "110\t75\t0.2\t1\t60 15\n",
      "111\t84\t0.2\t1\t74 10\n",
      "112\t59\t0.2\t1\t54 5\n",
      "113\t52\t0.2\t1\t44 8\n",
      "114\t43\t0.2\t1\t34 9\n",
      "115\t49\t0.2\t1\t38 11\n",
      "116\t43\t0.2\t1\t33 10\n",
      "117\t39\t0.2\t1\t24 15\n",
      "118\t38\t0.2\t1\t32 6\n",
      "119\t50\t0.2\t1\t34 16\n",
      "120\t58\t0.2\t1\t40 18\n",
      "121\t45\t0.2\t1\t31 14\n",
      "122\t38\t0.2\t1\t24 14\n",
      "123\t46\t0.2\t1\t26 20\n",
      "124\t22\t0.2\t1\t14 8\n",
      "125\t32\t0.2\t1\t20 12\n",
      "126\t25\t0.2\t1\t16 9\n",
      "127\t35\t0.2\t1\t23 12\n",
      "128\t13\t0.2\t1\t10 3\n",
      "129\t17\t0.2\t1\t16 1\n",
      "130\t12\t0.2\t1\t6 6\n",
      "131\t11\t0.2\t1\t8 3\n",
      "132\t13\t0.2\t1\t10 3\n",
      "133\t5\t0.2\t1\t2 3\n",
      "134\t7\t0.2\t1\t5 2\n",
      "135\t9\t0.2\t1\t9\n",
      "136\t7\t0.2\t1\t5 2\n",
      "137\t4\t0.2\t1\t3 1\n",
      "138\t5\t0.2\t1\t3 2\n",
      "139\t3\t0.2\t1\t2 1\n",
      "140\t4\t0.2\t1\t1 3\n",
      "141\t4\t0.2\t1\t4\n",
      "143\t4\t0.2\t1\t3 1\n",
      "144\t2\t0.2\t1\t0 2\n",
      "145\t7\t0.2\t1\t5 2\n",
      "146\t16\t0.2\t1\t13 3\n",
      "147\t9\t0.2\t1\t7 2\n",
      "148\t8\t0.2\t1\t8\n",
      "149\t2\t0.2\t1\t1 1\n",
      "150\t5\t0.2\t1\t5\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/U.2.fastq.gz\n",
      "=============================================\n",
      "15702314 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files U.1_trimmed.fq.gz and U.2_trimmed.fq.gz\n",
      "file_1: U.1_trimmed.fq.gz, file_2: U.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: U.1_trimmed.fq.gz and U.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to U.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to U.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to U.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to U.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 15702314\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1353293 (8.62%)\n",
      "\n",
      "Deleting both intermediate output files U.1_trimmed.fq.gz and U.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/U.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/U.1.fastq.gz ./data_raw/U.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/V.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t334664\tAGATCGGAAGAGC\t1000000\t33.47\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 334664). Second best hit was Nextera (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/V.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/V.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/V.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to V.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/V.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/V.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1002.93 s (41 us/read; 1.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              24,749,848\n",
      "Reads with adapters:                15,267,214 (61.7%)\n",
      "Reads written (passing filters):    24,749,848 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,712,477,200 bp\n",
      "Quality-trimmed:              98,785,116 bp (2.7%)\n",
      "Total written (filtered):  3,227,157,577 bp (86.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 15267214 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.0%\n",
      "  C: 35.6%\n",
      "  G: 28.8%\n",
      "  T: 18.2%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3191282\t6187462.0\t0\t3191282\n",
      "2\t1064861\t1546865.5\t0\t1064861\n",
      "3\t430254\t386716.4\t0\t430254\n",
      "4\t211404\t96679.1\t0\t211404\n",
      "5\t145787\t24169.8\t0\t145787\n",
      "6\t119789\t6042.4\t0\t119789\n",
      "7\t118515\t1510.6\t0\t118515\n",
      "8\t144065\t377.7\t0\t144065\n",
      "9\t188640\t94.4\t0\t188360 280\n",
      "10\t212030\t23.6\t1\t189949 22081\n",
      "11\t150052\t5.9\t1\t131935 18117\n",
      "12\t149468\t1.5\t1\t132190 17278\n",
      "13\t186779\t0.4\t1\t166771 20008\n",
      "14\t135602\t0.4\t1\t119576 16026\n",
      "15\t192778\t0.4\t1\t172229 20549\n",
      "16\t100190\t0.4\t1\t87296 12894\n",
      "17\t161126\t0.4\t1\t143249 17877\n",
      "18\t141012\t0.4\t1\t126166 14846\n",
      "19\t175320\t0.4\t1\t156997 18323\n",
      "20\t132918\t0.4\t1\t118618 14300\n",
      "21\t177594\t0.4\t1\t158970 18624\n",
      "22\t187110\t0.4\t1\t166694 20416\n",
      "23\t175375\t0.4\t1\t156813 18562\n",
      "24\t172753\t0.4\t1\t152672 20081\n",
      "25\t166416\t0.4\t1\t148836 17580\n",
      "26\t139269\t0.4\t1\t124187 15082\n",
      "27\t177840\t0.4\t1\t158477 19363\n",
      "28\t163823\t0.4\t1\t147049 16774\n",
      "29\t175228\t0.4\t1\t155970 19258\n",
      "30\t163322\t0.4\t1\t145676 17646\n",
      "31\t178164\t0.4\t1\t159095 19069\n",
      "32\t193797\t0.4\t1\t173651 20146\n",
      "33\t152510\t0.4\t1\t135484 17026\n",
      "34\t154316\t0.4\t1\t138443 15873\n",
      "35\t190031\t0.4\t1\t168669 21362\n",
      "36\t177510\t0.4\t1\t157603 19907\n",
      "37\t210330\t0.4\t1\t187346 22984\n",
      "38\t139303\t0.4\t1\t124101 15202\n",
      "39\t230781\t0.4\t1\t204928 25853\n",
      "40\t206893\t0.4\t1\t183785 23108\n",
      "41\t294222\t0.4\t1\t261251 32971\n",
      "42\t424009\t0.4\t1\t375570 48439\n",
      "43\t385007\t0.4\t1\t338798 46209\n",
      "44\t406094\t0.4\t1\t362129 43965\n",
      "45\t55520\t0.4\t1\t48981 6539\n",
      "46\t78416\t0.4\t1\t69894 8522\n",
      "47\t81732\t0.4\t1\t72896 8836\n",
      "48\t111806\t0.4\t1\t99723 12083\n",
      "49\t190115\t0.4\t1\t170667 19448\n",
      "50\t57242\t0.4\t1\t50456 6786\n",
      "51\t188683\t0.4\t1\t167967 20716\n",
      "52\t147170\t0.4\t1\t131370 15800\n",
      "53\t94263\t0.4\t1\t84077 10186\n",
      "54\t109508\t0.4\t1\t97788 11720\n",
      "55\t158881\t0.4\t1\t141812 17069\n",
      "56\t111607\t0.4\t1\t99755 11852\n",
      "57\t30814\t0.4\t1\t27349 3465\n",
      "58\t84511\t0.4\t1\t75637 8874\n",
      "59\t170304\t0.4\t1\t152421 17883\n",
      "60\t30575\t0.4\t1\t27099 3476\n",
      "61\t73360\t0.4\t1\t65545 7815\n",
      "62\t100754\t0.4\t1\t89854 10900\n",
      "63\t25633\t0.4\t1\t22823 2810\n",
      "64\t59246\t0.4\t1\t52789 6457\n",
      "65\t91793\t0.4\t1\t81961 9832\n",
      "66\t38531\t0.4\t1\t34267 4264\n",
      "67\t47291\t0.4\t1\t42244 5047\n",
      "68\t54684\t0.4\t1\t48769 5915\n",
      "69\t48651\t0.4\t1\t43179 5472\n",
      "70\t47856\t0.4\t1\t42479 5377\n",
      "71\t49328\t0.4\t1\t43484 5844\n",
      "72\t67738\t0.4\t1\t58582 9156\n",
      "73\t81683\t0.4\t1\t71094 10589\n",
      "74\t90471\t0.4\t1\t77347 13124\n",
      "75\t69821\t0.4\t1\t61136 8685\n",
      "76\t6835\t0.4\t1\t5828 1007\n",
      "77\t313\t0.4\t1\t280 33\n",
      "78\t1825\t0.4\t1\t1625 200\n",
      "79\t5525\t0.4\t1\t4887 638\n",
      "80\t9268\t0.4\t1\t8261 1007\n",
      "81\t11321\t0.4\t1\t10025 1296\n",
      "82\t12084\t0.4\t1\t10788 1296\n",
      "83\t12328\t0.4\t1\t10920 1408\n",
      "84\t12132\t0.4\t1\t10734 1398\n",
      "85\t11474\t0.4\t1\t10227 1247\n",
      "86\t11186\t0.4\t1\t10015 1171\n",
      "87\t10224\t0.4\t1\t9033 1191\n",
      "88\t9692\t0.4\t1\t8611 1081\n",
      "89\t8920\t0.4\t1\t7938 982\n",
      "90\t8073\t0.4\t1\t7216 857\n",
      "91\t7127\t0.4\t1\t6309 818\n",
      "92\t6329\t0.4\t1\t5669 660\n",
      "93\t5392\t0.4\t1\t4758 634\n",
      "94\t4909\t0.4\t1\t4312 597\n",
      "95\t4396\t0.4\t1\t3895 501\n",
      "96\t4257\t0.4\t1\t3796 461\n",
      "97\t3876\t0.4\t1\t3438 438\n",
      "98\t3759\t0.4\t1\t3279 480\n",
      "99\t3381\t0.4\t1\t3016 365\n",
      "100\t2890\t0.4\t1\t2561 329\n",
      "101\t2518\t0.4\t1\t2228 290\n",
      "102\t2210\t0.4\t1\t1917 293\n",
      "103\t1772\t0.4\t1\t1572 200\n",
      "104\t1577\t0.4\t1\t1377 200\n",
      "105\t1431\t0.4\t1\t1249 182\n",
      "106\t1191\t0.4\t1\t1051 140\n",
      "107\t1178\t0.4\t1\t985 193\n",
      "108\t1120\t0.4\t1\t941 179\n",
      "109\t1058\t0.4\t1\t926 132\n",
      "110\t916\t0.4\t1\t792 124\n",
      "111\t733\t0.4\t1\t629 104\n",
      "112\t661\t0.4\t1\t562 99\n",
      "113\t627\t0.4\t1\t540 87\n",
      "114\t480\t0.4\t1\t400 80\n",
      "115\t443\t0.4\t1\t382 61\n",
      "116\t416\t0.4\t1\t366 50\n",
      "117\t405\t0.4\t1\t348 57\n",
      "118\t295\t0.4\t1\t243 52\n",
      "119\t321\t0.4\t1\t262 59\n",
      "120\t238\t0.4\t1\t186 52\n",
      "121\t297\t0.4\t1\t236 61\n",
      "122\t232\t0.4\t1\t180 52\n",
      "123\t243\t0.4\t1\t186 57\n",
      "124\t166\t0.4\t1\t122 44\n",
      "125\t215\t0.4\t1\t151 64\n",
      "126\t158\t0.4\t1\t106 52\n",
      "127\t181\t0.4\t1\t132 49\n",
      "128\t137\t0.4\t1\t99 38\n",
      "129\t142\t0.4\t1\t104 38\n",
      "130\t110\t0.4\t1\t72 38\n",
      "131\t107\t0.4\t1\t61 46\n",
      "132\t71\t0.4\t1\t52 19\n",
      "133\t71\t0.4\t1\t57 14\n",
      "134\t76\t0.4\t1\t66 10\n",
      "135\t77\t0.4\t1\t72 5\n",
      "136\t60\t0.4\t1\t53 7\n",
      "137\t56\t0.4\t1\t50 6\n",
      "138\t49\t0.4\t1\t47 2\n",
      "139\t54\t0.4\t1\t43 11\n",
      "140\t47\t0.4\t1\t35 12\n",
      "141\t107\t0.4\t1\t99 8\n",
      "142\t67\t0.4\t1\t51 16\n",
      "143\t70\t0.4\t1\t63 7\n",
      "144\t40\t0.4\t1\t35 5\n",
      "145\t149\t0.4\t1\t145 4\n",
      "146\t312\t0.4\t1\t301 11\n",
      "147\t154\t0.4\t1\t145 9\n",
      "148\t385\t0.4\t1\t378 7\n",
      "149\t132\t0.4\t1\t98 34\n",
      "150\t24587\t0.4\t1\t81 24506\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/V.1.fastq.gz\n",
      "=============================================\n",
      "24749848 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/V.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/V.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to V.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/V.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/V.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1296.33 s (52 us/read; 1.15 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              24,749,848\n",
      "Reads with adapters:                14,375,640 (58.1%)\n",
      "Reads written (passing filters):    24,749,848 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,712,477,200 bp\n",
      "Quality-trimmed:             417,798,617 bp (11.3%)\n",
      "Total written (filtered):  2,955,979,300 bp (79.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 14375640 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.7%\n",
      "  C: 36.6%\n",
      "  G: 33.4%\n",
      "  T: 13.0%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3441420\t6187462.0\t0\t3441420\n",
      "2\t980446\t1546865.5\t0\t980446\n",
      "3\t460101\t386716.4\t0\t460101\n",
      "4\t188867\t96679.1\t0\t188867\n",
      "5\t187545\t24169.8\t0\t187545\n",
      "6\t107979\t6042.4\t0\t107979\n",
      "7\t112590\t1510.6\t0\t112590\n",
      "8\t152235\t377.7\t0\t152235\n",
      "9\t199639\t94.4\t0\t199328 311\n",
      "10\t206930\t23.6\t1\t195046 11884\n",
      "11\t109617\t5.9\t1\t96912 12705\n",
      "12\t109037\t1.5\t1\t98885 10152\n",
      "13\t146578\t0.4\t1\t135738 10840\n",
      "14\t146505\t0.4\t1\t136608 9897\n",
      "15\t126122\t0.4\t1\t118353 7769\n",
      "16\t163782\t0.4\t1\t151002 12780\n",
      "17\t185208\t0.4\t1\t172910 12298\n",
      "18\t200517\t0.4\t1\t188651 11866\n",
      "19\t165472\t0.4\t1\t154946 10526\n",
      "20\t187920\t0.4\t1\t177882 10038\n",
      "21\t90524\t0.4\t1\t82770 7754\n",
      "22\t116808\t0.4\t1\t111276 5532\n",
      "23\t125948\t0.4\t1\t119003 6945\n",
      "24\t182585\t0.4\t1\t172136 10449\n",
      "25\t130907\t0.4\t1\t122986 7921\n",
      "26\t179471\t0.4\t1\t168062 11409\n",
      "27\t268327\t0.4\t1\t250184 18143\n",
      "28\t148348\t0.4\t1\t140243 8105\n",
      "29\t176708\t0.4\t1\t160035 16673\n",
      "30\t160386\t0.4\t1\t151700 8686\n",
      "31\t207311\t0.4\t1\t196678 10633\n",
      "32\t132517\t0.4\t1\t125268 7249\n",
      "33\t191528\t0.4\t1\t182508 9020\n",
      "34\t131020\t0.4\t1\t123655 7365\n",
      "35\t210192\t0.4\t1\t200846 9346\n",
      "36\t87308\t0.4\t1\t82747 4561\n",
      "37\t142732\t0.4\t1\t133483 9249\n",
      "38\t128871\t0.4\t1\t123249 5622\n",
      "39\t181120\t0.4\t1\t173553 7567\n",
      "40\t181830\t0.4\t1\t170141 11689\n",
      "41\t133603\t0.4\t1\t126126 7477\n",
      "42\t173653\t0.4\t1\t167164 6489\n",
      "43\t162604\t0.4\t1\t155666 6938\n",
      "44\t104744\t0.4\t1\t100135 4609\n",
      "45\t220510\t0.4\t1\t211723 8787\n",
      "46\t131453\t0.4\t1\t124009 7444\n",
      "47\t186842\t0.4\t1\t180551 6291\n",
      "48\t99765\t0.4\t1\t92745 7020\n",
      "49\t182334\t0.4\t1\t174790 7544\n",
      "50\t64490\t0.4\t1\t60562 3928\n",
      "51\t130665\t0.4\t1\t124849 5816\n",
      "52\t77877\t0.4\t1\t73717 4160\n",
      "53\t103560\t0.4\t1\t99803 3757\n",
      "54\t148080\t0.4\t1\t143303 4777\n",
      "55\t94904\t0.4\t1\t90909 3995\n",
      "56\t99876\t0.4\t1\t96129 3747\n",
      "57\t156282\t0.4\t1\t151302 4980\n",
      "58\t122372\t0.4\t1\t117702 4670\n",
      "59\t59679\t0.4\t1\t56628 3051\n",
      "60\t120455\t0.4\t1\t117033 3422\n",
      "61\t57418\t0.4\t1\t54764 2654\n",
      "62\t63767\t0.4\t1\t60618 3149\n",
      "63\t100363\t0.4\t1\t97153 3210\n",
      "64\t74129\t0.4\t1\t71193 2936\n",
      "65\t35526\t0.4\t1\t33506 2020\n",
      "66\t74700\t0.4\t1\t71941 2759\n",
      "67\t22344\t0.4\t1\t21164 1180\n",
      "68\t55402\t0.4\t1\t52769 2633\n",
      "69\t49678\t0.4\t1\t47697 1981\n",
      "70\t91741\t0.4\t1\t88654 3087\n",
      "71\t17202\t0.4\t1\t16335 867\n",
      "72\t30283\t0.4\t1\t28983 1300\n",
      "73\t32933\t0.4\t1\t31533 1400\n",
      "74\t33766\t0.4\t1\t32449 1317\n",
      "75\t36520\t0.4\t1\t35093 1427\n",
      "76\t47714\t0.4\t1\t46095 1619\n",
      "77\t82616\t0.4\t1\t79797 2819\n",
      "78\t67305\t0.4\t1\t63806 3499\n",
      "79\t19731\t0.4\t1\t17876 1855\n",
      "80\t4499\t0.4\t1\t3803 696\n",
      "81\t544\t0.4\t1\t473 71\n",
      "82\t642\t0.4\t1\t580 62\n",
      "83\t1623\t0.4\t1\t1570 53\n",
      "84\t2851\t0.4\t1\t2756 95\n",
      "85\t4464\t0.4\t1\t4315 149\n",
      "86\t5724\t0.4\t1\t5563 161\n",
      "87\t5752\t0.4\t1\t5586 166\n",
      "88\t4689\t0.4\t1\t4540 149\n",
      "89\t3536\t0.4\t1\t3414 122\n",
      "90\t2976\t0.4\t1\t2881 95\n",
      "91\t2535\t0.4\t1\t2445 90\n",
      "92\t2117\t0.4\t1\t2030 87\n",
      "93\t1654\t0.4\t1\t1573 81\n",
      "94\t1481\t0.4\t1\t1430 51\n",
      "95\t1254\t0.4\t1\t1186 68\n",
      "96\t1166\t0.4\t1\t1114 52\n",
      "97\t1003\t0.4\t1\t961 42\n",
      "98\t901\t0.4\t1\t852 49\n",
      "99\t779\t0.4\t1\t741 38\n",
      "100\t651\t0.4\t1\t617 34\n",
      "101\t572\t0.4\t1\t542 30\n",
      "102\t507\t0.4\t1\t472 35\n",
      "103\t350\t0.4\t1\t329 21\n",
      "104\t322\t0.4\t1\t300 22\n",
      "105\t260\t0.4\t1\t241 19\n",
      "106\t215\t0.4\t1\t199 16\n",
      "107\t222\t0.4\t1\t204 18\n",
      "108\t211\t0.4\t1\t198 13\n",
      "109\t193\t0.4\t1\t171 22\n",
      "110\t174\t0.4\t1\t159 15\n",
      "111\t155\t0.4\t1\t139 16\n",
      "112\t143\t0.4\t1\t129 14\n",
      "113\t121\t0.4\t1\t100 21\n",
      "114\t102\t0.4\t1\t86 16\n",
      "115\t91\t0.4\t1\t77 14\n",
      "116\t98\t0.4\t1\t80 18\n",
      "117\t102\t0.4\t1\t76 26\n",
      "118\t77\t0.4\t1\t62 15\n",
      "119\t82\t0.4\t1\t54 28\n",
      "120\t77\t0.4\t1\t56 21\n",
      "121\t80\t0.4\t1\t59 21\n",
      "122\t83\t0.4\t1\t58 25\n",
      "123\t74\t0.4\t1\t48 26\n",
      "124\t59\t0.4\t1\t36 23\n",
      "125\t72\t0.4\t1\t40 32\n",
      "126\t53\t0.4\t1\t28 25\n",
      "127\t57\t0.4\t1\t32 25\n",
      "128\t43\t0.4\t1\t25 18\n",
      "129\t37\t0.4\t1\t30 7\n",
      "130\t32\t0.4\t1\t18 14\n",
      "131\t24\t0.4\t1\t18 6\n",
      "132\t14\t0.4\t1\t9 5\n",
      "133\t14\t0.4\t1\t8 6\n",
      "134\t18\t0.4\t1\t10 8\n",
      "135\t11\t0.4\t1\t8 3\n",
      "136\t10\t0.4\t1\t5 5\n",
      "137\t12\t0.4\t1\t9 3\n",
      "138\t8\t0.4\t1\t5 3\n",
      "139\t6\t0.4\t1\t4 2\n",
      "140\t5\t0.4\t1\t4 1\n",
      "141\t18\t0.4\t1\t14 4\n",
      "142\t10\t0.4\t1\t9 1\n",
      "143\t11\t0.4\t1\t8 3\n",
      "144\t3\t0.4\t1\t2 1\n",
      "145\t15\t0.4\t1\t10 5\n",
      "146\t48\t0.4\t1\t40 8\n",
      "147\t15\t0.4\t1\t9 6\n",
      "148\t38\t0.4\t1\t37 1\n",
      "149\t14\t0.4\t1\t9 5\n",
      "150\t4\t0.4\t1\t4\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/V.2.fastq.gz\n",
      "=============================================\n",
      "24749848 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files V.1_trimmed.fq.gz and V.2_trimmed.fq.gz\n",
      "file_1: V.1_trimmed.fq.gz, file_2: V.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: V.1_trimmed.fq.gz and V.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to V.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to V.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to V.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to V.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 24749848\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1904811 (7.70%)\n",
      "\n",
      "Deleting both intermediate output files V.1_trimmed.fq.gz and V.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/V.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/V.1.fastq.gz ./data_raw/V.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/W.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t384971\tAGATCGGAAGAGC\t1000000\t38.50\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 384971). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/W.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/W.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/W.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to W.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/W.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/W.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 681.80 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,366,840\n",
      "Reads with adapters:                10,654,905 (65.1%)\n",
      "Reads written (passing filters):    16,366,840 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,455,026,000 bp\n",
      "Quality-trimmed:              66,400,399 bp (2.7%)\n",
      "Total written (filtered):  2,095,470,066 bp (85.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10654905 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.1%\n",
      "  C: 34.2%\n",
      "  G: 31.1%\n",
      "  T: 18.1%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1945347\t4091710.0\t0\t1945347\n",
      "2\t659004\t1022927.5\t0\t659004\n",
      "3\t277141\t255731.9\t0\t277141\n",
      "4\t141222\t63933.0\t0\t141222\n",
      "5\t99130\t15983.2\t0\t99130\n",
      "6\t83904\t3995.8\t0\t83904\n",
      "7\t83751\t999.0\t0\t83751\n",
      "8\t100097\t249.7\t0\t100097\n",
      "9\t130099\t62.4\t0\t129941 158\n",
      "10\t144648\t15.6\t1\t132792 11856\n",
      "11\t100755\t3.9\t1\t90656 10099\n",
      "12\t103791\t1.0\t1\t94048 9743\n",
      "13\t128894\t0.2\t1\t117749 11145\n",
      "14\t93254\t0.2\t1\t84330 8924\n",
      "15\t133334\t0.2\t1\t122053 11281\n",
      "16\t68947\t0.2\t1\t61750 7197\n",
      "17\t111352\t0.2\t1\t101355 9997\n",
      "18\t98495\t0.2\t1\t90531 7964\n",
      "19\t120512\t0.2\t1\t110358 10154\n",
      "20\t91931\t0.2\t1\t84264 7667\n",
      "21\t122673\t0.2\t1\t112310 10363\n",
      "22\t129259\t0.2\t1\t118068 11191\n",
      "23\t121209\t0.2\t1\t110866 10343\n",
      "24\t118759\t0.2\t1\t107779 10980\n",
      "25\t116999\t0.2\t1\t107014 9985\n",
      "26\t97036\t0.2\t1\t88559 8477\n",
      "27\t124901\t0.2\t1\t114211 10690\n",
      "28\t115695\t0.2\t1\t106510 9185\n",
      "29\t122182\t0.2\t1\t111682 10500\n",
      "30\t115355\t0.2\t1\t105532 9823\n",
      "31\t125143\t0.2\t1\t114589 10554\n",
      "32\t136143\t0.2\t1\t125132 11011\n",
      "33\t107840\t0.2\t1\t98269 9571\n",
      "34\t108754\t0.2\t1\t100133 8621\n",
      "35\t133833\t0.2\t1\t122144 11689\n",
      "36\t125292\t0.2\t1\t114097 11195\n",
      "37\t151208\t0.2\t1\t138533 12675\n",
      "38\t99857\t0.2\t1\t91511 8346\n",
      "39\t164650\t0.2\t1\t150097 14553\n",
      "40\t153643\t0.2\t1\t140501 13142\n",
      "41\t217531\t0.2\t1\t198293 19238\n",
      "42\t316517\t0.2\t1\t288441 28076\n",
      "43\t284561\t0.2\t1\t257745 26816\n",
      "44\t307405\t0.2\t1\t281438 25967\n",
      "45\t41227\t0.2\t1\t37524 3703\n",
      "46\t58082\t0.2\t1\t53061 5021\n",
      "47\t60114\t0.2\t1\t55087 5027\n",
      "48\t83803\t0.2\t1\t77039 6764\n",
      "49\t145364\t0.2\t1\t134096 11268\n",
      "50\t42415\t0.2\t1\t38432 3983\n",
      "51\t143679\t0.2\t1\t131512 12167\n",
      "52\t113066\t0.2\t1\t103911 9155\n",
      "53\t72998\t0.2\t1\t66926 6072\n",
      "54\t85609\t0.2\t1\t78645 6964\n",
      "55\t127117\t0.2\t1\t116936 10181\n",
      "56\t89752\t0.2\t1\t82532 7220\n",
      "57\t23913\t0.2\t1\t21809 2104\n",
      "58\t68295\t0.2\t1\t62821 5474\n",
      "59\t141234\t0.2\t1\t129685 11549\n",
      "60\t24354\t0.2\t1\t22237 2117\n",
      "61\t61485\t0.2\t1\t56639 4846\n",
      "62\t86117\t0.2\t1\t79232 6885\n",
      "63\t20747\t0.2\t1\t19004 1743\n",
      "64\t50853\t0.2\t1\t46635 4218\n",
      "65\t78129\t0.2\t1\t71830 6299\n",
      "66\t33138\t0.2\t1\t30337 2801\n",
      "67\t40809\t0.2\t1\t37420 3389\n",
      "68\t47499\t0.2\t1\t43737 3762\n",
      "69\t43385\t0.2\t1\t39647 3738\n",
      "70\t42486\t0.2\t1\t38750 3736\n",
      "71\t43955\t0.2\t1\t39760 4195\n",
      "72\t59947\t0.2\t1\t53827 6120\n",
      "73\t72243\t0.2\t1\t65044 7199\n",
      "74\t78831\t0.2\t1\t70147 8684\n",
      "75\t60900\t0.2\t1\t54934 5966\n",
      "76\t5456\t0.2\t1\t4753 703\n",
      "77\t314\t0.2\t1\t272 42\n",
      "78\t1620\t0.2\t1\t1471 149\n",
      "79\t5247\t0.2\t1\t4814 433\n",
      "80\t8473\t0.2\t1\t7739 734\n",
      "81\t10053\t0.2\t1\t9186 867\n",
      "82\t10820\t0.2\t1\t9905 915\n",
      "83\t10947\t0.2\t1\t10002 945\n",
      "84\t10592\t0.2\t1\t9630 962\n",
      "85\t9926\t0.2\t1\t9085 841\n",
      "86\t9625\t0.2\t1\t8802 823\n",
      "87\t8882\t0.2\t1\t8100 782\n",
      "88\t7843\t0.2\t1\t7176 667\n",
      "89\t7017\t0.2\t1\t6400 617\n",
      "90\t6255\t0.2\t1\t5695 560\n",
      "91\t5729\t0.2\t1\t5178 551\n",
      "92\t4933\t0.2\t1\t4454 479\n",
      "93\t4415\t0.2\t1\t3972 443\n",
      "94\t3760\t0.2\t1\t3431 329\n",
      "95\t3426\t0.2\t1\t3100 326\n",
      "96\t3188\t0.2\t1\t2893 295\n",
      "97\t3066\t0.2\t1\t2794 272\n",
      "98\t2732\t0.2\t1\t2502 230\n",
      "99\t2468\t0.2\t1\t2224 244\n",
      "100\t2130\t0.2\t1\t1929 201\n",
      "101\t1818\t0.2\t1\t1651 167\n",
      "102\t1490\t0.2\t1\t1350 140\n",
      "103\t1249\t0.2\t1\t1119 130\n",
      "104\t1115\t0.2\t1\t1013 102\n",
      "105\t937\t0.2\t1\t847 90\n",
      "106\t873\t0.2\t1\t794 79\n",
      "107\t874\t0.2\t1\t747 127\n",
      "108\t712\t0.2\t1\t616 96\n",
      "109\t630\t0.2\t1\t557 73\n",
      "110\t588\t0.2\t1\t530 58\n",
      "111\t486\t0.2\t1\t422 64\n",
      "112\t441\t0.2\t1\t382 59\n",
      "113\t365\t0.2\t1\t316 49\n",
      "114\t327\t0.2\t1\t277 50\n",
      "115\t341\t0.2\t1\t295 46\n",
      "116\t246\t0.2\t1\t210 36\n",
      "117\t249\t0.2\t1\t201 48\n",
      "118\t213\t0.2\t1\t173 40\n",
      "119\t189\t0.2\t1\t145 44\n",
      "120\t199\t0.2\t1\t166 33\n",
      "121\t200\t0.2\t1\t154 46\n",
      "122\t161\t0.2\t1\t122 39\n",
      "123\t171\t0.2\t1\t130 41\n",
      "124\t145\t0.2\t1\t109 36\n",
      "125\t181\t0.2\t1\t133 48\n",
      "126\t124\t0.2\t1\t80 44\n",
      "127\t125\t0.2\t1\t81 44\n",
      "128\t96\t0.2\t1\t68 28\n",
      "129\t92\t0.2\t1\t62 30\n",
      "130\t74\t0.2\t1\t54 20\n",
      "131\t92\t0.2\t1\t62 30\n",
      "132\t54\t0.2\t1\t43 11\n",
      "133\t64\t0.2\t1\t54 10\n",
      "134\t70\t0.2\t1\t62 8\n",
      "135\t59\t0.2\t1\t54 5\n",
      "136\t43\t0.2\t1\t39 4\n",
      "137\t66\t0.2\t1\t57 9\n",
      "138\t35\t0.2\t1\t28 7\n",
      "139\t47\t0.2\t1\t43 4\n",
      "140\t72\t0.2\t1\t63 9\n",
      "141\t87\t0.2\t1\t82 5\n",
      "142\t44\t0.2\t1\t34 10\n",
      "143\t42\t0.2\t1\t37 5\n",
      "144\t27\t0.2\t1\t25 2\n",
      "145\t109\t0.2\t1\t102 7\n",
      "146\t190\t0.2\t1\t186 4\n",
      "147\t106\t0.2\t1\t102 4\n",
      "148\t285\t0.2\t1\t281 4\n",
      "149\t78\t0.2\t1\t57 21\n",
      "150\t16108\t0.2\t1\t39 16069\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/W.1.fastq.gz\n",
      "=============================================\n",
      "16366840 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/W.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/W.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to W.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/W.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/W.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 664.33 s (41 us/read; 1.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,366,840\n",
      "Reads with adapters:                10,171,841 (62.1%)\n",
      "Reads written (passing filters):    16,366,840 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,455,026,000 bp\n",
      "Quality-trimmed:             240,853,627 bp (9.8%)\n",
      "Total written (filtered):  1,949,222,817 bp (79.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10171841 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.2%\n",
      "  C: 34.8%\n",
      "  G: 36.1%\n",
      "  T: 12.6%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2157004\t4091710.0\t0\t2157004\n",
      "2\t588565\t1022927.5\t0\t588565\n",
      "3\t307882\t255731.9\t0\t307882\n",
      "4\t129017\t63933.0\t0\t129017\n",
      "5\t132552\t15983.2\t0\t132552\n",
      "6\t76733\t3995.8\t0\t76733\n",
      "7\t80404\t999.0\t0\t80404\n",
      "8\t102084\t249.7\t0\t102084\n",
      "9\t136906\t62.4\t0\t136712 194\n",
      "10\t131683\t15.6\t1\t125232 6451\n",
      "11\t82114\t3.9\t1\t74007 8107\n",
      "12\t73326\t1.0\t1\t67117 6209\n",
      "13\t103974\t0.2\t1\t97530 6444\n",
      "14\t97658\t0.2\t1\t92213 5445\n",
      "15\t88920\t0.2\t1\t84267 4653\n",
      "16\t117572\t0.2\t1\t109613 7959\n",
      "17\t126027\t0.2\t1\t118944 7083\n",
      "18\t146071\t0.2\t1\t138641 7430\n",
      "19\t108165\t0.2\t1\t102072 6093\n",
      "20\t146928\t0.2\t1\t140300 6628\n",
      "21\t65453\t0.2\t1\t60798 4655\n",
      "22\t73174\t0.2\t1\t70366 2808\n",
      "23\t87056\t0.2\t1\t83381 3675\n",
      "24\t119216\t0.2\t1\t113668 5548\n",
      "25\t110771\t0.2\t1\t105629 5142\n",
      "26\t117932\t0.2\t1\t111894 6038\n",
      "27\t181519\t0.2\t1\t171713 9806\n",
      "28\t74367\t0.2\t1\t70621 3746\n",
      "29\t146246\t0.2\t1\t135674 10572\n",
      "30\t100581\t0.2\t1\t96275 4306\n",
      "31\t153015\t0.2\t1\t146968 6047\n",
      "32\t73548\t0.2\t1\t69895 3653\n",
      "33\t109775\t0.2\t1\t105030 4745\n",
      "34\t210870\t0.2\t1\t202985 7885\n",
      "35\t54137\t0.2\t1\t50841 3296\n",
      "36\t141247\t0.2\t1\t134797 6450\n",
      "37\t52641\t0.2\t1\t50318 2323\n",
      "38\t127564\t0.2\t1\t119885 7679\n",
      "39\t126683\t0.2\t1\t120855 5828\n",
      "40\t239814\t0.2\t1\t231616 8198\n",
      "41\t68363\t0.2\t1\t63886 4477\n",
      "42\t54375\t0.2\t1\t51835 2540\n",
      "43\t150482\t0.2\t1\t145822 4660\n",
      "44\t47221\t0.2\t1\t44959 2262\n",
      "45\t188568\t0.2\t1\t182899 5669\n",
      "46\t110671\t0.2\t1\t105888 4783\n",
      "47\t89231\t0.2\t1\t86406 2825\n",
      "48\t98598\t0.2\t1\t93229 5369\n",
      "49\t165596\t0.2\t1\t159887 5709\n",
      "50\t41978\t0.2\t1\t39633 2345\n",
      "51\t122914\t0.2\t1\t118626 4288\n",
      "52\t52853\t0.2\t1\t50490 2363\n",
      "53\t88863\t0.2\t1\t86445 2418\n",
      "54\t102460\t0.2\t1\t99943 2517\n",
      "55\t95385\t0.2\t1\t92339 3046\n",
      "56\t78099\t0.2\t1\t75825 2274\n",
      "57\t97376\t0.2\t1\t94672 2704\n",
      "58\t118767\t0.2\t1\t115251 3516\n",
      "59\t48580\t0.2\t1\t46721 1859\n",
      "60\t93783\t0.2\t1\t91646 2137\n",
      "61\t51419\t0.2\t1\t49652 1767\n",
      "62\t56601\t0.2\t1\t54579 2022\n",
      "63\t76998\t0.2\t1\t75051 1947\n",
      "64\t74653\t0.2\t1\t72592 2061\n",
      "65\t31296\t0.2\t1\t29800 1496\n",
      "66\t73206\t0.2\t1\t71237 1969\n",
      "67\t16591\t0.2\t1\t15844 747\n",
      "68\t54235\t0.2\t1\t52259 1976\n",
      "69\t47272\t0.2\t1\t45737 1535\n",
      "70\t93543\t0.2\t1\t91177 2366\n",
      "71\t16591\t0.2\t1\t15945 646\n",
      "72\t28217\t0.2\t1\t27332 885\n",
      "73\t29716\t0.2\t1\t28754 962\n",
      "74\t30273\t0.2\t1\t29287 986\n",
      "75\t31794\t0.2\t1\t30797 997\n",
      "76\t40942\t0.2\t1\t39767 1175\n",
      "77\t72231\t0.2\t1\t70366 1865\n",
      "78\t64732\t0.2\t1\t62267 2465\n",
      "79\t22022\t0.2\t1\t20481 1541\n",
      "80\t5292\t0.2\t1\t4627 665\n",
      "81\t506\t0.2\t1\t434 72\n",
      "82\t478\t0.2\t1\t443 35\n",
      "83\t1077\t0.2\t1\t1045 32\n",
      "84\t2130\t0.2\t1\t2082 48\n",
      "85\t3298\t0.2\t1\t3210 88\n",
      "86\t4327\t0.2\t1\t4245 82\n",
      "87\t4531\t0.2\t1\t4418 113\n",
      "88\t4025\t0.2\t1\t3932 93\n",
      "89\t2857\t0.2\t1\t2786 71\n",
      "90\t2493\t0.2\t1\t2442 51\n",
      "91\t2162\t0.2\t1\t2102 60\n",
      "92\t1792\t0.2\t1\t1738 54\n",
      "93\t1456\t0.2\t1\t1407 49\n",
      "94\t1249\t0.2\t1\t1209 40\n",
      "95\t1041\t0.2\t1\t1003 38\n",
      "96\t897\t0.2\t1\t866 31\n",
      "97\t885\t0.2\t1\t865 20\n",
      "98\t742\t0.2\t1\t723 19\n",
      "99\t630\t0.2\t1\t600 30\n",
      "100\t500\t0.2\t1\t474 26\n",
      "101\t468\t0.2\t1\t450 18\n",
      "102\t369\t0.2\t1\t357 12\n",
      "103\t277\t0.2\t1\t265 12\n",
      "104\t262\t0.2\t1\t253 9\n",
      "105\t214\t0.2\t1\t203 11\n",
      "106\t170\t0.2\t1\t159 11\n",
      "107\t188\t0.2\t1\t178 10\n",
      "108\t154\t0.2\t1\t137 17\n",
      "109\t131\t0.2\t1\t122 9\n",
      "110\t120\t0.2\t1\t112 8\n",
      "111\t116\t0.2\t1\t100 16\n",
      "112\t97\t0.2\t1\t87 10\n",
      "113\t96\t0.2\t1\t85 11\n",
      "114\t77\t0.2\t1\t65 12\n",
      "115\t90\t0.2\t1\t77 13\n",
      "116\t63\t0.2\t1\t53 10\n",
      "117\t67\t0.2\t1\t48 19\n",
      "118\t60\t0.2\t1\t49 11\n",
      "119\t57\t0.2\t1\t39 18\n",
      "120\t57\t0.2\t1\t44 13\n",
      "121\t57\t0.2\t1\t42 15\n",
      "122\t56\t0.2\t1\t39 17\n",
      "123\t62\t0.2\t1\t40 22\n",
      "124\t44\t0.2\t1\t29 15\n",
      "125\t58\t0.2\t1\t41 17\n",
      "126\t49\t0.2\t1\t35 14\n",
      "127\t42\t0.2\t1\t25 17\n",
      "128\t29\t0.2\t1\t21 8\n",
      "129\t17\t0.2\t1\t11 6\n",
      "130\t18\t0.2\t1\t14 4\n",
      "131\t18\t0.2\t1\t10 8\n",
      "132\t8\t0.2\t1\t6 2\n",
      "133\t15\t0.2\t1\t12 3\n",
      "134\t14\t0.2\t1\t12 2\n",
      "135\t15\t0.2\t1\t10 5\n",
      "136\t7\t0.2\t1\t6 1\n",
      "137\t11\t0.2\t1\t10 1\n",
      "138\t6\t0.2\t1\t3 3\n",
      "139\t7\t0.2\t1\t6 1\n",
      "140\t11\t0.2\t1\t10 1\n",
      "141\t9\t0.2\t1\t8 1\n",
      "142\t3\t0.2\t1\t2 1\n",
      "143\t7\t0.2\t1\t4 3\n",
      "144\t4\t0.2\t1\t4\n",
      "145\t11\t0.2\t1\t9 2\n",
      "146\t23\t0.2\t1\t22 1\n",
      "147\t11\t0.2\t1\t9 2\n",
      "148\t27\t0.2\t1\t22 5\n",
      "149\t9\t0.2\t1\t8 1\n",
      "150\t3\t0.2\t1\t2 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/W.2.fastq.gz\n",
      "=============================================\n",
      "16366840 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files W.1_trimmed.fq.gz and W.2_trimmed.fq.gz\n",
      "file_1: W.1_trimmed.fq.gz, file_2: W.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: W.1_trimmed.fq.gz and W.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to W.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to W.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to W.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to W.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16366840\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1020529 (6.24%)\n",
      "\n",
      "Deleting both intermediate output files W.1_trimmed.fq.gz and W.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/W.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/W.1.fastq.gz ./data_raw/W.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data_raw/X.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t385170\tAGATCGGAAGAGC\t1000000\t38.52\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 385170). Second best hit was Nextera (count: 0)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/X.1.fastq.gz: uncompress failed\n",
      "Writing report to './data_raw/X.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/X.1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to X.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/X.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/X.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1017.98 s (63 us/read; 0.95 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,200,512\n",
      "Reads with adapters:                10,560,855 (65.2%)\n",
      "Reads written (passing filters):    16,200,512 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,430,076,800 bp\n",
      "Quality-trimmed:              64,723,089 bp (2.7%)\n",
      "Total written (filtered):  2,082,354,507 bp (85.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10560855 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.3%\n",
      "  C: 35.6%\n",
      "  G: 30.0%\n",
      "  T: 18.8%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1920384\t4050128.0\t0\t1920384\n",
      "2\t654057\t1012532.0\t0\t654057\n",
      "3\t276215\t253133.0\t0\t276215\n",
      "4\t144418\t63283.2\t0\t144418\n",
      "5\t101569\t15820.8\t0\t101569\n",
      "6\t83890\t3955.2\t0\t83890\n",
      "7\t83089\t988.8\t0\t83089\n",
      "8\t103235\t247.2\t0\t103235\n",
      "9\t134711\t61.8\t0\t134550 161\n",
      "10\t150069\t15.5\t1\t136844 13225\n",
      "11\t103411\t3.9\t1\t92544 10867\n",
      "12\t106086\t1.0\t1\t95551 10535\n",
      "13\t131301\t0.2\t1\t119306 11995\n",
      "14\t94635\t0.2\t1\t85124 9511\n",
      "15\t134671\t0.2\t1\t122801 11870\n",
      "16\t70270\t0.2\t1\t62518 7752\n",
      "17\t113692\t0.2\t1\t103228 10464\n",
      "18\t100496\t0.2\t1\t91833 8663\n",
      "19\t124503\t0.2\t1\t113734 10769\n",
      "20\t95257\t0.2\t1\t87035 8222\n",
      "21\t128012\t0.2\t1\t116904 11108\n",
      "22\t135346\t0.2\t1\t123102 12244\n",
      "23\t125853\t0.2\t1\t114830 11023\n",
      "24\t122596\t0.2\t1\t110962 11634\n",
      "25\t119412\t0.2\t1\t109199 10213\n",
      "26\t99213\t0.2\t1\t90303 8910\n",
      "27\t128467\t0.2\t1\t117169 11298\n",
      "28\t119363\t0.2\t1\t109648 9715\n",
      "29\t126563\t0.2\t1\t115673 10890\n",
      "30\t119637\t0.2\t1\t109445 10192\n",
      "31\t129750\t0.2\t1\t118724 11026\n",
      "32\t143681\t0.2\t1\t131957 11724\n",
      "33\t111069\t0.2\t1\t101303 9766\n",
      "34\t112549\t0.2\t1\t103305 9244\n",
      "35\t138770\t0.2\t1\t126427 12343\n",
      "36\t130598\t0.2\t1\t119122 11476\n",
      "37\t155625\t0.2\t1\t142583 13042\n",
      "38\t103163\t0.2\t1\t94576 8587\n",
      "39\t169119\t0.2\t1\t154638 14481\n",
      "40\t156751\t0.2\t1\t143425 13326\n",
      "41\t220874\t0.2\t1\t201974 18900\n",
      "42\t316690\t0.2\t1\t289550 27140\n",
      "43\t282547\t0.2\t1\t256721 25826\n",
      "44\t296099\t0.2\t1\t272294 23805\n",
      "45\t41402\t0.2\t1\t37788 3614\n",
      "46\t58305\t0.2\t1\t53437 4868\n",
      "47\t60325\t0.2\t1\t55426 4899\n",
      "48\t82605\t0.2\t1\t76176 6429\n",
      "49\t141340\t0.2\t1\t130927 10413\n",
      "50\t43888\t0.2\t1\t39832 4056\n",
      "51\t141986\t0.2\t1\t130583 11403\n",
      "52\t111195\t0.2\t1\t102658 8537\n",
      "53\t71573\t0.2\t1\t65704 5869\n",
      "54\t82601\t0.2\t1\t76172 6429\n",
      "55\t119826\t0.2\t1\t110535 9291\n",
      "56\t83818\t0.2\t1\t77533 6285\n",
      "57\t23777\t0.2\t1\t21812 1965\n",
      "58\t64291\t0.2\t1\t59342 4949\n",
      "59\t128376\t0.2\t1\t118792 9584\n",
      "60\t23526\t0.2\t1\t21529 1997\n",
      "61\t54779\t0.2\t1\t50694 4085\n",
      "62\t76349\t0.2\t1\t70606 5743\n",
      "63\t19374\t0.2\t1\t17933 1441\n",
      "64\t44959\t0.2\t1\t41650 3309\n",
      "65\t69231\t0.2\t1\t64033 5198\n",
      "66\t29198\t0.2\t1\t26969 2229\n",
      "67\t35459\t0.2\t1\t32812 2647\n",
      "68\t40390\t0.2\t1\t37328 3062\n",
      "69\t36250\t0.2\t1\t33367 2883\n",
      "70\t35381\t0.2\t1\t32641 2740\n",
      "71\t37313\t0.2\t1\t34314 2999\n",
      "72\t51245\t0.2\t1\t46742 4503\n",
      "73\t61634\t0.2\t1\t56248 5386\n",
      "74\t66736\t0.2\t1\t60630 6106\n",
      "75\t51684\t0.2\t1\t47497 4187\n",
      "76\t5059\t0.2\t1\t4524 535\n",
      "77\t277\t0.2\t1\t246 31\n",
      "78\t1350\t0.2\t1\t1243 107\n",
      "79\t4029\t0.2\t1\t3740 289\n",
      "80\t6790\t0.2\t1\t6241 549\n",
      "81\t8328\t0.2\t1\t7646 682\n",
      "82\t8728\t0.2\t1\t8088 640\n",
      "83\t9216\t0.2\t1\t8504 712\n",
      "84\t8790\t0.2\t1\t8055 735\n",
      "85\t8358\t0.2\t1\t7749 609\n",
      "86\t8088\t0.2\t1\t7510 578\n",
      "87\t7550\t0.2\t1\t6956 594\n",
      "88\t6952\t0.2\t1\t6438 514\n",
      "89\t6446\t0.2\t1\t5915 531\n",
      "90\t5800\t0.2\t1\t5323 477\n",
      "91\t5095\t0.2\t1\t4689 406\n",
      "92\t4573\t0.2\t1\t4203 370\n",
      "93\t3931\t0.2\t1\t3607 324\n",
      "94\t3508\t0.2\t1\t3231 277\n",
      "95\t3204\t0.2\t1\t2972 232\n",
      "96\t3051\t0.2\t1\t2835 216\n",
      "97\t2868\t0.2\t1\t2663 205\n",
      "98\t2700\t0.2\t1\t2486 214\n",
      "99\t2451\t0.2\t1\t2253 198\n",
      "100\t2104\t0.2\t1\t1955 149\n",
      "101\t1847\t0.2\t1\t1716 131\n",
      "102\t1577\t0.2\t1\t1461 116\n",
      "103\t1360\t0.2\t1\t1262 98\n",
      "104\t1108\t0.2\t1\t1026 82\n",
      "105\t1002\t0.2\t1\t910 92\n",
      "106\t952\t0.2\t1\t847 105\n",
      "107\t915\t0.2\t1\t809 106\n",
      "108\t815\t0.2\t1\t745 70\n",
      "109\t775\t0.2\t1\t713 62\n",
      "110\t635\t0.2\t1\t562 73\n",
      "111\t634\t0.2\t1\t580 54\n",
      "112\t484\t0.2\t1\t430 54\n",
      "113\t487\t0.2\t1\t431 56\n",
      "114\t385\t0.2\t1\t333 52\n",
      "115\t386\t0.2\t1\t330 56\n",
      "116\t273\t0.2\t1\t241 32\n",
      "117\t293\t0.2\t1\t249 44\n",
      "118\t262\t0.2\t1\t228 34\n",
      "119\t293\t0.2\t1\t247 46\n",
      "120\t192\t0.2\t1\t164 28\n",
      "121\t203\t0.2\t1\t171 32\n",
      "122\t167\t0.2\t1\t131 36\n",
      "123\t189\t0.2\t1\t143 46\n",
      "124\t141\t0.2\t1\t115 26\n",
      "125\t168\t0.2\t1\t136 32\n",
      "126\t136\t0.2\t1\t104 32\n",
      "127\t135\t0.2\t1\t91 44\n",
      "128\t104\t0.2\t1\t72 32\n",
      "129\t107\t0.2\t1\t79 28\n",
      "130\t68\t0.2\t1\t49 19\n",
      "131\t86\t0.2\t1\t50 36\n",
      "132\t35\t0.2\t1\t25 10\n",
      "133\t54\t0.2\t1\t45 9\n",
      "134\t57\t0.2\t1\t50 7\n",
      "135\t53\t0.2\t1\t48 5\n",
      "136\t41\t0.2\t1\t30 11\n",
      "137\t44\t0.2\t1\t39 5\n",
      "138\t37\t0.2\t1\t35 2\n",
      "139\t36\t0.2\t1\t25 11\n",
      "140\t24\t0.2\t1\t20 4\n",
      "141\t67\t0.2\t1\t60 7\n",
      "142\t35\t0.2\t1\t25 10\n",
      "143\t53\t0.2\t1\t52 1\n",
      "144\t20\t0.2\t1\t19 1\n",
      "145\t137\t0.2\t1\t94 43\n",
      "146\t193\t0.2\t1\t187 6\n",
      "147\t75\t0.2\t1\t67 8\n",
      "148\t236\t0.2\t1\t233 3\n",
      "149\t82\t0.2\t1\t60 22\n",
      "150\t7629\t0.2\t1\t39 7590\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/X.1.fastq.gz\n",
      "=============================================\n",
      "16200512 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to './data_raw/X.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data_raw/X.2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Length cut-off for read 1: 35 bp (default)\n",
      "Length cut-off for read 2: 35 bb (default)\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to X.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data_raw/X.2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data_raw/X.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1056.79 s (65 us/read; 0.92 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,200,512\n",
      "Reads with adapters:                10,026,090 (61.9%)\n",
      "Reads written (passing filters):    16,200,512 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,430,076,800 bp\n",
      "Quality-trimmed:             254,186,044 bp (10.5%)\n",
      "Total written (filtered):  1,933,043,734 bp (79.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10026090 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.5%\n",
      "  C: 37.7%\n",
      "  G: 34.6%\n",
      "  T: 13.0%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2054942\t4050128.0\t0\t2054942\n",
      "2\t605485\t1012532.0\t0\t605485\n",
      "3\t284178\t253133.0\t0\t284178\n",
      "4\t122154\t63283.2\t0\t122154\n",
      "5\t130328\t15820.8\t0\t130328\n",
      "6\t75398\t3955.2\t0\t75398\n",
      "7\t80263\t988.8\t0\t80263\n",
      "8\t110540\t247.2\t0\t110540\n",
      "9\t141262\t61.8\t0\t141036 226\n",
      "10\t149072\t15.5\t1\t140714 8358\n",
      "11\t75821\t3.9\t1\t66937 8884\n",
      "12\t78951\t1.0\t1\t71626 7325\n",
      "13\t103217\t0.2\t1\t95416 7801\n",
      "14\t105402\t0.2\t1\t98148 7254\n",
      "15\t91968\t0.2\t1\t86140 5828\n",
      "16\t118052\t0.2\t1\t108383 9669\n",
      "17\t135824\t0.2\t1\t126601 9223\n",
      "18\t146032\t0.2\t1\t137027 9005\n",
      "19\t125619\t0.2\t1\t117231 8388\n",
      "20\t137116\t0.2\t1\t129773 7343\n",
      "21\t65963\t0.2\t1\t60151 5812\n",
      "22\t88190\t0.2\t1\t83812 4378\n",
      "23\t93621\t0.2\t1\t88278 5343\n",
      "24\t143342\t0.2\t1\t134631 8711\n",
      "25\t93640\t0.2\t1\t87639 6001\n",
      "26\t135939\t0.2\t1\t127166 8773\n",
      "27\t201385\t0.2\t1\t187594 13791\n",
      "28\t125254\t0.2\t1\t118441 6813\n",
      "29\t124089\t0.2\t1\t112276 11813\n",
      "30\t125940\t0.2\t1\t119072 6868\n",
      "31\t147979\t0.2\t1\t140452 7527\n",
      "32\t100831\t0.2\t1\t94686 6145\n",
      "33\t114019\t0.2\t1\t107294 6725\n",
      "34\t245927\t0.2\t1\t235091 10836\n",
      "35\t79557\t0.2\t1\t75261 4296\n",
      "36\t146836\t0.2\t1\t137891 8945\n",
      "37\t187403\t0.2\t1\t180673 6730\n",
      "38\t59765\t0.2\t1\t54214 5551\n",
      "39\t94844\t0.2\t1\t90508 4336\n",
      "40\t120525\t0.2\t1\t114581 5944\n",
      "41\t168072\t0.2\t1\t160294 7778\n",
      "42\t189174\t0.2\t1\t182482 6692\n",
      "43\t140465\t0.2\t1\t134379 6086\n",
      "44\t500974\t0.2\t1\t489301 11673\n",
      "45\t156677\t0.2\t1\t150195 6482\n",
      "46\t101257\t0.2\t1\t95632 5625\n",
      "47\t124546\t0.2\t1\t120103 4443\n",
      "48\t49610\t0.2\t1\t45822 3788\n",
      "49\t94950\t0.2\t1\t91100 3850\n",
      "50\t34704\t0.2\t1\t32825 1879\n",
      "51\t54698\t0.2\t1\t52312 2386\n",
      "52\t29202\t0.2\t1\t27668 1534\n",
      "53\t40550\t0.2\t1\t39046 1504\n",
      "54\t88272\t0.2\t1\t85576 2696\n",
      "55\t37487\t0.2\t1\t35813 1674\n",
      "56\t51536\t0.2\t1\t49751 1785\n",
      "57\t137535\t0.2\t1\t134002 3533\n",
      "58\t55654\t0.2\t1\t53486 2168\n",
      "59\t23578\t0.2\t1\t22465 1113\n",
      "60\t69862\t0.2\t1\t68025 1837\n",
      "61\t24983\t0.2\t1\t23831 1152\n",
      "62\t36243\t0.2\t1\t34573 1670\n",
      "63\t66889\t0.2\t1\t64998 1891\n",
      "64\t27095\t0.2\t1\t25961 1134\n",
      "65\t14455\t0.2\t1\t13568 887\n",
      "66\t31194\t0.2\t1\t30145 1049\n",
      "67\t8701\t0.2\t1\t8275 426\n",
      "68\t21740\t0.2\t1\t20642 1098\n",
      "69\t18678\t0.2\t1\t17912 766\n",
      "70\t43549\t0.2\t1\t42074 1475\n",
      "71\t6109\t0.2\t1\t5800 309\n",
      "72\t11897\t0.2\t1\t11402 495\n",
      "73\t14545\t0.2\t1\t13972 573\n",
      "74\t15884\t0.2\t1\t15284 600\n",
      "75\t17818\t0.2\t1\t17158 660\n",
      "76\t22260\t0.2\t1\t21454 806\n",
      "77\t38742\t0.2\t1\t37605 1137\n",
      "78\t42265\t0.2\t1\t40603 1662\n",
      "79\t14598\t0.2\t1\t13564 1034\n",
      "80\t3255\t0.2\t1\t2851 404\n",
      "81\t375\t0.2\t1\t334 41\n",
      "82\t364\t0.2\t1\t341 23\n",
      "83\t908\t0.2\t1\t881 27\n",
      "84\t1656\t0.2\t1\t1603 53\n",
      "85\t2596\t0.2\t1\t2517 79\n",
      "86\t3458\t0.2\t1\t3352 106\n",
      "87\t3535\t0.2\t1\t3430 105\n",
      "88\t2814\t0.2\t1\t2710 104\n",
      "89\t1887\t0.2\t1\t1828 59\n",
      "90\t1604\t0.2\t1\t1564 40\n",
      "91\t1348\t0.2\t1\t1307 41\n",
      "92\t1134\t0.2\t1\t1087 47\n",
      "93\t901\t0.2\t1\t868 33\n",
      "94\t816\t0.2\t1\t790 26\n",
      "95\t717\t0.2\t1\t686 31\n",
      "96\t573\t0.2\t1\t552 21\n",
      "97\t565\t0.2\t1\t540 25\n",
      "98\t495\t0.2\t1\t478 17\n",
      "99\t447\t0.2\t1\t422 25\n",
      "100\t363\t0.2\t1\t345 18\n",
      "101\t326\t0.2\t1\t315 11\n",
      "102\t231\t0.2\t1\t218 13\n",
      "103\t245\t0.2\t1\t230 15\n",
      "104\t176\t0.2\t1\t169 7\n",
      "105\t159\t0.2\t1\t144 15\n",
      "106\t155\t0.2\t1\t140 15\n",
      "107\t161\t0.2\t1\t148 13\n",
      "108\t131\t0.2\t1\t119 12\n",
      "109\t109\t0.2\t1\t102 7\n",
      "110\t110\t0.2\t1\t96 14\n",
      "111\t111\t0.2\t1\t97 14\n",
      "112\t85\t0.2\t1\t73 12\n",
      "113\t101\t0.2\t1\t89 12\n",
      "114\t76\t0.2\t1\t68 8\n",
      "115\t62\t0.2\t1\t55 7\n",
      "116\t63\t0.2\t1\t52 11\n",
      "117\t68\t0.2\t1\t54 14\n",
      "118\t59\t0.2\t1\t44 15\n",
      "119\t79\t0.2\t1\t53 26\n",
      "120\t38\t0.2\t1\t30 8\n",
      "121\t69\t0.2\t1\t53 16\n",
      "122\t53\t0.2\t1\t39 14\n",
      "123\t71\t0.2\t1\t46 25\n",
      "124\t38\t0.2\t1\t28 10\n",
      "125\t43\t0.2\t1\t28 15\n",
      "126\t56\t0.2\t1\t39 17\n",
      "127\t41\t0.2\t1\t23 18\n",
      "128\t31\t0.2\t1\t23 8\n",
      "129\t31\t0.2\t1\t20 11\n",
      "130\t15\t0.2\t1\t4 11\n",
      "131\t12\t0.2\t1\t9 3\n",
      "132\t4\t0.2\t1\t4\n",
      "133\t10\t0.2\t1\t4 6\n",
      "134\t8\t0.2\t1\t8\n",
      "135\t5\t0.2\t1\t4 1\n",
      "136\t11\t0.2\t1\t8 3\n",
      "137\t10\t0.2\t1\t7 3\n",
      "138\t9\t0.2\t1\t7 2\n",
      "139\t6\t0.2\t1\t2 4\n",
      "140\t5\t0.2\t1\t2 3\n",
      "141\t8\t0.2\t1\t7 1\n",
      "142\t3\t0.2\t1\t1 2\n",
      "143\t4\t0.2\t1\t1 3\n",
      "144\t8\t0.2\t1\t6 2\n",
      "145\t9\t0.2\t1\t7 2\n",
      "146\t25\t0.2\t1\t20 5\n",
      "147\t6\t0.2\t1\t3 3\n",
      "148\t14\t0.2\t1\t14\n",
      "149\t7\t0.2\t1\t5 2\n",
      "150\t1\t0.2\t1\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data_raw/X.2.fastq.gz\n",
      "=============================================\n",
      "16200512 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files X.1_trimmed.fq.gz and X.2_trimmed.fq.gz\n",
      "file_1: X.1_trimmed.fq.gz, file_2: X.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: X.1_trimmed.fq.gz and X.2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to X.1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to X.2_val_2.fq.gz\n",
      "\n",
      "Writing unpaired read 1 reads to X.1_unpaired_1.fq.gz\n",
      "Writing unpaired read 2 reads to X.2_unpaired_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 16200512\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1004813 (6.20%)\n",
      "\n",
      "Deleting both intermediate output files X.1_trimmed.fq.gz and X.2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data_raw/X.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore --paired ./data_raw/X.1.fastq.gz ./data_raw/X.2.fastq.gz --retain_unpaired --output_dir ./data_raw/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/andrewspanjer/RNAseq_Coho\n"
     ]
    }
   ],
   "source": [
    "cd /Users/andrewspanjer/RNAseq_Coho/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "and re-fun FastQc to see how the reads have improved..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of A.1_trimmed.fq.gz\n",
      "Approx 5% complete for A.1_trimmed.fq.gz\n",
      "Approx 10% complete for A.1_trimmed.fq.gz\n",
      "Approx 15% complete for A.1_trimmed.fq.gz\n",
      "Approx 20% complete for A.1_trimmed.fq.gz\n",
      "Approx 25% complete for A.1_trimmed.fq.gz\n",
      "Approx 30% complete for A.1_trimmed.fq.gz\n",
      "Approx 35% complete for A.1_trimmed.fq.gz\n",
      "Approx 40% complete for A.1_trimmed.fq.gz\n",
      "Approx 45% complete for A.1_trimmed.fq.gz\n",
      "Approx 50% complete for A.1_trimmed.fq.gz\n",
      "Approx 55% complete for A.1_trimmed.fq.gz\n",
      "Approx 60% complete for A.1_trimmed.fq.gz\n",
      "Approx 65% complete for A.1_trimmed.fq.gz\n",
      "Approx 70% complete for A.1_trimmed.fq.gz\n",
      "Approx 75% complete for A.1_trimmed.fq.gz\n",
      "Approx 80% complete for A.1_trimmed.fq.gz\n",
      "Approx 85% complete for A.1_trimmed.fq.gz\n",
      "Approx 90% complete for A.1_trimmed.fq.gz\n",
      "Approx 95% complete for A.1_trimmed.fq.gz\n",
      "Analysis complete for A.1_trimmed.fq.gz\n",
      "Started analysis of A.2_trimmed.fq.gz\n",
      "Approx 5% complete for A.2_trimmed.fq.gz\n",
      "Approx 10% complete for A.2_trimmed.fq.gz\n",
      "Approx 15% complete for A.2_trimmed.fq.gz\n",
      "Approx 20% complete for A.2_trimmed.fq.gz\n",
      "Approx 25% complete for A.2_trimmed.fq.gz\n",
      "Approx 30% complete for A.2_trimmed.fq.gz\n",
      "Approx 35% complete for A.2_trimmed.fq.gz\n",
      "Approx 40% complete for A.2_trimmed.fq.gz\n",
      "Approx 45% complete for A.2_trimmed.fq.gz\n",
      "Approx 50% complete for A.2_trimmed.fq.gz\n",
      "Approx 55% complete for A.2_trimmed.fq.gz\n",
      "Approx 60% complete for A.2_trimmed.fq.gz\n",
      "Approx 65% complete for A.2_trimmed.fq.gz\n",
      "Approx 70% complete for A.2_trimmed.fq.gz\n",
      "Approx 75% complete for A.2_trimmed.fq.gz\n",
      "Approx 80% complete for A.2_trimmed.fq.gz\n",
      "Approx 85% complete for A.2_trimmed.fq.gz\n",
      "Approx 90% complete for A.2_trimmed.fq.gz\n",
      "Approx 95% complete for A.2_trimmed.fq.gz\n",
      "Analysis complete for A.2_trimmed.fq.gz\n",
      "Started analysis of B.1_trimmed.fq.gz\n",
      "Approx 5% complete for B.1_trimmed.fq.gz\n",
      "Approx 10% complete for B.1_trimmed.fq.gz\n",
      "Approx 15% complete for B.1_trimmed.fq.gz\n",
      "Approx 20% complete for B.1_trimmed.fq.gz\n",
      "Approx 25% complete for B.1_trimmed.fq.gz\n",
      "Approx 30% complete for B.1_trimmed.fq.gz\n",
      "Approx 35% complete for B.1_trimmed.fq.gz\n",
      "Approx 40% complete for B.1_trimmed.fq.gz\n",
      "Approx 45% complete for B.1_trimmed.fq.gz\n",
      "Approx 50% complete for B.1_trimmed.fq.gz\n",
      "Approx 55% complete for B.1_trimmed.fq.gz\n",
      "Approx 60% complete for B.1_trimmed.fq.gz\n",
      "Approx 65% complete for B.1_trimmed.fq.gz\n",
      "Approx 70% complete for B.1_trimmed.fq.gz\n",
      "Approx 75% complete for B.1_trimmed.fq.gz\n",
      "Approx 80% complete for B.1_trimmed.fq.gz\n",
      "Approx 85% complete for B.1_trimmed.fq.gz\n",
      "Approx 90% complete for B.1_trimmed.fq.gz\n",
      "Approx 95% complete for B.1_trimmed.fq.gz\n",
      "Analysis complete for B.1_trimmed.fq.gz\n",
      "Started analysis of B.2_trimmed.fq.gz\n",
      "Approx 5% complete for B.2_trimmed.fq.gz\n",
      "Approx 10% complete for B.2_trimmed.fq.gz\n",
      "Approx 15% complete for B.2_trimmed.fq.gz\n",
      "Approx 20% complete for B.2_trimmed.fq.gz\n",
      "Approx 25% complete for B.2_trimmed.fq.gz\n",
      "Approx 30% complete for B.2_trimmed.fq.gz\n",
      "Approx 35% complete for B.2_trimmed.fq.gz\n",
      "Approx 40% complete for B.2_trimmed.fq.gz\n",
      "Approx 45% complete for B.2_trimmed.fq.gz\n",
      "Approx 50% complete for B.2_trimmed.fq.gz\n",
      "Approx 55% complete for B.2_trimmed.fq.gz\n",
      "Approx 60% complete for B.2_trimmed.fq.gz\n",
      "Approx 65% complete for B.2_trimmed.fq.gz\n",
      "Approx 70% complete for B.2_trimmed.fq.gz\n",
      "Approx 75% complete for B.2_trimmed.fq.gz\n",
      "Approx 80% complete for B.2_trimmed.fq.gz\n",
      "Approx 85% complete for B.2_trimmed.fq.gz\n",
      "Approx 90% complete for B.2_trimmed.fq.gz\n",
      "Approx 95% complete for B.2_trimmed.fq.gz\n",
      "Analysis complete for B.2_trimmed.fq.gz\n",
      "Started analysis of C.1_trimmed.fq.gz\n",
      "Approx 5% complete for C.1_trimmed.fq.gz\n",
      "Approx 10% complete for C.1_trimmed.fq.gz\n",
      "Approx 15% complete for C.1_trimmed.fq.gz\n",
      "Approx 20% complete for C.1_trimmed.fq.gz\n",
      "Approx 25% complete for C.1_trimmed.fq.gz\n",
      "Approx 30% complete for C.1_trimmed.fq.gz\n",
      "Approx 35% complete for C.1_trimmed.fq.gz\n",
      "Approx 40% complete for C.1_trimmed.fq.gz\n",
      "Approx 45% complete for C.1_trimmed.fq.gz\n",
      "Approx 50% complete for C.1_trimmed.fq.gz\n",
      "Approx 55% complete for C.1_trimmed.fq.gz\n",
      "Approx 60% complete for C.1_trimmed.fq.gz\n",
      "Approx 65% complete for C.1_trimmed.fq.gz\n",
      "Approx 70% complete for C.1_trimmed.fq.gz\n",
      "Approx 75% complete for C.1_trimmed.fq.gz\n",
      "Approx 80% complete for C.1_trimmed.fq.gz\n",
      "Approx 85% complete for C.1_trimmed.fq.gz\n",
      "Approx 90% complete for C.1_trimmed.fq.gz\n",
      "Approx 95% complete for C.1_trimmed.fq.gz\n",
      "Analysis complete for C.1_trimmed.fq.gz\n",
      "Started analysis of C.2_trimmed.fq.gz\n",
      "Approx 5% complete for C.2_trimmed.fq.gz\n",
      "Approx 10% complete for C.2_trimmed.fq.gz\n",
      "Approx 15% complete for C.2_trimmed.fq.gz\n",
      "Approx 20% complete for C.2_trimmed.fq.gz\n",
      "Approx 25% complete for C.2_trimmed.fq.gz\n",
      "Approx 30% complete for C.2_trimmed.fq.gz\n",
      "Approx 35% complete for C.2_trimmed.fq.gz\n",
      "Approx 40% complete for C.2_trimmed.fq.gz\n",
      "Approx 45% complete for C.2_trimmed.fq.gz\n",
      "Approx 50% complete for C.2_trimmed.fq.gz\n",
      "Approx 55% complete for C.2_trimmed.fq.gz\n",
      "Approx 60% complete for C.2_trimmed.fq.gz\n",
      "Approx 65% complete for C.2_trimmed.fq.gz\n",
      "Approx 70% complete for C.2_trimmed.fq.gz\n",
      "Approx 75% complete for C.2_trimmed.fq.gz\n",
      "Approx 80% complete for C.2_trimmed.fq.gz\n",
      "Approx 85% complete for C.2_trimmed.fq.gz\n",
      "Approx 90% complete for C.2_trimmed.fq.gz\n",
      "Approx 95% complete for C.2_trimmed.fq.gz\n",
      "Analysis complete for C.2_trimmed.fq.gz\n",
      "Started analysis of D.1_trimmed.fq.gz\n",
      "Approx 5% complete for D.1_trimmed.fq.gz\n",
      "Approx 10% complete for D.1_trimmed.fq.gz\n",
      "Approx 15% complete for D.1_trimmed.fq.gz\n",
      "Approx 20% complete for D.1_trimmed.fq.gz\n",
      "Approx 25% complete for D.1_trimmed.fq.gz\n",
      "Approx 30% complete for D.1_trimmed.fq.gz\n",
      "Approx 35% complete for D.1_trimmed.fq.gz\n",
      "Approx 40% complete for D.1_trimmed.fq.gz\n",
      "Approx 45% complete for D.1_trimmed.fq.gz\n",
      "Approx 50% complete for D.1_trimmed.fq.gz\n",
      "Approx 55% complete for D.1_trimmed.fq.gz\n",
      "Approx 60% complete for D.1_trimmed.fq.gz\n",
      "Approx 65% complete for D.1_trimmed.fq.gz\n",
      "Approx 70% complete for D.1_trimmed.fq.gz\n",
      "Approx 75% complete for D.1_trimmed.fq.gz\n",
      "Approx 80% complete for D.1_trimmed.fq.gz\n",
      "Approx 85% complete for D.1_trimmed.fq.gz\n",
      "Approx 90% complete for D.1_trimmed.fq.gz\n",
      "Approx 95% complete for D.1_trimmed.fq.gz\n",
      "Analysis complete for D.1_trimmed.fq.gz\n",
      "Started analysis of D.2_trimmed.fq.gz\n",
      "Approx 5% complete for D.2_trimmed.fq.gz\n",
      "Approx 10% complete for D.2_trimmed.fq.gz\n",
      "Approx 15% complete for D.2_trimmed.fq.gz\n",
      "Approx 20% complete for D.2_trimmed.fq.gz\n",
      "Approx 25% complete for D.2_trimmed.fq.gz\n",
      "Approx 30% complete for D.2_trimmed.fq.gz\n",
      "Approx 35% complete for D.2_trimmed.fq.gz\n",
      "Approx 40% complete for D.2_trimmed.fq.gz\n",
      "Approx 45% complete for D.2_trimmed.fq.gz\n",
      "Approx 50% complete for D.2_trimmed.fq.gz\n",
      "Approx 55% complete for D.2_trimmed.fq.gz\n",
      "Approx 60% complete for D.2_trimmed.fq.gz\n",
      "Approx 65% complete for D.2_trimmed.fq.gz\n",
      "Approx 70% complete for D.2_trimmed.fq.gz\n",
      "Approx 75% complete for D.2_trimmed.fq.gz\n",
      "Approx 80% complete for D.2_trimmed.fq.gz\n",
      "Approx 85% complete for D.2_trimmed.fq.gz\n",
      "Approx 90% complete for D.2_trimmed.fq.gz\n",
      "Approx 95% complete for D.2_trimmed.fq.gz\n",
      "Analysis complete for D.2_trimmed.fq.gz\n",
      "Started analysis of E.1_trimmed.fq.gz\n",
      "Approx 5% complete for E.1_trimmed.fq.gz\n",
      "Approx 10% complete for E.1_trimmed.fq.gz\n",
      "Approx 15% complete for E.1_trimmed.fq.gz\n",
      "Approx 20% complete for E.1_trimmed.fq.gz\n",
      "Approx 25% complete for E.1_trimmed.fq.gz\n",
      "Approx 30% complete for E.1_trimmed.fq.gz\n",
      "Approx 35% complete for E.1_trimmed.fq.gz\n",
      "Approx 40% complete for E.1_trimmed.fq.gz\n",
      "Approx 45% complete for E.1_trimmed.fq.gz\n",
      "Approx 50% complete for E.1_trimmed.fq.gz\n",
      "Approx 55% complete for E.1_trimmed.fq.gz\n",
      "Approx 60% complete for E.1_trimmed.fq.gz\n",
      "Approx 65% complete for E.1_trimmed.fq.gz\n",
      "Approx 70% complete for E.1_trimmed.fq.gz\n",
      "Approx 75% complete for E.1_trimmed.fq.gz\n",
      "Approx 80% complete for E.1_trimmed.fq.gz\n",
      "Approx 85% complete for E.1_trimmed.fq.gz\n",
      "Approx 90% complete for E.1_trimmed.fq.gz\n",
      "Approx 95% complete for E.1_trimmed.fq.gz\n",
      "Analysis complete for E.1_trimmed.fq.gz\n",
      "Started analysis of E.2_trimmed.fq.gz\n",
      "Approx 5% complete for E.2_trimmed.fq.gz\n",
      "Approx 10% complete for E.2_trimmed.fq.gz\n",
      "Approx 15% complete for E.2_trimmed.fq.gz\n",
      "Approx 20% complete for E.2_trimmed.fq.gz\n",
      "Approx 25% complete for E.2_trimmed.fq.gz\n",
      "Approx 30% complete for E.2_trimmed.fq.gz\n",
      "Approx 35% complete for E.2_trimmed.fq.gz\n",
      "Approx 40% complete for E.2_trimmed.fq.gz\n",
      "Approx 45% complete for E.2_trimmed.fq.gz\n",
      "Approx 50% complete for E.2_trimmed.fq.gz\n",
      "Approx 55% complete for E.2_trimmed.fq.gz\n",
      "Approx 60% complete for E.2_trimmed.fq.gz\n",
      "Approx 65% complete for E.2_trimmed.fq.gz\n",
      "Approx 70% complete for E.2_trimmed.fq.gz\n",
      "Approx 75% complete for E.2_trimmed.fq.gz\n",
      "Approx 80% complete for E.2_trimmed.fq.gz\n",
      "Approx 85% complete for E.2_trimmed.fq.gz\n",
      "Approx 90% complete for E.2_trimmed.fq.gz\n",
      "Approx 95% complete for E.2_trimmed.fq.gz\n",
      "Analysis complete for E.2_trimmed.fq.gz\n",
      "Started analysis of F.1_trimmed.fq.gz\n",
      "Approx 5% complete for F.1_trimmed.fq.gz\n",
      "Approx 10% complete for F.1_trimmed.fq.gz\n",
      "Approx 15% complete for F.1_trimmed.fq.gz\n",
      "Approx 20% complete for F.1_trimmed.fq.gz\n",
      "Approx 25% complete for F.1_trimmed.fq.gz\n",
      "Approx 30% complete for F.1_trimmed.fq.gz\n",
      "Approx 35% complete for F.1_trimmed.fq.gz\n",
      "Approx 40% complete for F.1_trimmed.fq.gz\n",
      "Approx 45% complete for F.1_trimmed.fq.gz\n",
      "Approx 50% complete for F.1_trimmed.fq.gz\n",
      "Approx 55% complete for F.1_trimmed.fq.gz\n",
      "Approx 60% complete for F.1_trimmed.fq.gz\n",
      "Approx 65% complete for F.1_trimmed.fq.gz\n",
      "Approx 70% complete for F.1_trimmed.fq.gz\n",
      "Approx 75% complete for F.1_trimmed.fq.gz\n",
      "Approx 80% complete for F.1_trimmed.fq.gz\n",
      "Approx 85% complete for F.1_trimmed.fq.gz\n",
      "Approx 90% complete for F.1_trimmed.fq.gz\n",
      "Approx 95% complete for F.1_trimmed.fq.gz\n",
      "Analysis complete for F.1_trimmed.fq.gz\n",
      "Started analysis of F.2_trimmed.fq.gz\n",
      "Approx 5% complete for F.2_trimmed.fq.gz\n",
      "Approx 10% complete for F.2_trimmed.fq.gz\n",
      "Approx 15% complete for F.2_trimmed.fq.gz\n",
      "Approx 20% complete for F.2_trimmed.fq.gz\n",
      "Approx 25% complete for F.2_trimmed.fq.gz\n",
      "Approx 30% complete for F.2_trimmed.fq.gz\n",
      "Approx 35% complete for F.2_trimmed.fq.gz\n",
      "Approx 40% complete for F.2_trimmed.fq.gz\n",
      "Approx 45% complete for F.2_trimmed.fq.gz\n",
      "Approx 50% complete for F.2_trimmed.fq.gz\n",
      "Approx 55% complete for F.2_trimmed.fq.gz\n",
      "Approx 60% complete for F.2_trimmed.fq.gz\n",
      "Approx 65% complete for F.2_trimmed.fq.gz\n",
      "Approx 70% complete for F.2_trimmed.fq.gz\n",
      "Approx 75% complete for F.2_trimmed.fq.gz\n",
      "Approx 80% complete for F.2_trimmed.fq.gz\n",
      "Approx 85% complete for F.2_trimmed.fq.gz\n",
      "Approx 90% complete for F.2_trimmed.fq.gz\n",
      "Approx 95% complete for F.2_trimmed.fq.gz\n",
      "Analysis complete for F.2_trimmed.fq.gz\n",
      "Started analysis of G.1_trimmed.fq.gz\n",
      "Approx 5% complete for G.1_trimmed.fq.gz\n",
      "Approx 10% complete for G.1_trimmed.fq.gz\n",
      "Approx 15% complete for G.1_trimmed.fq.gz\n",
      "Approx 20% complete for G.1_trimmed.fq.gz\n",
      "Approx 25% complete for G.1_trimmed.fq.gz\n",
      "Approx 30% complete for G.1_trimmed.fq.gz\n",
      "Approx 35% complete for G.1_trimmed.fq.gz\n",
      "Approx 40% complete for G.1_trimmed.fq.gz\n",
      "Approx 45% complete for G.1_trimmed.fq.gz\n",
      "Approx 50% complete for G.1_trimmed.fq.gz\n",
      "Approx 55% complete for G.1_trimmed.fq.gz\n",
      "Approx 60% complete for G.1_trimmed.fq.gz\n",
      "Approx 65% complete for G.1_trimmed.fq.gz\n",
      "Approx 70% complete for G.1_trimmed.fq.gz\n",
      "Approx 75% complete for G.1_trimmed.fq.gz\n",
      "Approx 80% complete for G.1_trimmed.fq.gz\n",
      "Approx 85% complete for G.1_trimmed.fq.gz\n",
      "Approx 90% complete for G.1_trimmed.fq.gz\n",
      "Approx 95% complete for G.1_trimmed.fq.gz\n",
      "Analysis complete for G.1_trimmed.fq.gz\n",
      "Started analysis of G.2_trimmed.fq.gz\n",
      "Approx 5% complete for G.2_trimmed.fq.gz\n",
      "Approx 10% complete for G.2_trimmed.fq.gz\n",
      "Approx 15% complete for G.2_trimmed.fq.gz\n",
      "Approx 20% complete for G.2_trimmed.fq.gz\n",
      "Approx 25% complete for G.2_trimmed.fq.gz\n",
      "Approx 30% complete for G.2_trimmed.fq.gz\n",
      "Approx 35% complete for G.2_trimmed.fq.gz\n",
      "Approx 40% complete for G.2_trimmed.fq.gz\n",
      "Approx 45% complete for G.2_trimmed.fq.gz\n",
      "Approx 50% complete for G.2_trimmed.fq.gz\n",
      "Approx 55% complete for G.2_trimmed.fq.gz\n",
      "Approx 60% complete for G.2_trimmed.fq.gz\n",
      "Approx 65% complete for G.2_trimmed.fq.gz\n",
      "Approx 70% complete for G.2_trimmed.fq.gz\n",
      "Approx 75% complete for G.2_trimmed.fq.gz\n",
      "Approx 80% complete for G.2_trimmed.fq.gz\n",
      "Approx 85% complete for G.2_trimmed.fq.gz\n",
      "Approx 90% complete for G.2_trimmed.fq.gz\n",
      "Approx 95% complete for G.2_trimmed.fq.gz\n",
      "Analysis complete for G.2_trimmed.fq.gz\n",
      "Started analysis of H.1_trimmed.fq.gz\n",
      "Approx 5% complete for H.1_trimmed.fq.gz\n",
      "Approx 10% complete for H.1_trimmed.fq.gz\n",
      "Approx 15% complete for H.1_trimmed.fq.gz\n",
      "Approx 20% complete for H.1_trimmed.fq.gz\n",
      "Approx 25% complete for H.1_trimmed.fq.gz\n",
      "Approx 30% complete for H.1_trimmed.fq.gz\n",
      "Approx 35% complete for H.1_trimmed.fq.gz\n",
      "Approx 40% complete for H.1_trimmed.fq.gz\n",
      "Approx 45% complete for H.1_trimmed.fq.gz\n",
      "Approx 50% complete for H.1_trimmed.fq.gz\n",
      "Approx 55% complete for H.1_trimmed.fq.gz\n",
      "Approx 60% complete for H.1_trimmed.fq.gz\n",
      "Approx 65% complete for H.1_trimmed.fq.gz\n",
      "Approx 70% complete for H.1_trimmed.fq.gz\n",
      "Approx 75% complete for H.1_trimmed.fq.gz\n",
      "Approx 80% complete for H.1_trimmed.fq.gz\n",
      "Approx 85% complete for H.1_trimmed.fq.gz\n",
      "Approx 90% complete for H.1_trimmed.fq.gz\n",
      "Approx 95% complete for H.1_trimmed.fq.gz\n",
      "Analysis complete for H.1_trimmed.fq.gz\n",
      "Started analysis of H.2_trimmed.fq.gz\n",
      "Approx 5% complete for H.2_trimmed.fq.gz\n",
      "Approx 10% complete for H.2_trimmed.fq.gz\n",
      "Approx 15% complete for H.2_trimmed.fq.gz\n",
      "Approx 20% complete for H.2_trimmed.fq.gz\n",
      "Approx 25% complete for H.2_trimmed.fq.gz\n",
      "Approx 30% complete for H.2_trimmed.fq.gz\n",
      "Approx 35% complete for H.2_trimmed.fq.gz\n",
      "Approx 40% complete for H.2_trimmed.fq.gz\n",
      "Approx 45% complete for H.2_trimmed.fq.gz\n",
      "Approx 50% complete for H.2_trimmed.fq.gz\n",
      "Approx 55% complete for H.2_trimmed.fq.gz\n",
      "Approx 60% complete for H.2_trimmed.fq.gz\n",
      "Approx 65% complete for H.2_trimmed.fq.gz\n",
      "Approx 70% complete for H.2_trimmed.fq.gz\n",
      "Approx 75% complete for H.2_trimmed.fq.gz\n",
      "Approx 80% complete for H.2_trimmed.fq.gz\n",
      "Approx 85% complete for H.2_trimmed.fq.gz\n",
      "Approx 90% complete for H.2_trimmed.fq.gz\n",
      "Approx 95% complete for H.2_trimmed.fq.gz\n",
      "Analysis complete for H.2_trimmed.fq.gz\n",
      "Started analysis of I.1_trimmed.fq.gz\n",
      "Approx 5% complete for I.1_trimmed.fq.gz\n",
      "Approx 10% complete for I.1_trimmed.fq.gz\n",
      "Approx 15% complete for I.1_trimmed.fq.gz\n",
      "Approx 20% complete for I.1_trimmed.fq.gz\n",
      "Approx 25% complete for I.1_trimmed.fq.gz\n",
      "Approx 30% complete for I.1_trimmed.fq.gz\n",
      "Approx 35% complete for I.1_trimmed.fq.gz\n",
      "Approx 40% complete for I.1_trimmed.fq.gz\n",
      "Approx 45% complete for I.1_trimmed.fq.gz\n",
      "Approx 50% complete for I.1_trimmed.fq.gz\n",
      "Approx 55% complete for I.1_trimmed.fq.gz\n",
      "Approx 60% complete for I.1_trimmed.fq.gz\n",
      "Approx 65% complete for I.1_trimmed.fq.gz\n",
      "Approx 70% complete for I.1_trimmed.fq.gz\n",
      "Approx 75% complete for I.1_trimmed.fq.gz\n",
      "Approx 80% complete for I.1_trimmed.fq.gz\n",
      "Approx 85% complete for I.1_trimmed.fq.gz\n",
      "Approx 90% complete for I.1_trimmed.fq.gz\n",
      "Approx 95% complete for I.1_trimmed.fq.gz\n",
      "Analysis complete for I.1_trimmed.fq.gz\n",
      "Started analysis of I.2_trimmed.fq.gz\n",
      "Approx 5% complete for I.2_trimmed.fq.gz\n",
      "Approx 10% complete for I.2_trimmed.fq.gz\n",
      "Approx 15% complete for I.2_trimmed.fq.gz\n",
      "Approx 20% complete for I.2_trimmed.fq.gz\n",
      "Approx 25% complete for I.2_trimmed.fq.gz\n",
      "Approx 30% complete for I.2_trimmed.fq.gz\n",
      "Approx 35% complete for I.2_trimmed.fq.gz\n",
      "Approx 40% complete for I.2_trimmed.fq.gz\n",
      "Approx 45% complete for I.2_trimmed.fq.gz\n",
      "Approx 50% complete for I.2_trimmed.fq.gz\n",
      "Approx 55% complete for I.2_trimmed.fq.gz\n",
      "Approx 60% complete for I.2_trimmed.fq.gz\n",
      "Approx 65% complete for I.2_trimmed.fq.gz\n",
      "Approx 70% complete for I.2_trimmed.fq.gz\n",
      "Approx 75% complete for I.2_trimmed.fq.gz\n",
      "Approx 80% complete for I.2_trimmed.fq.gz\n",
      "Approx 85% complete for I.2_trimmed.fq.gz\n",
      "Approx 90% complete for I.2_trimmed.fq.gz\n",
      "Approx 95% complete for I.2_trimmed.fq.gz\n",
      "Analysis complete for I.2_trimmed.fq.gz\n",
      "Started analysis of J.1_trimmed.fq.gz\n",
      "Approx 5% complete for J.1_trimmed.fq.gz\n",
      "Approx 10% complete for J.1_trimmed.fq.gz\n",
      "Approx 15% complete for J.1_trimmed.fq.gz\n",
      "Approx 20% complete for J.1_trimmed.fq.gz\n",
      "Approx 25% complete for J.1_trimmed.fq.gz\n",
      "Approx 30% complete for J.1_trimmed.fq.gz\n",
      "Approx 35% complete for J.1_trimmed.fq.gz\n",
      "Approx 40% complete for J.1_trimmed.fq.gz\n",
      "Approx 45% complete for J.1_trimmed.fq.gz\n",
      "Approx 50% complete for J.1_trimmed.fq.gz\n",
      "Approx 55% complete for J.1_trimmed.fq.gz\n",
      "Approx 60% complete for J.1_trimmed.fq.gz\n",
      "Approx 65% complete for J.1_trimmed.fq.gz\n",
      "Approx 70% complete for J.1_trimmed.fq.gz\n",
      "Approx 75% complete for J.1_trimmed.fq.gz\n",
      "Approx 80% complete for J.1_trimmed.fq.gz\n",
      "Approx 85% complete for J.1_trimmed.fq.gz\n",
      "Approx 90% complete for J.1_trimmed.fq.gz\n",
      "Approx 95% complete for J.1_trimmed.fq.gz\n",
      "Analysis complete for J.1_trimmed.fq.gz\n",
      "Started analysis of J.2_trimmed.fq.gz\n",
      "Approx 5% complete for J.2_trimmed.fq.gz\n",
      "Approx 10% complete for J.2_trimmed.fq.gz\n",
      "Approx 15% complete for J.2_trimmed.fq.gz\n",
      "Approx 20% complete for J.2_trimmed.fq.gz\n",
      "Approx 25% complete for J.2_trimmed.fq.gz\n",
      "Approx 30% complete for J.2_trimmed.fq.gz\n",
      "Approx 35% complete for J.2_trimmed.fq.gz\n",
      "Approx 40% complete for J.2_trimmed.fq.gz\n",
      "Approx 45% complete for J.2_trimmed.fq.gz\n",
      "Approx 50% complete for J.2_trimmed.fq.gz\n",
      "Approx 55% complete for J.2_trimmed.fq.gz\n",
      "Approx 60% complete for J.2_trimmed.fq.gz\n",
      "Approx 65% complete for J.2_trimmed.fq.gz\n",
      "Approx 70% complete for J.2_trimmed.fq.gz\n",
      "Approx 75% complete for J.2_trimmed.fq.gz\n",
      "Approx 80% complete for J.2_trimmed.fq.gz\n",
      "Approx 85% complete for J.2_trimmed.fq.gz\n",
      "Approx 90% complete for J.2_trimmed.fq.gz\n",
      "Approx 95% complete for J.2_trimmed.fq.gz\n",
      "Analysis complete for J.2_trimmed.fq.gz\n",
      "Started analysis of K.1_trimmed.fq.gz\n",
      "Approx 5% complete for K.1_trimmed.fq.gz\n",
      "Approx 10% complete for K.1_trimmed.fq.gz\n",
      "Approx 15% complete for K.1_trimmed.fq.gz\n",
      "Approx 20% complete for K.1_trimmed.fq.gz\n",
      "Approx 25% complete for K.1_trimmed.fq.gz\n",
      "Approx 30% complete for K.1_trimmed.fq.gz\n",
      "Approx 35% complete for K.1_trimmed.fq.gz\n",
      "Approx 40% complete for K.1_trimmed.fq.gz\n",
      "Approx 45% complete for K.1_trimmed.fq.gz\n",
      "Approx 50% complete for K.1_trimmed.fq.gz\n",
      "Approx 55% complete for K.1_trimmed.fq.gz\n",
      "Approx 60% complete for K.1_trimmed.fq.gz\n",
      "Approx 65% complete for K.1_trimmed.fq.gz\n",
      "Approx 70% complete for K.1_trimmed.fq.gz\n",
      "Approx 75% complete for K.1_trimmed.fq.gz\n",
      "Approx 80% complete for K.1_trimmed.fq.gz\n",
      "Approx 85% complete for K.1_trimmed.fq.gz\n",
      "Approx 90% complete for K.1_trimmed.fq.gz\n",
      "Approx 95% complete for K.1_trimmed.fq.gz\n",
      "Analysis complete for K.1_trimmed.fq.gz\n",
      "Started analysis of K.2_trimmed.fq.gz\n",
      "Approx 5% complete for K.2_trimmed.fq.gz\n",
      "Approx 10% complete for K.2_trimmed.fq.gz\n",
      "Approx 15% complete for K.2_trimmed.fq.gz\n",
      "Approx 20% complete for K.2_trimmed.fq.gz\n",
      "Approx 25% complete for K.2_trimmed.fq.gz\n",
      "Approx 30% complete for K.2_trimmed.fq.gz\n",
      "Approx 35% complete for K.2_trimmed.fq.gz\n",
      "Approx 40% complete for K.2_trimmed.fq.gz\n",
      "Approx 45% complete for K.2_trimmed.fq.gz\n",
      "Approx 50% complete for K.2_trimmed.fq.gz\n",
      "Approx 55% complete for K.2_trimmed.fq.gz\n",
      "Approx 60% complete for K.2_trimmed.fq.gz\n",
      "Approx 65% complete for K.2_trimmed.fq.gz\n",
      "Approx 70% complete for K.2_trimmed.fq.gz\n",
      "Approx 75% complete for K.2_trimmed.fq.gz\n",
      "Approx 80% complete for K.2_trimmed.fq.gz\n",
      "Approx 85% complete for K.2_trimmed.fq.gz\n",
      "Approx 90% complete for K.2_trimmed.fq.gz\n",
      "Approx 95% complete for K.2_trimmed.fq.gz\n",
      "Analysis complete for K.2_trimmed.fq.gz\n",
      "Started analysis of L.1_trimmed.fq.gz\n",
      "Approx 5% complete for L.1_trimmed.fq.gz\n",
      "Approx 10% complete for L.1_trimmed.fq.gz\n",
      "Approx 15% complete for L.1_trimmed.fq.gz\n",
      "Approx 20% complete for L.1_trimmed.fq.gz\n",
      "Approx 25% complete for L.1_trimmed.fq.gz\n",
      "Approx 30% complete for L.1_trimmed.fq.gz\n",
      "Approx 35% complete for L.1_trimmed.fq.gz\n",
      "Approx 40% complete for L.1_trimmed.fq.gz\n",
      "Approx 45% complete for L.1_trimmed.fq.gz\n",
      "Approx 50% complete for L.1_trimmed.fq.gz\n",
      "Approx 55% complete for L.1_trimmed.fq.gz\n",
      "Approx 60% complete for L.1_trimmed.fq.gz\n",
      "Approx 65% complete for L.1_trimmed.fq.gz\n",
      "Approx 70% complete for L.1_trimmed.fq.gz\n",
      "Approx 75% complete for L.1_trimmed.fq.gz\n",
      "Approx 80% complete for L.1_trimmed.fq.gz\n",
      "Approx 85% complete for L.1_trimmed.fq.gz\n",
      "Approx 90% complete for L.1_trimmed.fq.gz\n",
      "Approx 95% complete for L.1_trimmed.fq.gz\n",
      "Analysis complete for L.1_trimmed.fq.gz\n",
      "Started analysis of L.2_trimmed.fq.gz\n",
      "Approx 5% complete for L.2_trimmed.fq.gz\n",
      "Approx 10% complete for L.2_trimmed.fq.gz\n",
      "Approx 15% complete for L.2_trimmed.fq.gz\n",
      "Approx 20% complete for L.2_trimmed.fq.gz\n",
      "Approx 25% complete for L.2_trimmed.fq.gz\n",
      "Approx 30% complete for L.2_trimmed.fq.gz\n",
      "Approx 35% complete for L.2_trimmed.fq.gz\n",
      "Approx 40% complete for L.2_trimmed.fq.gz\n",
      "Approx 45% complete for L.2_trimmed.fq.gz\n",
      "Approx 50% complete for L.2_trimmed.fq.gz\n",
      "Approx 55% complete for L.2_trimmed.fq.gz\n",
      "Approx 60% complete for L.2_trimmed.fq.gz\n",
      "Approx 65% complete for L.2_trimmed.fq.gz\n",
      "Approx 70% complete for L.2_trimmed.fq.gz\n",
      "Approx 75% complete for L.2_trimmed.fq.gz\n",
      "Approx 80% complete for L.2_trimmed.fq.gz\n",
      "Approx 85% complete for L.2_trimmed.fq.gz\n",
      "Approx 90% complete for L.2_trimmed.fq.gz\n",
      "Approx 95% complete for L.2_trimmed.fq.gz\n",
      "Analysis complete for L.2_trimmed.fq.gz\n",
      "Started analysis of M.1_trimmed.fq.gz\n",
      "Approx 5% complete for M.1_trimmed.fq.gz\n",
      "Approx 10% complete for M.1_trimmed.fq.gz\n",
      "Approx 15% complete for M.1_trimmed.fq.gz\n",
      "Approx 20% complete for M.1_trimmed.fq.gz\n",
      "Approx 25% complete for M.1_trimmed.fq.gz\n",
      "Approx 30% complete for M.1_trimmed.fq.gz\n",
      "Approx 35% complete for M.1_trimmed.fq.gz\n",
      "Approx 40% complete for M.1_trimmed.fq.gz\n",
      "Approx 45% complete for M.1_trimmed.fq.gz\n",
      "Approx 50% complete for M.1_trimmed.fq.gz\n",
      "Approx 55% complete for M.1_trimmed.fq.gz\n",
      "Approx 60% complete for M.1_trimmed.fq.gz\n",
      "Approx 65% complete for M.1_trimmed.fq.gz\n",
      "Approx 70% complete for M.1_trimmed.fq.gz\n",
      "Approx 75% complete for M.1_trimmed.fq.gz\n",
      "Approx 80% complete for M.1_trimmed.fq.gz\n",
      "Approx 85% complete for M.1_trimmed.fq.gz\n",
      "Approx 90% complete for M.1_trimmed.fq.gz\n",
      "Approx 95% complete for M.1_trimmed.fq.gz\n",
      "Analysis complete for M.1_trimmed.fq.gz\n",
      "Started analysis of M.2_trimmed.fq.gz\n",
      "Approx 5% complete for M.2_trimmed.fq.gz\n",
      "Approx 10% complete for M.2_trimmed.fq.gz\n",
      "Approx 15% complete for M.2_trimmed.fq.gz\n",
      "Approx 20% complete for M.2_trimmed.fq.gz\n",
      "Approx 25% complete for M.2_trimmed.fq.gz\n",
      "Approx 30% complete for M.2_trimmed.fq.gz\n",
      "Approx 35% complete for M.2_trimmed.fq.gz\n",
      "Approx 40% complete for M.2_trimmed.fq.gz\n",
      "Approx 45% complete for M.2_trimmed.fq.gz\n",
      "Approx 50% complete for M.2_trimmed.fq.gz\n",
      "Approx 55% complete for M.2_trimmed.fq.gz\n",
      "Approx 60% complete for M.2_trimmed.fq.gz\n",
      "Approx 65% complete for M.2_trimmed.fq.gz\n",
      "Approx 70% complete for M.2_trimmed.fq.gz\n",
      "Approx 75% complete for M.2_trimmed.fq.gz\n",
      "Approx 80% complete for M.2_trimmed.fq.gz\n",
      "Approx 85% complete for M.2_trimmed.fq.gz\n",
      "Approx 90% complete for M.2_trimmed.fq.gz\n",
      "Approx 95% complete for M.2_trimmed.fq.gz\n",
      "Analysis complete for M.2_trimmed.fq.gz\n",
      "Started analysis of N.1_trimmed.fq.gz\n",
      "Approx 5% complete for N.1_trimmed.fq.gz\n",
      "Approx 10% complete for N.1_trimmed.fq.gz\n",
      "Approx 15% complete for N.1_trimmed.fq.gz\n",
      "Approx 20% complete for N.1_trimmed.fq.gz\n",
      "Approx 25% complete for N.1_trimmed.fq.gz\n",
      "Approx 30% complete for N.1_trimmed.fq.gz\n",
      "Approx 35% complete for N.1_trimmed.fq.gz\n",
      "Approx 40% complete for N.1_trimmed.fq.gz\n",
      "Approx 45% complete for N.1_trimmed.fq.gz\n",
      "Approx 50% complete for N.1_trimmed.fq.gz\n",
      "Approx 55% complete for N.1_trimmed.fq.gz\n",
      "Approx 60% complete for N.1_trimmed.fq.gz\n",
      "Approx 65% complete for N.1_trimmed.fq.gz\n",
      "Approx 70% complete for N.1_trimmed.fq.gz\n",
      "Approx 75% complete for N.1_trimmed.fq.gz\n",
      "Approx 80% complete for N.1_trimmed.fq.gz\n",
      "Approx 85% complete for N.1_trimmed.fq.gz\n",
      "Approx 90% complete for N.1_trimmed.fq.gz\n",
      "Approx 95% complete for N.1_trimmed.fq.gz\n",
      "Analysis complete for N.1_trimmed.fq.gz\n",
      "Started analysis of N.2_trimmed.fq.gz\n",
      "Approx 5% complete for N.2_trimmed.fq.gz\n",
      "Approx 10% complete for N.2_trimmed.fq.gz\n",
      "Approx 15% complete for N.2_trimmed.fq.gz\n",
      "Approx 20% complete for N.2_trimmed.fq.gz\n",
      "Approx 25% complete for N.2_trimmed.fq.gz\n",
      "Approx 30% complete for N.2_trimmed.fq.gz\n",
      "Approx 35% complete for N.2_trimmed.fq.gz\n",
      "Approx 40% complete for N.2_trimmed.fq.gz\n",
      "Approx 45% complete for N.2_trimmed.fq.gz\n",
      "Approx 50% complete for N.2_trimmed.fq.gz\n",
      "Approx 55% complete for N.2_trimmed.fq.gz\n",
      "Approx 60% complete for N.2_trimmed.fq.gz\n",
      "Approx 65% complete for N.2_trimmed.fq.gz\n",
      "Approx 70% complete for N.2_trimmed.fq.gz\n",
      "Approx 75% complete for N.2_trimmed.fq.gz\n",
      "Approx 80% complete for N.2_trimmed.fq.gz\n",
      "Approx 85% complete for N.2_trimmed.fq.gz\n",
      "Approx 90% complete for N.2_trimmed.fq.gz\n",
      "Approx 95% complete for N.2_trimmed.fq.gz\n",
      "Analysis complete for N.2_trimmed.fq.gz\n",
      "Started analysis of O.1_trimmed.fq.gz\n",
      "Approx 5% complete for O.1_trimmed.fq.gz\n",
      "Approx 10% complete for O.1_trimmed.fq.gz\n",
      "Approx 15% complete for O.1_trimmed.fq.gz\n",
      "Approx 20% complete for O.1_trimmed.fq.gz\n",
      "Approx 25% complete for O.1_trimmed.fq.gz\n",
      "Approx 30% complete for O.1_trimmed.fq.gz\n",
      "Approx 35% complete for O.1_trimmed.fq.gz\n",
      "Approx 40% complete for O.1_trimmed.fq.gz\n",
      "Approx 45% complete for O.1_trimmed.fq.gz\n",
      "Approx 50% complete for O.1_trimmed.fq.gz\n",
      "Approx 55% complete for O.1_trimmed.fq.gz\n",
      "Approx 60% complete for O.1_trimmed.fq.gz\n",
      "Approx 65% complete for O.1_trimmed.fq.gz\n",
      "Approx 70% complete for O.1_trimmed.fq.gz\n",
      "Approx 75% complete for O.1_trimmed.fq.gz\n",
      "Approx 80% complete for O.1_trimmed.fq.gz\n",
      "Approx 85% complete for O.1_trimmed.fq.gz\n",
      "Approx 90% complete for O.1_trimmed.fq.gz\n",
      "Approx 95% complete for O.1_trimmed.fq.gz\n",
      "Analysis complete for O.1_trimmed.fq.gz\n",
      "Started analysis of O.2_trimmed.fq.gz\n",
      "Approx 5% complete for O.2_trimmed.fq.gz\n",
      "Approx 10% complete for O.2_trimmed.fq.gz\n",
      "Approx 15% complete for O.2_trimmed.fq.gz\n",
      "Approx 20% complete for O.2_trimmed.fq.gz\n",
      "Approx 25% complete for O.2_trimmed.fq.gz\n",
      "Approx 30% complete for O.2_trimmed.fq.gz\n",
      "Approx 35% complete for O.2_trimmed.fq.gz\n",
      "Approx 40% complete for O.2_trimmed.fq.gz\n",
      "Approx 45% complete for O.2_trimmed.fq.gz\n",
      "Approx 50% complete for O.2_trimmed.fq.gz\n",
      "Approx 55% complete for O.2_trimmed.fq.gz\n",
      "Approx 60% complete for O.2_trimmed.fq.gz\n",
      "Approx 65% complete for O.2_trimmed.fq.gz\n",
      "Approx 70% complete for O.2_trimmed.fq.gz\n",
      "Approx 75% complete for O.2_trimmed.fq.gz\n",
      "Approx 80% complete for O.2_trimmed.fq.gz\n",
      "Approx 85% complete for O.2_trimmed.fq.gz\n",
      "Approx 90% complete for O.2_trimmed.fq.gz\n",
      "Approx 95% complete for O.2_trimmed.fq.gz\n",
      "Analysis complete for O.2_trimmed.fq.gz\n",
      "Started analysis of P.1_trimmed.fq.gz\n",
      "Approx 5% complete for P.1_trimmed.fq.gz\n",
      "Approx 10% complete for P.1_trimmed.fq.gz\n",
      "Approx 15% complete for P.1_trimmed.fq.gz\n",
      "Approx 20% complete for P.1_trimmed.fq.gz\n",
      "Approx 25% complete for P.1_trimmed.fq.gz\n",
      "Approx 30% complete for P.1_trimmed.fq.gz\n",
      "Approx 35% complete for P.1_trimmed.fq.gz\n",
      "Approx 40% complete for P.1_trimmed.fq.gz\n",
      "Approx 45% complete for P.1_trimmed.fq.gz\n",
      "Approx 50% complete for P.1_trimmed.fq.gz\n",
      "Approx 55% complete for P.1_trimmed.fq.gz\n",
      "Approx 60% complete for P.1_trimmed.fq.gz\n",
      "Approx 65% complete for P.1_trimmed.fq.gz\n",
      "Approx 70% complete for P.1_trimmed.fq.gz\n",
      "Approx 75% complete for P.1_trimmed.fq.gz\n",
      "Approx 80% complete for P.1_trimmed.fq.gz\n",
      "Approx 85% complete for P.1_trimmed.fq.gz\n",
      "Approx 90% complete for P.1_trimmed.fq.gz\n",
      "Approx 95% complete for P.1_trimmed.fq.gz\n",
      "Analysis complete for P.1_trimmed.fq.gz\n",
      "Started analysis of P.2_trimmed.fq.gz\n",
      "Approx 5% complete for P.2_trimmed.fq.gz\n",
      "Approx 10% complete for P.2_trimmed.fq.gz\n",
      "Approx 15% complete for P.2_trimmed.fq.gz\n",
      "Approx 20% complete for P.2_trimmed.fq.gz\n",
      "Approx 25% complete for P.2_trimmed.fq.gz\n",
      "Approx 30% complete for P.2_trimmed.fq.gz\n",
      "Approx 35% complete for P.2_trimmed.fq.gz\n",
      "Approx 40% complete for P.2_trimmed.fq.gz\n",
      "Approx 45% complete for P.2_trimmed.fq.gz\n",
      "Approx 50% complete for P.2_trimmed.fq.gz\n",
      "Approx 55% complete for P.2_trimmed.fq.gz\n",
      "Approx 60% complete for P.2_trimmed.fq.gz\n",
      "Approx 65% complete for P.2_trimmed.fq.gz\n",
      "Approx 70% complete for P.2_trimmed.fq.gz\n",
      "Approx 75% complete for P.2_trimmed.fq.gz\n",
      "Approx 80% complete for P.2_trimmed.fq.gz\n",
      "Approx 85% complete for P.2_trimmed.fq.gz\n",
      "Approx 90% complete for P.2_trimmed.fq.gz\n",
      "Approx 95% complete for P.2_trimmed.fq.gz\n",
      "Analysis complete for P.2_trimmed.fq.gz\n",
      "Started analysis of Q.1_trimmed.fq.gz\n",
      "Approx 5% complete for Q.1_trimmed.fq.gz\n",
      "Approx 10% complete for Q.1_trimmed.fq.gz\n",
      "Approx 15% complete for Q.1_trimmed.fq.gz\n",
      "Approx 20% complete for Q.1_trimmed.fq.gz\n",
      "Approx 25% complete for Q.1_trimmed.fq.gz\n",
      "Approx 30% complete for Q.1_trimmed.fq.gz\n",
      "Approx 35% complete for Q.1_trimmed.fq.gz\n",
      "Approx 40% complete for Q.1_trimmed.fq.gz\n",
      "Approx 45% complete for Q.1_trimmed.fq.gz\n",
      "Approx 50% complete for Q.1_trimmed.fq.gz\n",
      "Approx 55% complete for Q.1_trimmed.fq.gz\n",
      "Approx 60% complete for Q.1_trimmed.fq.gz\n",
      "Approx 65% complete for Q.1_trimmed.fq.gz\n",
      "Approx 70% complete for Q.1_trimmed.fq.gz\n",
      "Approx 75% complete for Q.1_trimmed.fq.gz\n",
      "Approx 80% complete for Q.1_trimmed.fq.gz\n",
      "Approx 85% complete for Q.1_trimmed.fq.gz\n",
      "Approx 90% complete for Q.1_trimmed.fq.gz\n",
      "Approx 95% complete for Q.1_trimmed.fq.gz\n",
      "Analysis complete for Q.1_trimmed.fq.gz\n",
      "Started analysis of Q.2_trimmed.fq.gz\n",
      "Approx 5% complete for Q.2_trimmed.fq.gz\n",
      "Approx 10% complete for Q.2_trimmed.fq.gz\n",
      "Approx 15% complete for Q.2_trimmed.fq.gz\n",
      "Approx 20% complete for Q.2_trimmed.fq.gz\n",
      "Approx 25% complete for Q.2_trimmed.fq.gz\n",
      "Approx 30% complete for Q.2_trimmed.fq.gz\n",
      "Approx 35% complete for Q.2_trimmed.fq.gz\n",
      "Approx 40% complete for Q.2_trimmed.fq.gz\n",
      "Approx 45% complete for Q.2_trimmed.fq.gz\n",
      "Approx 50% complete for Q.2_trimmed.fq.gz\n",
      "Approx 55% complete for Q.2_trimmed.fq.gz\n",
      "Approx 60% complete for Q.2_trimmed.fq.gz\n",
      "Approx 65% complete for Q.2_trimmed.fq.gz\n",
      "Approx 70% complete for Q.2_trimmed.fq.gz\n",
      "Approx 75% complete for Q.2_trimmed.fq.gz\n",
      "Approx 80% complete for Q.2_trimmed.fq.gz\n",
      "Approx 85% complete for Q.2_trimmed.fq.gz\n",
      "Approx 90% complete for Q.2_trimmed.fq.gz\n",
      "Approx 95% complete for Q.2_trimmed.fq.gz\n",
      "Analysis complete for Q.2_trimmed.fq.gz\n",
      "Started analysis of R.1_trimmed.fq.gz\n",
      "Approx 5% complete for R.1_trimmed.fq.gz\n",
      "Approx 10% complete for R.1_trimmed.fq.gz\n",
      "Approx 15% complete for R.1_trimmed.fq.gz\n",
      "Approx 20% complete for R.1_trimmed.fq.gz\n",
      "Approx 25% complete for R.1_trimmed.fq.gz\n",
      "Approx 30% complete for R.1_trimmed.fq.gz\n",
      "Approx 35% complete for R.1_trimmed.fq.gz\n",
      "Approx 40% complete for R.1_trimmed.fq.gz\n",
      "Approx 45% complete for R.1_trimmed.fq.gz\n",
      "Approx 50% complete for R.1_trimmed.fq.gz\n",
      "Approx 55% complete for R.1_trimmed.fq.gz\n",
      "Approx 60% complete for R.1_trimmed.fq.gz\n",
      "Approx 65% complete for R.1_trimmed.fq.gz\n",
      "Approx 70% complete for R.1_trimmed.fq.gz\n",
      "Approx 75% complete for R.1_trimmed.fq.gz\n",
      "Approx 80% complete for R.1_trimmed.fq.gz\n",
      "Approx 85% complete for R.1_trimmed.fq.gz\n",
      "Approx 90% complete for R.1_trimmed.fq.gz\n",
      "Approx 95% complete for R.1_trimmed.fq.gz\n",
      "Analysis complete for R.1_trimmed.fq.gz\n",
      "Started analysis of R.2_trimmed.fq.gz\n",
      "Approx 5% complete for R.2_trimmed.fq.gz\n",
      "Approx 10% complete for R.2_trimmed.fq.gz\n",
      "Approx 15% complete for R.2_trimmed.fq.gz\n",
      "Approx 20% complete for R.2_trimmed.fq.gz\n",
      "Approx 25% complete for R.2_trimmed.fq.gz\n",
      "Approx 30% complete for R.2_trimmed.fq.gz\n",
      "Approx 35% complete for R.2_trimmed.fq.gz\n",
      "Approx 40% complete for R.2_trimmed.fq.gz\n",
      "Approx 45% complete for R.2_trimmed.fq.gz\n",
      "Approx 50% complete for R.2_trimmed.fq.gz\n",
      "Approx 55% complete for R.2_trimmed.fq.gz\n",
      "Approx 60% complete for R.2_trimmed.fq.gz\n",
      "Approx 65% complete for R.2_trimmed.fq.gz\n",
      "Approx 70% complete for R.2_trimmed.fq.gz\n",
      "Approx 75% complete for R.2_trimmed.fq.gz\n",
      "Approx 80% complete for R.2_trimmed.fq.gz\n",
      "Approx 85% complete for R.2_trimmed.fq.gz\n",
      "Approx 90% complete for R.2_trimmed.fq.gz\n",
      "Approx 95% complete for R.2_trimmed.fq.gz\n",
      "Analysis complete for R.2_trimmed.fq.gz\n",
      "Started analysis of S.1_trimmed.fq.gz\n",
      "Approx 5% complete for S.1_trimmed.fq.gz\n",
      "Approx 10% complete for S.1_trimmed.fq.gz\n",
      "Approx 15% complete for S.1_trimmed.fq.gz\n",
      "Approx 20% complete for S.1_trimmed.fq.gz\n",
      "Approx 25% complete for S.1_trimmed.fq.gz\n",
      "Approx 30% complete for S.1_trimmed.fq.gz\n",
      "Approx 35% complete for S.1_trimmed.fq.gz\n",
      "Approx 40% complete for S.1_trimmed.fq.gz\n",
      "Approx 45% complete for S.1_trimmed.fq.gz\n",
      "Approx 50% complete for S.1_trimmed.fq.gz\n",
      "Approx 55% complete for S.1_trimmed.fq.gz\n",
      "Approx 60% complete for S.1_trimmed.fq.gz\n",
      "Approx 65% complete for S.1_trimmed.fq.gz\n",
      "Approx 70% complete for S.1_trimmed.fq.gz\n",
      "Approx 75% complete for S.1_trimmed.fq.gz\n",
      "Approx 80% complete for S.1_trimmed.fq.gz\n",
      "Approx 85% complete for S.1_trimmed.fq.gz\n",
      "Approx 90% complete for S.1_trimmed.fq.gz\n",
      "Approx 95% complete for S.1_trimmed.fq.gz\n",
      "Analysis complete for S.1_trimmed.fq.gz\n",
      "Started analysis of S.2_trimmed.fq.gz\n",
      "Approx 5% complete for S.2_trimmed.fq.gz\n",
      "Approx 10% complete for S.2_trimmed.fq.gz\n",
      "Approx 15% complete for S.2_trimmed.fq.gz\n",
      "Approx 20% complete for S.2_trimmed.fq.gz\n",
      "Approx 25% complete for S.2_trimmed.fq.gz\n",
      "Approx 30% complete for S.2_trimmed.fq.gz\n",
      "Approx 35% complete for S.2_trimmed.fq.gz\n",
      "Approx 40% complete for S.2_trimmed.fq.gz\n",
      "Approx 45% complete for S.2_trimmed.fq.gz\n",
      "Approx 50% complete for S.2_trimmed.fq.gz\n",
      "Approx 55% complete for S.2_trimmed.fq.gz\n",
      "Approx 60% complete for S.2_trimmed.fq.gz\n",
      "Approx 65% complete for S.2_trimmed.fq.gz\n",
      "Approx 70% complete for S.2_trimmed.fq.gz\n",
      "Approx 75% complete for S.2_trimmed.fq.gz\n",
      "Approx 80% complete for S.2_trimmed.fq.gz\n",
      "Approx 85% complete for S.2_trimmed.fq.gz\n",
      "Approx 90% complete for S.2_trimmed.fq.gz\n",
      "Approx 95% complete for S.2_trimmed.fq.gz\n",
      "Analysis complete for S.2_trimmed.fq.gz\n",
      "Started analysis of T.1_trimmed.fq.gz\n",
      "Approx 5% complete for T.1_trimmed.fq.gz\n",
      "Approx 10% complete for T.1_trimmed.fq.gz\n",
      "Approx 15% complete for T.1_trimmed.fq.gz\n",
      "Approx 20% complete for T.1_trimmed.fq.gz\n",
      "Approx 25% complete for T.1_trimmed.fq.gz\n",
      "Approx 30% complete for T.1_trimmed.fq.gz\n",
      "Approx 35% complete for T.1_trimmed.fq.gz\n",
      "Approx 40% complete for T.1_trimmed.fq.gz\n",
      "Approx 45% complete for T.1_trimmed.fq.gz\n",
      "Approx 50% complete for T.1_trimmed.fq.gz\n",
      "Approx 55% complete for T.1_trimmed.fq.gz\n",
      "Approx 60% complete for T.1_trimmed.fq.gz\n",
      "Approx 65% complete for T.1_trimmed.fq.gz\n",
      "Approx 70% complete for T.1_trimmed.fq.gz\n",
      "Approx 75% complete for T.1_trimmed.fq.gz\n",
      "Approx 80% complete for T.1_trimmed.fq.gz\n",
      "Approx 85% complete for T.1_trimmed.fq.gz\n",
      "Approx 90% complete for T.1_trimmed.fq.gz\n",
      "Approx 95% complete for T.1_trimmed.fq.gz\n",
      "Analysis complete for T.1_trimmed.fq.gz\n",
      "Started analysis of T.2_trimmed.fq.gz\n",
      "Approx 5% complete for T.2_trimmed.fq.gz\n",
      "Approx 10% complete for T.2_trimmed.fq.gz\n",
      "Approx 15% complete for T.2_trimmed.fq.gz\n",
      "Approx 20% complete for T.2_trimmed.fq.gz\n",
      "Approx 25% complete for T.2_trimmed.fq.gz\n",
      "Approx 30% complete for T.2_trimmed.fq.gz\n",
      "Approx 35% complete for T.2_trimmed.fq.gz\n",
      "Approx 40% complete for T.2_trimmed.fq.gz\n",
      "Approx 45% complete for T.2_trimmed.fq.gz\n",
      "Approx 50% complete for T.2_trimmed.fq.gz\n",
      "Approx 55% complete for T.2_trimmed.fq.gz\n",
      "Approx 60% complete for T.2_trimmed.fq.gz\n",
      "Approx 65% complete for T.2_trimmed.fq.gz\n",
      "Approx 70% complete for T.2_trimmed.fq.gz\n",
      "Approx 75% complete for T.2_trimmed.fq.gz\n",
      "Approx 80% complete for T.2_trimmed.fq.gz\n",
      "Approx 85% complete for T.2_trimmed.fq.gz\n",
      "Approx 90% complete for T.2_trimmed.fq.gz\n",
      "Approx 95% complete for T.2_trimmed.fq.gz\n",
      "Analysis complete for T.2_trimmed.fq.gz\n",
      "Started analysis of U.1_trimmed.fq.gz\n",
      "Approx 5% complete for U.1_trimmed.fq.gz\n",
      "Approx 10% complete for U.1_trimmed.fq.gz\n",
      "Approx 15% complete for U.1_trimmed.fq.gz\n",
      "Approx 20% complete for U.1_trimmed.fq.gz\n",
      "Approx 25% complete for U.1_trimmed.fq.gz\n",
      "Approx 30% complete for U.1_trimmed.fq.gz\n",
      "Approx 35% complete for U.1_trimmed.fq.gz\n",
      "Approx 40% complete for U.1_trimmed.fq.gz\n",
      "Approx 45% complete for U.1_trimmed.fq.gz\n",
      "Approx 50% complete for U.1_trimmed.fq.gz\n",
      "Approx 55% complete for U.1_trimmed.fq.gz\n",
      "Approx 60% complete for U.1_trimmed.fq.gz\n",
      "Approx 65% complete for U.1_trimmed.fq.gz\n",
      "Approx 70% complete for U.1_trimmed.fq.gz\n",
      "Approx 75% complete for U.1_trimmed.fq.gz\n",
      "Approx 80% complete for U.1_trimmed.fq.gz\n",
      "Approx 85% complete for U.1_trimmed.fq.gz\n",
      "Approx 90% complete for U.1_trimmed.fq.gz\n",
      "Approx 95% complete for U.1_trimmed.fq.gz\n",
      "Analysis complete for U.1_trimmed.fq.gz\n",
      "Started analysis of U.2_trimmed.fq.gz\n",
      "Approx 5% complete for U.2_trimmed.fq.gz\n",
      "Approx 10% complete for U.2_trimmed.fq.gz\n",
      "Approx 15% complete for U.2_trimmed.fq.gz\n",
      "Approx 20% complete for U.2_trimmed.fq.gz\n",
      "Approx 25% complete for U.2_trimmed.fq.gz\n",
      "Approx 30% complete for U.2_trimmed.fq.gz\n",
      "Approx 35% complete for U.2_trimmed.fq.gz\n",
      "Approx 40% complete for U.2_trimmed.fq.gz\n",
      "Approx 45% complete for U.2_trimmed.fq.gz\n",
      "Approx 50% complete for U.2_trimmed.fq.gz\n",
      "Approx 55% complete for U.2_trimmed.fq.gz\n",
      "Approx 60% complete for U.2_trimmed.fq.gz\n",
      "Approx 65% complete for U.2_trimmed.fq.gz\n",
      "Approx 70% complete for U.2_trimmed.fq.gz\n",
      "Approx 75% complete for U.2_trimmed.fq.gz\n",
      "Approx 80% complete for U.2_trimmed.fq.gz\n",
      "Approx 85% complete for U.2_trimmed.fq.gz\n",
      "Approx 90% complete for U.2_trimmed.fq.gz\n",
      "Approx 95% complete for U.2_trimmed.fq.gz\n",
      "Analysis complete for U.2_trimmed.fq.gz\n",
      "Started analysis of V.1_trimmed.fq.gz\n",
      "Approx 5% complete for V.1_trimmed.fq.gz\n",
      "Approx 10% complete for V.1_trimmed.fq.gz\n",
      "Approx 15% complete for V.1_trimmed.fq.gz\n",
      "Approx 20% complete for V.1_trimmed.fq.gz\n",
      "Approx 25% complete for V.1_trimmed.fq.gz\n",
      "Approx 30% complete for V.1_trimmed.fq.gz\n",
      "Approx 35% complete for V.1_trimmed.fq.gz\n",
      "Approx 40% complete for V.1_trimmed.fq.gz\n",
      "Approx 45% complete for V.1_trimmed.fq.gz\n",
      "Approx 50% complete for V.1_trimmed.fq.gz\n",
      "Approx 55% complete for V.1_trimmed.fq.gz\n",
      "Approx 60% complete for V.1_trimmed.fq.gz\n",
      "Approx 65% complete for V.1_trimmed.fq.gz\n",
      "Approx 70% complete for V.1_trimmed.fq.gz\n",
      "Approx 75% complete for V.1_trimmed.fq.gz\n",
      "Approx 80% complete for V.1_trimmed.fq.gz\n",
      "Approx 85% complete for V.1_trimmed.fq.gz\n",
      "Approx 90% complete for V.1_trimmed.fq.gz\n",
      "Approx 95% complete for V.1_trimmed.fq.gz\n",
      "Analysis complete for V.1_trimmed.fq.gz\n",
      "Started analysis of V.2_trimmed.fq.gz\n",
      "Approx 5% complete for V.2_trimmed.fq.gz\n",
      "Approx 10% complete for V.2_trimmed.fq.gz\n",
      "Approx 15% complete for V.2_trimmed.fq.gz\n",
      "Approx 20% complete for V.2_trimmed.fq.gz\n",
      "Approx 25% complete for V.2_trimmed.fq.gz\n",
      "Approx 30% complete for V.2_trimmed.fq.gz\n",
      "Approx 35% complete for V.2_trimmed.fq.gz\n",
      "Approx 40% complete for V.2_trimmed.fq.gz\n",
      "Approx 45% complete for V.2_trimmed.fq.gz\n",
      "Approx 50% complete for V.2_trimmed.fq.gz\n",
      "Approx 55% complete for V.2_trimmed.fq.gz\n",
      "Approx 60% complete for V.2_trimmed.fq.gz\n",
      "Approx 65% complete for V.2_trimmed.fq.gz\n",
      "Approx 70% complete for V.2_trimmed.fq.gz\n",
      "Approx 75% complete for V.2_trimmed.fq.gz\n",
      "Approx 80% complete for V.2_trimmed.fq.gz\n",
      "Approx 85% complete for V.2_trimmed.fq.gz\n",
      "Approx 90% complete for V.2_trimmed.fq.gz\n",
      "Approx 95% complete for V.2_trimmed.fq.gz\n",
      "Analysis complete for V.2_trimmed.fq.gz\n",
      "Started analysis of W.1_trimmed.fq.gz\n",
      "Approx 5% complete for W.1_trimmed.fq.gz\n",
      "Approx 10% complete for W.1_trimmed.fq.gz\n",
      "Approx 15% complete for W.1_trimmed.fq.gz\n",
      "Approx 20% complete for W.1_trimmed.fq.gz\n",
      "Approx 25% complete for W.1_trimmed.fq.gz\n",
      "Approx 30% complete for W.1_trimmed.fq.gz\n",
      "Approx 35% complete for W.1_trimmed.fq.gz\n",
      "Approx 40% complete for W.1_trimmed.fq.gz\n",
      "Approx 45% complete for W.1_trimmed.fq.gz\n",
      "Approx 50% complete for W.1_trimmed.fq.gz\n",
      "Approx 55% complete for W.1_trimmed.fq.gz\n",
      "Approx 60% complete for W.1_trimmed.fq.gz\n",
      "Approx 65% complete for W.1_trimmed.fq.gz\n",
      "Approx 70% complete for W.1_trimmed.fq.gz\n",
      "Approx 75% complete for W.1_trimmed.fq.gz\n",
      "Approx 80% complete for W.1_trimmed.fq.gz\n",
      "Approx 85% complete for W.1_trimmed.fq.gz\n",
      "Approx 90% complete for W.1_trimmed.fq.gz\n",
      "Approx 95% complete for W.1_trimmed.fq.gz\n",
      "Analysis complete for W.1_trimmed.fq.gz\n",
      "Started analysis of W.2_trimmed.fq.gz\n",
      "Approx 5% complete for W.2_trimmed.fq.gz\n",
      "Approx 10% complete for W.2_trimmed.fq.gz\n",
      "Approx 15% complete for W.2_trimmed.fq.gz\n",
      "Approx 20% complete for W.2_trimmed.fq.gz\n",
      "Approx 25% complete for W.2_trimmed.fq.gz\n",
      "Approx 30% complete for W.2_trimmed.fq.gz\n",
      "Approx 35% complete for W.2_trimmed.fq.gz\n",
      "Approx 40% complete for W.2_trimmed.fq.gz\n",
      "Approx 45% complete for W.2_trimmed.fq.gz\n",
      "Approx 50% complete for W.2_trimmed.fq.gz\n",
      "Approx 55% complete for W.2_trimmed.fq.gz\n",
      "Approx 60% complete for W.2_trimmed.fq.gz\n",
      "Approx 65% complete for W.2_trimmed.fq.gz\n",
      "Approx 70% complete for W.2_trimmed.fq.gz\n",
      "Approx 75% complete for W.2_trimmed.fq.gz\n",
      "Approx 80% complete for W.2_trimmed.fq.gz\n",
      "Approx 85% complete for W.2_trimmed.fq.gz\n",
      "Approx 90% complete for W.2_trimmed.fq.gz\n",
      "Approx 95% complete for W.2_trimmed.fq.gz\n",
      "Analysis complete for W.2_trimmed.fq.gz\n",
      "Started analysis of X.1_trimmed.fq.gz\n",
      "Approx 5% complete for X.1_trimmed.fq.gz\n",
      "Approx 10% complete for X.1_trimmed.fq.gz\n",
      "Approx 15% complete for X.1_trimmed.fq.gz\n",
      "Approx 20% complete for X.1_trimmed.fq.gz\n",
      "Approx 25% complete for X.1_trimmed.fq.gz\n",
      "Approx 30% complete for X.1_trimmed.fq.gz\n",
      "Approx 35% complete for X.1_trimmed.fq.gz\n",
      "Approx 40% complete for X.1_trimmed.fq.gz\n",
      "Approx 45% complete for X.1_trimmed.fq.gz\n",
      "Approx 50% complete for X.1_trimmed.fq.gz\n",
      "Approx 55% complete for X.1_trimmed.fq.gz\n",
      "Approx 60% complete for X.1_trimmed.fq.gz\n",
      "Approx 65% complete for X.1_trimmed.fq.gz\n",
      "Approx 70% complete for X.1_trimmed.fq.gz\n",
      "Approx 75% complete for X.1_trimmed.fq.gz\n",
      "Approx 80% complete for X.1_trimmed.fq.gz\n",
      "Approx 85% complete for X.1_trimmed.fq.gz\n",
      "Approx 90% complete for X.1_trimmed.fq.gz\n",
      "Approx 95% complete for X.1_trimmed.fq.gz\n",
      "Analysis complete for X.1_trimmed.fq.gz\n",
      "Started analysis of X.2_trimmed.fq.gz\n",
      "Approx 5% complete for X.2_trimmed.fq.gz\n",
      "Approx 10% complete for X.2_trimmed.fq.gz\n",
      "Approx 15% complete for X.2_trimmed.fq.gz\n",
      "Approx 20% complete for X.2_trimmed.fq.gz\n",
      "Approx 25% complete for X.2_trimmed.fq.gz\n",
      "Approx 30% complete for X.2_trimmed.fq.gz\n",
      "Approx 35% complete for X.2_trimmed.fq.gz\n",
      "Approx 40% complete for X.2_trimmed.fq.gz\n",
      "Approx 45% complete for X.2_trimmed.fq.gz\n",
      "Approx 50% complete for X.2_trimmed.fq.gz\n",
      "Approx 55% complete for X.2_trimmed.fq.gz\n",
      "Approx 60% complete for X.2_trimmed.fq.gz\n",
      "Approx 65% complete for X.2_trimmed.fq.gz\n",
      "Approx 70% complete for X.2_trimmed.fq.gz\n",
      "Approx 75% complete for X.2_trimmed.fq.gz\n",
      "Approx 80% complete for X.2_trimmed.fq.gz\n",
      "Approx 85% complete for X.2_trimmed.fq.gz\n",
      "Approx 90% complete for X.2_trimmed.fq.gz\n",
      "Approx 95% complete for X.2_trimmed.fq.gz\n",
      "Analysis complete for X.2_trimmed.fq.gz\n"
     ]
    }
   ],
   "source": [
    "!scripts/FastQC/fastqc /Volumes/Untitled/trimed_fastq/*.fq.gz -o analyses/fastQC_after/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "re-run multi-qc..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v0.8\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching 'analyses/fastQC_after/'\n",
      "[INFO   ]          fastqc : Found 48 reports\n",
      "[INFO   ]         multiqc : Report      : analyses/multiqc_after.html\n",
      "[INFO   ]         multiqc : Data        : analyses/multiqc_after_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc analyses/fastQC_after/ -o ./analyses/ -n multiqc_after"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I'm happy with the results from above, next I need to concatenate all of my front reads and reverse reads in the same order so they can be run in Trinity for assembly..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat /Volumes/Untitled/trimed_fastq/*.1.fq.gz > /Volumes/Untitled/all_1.fq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat /Volumes/Untitled/trimed_fastq/*.2.fq.gz > /Volumes/Untitled/all_1.fq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Upload to Galaxy Indiana for assembly \n",
    "\n",
    "![upload galaxy](https://github.com/aspanjer/RNAseq_Coho/blob/master/jupyter/upload.png?raw=true)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Start the assembly process..\n",
    "\n",
    "![Galaxy assembly](https://github.com/aspanjer/RNAseq_Coho/blob/master/analyses/Trinity/Trinity%20Galaxy.png?raw=true)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "Next step [Differential Expression](https://github.com/aspanjer/RNAseq_Coho/blob/master/jupyter_notebooks/Differential%20Expression.ipynb) (new journal)"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
