## Final Figures for class project
### Below are the captions for each of the numbered figures

**Table 1. Summary statistics for Trinity assembled hepatic coho salmon (_Oncorhynhcus kitsch_) transcriptome.** Transcriptome was assembled from >600 million 150bp PE short reads from total RNA isolated from 24 individual juvenile (year-0) fish livers sampled at 4 different small perrenial streams that drain into the Salish Sea. 

**Figure 1. Principal Component Analysis of raw Kalisto mapped counts for 24 juvenile coho salmon (_Oncorhynhcus kitsch_) grouped by sample location.** All reads that mapped back to the assembled transcriptome were used to investigate how samples varried in gene expresion by location, thus showing that some varience in gene expression is likely due to sampling location. 

**Figure 2. Log2 fold change for differentially expressed genes (DEGs) as compated to reference site (Coulter Creek).** All comparisons are pairwise to the reference site at padj=.05. If a gene was significant for any of the three comparisons, the log2 fold change is included for all three sites in this heatmap. Contigs are roughly even split by up/down regulation. Fish from Jenkins and Swamp are more similar in their DEGs than Issaquah.  

**Figure 3. Volcano plot showing proportion of significant (padj <=.05) differentially expressed genes (DEGs) as compared to total expression pattern among all sites.** Issaquah shows a much higher proportion of significant DEGs than Jenkins or Swamp. 

**Figure 4. Top Gene Ontology terms for significant differentially expressed genes between Issaquah and Coulter creeks.** Top 15 terms for each GO category (molecular function, cellular component, and biological process). Top 15 terms for each were determined by lowest enrichment pvalue reported by DAVID as compared to GO terms representing whole transcriptome. Bar length indicates the number of DEGs belonging to each term. 

**Figure 5. Top Gene Ontology terms for significant differentially expressed genes between Jenkins and Coulter creeks.** Same as above (less than 15 total terms for DEGs). 

**Figure 6. Top Gene Ontology terms for significant differentially expressed genes between Swamp and Coulter creeks.** Same as figure 4. 

**Figure 7. REVIGO plot showing the similarity in biological process for enriched GO terms in DEGs from each sampling stream.** Enrichment of GO terms was first determined by comparing biological process GO terms represented by DEGs for each pairwise comparison to the reference site in DAVID. List of enriched genes was submitted to REVIGO for further similarity analysis represented by this plot. 
