{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "   # RNAseq Coho Differential Expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/andrewspanjer/RNAseq_Coho/data\n"
     ]
    }
   ],
   "source": [
    "cd data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A.1.fastq.gz  F.1.fastq.gz  K.1.fastq.gz  P.1.fastq.gz  T.2.fastq.gz\r\n",
      "A.2.fastq.gz  F.2.fastq.gz  K.2.fastq.gz  P.2.fastq.gz  U.1.fastq.gz\r\n",
      "B.1.fastq.gz  G.1.fastq.gz  L.1.fastq.gz  Q.1.fastq.gz  U.2.fastq.gz\r\n",
      "B.2.fastq.gz  G.2.fastq.gz  L.2.fastq.gz  Q.2.fastq.gz  V.1.fastq.gz\r\n",
      "C.1.fastq.gz  H.1.fastq.gz  M.1.fastq.gz  R.1.fastq.gz  V.2.fastq.gz\r\n",
      "C.2.fastq.gz  H.2.fastq.gz  M.2.fastq.gz  R.2.fastq.gz  W.1.fastq.gz\r\n",
      "D.1.fastq.gz  I.1.fastq.gz  N.1.fastq.gz  README.md     W.2.fastq.gz\r\n",
      "D.2.fastq.gz  I.2.fastq.gz  N.2.fastq.gz  S.1.fastq.gz  X.1.fastq.gz\r\n",
      "E.1.fastq.gz  J.1.fastq.gz  O.1.fastq.gz  S.2.fastq.gz  X.2.fastq.gz\r\n",
      "E.2.fastq.gz  J.2.fastq.gz  O.2.fastq.gz  T.1.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Unzip first fastq file to look at the first 10 lines to see how the raw sequencng data looks. Rezip the file to save space: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "head: A.1.fastq: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!gunzip A.1.fastq.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For a complete description of the data files see the README.md in /data. This file is one of 48 sequencing files representing PE reads of RNA from Juvenile Coho fish livers. Sequencing was performed by the UW Genomic Core facility."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@HWKWKBGXX:1:11101:10000:19471/1\r\n",
      "TTTTTTTGTACAAAAGCTGAGTAAGTTGGCCTGCAAAGCCCTAATTAAATGCCCCCACAACCAATAGAGAAGCAACAATTACAATTGTCAACTCTTTTTGCAGGCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACGAATTCGTATA\r\n",
      "+\r\n",
      "AAAAAEEAEEEEAEEEEA<EEEEE6E//6E/AA<EEEEEE/EEE/EEAEEE<EA/EE/E666<EAAA6<EEEEEEEAEE<E/EEA/A/<EE/EE<EE/<//E//AE/<EAEEEEEEE/6/A/<E/E<E/AA</AAE<A//EEE<A/AE##\r\n",
      "@HWKWKBGXX:1:11101:10002:10211/1\r\n",
      "GAAGTGTTCTAGGGGTACCGCTTCAACAACGATGGGTATGAAGCTATGGTTAGCCCCGCAGATTTCAGAACATTGTCCGTAGAATACTCCAGGTCGAGAGATCGGAAGAGCACACGTCTGAACTCCAGTCACGAATTCGTATCTCGTATG\r\n",
      "+\r\n",
      "6A/AA6/EEEEEAEEEEAAEEEEEEAE/EEAEEEEAEE//E//AEEEEA/EEE6/EEEEEEA//E//</<EEAE/AA6EEEA<AEEEE/6E<6AE<EEEE6/AAEAEAEEAEEE<A/EAA<E/EE/66EEA66AAE/</AE<<66<A###\r\n",
      "@HWKWKBGXX:1:11101:10005:13424/1\r\n",
      "CCACCGCCGGCTTGTACCTGTTGAGGTAGTCAGCCACGCCACATATCGTCGGGCAGTACTTGCCAAATCCATCCCTTGGTGTGCAGTCCTCAGAATAGTCTCCCCTCGTTTGTGCAGAGGAGAGGGAGAACAGCAAAAGAACGCCTGCGG\r\n"
     ]
    }
   ],
   "source": [
    "!head A.1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The data files are structured as follows:\n",
    "\n",
    "line 1:  @ 'instrument':'run number':'flowcell ID':'lane':'tile':'x-pos':'y-pos' 'read':'is filtered':'control number':'index sequence'\n",
    "\n",
    "Line 2: This is the actual sequence\n",
    "\n",
    "Line 3: a line break signaled by '+'\n",
    "\n",
    "Line 4: These are quality codes for each base in the sequence. A guide for these codes can be found here: http://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/QualityScoreEncoding_swBS.htm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!gzip A.1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next I want to count the number of sequences in each file (representing each individual). I would like to select the largest sequencing data set from each site to use in constructing the de novo transcriptome. Perhaps an easier way of doing this would be to simply look at the file size, but this way I have the number of reads for each individual. I need to use the 'zgrep' script to work with compressed files (\"wc\" will give an incorrect result with compressed files). I'll need to divide the output by 4 to get the correct number of reads (4 lines per read). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A.1.fastq.gz:56752468\r\n",
      "A.2.fastq.gz:56752468\r\n",
      "B.1.fastq.gz:65017156\r\n",
      "B.2.fastq.gz:65017156\r\n",
      "C.1.fastq.gz:65433440\r\n",
      "C.2.fastq.gz:65433440\r\n",
      "D.1.fastq.gz:62594380\r\n",
      "D.2.fastq.gz:62594380\r\n",
      "E.1.fastq.gz:63222480\r\n",
      "E.2.fastq.gz:63222480\r\n",
      "F.1.fastq.gz:86837812\r\n",
      "F.2.fastq.gz:86837812\r\n",
      "G.1.fastq.gz:66037316\r\n",
      "G.2.fastq.gz:66037316\r\n",
      "H.1.fastq.gz:63267680\r\n",
      "H.2.fastq.gz:63267680\r\n",
      "I.1.fastq.gz:85395968\r\n",
      "I.2.fastq.gz:85395968\r\n",
      "J.1.fastq.gz:71119360\r\n",
      "J.2.fastq.gz:71119360\r\n",
      "K.1.fastq.gz:65728504\r\n",
      "K.2.fastq.gz:65728504\r\n",
      "L.1.fastq.gz:67399792\r\n",
      "L.2.fastq.gz:67399792\r\n",
      "M.1.fastq.gz:63365156\r\n",
      "M.2.fastq.gz:63365156\r\n",
      "N.1.fastq.gz:74193224\r\n",
      "N.2.fastq.gz:74193224\r\n",
      "O.1.fastq.gz:65143844\r\n",
      "O.2.fastq.gz:65143844\r\n",
      "P.1.fastq.gz:58907416\r\n",
      "P.2.fastq.gz:58907416\r\n",
      "Q.1.fastq.gz:73842176\r\n",
      "Q.2.fastq.gz:73842176\r\n",
      "R.1.fastq.gz:67564232\r\n",
      "R.2.fastq.gz:67564232\r\n",
      "S.1.fastq.gz:63299656\r\n",
      "S.2.fastq.gz:63299656\r\n",
      "T.1.fastq.gz:72148680\r\n",
      "T.2.fastq.gz:72148680\r\n",
      "U.1.fastq.gz:62809256\r\n",
      "U.2.fastq.gz:62809256\r\n",
      "V.1.fastq.gz:98999392\r\n",
      "V.2.fastq.gz:98999392\r\n",
      "W.1.fastq.gz:65467360\r\n",
      "W.2.fastq.gz:65467360\r\n",
      "X.1.fastq.gz:64802048\r\n",
      "X.2.fastq.gz:64802048\r\n"
     ]
    }
   ],
   "source": [
    "!zgrep -c ^ *.fastq.gz | tee reads.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/andrewspanjer/RNAseq_Coho/data\n"
     ]
    }
   ],
   "source": [
    "cd RNAseq_Coho/data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of A.1.fastq.gz\n",
      "Approx 5% complete for A.1.fastq.gz\n",
      "Approx 10% complete for A.1.fastq.gz\n",
      "Approx 15% complete for A.1.fastq.gz\n",
      "Approx 20% complete for A.1.fastq.gz\n",
      "Approx 25% complete for A.1.fastq.gz\n",
      "Approx 30% complete for A.1.fastq.gz\n",
      "Approx 35% complete for A.1.fastq.gz\n",
      "Approx 40% complete for A.1.fastq.gz\n",
      "Approx 45% complete for A.1.fastq.gz\n",
      "Approx 50% complete for A.1.fastq.gz\n",
      "Approx 55% complete for A.1.fastq.gz\n",
      "Approx 60% complete for A.1.fastq.gz\n",
      "Approx 65% complete for A.1.fastq.gz\n",
      "Approx 70% complete for A.1.fastq.gz\n",
      "Approx 75% complete for A.1.fastq.gz\n",
      "Approx 80% complete for A.1.fastq.gz\n",
      "Approx 85% complete for A.1.fastq.gz\n",
      "Approx 90% complete for A.1.fastq.gz\n",
      "Approx 95% complete for A.1.fastq.gz\n",
      "Analysis complete for A.1.fastq.gz\n",
      "Started analysis of A.2.fastq.gz\n",
      "Approx 5% complete for A.2.fastq.gz\n",
      "Approx 10% complete for A.2.fastq.gz\n",
      "Approx 15% complete for A.2.fastq.gz\n",
      "Approx 20% complete for A.2.fastq.gz\n",
      "Approx 25% complete for A.2.fastq.gz\n",
      "Approx 30% complete for A.2.fastq.gz\n",
      "Approx 35% complete for A.2.fastq.gz\n",
      "Approx 40% complete for A.2.fastq.gz\n",
      "Approx 45% complete for A.2.fastq.gz\n",
      "Approx 50% complete for A.2.fastq.gz\n",
      "Approx 55% complete for A.2.fastq.gz\n",
      "Approx 60% complete for A.2.fastq.gz\n",
      "Approx 65% complete for A.2.fastq.gz\n",
      "Approx 70% complete for A.2.fastq.gz\n",
      "Approx 75% complete for A.2.fastq.gz\n",
      "Approx 80% complete for A.2.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "! scripts/FastQC/fastqc data/*.fastq.gz -o analyses/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v0.8\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching './analyses/'\n",
      "[INFO   ]          fastqc : Found 48 reports\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc ./analyses/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.11\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./data/A.1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t273916\tAGATCGGAAGAGC\t1000000\t27.39\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 273916). Second best hit was Nextera (count: 2)\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data/A.1.fastq.gz: uncompress failed\n",
      "Writing report to '/Volumes/Untitled/trimed_fastq/A.1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data/A.1.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to A.1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data/A.1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data/A.1.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 803.98 s (57 us/read; 1.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,188,117\n",
      "Reads with adapters:                 8,628,250 (60.8%)\n",
      "Reads written (passing filters):    14,188,117 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,128,217,550 bp\n",
      "Quality-trimmed:             253,477,958 bp (11.9%)\n",
      "Total written (filtered):  1,692,805,170 bp (79.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 8628250 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 22.0%\n",
      "  C: 31.8%\n",
      "  G: 26.7%\n",
      "  T: 17.6%\n",
      "  none/other: 1.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2666963\t3547029.2\t0\t2666963\n",
      "2\t605006\t886757.3\t0\t605006\n",
      "3\t311682\t221689.3\t0\t311682\n",
      "4\t134798\t55422.3\t0\t134798\n",
      "5\t72606\t13855.6\t0\t72606\n",
      "6\t50292\t3463.9\t0\t50292\n",
      "7\t48711\t866.0\t0\t48711\n",
      "8\t64828\t216.5\t0\t64828\n",
      "9\t99184\t54.1\t0\t99023 161\n",
      "10\t95939\t13.5\t1\t84307 11632\n",
      "11\t106247\t3.4\t1\t84026 22221\n",
      "12\t85523\t0.8\t1\t70206 15317\n",
      "13\t81158\t0.2\t1\t70892 10266\n",
      "14\t78835\t0.2\t1\t63989 14846\n",
      "15\t81403\t0.2\t1\t70671 10732\n",
      "16\t89525\t0.2\t1\t66914 22611\n",
      "17\t81524\t0.2\t1\t68177 13347\n",
      "18\t60023\t0.2\t1\t52571 7452\n",
      "19\t75697\t0.2\t1\t65817 9880\n",
      "20\t59296\t0.2\t1\t50994 8302\n",
      "21\t79066\t0.2\t1\t68810 10256\n",
      "22\t79735\t0.2\t1\t70068 9667\n",
      "23\t80663\t0.2\t1\t69340 11323\n",
      "24\t101759\t0.2\t1\t83477 18282\n",
      "25\t84223\t0.2\t1\t71089 13134\n",
      "26\t80033\t0.2\t1\t66910 13123\n",
      "27\t82922\t0.2\t1\t71978 10944\n",
      "28\t65102\t0.2\t1\t57638 7464\n",
      "29\t96468\t0.2\t1\t81894 14574\n",
      "30\t73598\t0.2\t1\t64183 9415\n",
      "31\t89529\t0.2\t1\t77979 11550\n",
      "32\t76518\t0.2\t1\t67626 8892\n",
      "33\t84584\t0.2\t1\t72100 12484\n",
      "34\t80748\t0.2\t1\t70831 9917\n",
      "35\t72407\t0.2\t1\t62513 9894\n",
      "36\t84060\t0.2\t1\t71988 12072\n",
      "37\t106370\t0.2\t1\t89740 16630\n",
      "38\t97434\t0.2\t1\t82435 14999\n",
      "39\t122763\t0.2\t1\t104111 18652\n",
      "40\t123222\t0.2\t1\t106443 16779\n",
      "41\t149220\t0.2\t1\t131133 18087\n",
      "42\t173180\t0.2\t1\t152240 20940\n",
      "43\t179578\t0.2\t1\t152779 26799\n",
      "44\t113773\t0.2\t1\t98374 15399\n",
      "45\t41599\t0.2\t1\t36175 5424\n",
      "46\t39153\t0.2\t1\t33409 5744\n",
      "47\t38359\t0.2\t1\t33533 4826\n",
      "48\t47016\t0.2\t1\t42363 4653\n",
      "49\t61509\t0.2\t1\t55386 6123\n",
      "50\t52497\t0.2\t1\t45243 7254\n",
      "51\t77420\t0.2\t1\t68109 9311\n",
      "52\t61777\t0.2\t1\t55387 6390\n",
      "53\t47195\t0.2\t1\t42136 5059\n",
      "54\t48056\t0.2\t1\t43137 4919\n",
      "55\t55479\t0.2\t1\t50264 5215\n",
      "56\t41616\t0.2\t1\t37414 4202\n",
      "57\t25225\t0.2\t1\t22504 2721\n",
      "58\t41401\t0.2\t1\t37368 4033\n",
      "59\t54783\t0.2\t1\t49349 5434\n",
      "60\t23684\t0.2\t1\t21125 2559\n",
      "61\t32635\t0.2\t1\t29275 3360\n",
      "62\t34911\t0.2\t1\t31570 3341\n",
      "63\t18399\t0.2\t1\t16432 1967\n",
      "64\t27163\t0.2\t1\t24464 2699\n",
      "65\t32516\t0.2\t1\t29294 3222\n",
      "66\t20593\t0.2\t1\t18499 2094\n",
      "67\t22579\t0.2\t1\t20212 2367\n",
      "68\t23651\t0.2\t1\t21151 2500\n",
      "69\t22885\t0.2\t1\t20149 2736\n",
      "70\t23225\t0.2\t1\t20322 2903\n",
      "71\t27347\t0.2\t1\t23499 3848\n",
      "72\t35890\t0.2\t1\t30498 5392\n",
      "73\t38523\t0.2\t1\t32407 6116\n",
      "74\t41479\t0.2\t1\t33590 7889\n",
      "75\t33751\t0.2\t1\t27278 6473\n",
      "76\t10168\t0.2\t1\t7983 2185\n",
      "77\t1210\t0.2\t1\t925 285\n",
      "78\t1457\t0.2\t1\t1240 217\n",
      "79\t2636\t0.2\t1\t2340 296\n",
      "80\t3640\t0.2\t1\t3241 399\n",
      "81\t4351\t0.2\t1\t3876 475\n",
      "82\t4527\t0.2\t1\t4046 481\n",
      "83\t4630\t0.2\t1\t4174 456\n",
      "84\t4540\t0.2\t1\t4058 482\n",
      "85\t4221\t0.2\t1\t3829 392\n",
      "86\t4120\t0.2\t1\t3710 410\n",
      "87\t3824\t0.2\t1\t3430 394\n",
      "88\t3650\t0.2\t1\t3259 391\n",
      "89\t3383\t0.2\t1\t3033 350\n",
      "90\t2999\t0.2\t1\t2686 313\n",
      "91\t2670\t0.2\t1\t2412 258\n",
      "92\t2432\t0.2\t1\t2168 264\n",
      "93\t2001\t0.2\t1\t1798 203\n",
      "94\t1775\t0.2\t1\t1590 185\n",
      "95\t1672\t0.2\t1\t1507 165\n",
      "96\t1560\t0.2\t1\t1393 167\n",
      "97\t1538\t0.2\t1\t1376 162\n",
      "98\t1451\t0.2\t1\t1310 141\n",
      "99\t1272\t0.2\t1\t1139 133\n",
      "100\t1196\t0.2\t1\t1071 125\n",
      "101\t1048\t0.2\t1\t941 107\n",
      "102\t861\t0.2\t1\t784 77\n",
      "103\t688\t0.2\t1\t625 63\n",
      "104\t642\t0.2\t1\t570 72\n",
      "105\t593\t0.2\t1\t523 70\n",
      "106\t482\t0.2\t1\t428 54\n",
      "107\t525\t0.2\t1\t448 77\n",
      "108\t436\t0.2\t1\t380 56\n",
      "109\t461\t0.2\t1\t412 49\n",
      "110\t402\t0.2\t1\t356 46\n",
      "111\t411\t0.2\t1\t364 47\n",
      "112\t326\t0.2\t1\t294 32\n",
      "113\t319\t0.2\t1\t279 40\n",
      "114\t271\t0.2\t1\t242 29\n",
      "115\t286\t0.2\t1\t250 36\n",
      "116\t296\t0.2\t1\t254 42\n",
      "117\t292\t0.2\t1\t264 28\n",
      "118\t246\t0.2\t1\t209 37\n",
      "119\t272\t0.2\t1\t248 24\n",
      "120\t235\t0.2\t1\t203 32\n",
      "121\t211\t0.2\t1\t182 29\n",
      "122\t202\t0.2\t1\t181 21\n",
      "123\t229\t0.2\t1\t197 32\n",
      "124\t188\t0.2\t1\t170 18\n",
      "125\t191\t0.2\t1\t169 22\n",
      "126\t207\t0.2\t1\t179 28\n",
      "127\t139\t0.2\t1\t114 25\n",
      "128\t112\t0.2\t1\t93 19\n",
      "129\t84\t0.2\t1\t64 20\n",
      "130\t68\t0.2\t1\t56 12\n",
      "131\t90\t0.2\t1\t70 20\n",
      "132\t58\t0.2\t1\t41 17\n",
      "133\t96\t0.2\t1\t86 10\n",
      "134\t105\t0.2\t1\t93 12\n",
      "135\t89\t0.2\t1\t82 7\n",
      "136\t43\t0.2\t1\t38 5\n",
      "137\t44\t0.2\t1\t36 8\n",
      "138\t72\t0.2\t1\t69 3\n",
      "139\t58\t0.2\t1\t50 8\n",
      "140\t99\t0.2\t1\t91 8\n",
      "141\t78\t0.2\t1\t68 10\n",
      "142\t67\t0.2\t1\t54 13\n",
      "143\t57\t0.2\t1\t52 5\n",
      "144\t67\t0.2\t1\t63 4\n",
      "145\t251\t0.2\t1\t243 8\n",
      "146\t188\t0.2\t1\t179 9\n",
      "147\t54\t0.2\t1\t50 4\n",
      "148\t282\t0.2\t1\t206 76\n",
      "149\t48\t0.2\t1\t37 11\n",
      "150\t20247\t0.2\t1\t68 20179\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./data/A.1.fastq.gz\n",
      "=============================================\n",
      "14188117 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t454157 (3.2%)\n",
      "\n",
      "Writing report to '/Volumes/Untitled/trimed_fastq/A.2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./data/A.2.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.2\n",
      "Cutadapt version: 1.11\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to A.2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./data/A.2.fastq.gz <<< \n",
      "gzip: write: Input/output error\n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.11 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./data/A.2.fastq.gz\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 693.24 s (49 us/read; 1.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,188,117\n",
      "Reads with adapters:                 7,543,028 (53.2%)\n",
      "Reads written (passing filters):    14,188,117 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,128,217,550 bp\n",
      "Quality-trimmed:             630,200,205 bp (29.6%)\n",
      "Total written (filtered):  1,369,806,672 bp (64.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 7543028 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.4%\n",
      "  C: 31.3%\n",
      "  G: 27.2%\n",
      "  T: 13.7%\n",
      "  none/other: 0.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2967731\t3547029.2\t0\t2967731\n",
      "2\t560929\t886757.3\t0\t560929\n",
      "3\t302409\t221689.3\t0\t302409\n",
      "4\t112285\t55422.3\t0\t112285\n",
      "5\t86881\t13855.6\t0\t86881\n",
      "6\t41311\t3463.9\t0\t41311\n",
      "7\t40726\t866.0\t0\t40726\n",
      "8\t54764\t216.5\t0\t54764\n",
      "9\t77704\t54.1\t0\t77484 220\n",
      "10\t79336\t13.5\t1\t69326 10010\n",
      "11\t68221\t3.4\t1\t51710 16511\n",
      "12\t58245\t0.8\t1\t46957 11288\n",
      "13\t65601\t0.2\t1\t54075 11526\n",
      "14\t56142\t0.2\t1\t48609 7533\n",
      "15\t52642\t0.2\t1\t44516 8126\n",
      "16\t73882\t0.2\t1\t60556 13326\n",
      "17\t74111\t0.2\t1\t63614 10497\n",
      "18\t79164\t0.2\t1\t68157 11007\n",
      "19\t65866\t0.2\t1\t55639 10227\n",
      "20\t72102\t0.2\t1\t63853 8249\n",
      "21\t43786\t0.2\t1\t35462 8324\n",
      "22\t39125\t0.2\t1\t35522 3603\n",
      "23\t44613\t0.2\t1\t39966 4647\n",
      "24\t68735\t0.2\t1\t61362 7373\n",
      "25\t50571\t0.2\t1\t44482 6089\n",
      "26\t69475\t0.2\t1\t59810 9665\n",
      "27\t110827\t0.2\t1\t93618 17209\n",
      "28\t54591\t0.2\t1\t48584 6007\n",
      "29\t101840\t0.2\t1\t79810 22030\n",
      "30\t61493\t0.2\t1\t54186 7307\n",
      "31\t77264\t0.2\t1\t67487 9777\n",
      "32\t57474\t0.2\t1\t50929 6545\n",
      "33\t74178\t0.2\t1\t65473 8705\n",
      "34\t78371\t0.2\t1\t67802 10569\n",
      "35\t57806\t0.2\t1\t51765 6041\n",
      "36\t72945\t0.2\t1\t65158 7787\n",
      "37\t57573\t0.2\t1\t51042 6531\n",
      "38\t56618\t0.2\t1\t50207 6411\n",
      "39\t60104\t0.2\t1\t53330 6774\n",
      "40\t61318\t0.2\t1\t54277 7041\n",
      "41\t55858\t0.2\t1\t49998 5860\n",
      "42\t53793\t0.2\t1\t49771 4022\n",
      "43\t55792\t0.2\t1\t51282 4510\n",
      "44\t63877\t0.2\t1\t59646 4231\n",
      "45\t64926\t0.2\t1\t59324 5602\n",
      "46\t55983\t0.2\t1\t50242 5741\n",
      "47\t50453\t0.2\t1\t46897 3556\n",
      "48\t51413\t0.2\t1\t45004 6409\n",
      "49\t49939\t0.2\t1\t44514 5425\n",
      "50\t33642\t0.2\t1\t30553 3089\n",
      "51\t37681\t0.2\t1\t33835 3846\n",
      "52\t30389\t0.2\t1\t27966 2423\n",
      "53\t35016\t0.2\t1\t32937 2079\n",
      "54\t40578\t0.2\t1\t38106 2472\n",
      "55\t33748\t0.2\t1\t31452 2296\n",
      "56\t37577\t0.2\t1\t35428 2149\n",
      "57\t48084\t0.2\t1\t45295 2789\n",
      "58\t34919\t0.2\t1\t32206 2713\n",
      "59\t26133\t0.2\t1\t24515 1618\n",
      "60\t30568\t0.2\t1\t28680 1888\n",
      "61\t22545\t0.2\t1\t20889 1656\n",
      "62\t24586\t0.2\t1\t22706 1880\n",
      "63\t29573\t0.2\t1\t27674 1899\n",
      "64\t21457\t0.2\t1\t19973 1484\n",
      "65\t18073\t0.2\t1\t16631 1442\n",
      "66\t18598\t0.2\t1\t17128 1470\n",
      "67\t13848\t0.2\t1\t12969 879\n",
      "68\t18157\t0.2\t1\t16403 1754\n",
      "69\t20442\t0.2\t1\t19250 1192\n",
      "70\t21143\t0.2\t1\t19539 1604\n",
      "71\t10186\t0.2\t1\t9484 702\n",
      "72\t10410\t0.2\t1\t9603 807\n",
      "73\t11507\t0.2\t1\t10562 945\n",
      "74\t12561\t0.2\t1\t11597 964\n",
      "75\t14657\t0.2\t1\t13525 1132\n",
      "76\t19556\t0.2\t1\t18230 1326\n",
      "77\t24352\t0.2\t1\t22580 1772\n",
      "78\t20448\t0.2\t1\t18225 2223\n",
      "79\t10121\t0.2\t1\t8597 1524\n",
      "80\t4175\t0.2\t1\t3326 849\n",
      "81\t747\t0.2\t1\t582 165\n",
      "82\t462\t0.2\t1\t416 46\n",
      "83\t697\t0.2\t1\t654 43\n",
      "84\t1130\t0.2\t1\t1069 61\n",
      "85\t1411\t0.2\t1\t1333 78\n",
      "86\t1666\t0.2\t1\t1595 71\n",
      "87\t1603\t0.2\t1\t1522 81\n",
      "88\t1311\t0.2\t1\t1235 76\n",
      "89\t1013\t0.2\t1\t955 58\n",
      "90\t839\t0.2\t1\t786 53\n",
      "91\t729\t0.2\t1\t689 40\n",
      "92\t635\t0.2\t1\t594 41\n",
      "93\t526\t0.2\t1\t485 41\n",
      "94\t434\t0.2\t1\t404 30\n",
      "95\t362\t0.2\t1\t332 30\n",
      "96\t341\t0.2\t1\t318 23\n",
      "97\t332\t0.2\t1\t308 24\n",
      "98\t322\t0.2\t1\t294 28\n",
      "99\t248\t0.2\t1\t231 17\n",
      "100\t242\t0.2\t1\t224 18\n",
      "101\t213\t0.2\t1\t191 22\n",
      "102\t172\t0.2\t1\t158 14\n",
      "103\t131\t0.2\t1\t113 18\n",
      "104\t125\t0.2\t1\t111 14\n",
      "105\t108\t0.2\t1\t99 9\n",
      "106\t97\t0.2\t1\t85 12\n",
      "107\t95\t0.2\t1\t88 7\n",
      "108\t105\t0.2\t1\t91 14\n",
      "109\t82\t0.2\t1\t72 10\n",
      "110\t81\t0.2\t1\t72 9\n",
      "111\t87\t0.2\t1\t79 8\n",
      "112\t68\t0.2\t1\t61 7\n",
      "113\t72\t0.2\t1\t64 8\n",
      "114\t57\t0.2\t1\t46 11\n",
      "115\t74\t0.2\t1\t65 9\n",
      "116\t45\t0.2\t1\t39 6\n",
      "117\t61\t0.2\t1\t49 12\n",
      "118\t43\t0.2\t1\t35 8\n",
      "119\t60\t0.2\t1\t52 8\n",
      "120\t38\t0.2\t1\t30 8\n",
      "121\t42\t0.2\t1\t33 9\n",
      "122\t42\t0.2\t1\t35 7\n",
      "123\t53\t0.2\t1\t43 10\n",
      "124\t37\t0.2\t1\t28 9\n",
      "125\t44\t0.2\t1\t34 10\n",
      "126\t37\t0.2\t1\t30 7\n",
      "127\t37\t0.2\t1\t29 8\n",
      "128\t24\t0.2\t1\t18 6\n",
      "129\t12\t0.2\t1\t10 2\n",
      "130\t16\t0.2\t1\t10 6\n",
      "131\t14\t0.2\t1\t12 2\n",
      "132\t9\t0.2\t1\t8 1\n",
      "133\t11\t0.2\t1\t8 3\n",
      "134\t20\t0.2\t1\t17 3\n",
      "135\t14\t0.2\t1\t7 7\n",
      "136\t11\t0.2\t1\t5 6\n",
      "137\t10\t0.2\t1\t9 1\n",
      "138\t20\t0.2\t1\t18 2\n",
      "139\t8\t0.2\t1\t7 1\n",
      "140\t12\t0.2\t1\t11 1\n",
      "141\t16\t0.2\t1\t11 5\n",
      "142\t12\t0.2\t1\t8 4\n",
      "143\t6\t0.2\t1\t5 1\n",
      "144\t9\t0.2\t1\t6 3\n",
      "145\t29\t0.2\t1\t28 1\n",
      "146\t36\t0.2\t1\t32 4\n",
      "147\t10\t0.2\t1\t7 3\n",
      "148\t31\t0.2\t1\t22 9\n",
      "149\t8\t0.2\t1\t7 1\n",
      "150\t11\t0.2\t1\t5 6\n",
      "\n",
      "Unable to close OUT filehandle: Broken pipe\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: ./data/A.1.fastq.gz: uncompress failed\n"
     ]
    }
   ],
   "source": [
    "!./scripts/trim_galore_v0/trim_galore ./data/*.fastq.gz --output_dir /Volumes/Untitled/trimed_fastq/"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
